<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
    <head>
        <meta charset="utf-8"/>
        
        <link rel="shortcut icon" href="data:image/x-icon;base64,AAABAAMAEBAAAAEAIABoBAAANgAAABgYAAABACAAiAkAAJ4EAAAgIAAAAQAgAKgQAAAmDgAAKAAAABAAAAAgAAAAAQAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wCAgIAC////AP///wC0tKJHlZWSqI6OmuRxcXn9koaK9J2Ym8uNm5V73PPoFv///wAAAAAD////AP///wCAgIAC////AP//7xCHh4OodHSi/2xqw/9nZ9H/XVuU/9eFqv/njb7/yYiq/5GAiOifqKJV////AAAAAAL///8A////AO3t2w56eoHPdHLH/3Vz3vx4dtz7dXTb/mVimv/XhKn/8ZPG/fSVx/vvlcT/kG+E/32Himb///8AAAAAA////wB0dHageHbO/3585fl5d9r/dnTX/3V02/5mYpz80X2i/eaIuv/ukcD/wXak/I6Maf+OkXD9iIKjL////wCEhFo+c3Kv/4F+6P18etz/e3nd/3l33P57eeT/aWag/9F5oP/sib3+tmuZ/3uEXf/P1YX7zdKG/3x9dbX///8Ac3J9nYKA4f+CgOb7fnzf/3174P58et//bm2v8Vtgc7SdbIHNqWmM/4aLZP/L0YT/0dWH/tfdiP+go3P6YlyWJ3BvmduGg+n/g4Hk/oF+4/6AfuX/bm2k7pCUUUX///8Ajv/jCVBmXqPBxoH/1NmI/c7Uhf/b4Yn9wMWB/2BdcGBraqX6iYbs/4SB5P+Eguj7hYPm/2Vlcov///8AACRtB////wAAAIAWrLB29uHojv/X3on+3eSK+9PYh/9XWVN7bWum+ouI7/+Gg+f/hoTq+4iF6f9lZXKL////ACQAbQf///8AAABoFoF+Xva0snL/y818/uDmiPvZ34n/WVlTe3Jxm9uMiu//iofr/oiG6f6Jhu3/cnGo7oyQQ0X///8A/znjCXF8dKNotpL/Za2O/WuVe/+FkGv9mJ5w/2pqaGBzc3+djYrr/42K8fuKh+v/i4js/oqH6/93dbfxYmJztGaTec1yxJr/gOGu/4nptf+K6Lf+i+q8/2eXhfqDNG8ngIBKPnt6uP+Rjvf9jInr/4yJ7f+Miu3+jor0/25xoP920pz/gOGw/oHerv+H4rL/kPC9+5Ttvf98ioC1////AP///wB0dHSgioff/5SR+/mOi+7/jYvt/4+L8P5scJ78etKe/YXis/+H47P/kO+8/JLxv/92m4f9nXKNL////wD///8A7e3bDnt7hM+Ihdv/ko/4/JSS+PuUkPb+bnKg/4Lapv+Q7739kfO/+43puf9/qJH/mYeUZv///wAAAAADgICAAv///wD//+8QhoaBqIB/sf+Iht7/jYjw/2tvnf+D3Kf/ieS1/4bIpP+AkYfoqJOfVf///wAAAAAC////AP///wCAgIAC////AP///wCwsJdHlZWQqJORn+RzdHv9hZOK9Jecmcuej5l7/+jzFv///wAAAAAD////AP///wD4PwAA4A8AAMAHAACAAwAAgAEAAAABAAADgQAAA8EAAAPBAAADgQAAAAEAAIABAACAAwAAwAcAAOAPAAD4PwAAKAAAABgAAAAwAAAAAQAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wD///8A////AP///wEAAAAB////AP///wi6urdDubmxj6ennsuQkIzvc3Nz/YSIh/WWoZzWrrezoLe6t1X///8T////AP///wCAgIAC////AP///wD///8A////AP///wD///8A////Af///wD///8Ax8fDQJeYkLV4eH39WlmM/19dpP9dX6b/WFRr/611j/+/fJ//om2I/4N3ff+Jko7Ntri2Xv///wD///8AgICAAv///wD///8A////AP///wD///8B////AP///wOfn5t9c3N29GRjn/9oZ8P/c3LU/XVy3P9rbdH/YVt8/9eIrv/5mcz/75bE/t+Mt/67epz/fnN5/4uTkKP///8W////AAAAAAL///8A////AP///wH///8A////A5iYkZJoaH3/bWvB/3h23Px4dt3+dHPX/3Rx1v9sbcv/YFp6/82ApP/rj8D/65DA//KVxv/1mcn83424/5t1iP98gX7A////F////wCAgIAC////AAAAAAL///8AkJCHdWZlf/9zcs7/fHrj/Hh32/92ddn/dXTX/3Zz2P9vcM//YFt7/8t9of/pi73/54u7/+eNvf/vk8P/7ZbE/pxsh/xLVU7/iomJqP///wD///8A////Af///wCfn5Y4ZmV07XZ0zv9+fOX8enjc/3p43P94dtv/dnXZ/3d12f9xcdD/YFp6/8h5nf/mh7r/44e3/+qMvf/ljrv/jV59/3F+V/68wH3/gYNp/4uIlWf///8AAAAAAv///wJ3d22pc3G6/4KA5/t9e9//fHre/3t53f96eNz/eHbZ/3l23P50dNb9Ylx+/ch2m/3jg7b+44W2/+CItv+NXXz/bnpW/8XIgv/Y3Yv8ur57/3Fxbtni4uIa////AHl5XTdoZ4b4f3zb/4F+5P5+fN//fXvf/3x63v96edz/ennf/nt54P9xccb/Xlxy/7Rvj//gf7P/3IKw/olZeP9te1b/xsqD/9DVh//P1IX/09iG/pWYcf9nZ3Nt////AISEb35ycLb/hYPr/YB+4f9/feH/fnzh/3173/99e+L+enjX/2dmmfhlZXiwYGBiknJkaqGVZ37pg11x/3eBXP7GyoP/0NWH/87Uhf/R1ob/2eCK+8HFff9tbXWv////BG9vdb16eMn/hYPq/4F/4/+BfuP/gH7i/4B94v59e93/Y2KO+ISEbX7y8uQT////AP///weAjo5aT1tP5L7Cf//T2Ij+ztOF/9HWhv/S14b/2d+K/s3Sgv+BgnLcg4OSI11deeaBf9r/hYPo/4OB5P+DgeX/gX/i/4OB6v10c77/cXFjmv///wL///8A////AP///wD///8AeHN9apuecP/U2Yf/0teG/tPZhv/V24f/2uCJ/tTahv+KjGj6AABINV9egfuGg+P/hoTp/4WC5v+EgeX/goDk/4WD6/1wbq//WFgAQ////wAzMzMFAAAAAVVVVQP///8AAABjEoiLZ/Df5Y7/2d+J/dfch//W3If/2d+H/9vhif6ZnW3/AAA5P19egvuHheX/iIXq/4eE6P+Gg+f/hILl/4eE7P1xb7D/WFgAQ////wAzMzMFAAAAAVVVVQP///8AAABVEnd1WvDEx37/y9GD/dnfif/g54z/4OeL/97kiv6anWz/AAA1P15eeuaFg93/ioft/4iF6f+Hhen/hoPn/4mG7/14d8L/cXFimv///wP///8A////AP///wD///8AenV3a1dyZf9QY1//bGVW/pORY/+trnD/xch7/tTZhP+Pkmn6AABINXBwdb1/fc7/jYry/4mH6v+Jhur/iIbp/4iG6/6Gg+X/Z2aT+IKEaX7y8uQT////AP///wiXgY9bXndp5HXLnv95z6P+dsif/2Wrjv9RcWv/Z3Rk/nyBYv9xcWbbfHyDI4SEa354d73/kI31/YuI6/+LiOz/iofr/4mG6v+Lh+3+hoPi/21soPhnZ3qwYGBikmNza6JhinXpcsWa/3vdq/6C4a//iOi1/4zpt/+O6Lf/kOu8+3O1l/9jVmKv////BHl5WDdraor4iofn/4+M8v6Miez/i4nt/4uI7P+LiOv/i4jt/ouJ8P9+e83/XGBt/223jv952aj/e9uq/n3aqv+D36//huKy/4rntv+Q7r3/lvTC/nupj/9zYmxt////AP///wJ3d2qpfXvF/5KP9vyNiu7/jYru/4yJ7f+Miu7/i4jr/46L8P6FgN79XmZ5/XTMm/1+3q7+f9ys/4Lgr/+G5LP/iue2/4zot/+V9cL8iNWs/3BxcNni4uIa////AP///wCfn5I4aGd17YeE3v+Ukfj8jovu/46L7/+Oi+//jYru/46M8f+FgNv/XmV3/3jMnf+D4rH/g+Cw/4fks/+K57b/jem4/5TywP6Q57n/bYh5/5KIjWf///8AAAAAAgAAAAL///8AkJCFdWtqhf+IheL/lZL5/JCN8P+PjO//j4zv/5GO8/+Hgt3/X2Z4/33Rof+I5rb/h+Sz/4nmtf+N6rn/lPPB/pHruvxxm4T/h3yDqP///wD///8A////Af///wH///8A////A5aWj5JtbIP/g4DW/5SR9vyVkvj+kY7y/5KP9P+Ig97/X2d4/4DUpf+N6rn/jeq4/5Lwvv+U88D8h9qu/3KVgv+Lgoe/////F////wCAgIAC////AP///wD///8B////AP///wOfn5l9dHR39HVzsf+Fg97/k4/x/ZiW/v+Nh+f/YWl7/4ngrv+W+MX/kuy8/onesP52uZb/dH54/5aOk6P///8W////AAAAAAL///8A////AP///wD///8A////Af///wD///8Ax8fDQJWVjbV5eX79amic/3h2vv93crf/WV9r/3W1kP95vZn/aZ2C/3eCfP+Ui5DNu7i7Xv///wD///8AgICAAv///wD///8A////AP///wD///8A////AP///wEAAAAB////AP///wi6urdDuLitj6WmmcuPj4rvc3Nz/YiEh/Whlp3Wt66zoLq3ulX///8T////AP///wD///8C////AP///wD///8A////AP8B/wD8AH8A+AAfAOAADwDgAAcAwAAHAIAAAwCAAAMAgAABAAB8AQAAfgEAAP4BAAD+AQAAfgEAAHwBAIAAAQCAAAMAgAADAMAABwDgAAcA4AAPAPgAHwD8AH8A/wH/ACgAAAAgAAAAQAAAAAEAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A////AP///wD///8A////AP///wD///8A////AP///wD///8A////KO3t7XXFxcWwqqqq25eXl/RwcHD9mJiY9Kurq9vHx8ew8PDwdf///yj///8A////AP///wD///8A////AP///wD///8A////AP///wD///8AAAAAAP///wD///8A////AP///wD///8A////AP///wD///8A9/f3TLS0tLpvb3H/UVFR/1VVWf9bW27/WVlz/1FRUf96ZG//c2Nr/1lWV/9RUVH/d3V2/7m5ubr5+flM////AP///wD///8A////AP///wD///8A////AP///wAAAAAA////AP///wD///8A////AP///wD///8A/f39HLW1tatbW13/V1de/2FgkP9rabv/cG7S/29u0/9paL7/UVFR/9GIrv/wlcX/75bE/8+Irf+ZcYb/XVhb/2FfYP+8vLyr/v7+HP///wD///8A////AP///wD///8A////AAAAAAD///8A////AP///wD///8A////AOvr60l2dnnpVVVZ/2ZlnP9ycdL/c3LW/3Jx1f9xcNT/cG/T/2ppvv9RUVH/zoSs/+yRwf/uk8L/7pPC/++UxP/pk8D/pXaP/1hVV/9/fn/p7+/vSf///wD///8A////AP///wD///8AAAAAAP///wD///8A////AP///wDg4OBWX19h/11dcv9zccr/d3XZ/3Z02P91c9f/dHLX/3Nx1v9xcNT/a2q//1FRUf/Ngqn/6o6+/+uQwP/tksH/7pPD/++Vxf/xl8b/3I64/3Jiav9lY2T/2dnZVv///wD///8A////AP///wAAAAAA////AP///wD///8A5OTkQ1tbXf9hYIH/eHfY/3l32/94dtr/d3ba/3Z12f90c9f/c3LW/3Nx1f9tbMH/UVFR/8l/pf/mirr/6Iy8/+mOvv/qj7//7JHB/++VxP/xl8b/04qx/1RUUv9ZWVj/6+vrQ////wD///8A////AAAAAAD///8A////APX19Q9mZmjfYF98/3t53P97ed3/enjc/3l32/93dtr/dnXZ/3Z02P91c9f/dXTY/25tw/9RUVH/yHyj/+WIuf/libn/54q7/+mNvf/rj7//7JHB/9WJsf9XVlX/Z2hb/3BxX/9vb23f+fn5D////wD///8AAAAAAP///wD///8AkZCSnFpaZ/97edn/fHre/3t53v97ed3/enjc/3l33P94dtv/dnXZ/3V02P92dNj/cG7D/1FRUf/FeaH/4oW2/+OGt//liLj/54q6/+mOvv/Sha3/V1VV/2doW//FyYL/xcqC/19fWf+fn5+c////AP///wAAAAAA////ANjY2DhRUVL/dXPA/3994f9+fOD/fXvf/3x63v97ed3/enjc/3l32/94dtr/eHba/3d12f9xb8T/UVFR/8J1nf/fgLH/4IOz/+KEtf/liLn/zoCq/1dVVf9naFr/w8eA/83Shf/P1Yb/q652/1JSUf/h4eE4////AAAAAAD///8Afn6ApWNigv+BfuL/gH3h/3584P99e9//fHre/3x63v97ed3/eXjc/3l32/95d9v/d3ba/3Fwxf9RUVH/wXSc/9x9r//fgbL/4YO0/8t9pv9XVVT/Z2ha/8LHgf/N0oX/ztOF/8/Uhv/S14b/c3Vh/42Ni6X///8AAAAAAOfn5xJRUVH/eXbG/4F/4/+AfuL/f33h/3584P99e+D/fHrf/3x63v97ed3/e3ne/3p43f9ycMH/Y2OR/1FRUf+NY3j/w3Sd/9x9rv/IeaL/V1VU/2doWv/DyIH/zNGF/87Thf/Q1Yb/0daG/9TZh/+ytnj/UlJS/+/v7xIAAAAAq6urYF1cav+DgeX/goDk/4F/4/+AfuL/f33h/3994f9+fOD/fXvf/3x63v95d9X/YF99/1FRUb9sbG+AZmVmgG1qa4BRUVG/fWBv/1dVVP9naFr/xMiB/83Shv/O04b/z9WG/9HWhv/S14b/1NqH/9TZhv9fYFn/t7e3YAAAAAB6enyeaGeV/4SC5v+DgeX/goDk/4F+4/+BfuP/f33h/3584P9+fOD/fHrb/1xcbv9sbG6g39/fQP///wD///8A////ANra2kBbWlugVVVT/8LHgf/O04X/ztOF/8/Uhf/Q1Yb/0teG/9PZh//V24f/2N6I/4aIZv+IiIeeAAAAAFpaXMx0crX/hYLm/4SC5v+DgeX/goDk/4KA5P+Bf+P/gH7i/3994f9oZ5v/X19hv/Hx8SD///8A////AP///wD///8A////AOfn5yBcXFq/m55w/9DVhv/R1ob/0teG/9LXhv/T2Yb/1dqH/9jdiP/Z34j/o6Zy/2VlY8wAAAAAUVFR7Hx7zP+GhOj/hYLm/4SB5f+Egub/g4Hl/4KA5P+Bf+P/gX/j/1paZP+xsbFA////AP///wD///8A////AP///wD///8A////AKampkBqa17/0teG/9LXhv/T2Yf/1dqH/9bciP/Y3oj/2N6H/9rgiP+5vXr/UlJS7AAAAABRUVH7gX7W/4eE6P+GhOj/hYPn/4WC5v+EgeX/g4Hl/4OB5f9/fNv/UVFR/9zc3AD///8A////AP///wD///8A////AP///wD///8A0NDQAFRUU//R14b/1dqH/9Xbh//W3If/192H/9jeh//a4Ij/3OKI/8PJfv9RUVH7AAAAAFFRUfuCgNj/iIbq/4eE6P+HhOj/hoPn/4WC5v+Egub/hILm/4B+3f9RUVH/3NzcAP///wD///8A////AP///wD///8A////AP///wDLy8sAUlJR/87ThP/W3Ij/192H/9nfiP/Z34j/2uCI/9vhiP/d44j/xMh+/1FRUfsAAAAAUVFR7H99zv+Jh+v/iIXq/4iF6v+Hhen/hoTo/4aD5/+Fgub/hYLm/1taZP+xsbFA////AP///wD///8A////AP///wD///8A////AJWVlUBSUlH/UlJR/25vXf+Ul2z/ur97/9jeh//c44j/3uWJ/97lif+8wXr/UVFR7AAAAABaWlzMd3W4/4qH6/+Kh+v/iYbq/4iF6f+HhOj/iIXp/4aE6P+GhOj/a2qe/19fYr/x8fEg////AP///wD///8A////AP///wDp6ekgW1xcv2SWe/9nm4D/Wm9k/1daVv9RUVH/VlZT/3p8Yf+ipnH/ys+A/6ircv9lZmPMAAAAAHp6fJ5rapf/i4js/4uJ7f+KiOz/iYbq/4mG6v+Ihur/iIbq/4eE6P+FguP/Xl5w/2xsb6Df399A////AP///wD///8A2traQGVnZqBccGX/d9Sk/3vZqf+B367/g+Gw/33No/9qk33/W2lh/1ZXVP9RUVH/U1NR/3d3dZ4AAAAAq6urYF5ea/+Ni+//jInt/4uI7P+LiOz/iofr/4mG6v+Jhur/iIXp/4iF6f+EguD/ZGSB/1FRUb9ra2+AZWdmgGpubIBRUVG/Xn1u/3TOn/9516f/ftyr/4Herf+G47L/iOa1/4vot/+Q7bz/k+++/4PHo/9SVFP/oaGhYAAAAADn5+cSUVFR/4F/z/+Niu7/jInt/4yJ7f+LiOz/i4nt/4qI7P+LiOz/iofr/4mG6v+Jhur/fnzN/2ZljP9RUVH/YpB5/2++lf921KT/edem/3vZqf+A3q3/hOGw/4bjs/+J5rX/juu6/5Dsu/+U8cD/g8Wi/1JTUv/t7e0SAAAAAP///wB+foClZ2aG/46L8P+Oi/D/jYru/4yJ7f+Mie3/i4js/4uI7P+LiOz/iofr/4qH6/+KiOz/eXjA/1FRUf9zyJz/eden/3zaqf982qn/ftys/4Hfrv+G47L/iOW0/4rntv+N6rn/ku++/5XxwP9le2//iYqKpf///wAAAAAA////ANjY2DhRUVL/gH7L/4+M8P+Oi+//jovv/42K7v+Niu7/jIru/4yK7v+Mie3/jInt/4uI7P96eMD/UVFR/3bLoP972an/ftyr/4Herv+D4K//heOy/4jmtf+L6Lf/juu5/5DtvP+V8cD/gcGf/1NUU//l5eU4////AAAAAAD///8A////AJGRkpxdXGn/jYrp/5CN8f+PjPD/j4zw/46L7/+Oi+//jYru/42K7v+Niu7/jYru/3x5wv9RUVH/ec6i/4Herv+B367/hOGw/4bjsv+I5bT/i+i3/43quf+Q7bz/k+++/4/jtv9bY17/nZ6enP///wD///8AAAAAAP///wD///8A9vb2D2Zmad9mZYL/kI3w/5CN8f+QjfH/j4zw/4+M8P+PjPD/j4zw/46L7/+Oi+//fHrD/1FRUf9/0qb/hOGw/4Xjsv+F47L/iOa1/4vot/+O67r/kO28/5Lvvf+S7b3/ZHht/3N1dN/6+voP////AP///wAAAAAA////AP///wD///8A5OTkQ1xcXv9oZ4j/kI3v/5GN8f+RjfH/kI3x/5CN8f+QjfH/kI3x/5CN8f99e8P/UVFR/3/UqP+I5bT/iOW0/4vot/+L6Lf/juu5/4/su/+S773/kem6/2Z9cf9gYmH/6+vrQ////wD///8A////AAAAAAD///8A////AP///wD///8A4ODgVl9fYf9iYnf/iojh/5KP8/+Sj/P/kY7y/5GO8v+RjvL/kY7y/358xf9RUVH/hdmt/4vot/+O67n/juu5/5Dtu/+Q7bv/ku++/4rXrv9hcGj/aGpp/+np6Vb///8A////AP///wD///8AAAAAAP///wD///8A////AP///wD///8A7OzsSXd3eelWVlr/dXOr/5CO7v+TkPT/k5D0/5KP8/+Sj/P/f33F/1FRUf+I267/kO28/5Dsu/+Q7Lv/ku+9/5Douv9zoon/U1VU/4GDgunx8fFJ////AP///wD///8A////AP///wAAAAAA////AP///wD///8A////AP///wD///8A/f39HLW1tatbW13/WVlg/3Bunv+Gg9T/k4/y/5SR9f+Bfsf/UVFR/4vfsv+S773/kOu7/4TKpf9ul4H/VlpY/19hYP/AwMCr////HP///wD///8A////AP///wD///8A////AAAAAAD///8A////AP///wD///8A////AP///wD///8A////APf390y0tLS6cG9y/1FRUf9WVlr/Y2J2/2Fgdv9RUVH/ZYBx/2N1a/9VV1b/UVFR/3R2df+/v7+6+vr6TP///wD///8A////AP///wD///8A////AP///wD///8AAAAAAP///wD///8A////AP///wD///8A////AP///wD///8A////AP///wD///8o7e3tdcXFxbCqqqrbl5eX9HBwcP2YmJj0q6ur28fHx7Dv7+91////KP///wD///8A////AP///wD///8A////AP///wD///8A////AP///wAAAAAA//Af//+AA//+AAD//AAAf/gAAD/wAAAf4AAAD8AAAAfAAAAHgAAAA4AAAAOAAAADAAfAAQAP4AEAH/ABAB/wAQAf8AEAH/ABAA/gAQAHwAGAAAADgAAAA4AAAAPAAAAHwAAAB+AAAA/wAAAf+AAAP/wAAH/+AAD//4AD///wH/8="/>
        <script id="notfound">window.onload=function(){document.body.innerHTML="Could not get resources from \"http://krona.sourceforge.net\"."}</script>
        <script>
{//-----------------------------------------------------------------------------
// 
// PURPOSE
// 
// Krona is a flexible tool for exploring the relative proportions of
// hierarchical data, such as metagenomic classifications, using a
// radial, space-filling display. It is implemented using HTML5 and
// JavaScript, allowing charts to be explored locally or served over the
// Internet, requiring only a current version of any major web
// browser. Krona charts can be created using an Excel template or from
// common bioinformatic formats using the provided conversion scripts.
// 
// 
// COPYRIGHT LICENSE
// 
// Copyright (c) 2011, Battelle National Biodefense Institute (BNBI);
// all rights reserved. Authored by: Brian Ondov, Nicholas Bergman, and
// Adam Phillippy
// 
// This Software was prepared for the Department of Homeland Security
// (DHS) by the Battelle National Biodefense Institute, LLC (BNBI) as
// part of contract HSHQDC-07-C-00020 to manage and operate the National
// Biodefense Analysis and Countermeasures Center (NBACC), a Federally
// Funded Research and Development Center.
// 
// Redistribution and use in source and binary forms, with or without
// modification, are permitted provided that the following conditions are
// met:
// 
// * Redistributions of source code must retain the above copyright
//   notice, this list of conditions and the following disclaimer.
// 
// * Redistributions in binary form must reproduce the above copyright
//   notice, this list of conditions and the following disclaimer in the
//   documentation and/or other materials provided with the distribution.
// 
// * Neither the name of the Battelle National Biodefense Institute nor
//   the names of its contributors may be used to endorse or promote
//   products derived from this software without specific prior written
//   permission.
// 
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
// HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
// 
// 
// TRADEMARK LICENSE
// 
// KRONA(TM) is a trademark of the Department of Homeland Security, and use
// of the trademark is subject to the following conditions:
// 
// * Distribution of the unchanged, official code/software using the
//   KRONA(TM) mark is hereby permitted by the Department of Homeland
//   Security, provided that the software is distributed without charge
//   and modification.
// 
// * Distribution of altered source code/software using the KRONA(TM) mark
//   is not permitted unless written permission has been granted by the
//   Department of Homeland Security.
// 
// 
// FOR MORE INFORMATION VISIT
// 
// https://github.com/marbl/Krona/wiki/
// 
//-----------------------------------------------------------------------------
}


var canvas;
var context;
var svg; // for snapshot mode
var collapse = true;
var collapseCheckBox;
var collapseLast;
var compress;
var compressCheckBox;
var maxAbsoluteDepthText;
var maxAbsoluteDepthButtonDecrease;
var maxAbsoluteDepthButtonIncrease;
var fontSize = 11;
var fontSizeText;
var fontSizeButtonDecrease;
var fontSizeButtonIncrease;
var fontSizeLast;
var radiusButtonDecrease;
var radiusButtonIncrease;
var shorten;
var shortenCheckBox;
var maxAbsoluteDepth;
var backButton;
var upButton;
var forwardButton;
var snapshotButton;
var snapshotMode = false;
var details;
var detailsName;
var search;
var searchResults;
var nSearchResults;
var useHueCheckBox;
var useHueDiv;
var datasetDropDown;
var datasetButtonLast;
var datasetButtonPrev;
var datasetButtonNext;
var keyControl;
var showKeys = true;
var linkButton;
var linkText;
var frame;

// Node references. Note that the meanings of 'selected' and 'focused' are
// swapped in the docs.
//
var head; // the root of the entire tree
var selectedNode = 0; // the root of the current view
var focusNode = 0; // a node chosen for more info (single-click)
var highlightedNode = 0; // mouse hover node
var highlightingHidden = false;
var nodes = new Array();
var currentNodeID = 0; // to iterate while loading

var nodeHistory = new Array();
var nodeHistoryPosition = 0;

var dataEnabled = false; // true when supplemental files are present

// store non-Krona GET variables so they can be passed on to links
//
var getVariables = new Array();

// selectedNodeLast is separate from the history, since we need to check
// properties of the last node viewed when browsing through the history
//
var selectedNodeLast = 0;
var zoomOut = false;

// temporary zoom-in while holding the mouse button on a wedge
//
var quickLook = false; // true when in quick look state
var mouseDown = false;
var mouseDownTime; // to detect mouse button hold
var quickLookHoldLength = 200;

var imageWidth;
var imageHeight;
var centerX;
var centerY;
var gRadius;
var updateViewNeeded = false;

// Determines the angle that the pie chart starts at.  90 degrees makes the
// center label consistent with the children.
//
var rotationOffset = Math.PI / 2;

var buffer;
var bufferFactor = .1;

// The maps are the small pie charts showing the current slice being viewed.
//
var mapBuffer = 10;
var mapRadius = 0;
var maxMapRadius = 25;
var mapWidth = 150;
var maxLabelOverhang = Math.PI * 4.18;

// Keys are the labeled boxes for slices in the highest level that are too thin
// to label.
//
var maxKeySizeFactor = 2; // will be multiplied by font size
var keySize;
var keys;
var keyBuffer = 10;
var currentKey;
var keyMinTextLeft;
var keyMinAngle;

var minRingWidthFactor = 5; // will be multiplied by font size
var maxPossibleDepth; // the theoretical max that can be displayed
var maxDisplayDepth; // the actual depth that will be displayed
var headerHeight = 0;//document.getElementById('options').clientHeight;
var historySpacingFactor = 1.6; // will be multiplied by font size
var historyAlphaDelta = .25;

// appearance
//
var lineOpacity = 0.3;
var saturation = 0.5;
var lightnessBase = 0.6;
var lightnessMax = .8;
var thinLineWidth = .3;
var highlightLineWidth = 1.5;
var labelBoxBuffer = 6;
var labelBoxRounding = 15;
var labelWidthFudge = 1.05; // The width of unshortened labels are set slightly
							// longer than the name width so the animation
							// finishes faster.
var fontNormal;
var fontBold;
var fontFamily = 'sans-serif';
//var fontFaceBold = 'bold Arial';
var nodeRadius;
var angleFactor;
var tickLength;
var compressedRadii;

// colors
//
var highlightFill = 'rgba(255, 255, 255, .3)';
var colorUnclassified = 'rgb(220,220,220)';

// label staggering
//
var labelOffsets; // will store the current offset at each depth
//
// This will store pointers to the last node that had a label in each offset (or "track") of a
// each depth.  These will be used to shorten neighboring labels that would overlap.
// The [nLabelNodes] index will store the last node with a radial label.
// labelFirstNodes is the same, but to check for going all the way around and
// overlapping the first labels.
//
var labelLastNodes;
var labelFirstNodes;
//
var nLabelOffsets = 3; // the number of offsets to use

var mouseX = -1;
var mouseY = -1;
var mouseXRel = -1;
var mouseYRel = -1;

// tweening
//
var progress = 0; // for tweening; goes from 0 to 1.
var progressLast = 0;
var tweenFactor = 0; // progress converted by a curve for a smoother effect.
var tweenLength = 850; // in ms
var tweenCurvature = 13;
//
// tweenMax is used to scale the sigmoid function so its range is [0,1] for the
// domain [0,1]
//
var tweenMax = 1 / (1 + Math.exp(-tweenCurvature / 2));
//
var tweenStartTime;

// for framerate debug
//
var tweenFrames = 0;
var fpsDisplay = document.getElementById('frameRate');

// Arrays to translate xml attribute names into displayable attribute names
//
var attributes = new Array();
//
var magnitudeIndex; // the index of attribute arrays used for magnitude
var membersAssignedIndex;
var membersSummaryIndex;

// For defining gradients
//
var hueDisplayName;
var hueStopPositions;
var hueStopHues;
var hueStopText;

// multiple datasets
//
var currentDataset = 0;
var lastDataset = 0;
var datasets = 1;
var datasetNames;
var datasetSelectSize = 30;
var datasetAlpha = new Tween(0, 0);
var datasetWidths = new Array();
var datasetChanged;
var datasetSelectWidth = 50;

window.onload = load;

var image;
var hiddenPattern;
var loadingImage;
var logoImage;

function backingScale()
{
	if ('devicePixelRatio' in window)
	{
		if (window.devicePixelRatio > 1)
		{
			return window.devicePixelRatio;
		}
	}
	
	return 1;
}

function resize()
{
	imageWidth = window.innerWidth;
	imageHeight = window.innerHeight;
	
	if ( ! snapshotMode )
	{
		context.canvas.width = imageWidth * backingScale();
		context.canvas.height = imageHeight * backingScale();
		context.canvas.style.width = imageWidth + "px"
		context.canvas.style.height = imageHeight + "px"
		context.scale(backingScale(), backingScale());
	}
	
	if ( datasetDropDown )
	{
		var ratio = 
			(datasetDropDown.offsetTop + datasetDropDown.clientHeight) * 2 /
			imageHeight;
		
		if ( ratio > 1 )
		{
			ratio = 1;
		}
		
		ratio = Math.sqrt(ratio);
		
		datasetSelectWidth = 
			(datasetDropDown.offsetLeft + datasetDropDown.clientWidth) * ratio;
	}
	var leftMargin = datasets > 1 ? datasetSelectWidth + 30 : 0;
	var minDimension = imageWidth - mapWidth - leftMargin > imageHeight ?
		imageHeight :
		imageWidth - mapWidth - leftMargin;
	
	maxMapRadius = minDimension * .03;
	buffer = minDimension * bufferFactor;
	margin = minDimension * .015;
	centerX = (imageWidth - mapWidth - leftMargin) / 2 + leftMargin;
	centerY = imageHeight / 2;
	gRadius = minDimension / 2 - buffer;
	//context.font = '11px sans-serif';
}

function handleResize()
{
	updateViewNeeded = true;
}

function Attribute()
{
}

function Tween(start, end)
{
	this.start = start;
	this.end = end;
	this.current = this.start;
	
	this.current = function()
	{
		if ( progress == 1 || this.start == this.end )
		{
			return this.end;
		}
		else
		{
			return this.start + tweenFactor * (this.end - this.start);
		}
	};
	
	this.setTarget = function(target)
	{
		this.start = this.current();
		this.end = target;
	}
}

function Node()
{
	this.id = currentNodeID;
	currentNodeID++;
	nodes[this.id] = this;
	
	this.angleStart = new Tween(Math.PI, 0);
	this.angleEnd = new Tween(Math.PI, 0);
	this.radiusInner = new Tween(1, 1);
	this.labelRadius = new Tween(1, 1);
	this.labelWidth = new Tween(0, 0);
	this.scale = new Tween(1, 1); // TEMP
	this.radiusOuter = new Tween(1, 1);
	
	this.r = new Tween(255, 255);
	this.g = new Tween(255, 255);
	this.b = new Tween(255, 255);
	
	this.alphaLabel = new Tween(0, 1);
	this.alphaLine = new Tween(0, 1);
	this.alphaArc = new Tween(0, 0);
	this.alphaWedge = new Tween(0, 1);
	this.alphaOther = new Tween(0, 1);
	this.alphaPattern = new Tween(0, 0);
	this.children = Array();
	this.parent = 0;
	
	this.attributes = new Array(attributes.length);
	
	this.addChild = function(child)
	{
		this.children.push(child);
	};
	
	this.addLabelNode = function(depth, labelOffset)
	{
		if ( labelHeadNodes[depth][labelOffset] == 0 )
		{
			// this will become the head node for this list
			
			labelHeadNodes[depth][labelOffset] = this;
			this.labelPrev = this;
		}
		
		var head = labelHeadNodes[depth][labelOffset];
		
		this.labelNext = head;
		this.labelPrev = head.labelPrev;
		head.labelPrev.labelNext = this;
		head.labelPrev = this;
	}
	
	this.canDisplayDepth = function()
	{
		// whether this node is at a depth that can be displayed, according
		// to the max absolute depth
		
		return this.depth <= maxAbsoluteDepth;
	}
	
	this.canDisplayHistory = function()
	{
		var radiusInner;
		
		if ( compress )
		{
			radiusInner = compressedRadii[0];
		}
		else
		{
			radiusInner = nodeRadius;
		}
		
		return (
			-this.labelRadius.end * gRadius +
			historySpacingFactor * fontSize / 2 <
			radiusInner * gRadius
			);
	}
	
	this.canDisplayLabelCurrent = function()
	{
		return (
			(this.angleEnd.current() - this.angleStart.current()) *
			(this.radiusInner.current() * gRadius + gRadius) >=
			minWidth());
	}
	
	this.checkHighlight = function()
	{
		if ( this.children.length == 0 && this == focusNode )
		{
			//return false;
		}
		
		if ( this.hide )
		{
			return false;
		}
		
		if ( this.radiusInner.end == 1 )
		{
			// compressed to the outside; don't check
			
			return false;
		}
		
		var highlighted = false;
		
		var angleStartCurrent = this.angleStart.current() + rotationOffset;
		var angleEndCurrent = this.angleEnd.current() + rotationOffset;
		var radiusInner = this.radiusInner.current() * gRadius;
		
		for ( var i = 0; i < this.children.length; i++ )
		{
			highlighted = this.children[i].checkHighlight();
			
			if ( highlighted )
			{
				return true;
			}
		}
		
		if ( this.radial )
		{
			var angleText = (angleStartCurrent + angleEndCurrent) / 2;
			var radiusText = (gRadius + radiusInner) / 2;
			
			context.rotate(angleText);
			context.beginPath();
			context.moveTo(radiusText, -fontSize);
			context.lineTo(radiusText, fontSize);
			context.lineTo(radiusText + centerX, fontSize);
			context.lineTo(radiusText + centerX, -fontSize);
			context.closePath();
			context.rotate(-angleText);
			
			if ( context.isPointInPath(mouseXRel, mouseYRel) )
			{
				var label = String(this.getPercentage()) + '%' + '   ' + this.name;
				
				if ( this.searchResultChildren() )
			    {
					label += searchResultString(this.searchResultChildren());
				}
				
				if
				(
					Math.sqrt((mouseXRel) * (mouseXRel) + (mouseYRel) * (mouseYRel)) / backingScale() <
					radiusText + measureText(label)
				)
				{
					highlighted = true;
				}
			}
		}
		else
		{
		    for ( var i = 0; i < this.hiddenLabels.length; i++ )
		    {
		        var hiddenLabel = this.hiddenLabels[i];
		        
				context.rotate(hiddenLabel.angle);
				context.beginPath();
				context.moveTo(gRadius, -fontSize);
				context.lineTo(gRadius, fontSize);
				context.lineTo(gRadius + centerX, fontSize);
				context.lineTo(gRadius + centerX, -fontSize);
				context.closePath();
				context.rotate(-hiddenLabel.angle);
				
				if ( context.isPointInPath(mouseXRel, mouseYRel) )
				{
					var label = String(hiddenLabel.value) + ' more';
					
					if ( hiddenLabel.search )
				    {
						label += searchResultString(hiddenLabel.search);
					}
					
					if
					(
						Math.sqrt((mouseXRel) * (mouseXRel) + (mouseYRel) * (mouseYRel)) / backingScale() <
						gRadius + fontSize + measureText(label)
					)
					{
						highlighted = true;
						break;
					}
				}
			}
		}
		
		if ( ! highlighted && this != selectedNode && ! this.getCollapse() )
		{
			context.beginPath();
			context.arc(0, 0, radiusInner, angleStartCurrent, angleEndCurrent, false);
			context.arc(0, 0, gRadius, angleEndCurrent, angleStartCurrent, true);
			context.closePath();
			
			if ( context.isPointInPath(mouseXRel, mouseYRel) )
			{
				highlighted = true;
			}
			
			if
			(
				! highlighted &&
				(angleEndCurrent - angleStartCurrent) *
				(radiusInner + gRadius) <
				minWidth() &&
				this.getDepth() == selectedNode.getDepth() + 1
			)
			{
				if ( showKeys && this.checkHighlightKey() )
				{
					highlighted = true;
				}
			}
		}
		
		if ( highlighted )
		{
			if ( this != highlightedNode )
			{
			//	document.body.style.cursor='pointer';
			}
			
			highlightedNode = this;
		}
		
		return highlighted;
	}
	
	this.checkHighlightCenter = function()
	{
		if ( ! this.canDisplayHistory() )
		{
			return;
		}
		
		var cx = centerX;
		var cy = centerY - this.labelRadius.end * gRadius;
		//var dim = context.measureText(this.name);
		
		var width = this.nameWidth;
		
		if ( this.searchResultChildren() )
		{
			var results = searchResultString(this.searchResultChildren());
			var dim = context.measureText(results);
			width += dim.width;
		}
		
		if
		(
			mouseX > cx - width / 2 &&
			mouseX < cx + width / 2 &&
			mouseY > cy - historySpacingFactor * fontSize / 2 &&
			mouseY < cy + historySpacingFactor * fontSize / 2
		)
		{
			highlightedNode = this;
			return;
		}
		
		if ( this.getParent() )
		{
			this.getParent().checkHighlightCenter();
		}
	}
	
	this.checkHighlightKey = function()
	{
		var offset = keyOffset();
		
		var xMin = imageWidth - keySize - margin - this.keyNameWidth - keyBuffer;
		var xMax = imageWidth - margin;
		var yMin = offset;
		var yMax = offset + keySize;
		
		currentKey++;
		
		return (
			mouseX > xMin &&
			mouseX < xMax &&
			mouseY > yMin &&
			mouseY < yMax);
	}
	
	this.checkHighlightMap = function()
	{
		if ( this.parent )
		{
			this.parent.checkHighlightMap();
		}
		
		if ( this.getCollapse() || this == focusNode )
		{
			return;
		}
		
		var box = this.getMapPosition();
		
		if
		(
			mouseX > box.x - mapRadius &&
			mouseX < box.x + mapRadius &&
			mouseY > box.y - mapRadius &&
			mouseY < box.y + mapRadius
		)
		{
			highlightedNode = this;
		}
	}
	
/*	this.collapse = function()
	{
		for (var i = 0; i < this.children.length; i++ )
		{
			this.children[i] = this.children[i].collapse();
		}
		
		if
		(
			this.children.length == 1 &&
			this.children[0].magnitude == this.magnitude
		)
		{
			this.children[0].parent = this.parent;
			this.children[0].getDepth() = this.parent.getDepth() + 1;
			return this.children[0];
		}
		else
		{
			return this;
		}
	}
*/	
	this.draw = function(labelMode, selected, searchHighlighted)
	{
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
//		var hidden = false;
		
		if ( selectedNode == this )
		{
			selected = true;
		}
		
		var angleStartCurrent = this.angleStart.current() + rotationOffset;
		var angleEndCurrent = this.angleEnd.current() + rotationOffset;
		var radiusInner = this.radiusInner.current() * gRadius;
		var canDisplayLabelCurrent = this.canDisplayLabelCurrent();
		var hiddenSearchResults = false;
		
/*		if ( ! this.hide )
		{
			for ( var i = 0; i < this.children.length; i++ )
			{
				if ( this.children[i].hide && this.children[i].searchResults )
				{
					hiddenSearchResults = true;
				}
			}
		}
*/		
		var drawChildren =
			( ! this.hide || ! this.hidePrev && progress < 1 ) &&
			( ! this.hideAlone || ! this.hideAlonePrev && progress < 1 );
		
//		if ( this.alphaWedge.current() > 0 || this.alphaLabel.current() > 0 )
		{
			var lastChildAngleEnd;
			
			if ( this.hasChildren() )//canDisplayChildren )
			{
				lastChildAngleEnd =
					this.children[this.children.length - 1].angleEnd.current()
					+ rotationOffset;
			}
			
			if ( labelMode )
			{
				var drawRadial =
				!(
					this.parent &&
					this.parent != selectedNode &&
					angleEndCurrent == this.parent.angleEnd.current() + rotationOffset
				);
				
				if ( angleStartCurrent != angleEndCurrent )
				{
					this.drawLines(angleStartCurrent, angleEndCurrent, radiusInner, drawRadial, selected);
				}
				
				var alphaOtherCurrent = this.alphaOther.current();
				var childRadiusInner;
				
				if ( this == selectedNode || alphaOtherCurrent )
				{
					childRadiusInner =
						this.children[this.children.length - 1].radiusInner.current() * gRadius;
				}
				
				if ( this == selectedNode )
				{
					this.drawReferenceRings(childRadiusInner);
				}
				
				if
				(
					selected &&
					! searchHighlighted &&
					this != selectedNode &&
					(
						this.isSearchResult ||
						this.hideAlone && this.searchResultChildren() ||
						false
//						this.hide &&
//						this.containsSearchResult
					)
				)
				{
					context.globalAlpha = this.alphaWedge.current();
					
					drawWedge
					(
						angleStartCurrent,
						angleEndCurrent,
						radiusInner,
						gRadius,
						highlightFill,
						0,
						true
					);
					
					if
					(
						this.keyed &&
						! showKeys &&
						this.searchResults &&
						! searchHighlighted &&
						this != highlightedNode &&
						this != focusNode
					)
					{
						var angle = (angleEndCurrent + angleStartCurrent) / 2;
						this.drawLabel(angle, true, false, true, true);
					}
					
					//this.drawHighlight(false);
					searchHighlighted = true;
				}
				
				if
				(
					this == selectedNode ||
//					true
					//(canDisplayLabelCurrent) &&
					this != highlightedNode &&
					this != focusNode
				)
				{
					if ( this.radial != this.radialPrev && this.alphaLabel.end == 1 )
					{
						context.globalAlpha = tweenFactor;
					}
					else
					{
						context.globalAlpha = this.alphaLabel.current();
					}
					
					this.drawLabel
					(
						(angleStartCurrent + angleEndCurrent) / 2,
						this.hideAlone && this.searchResultChildren() ||
						(this.isSearchResult || hiddenSearchResults) && selected,
						this == selectedNode && ! this.radial,
						selected,
						this.radial
					);
					
					if ( this.radial != this.radialPrev && this.alphaLabel.start == 1 && progress < 1 )
					{
						context.globalAlpha = 1 - tweenFactor;
						
						this.drawLabel
						(
							(angleStartCurrent + angleEndCurrent) / 2,
							(this.isSearchResult || hiddenSearchResults) && selected,
							this == selectedNodeLast && ! this.radialPrev,
							selected,
							this.radialPrev
						);
					}
				}
				
				if
				(
					alphaOtherCurrent &&
					lastChildAngleEnd != null
				)
				{
					if
					(
						(angleEndCurrent - lastChildAngleEnd) *
						(childRadiusInner + gRadius) >=
						minWidth()
					)
					{
						//context.font = fontNormal;
						context.globalAlpha = this.alphaOther.current();
						
						drawTextPolar
						(
							this.getUnclassifiedText(),
							this.getUnclassifiedPercentage(),
							(lastChildAngleEnd + angleEndCurrent) / 2,
							(childRadiusInner + gRadius) / 2,
							true,
							false,
							false,
							0,
							0
						);
					}
				}
				
				if ( this == selectedNode && this.keyUnclassified && showKeys )
				{
					this.drawKey
					(
						(lastChildAngleEnd + angleEndCurrent) / 2,
						false,
						false
					);
				}
			}
			else
			{
				var alphaWedgeCurrent = this.alphaWedge.current();
				
				if ( alphaWedgeCurrent || this.alphaOther.current() )
				{
					var currentR = this.r.current();
					var currentG = this.g.current();
					var currentB = this.b.current();
						
					var fill = rgbText(currentR, currentG, currentB);
					
					var radiusOuter;
					var lastChildAngle;
					var truncateWedge =
					(
						this.hasChildren() &&
						! this.keyed &&
						(compress || depth < maxDisplayDepth) &&
						drawChildren
					);
					
					if ( truncateWedge )
					{
						radiusOuter = this.children[0].radiusInner.current() * gRadius;
					}
					else
					{
						radiusOuter = gRadius;
					}
					/*
					if ( this.hasChildren() )
					{
						radiusOuter = this.children[0].getUncollapsed().radiusInner.current() * gRadius + 1;
					}
					else
					{ // TEMP
						radiusOuter = radiusInner + nodeRadius * gRadius;
						
						if ( radiusOuter > gRadius )
						{
							radiusOuter = gRadius;
						}
					}
					*/
					context.globalAlpha = alphaWedgeCurrent;
					
					if ( radiusInner != radiusOuter )
					{
						drawWedge
						(
							angleStartCurrent,
							angleEndCurrent,
							radiusInner,
							radiusOuter,//this.radiusOuter.current() * gRadius,
							//'rgba(0, 200, 0, .1)',
							fill,
							this.alphaPattern.current()
						);
						
						if ( truncateWedge )
						{
							// fill in the extra space if the sum of our childrens'
							// magnitudes is less than ours
							
							if ( lastChildAngleEnd < angleEndCurrent )//&& false) // TEMP
							{
								if ( radiusOuter > 1 )
								{
									// overlap slightly to hide the seam
									
	//								radiusOuter -= 1;
								}
								
								if ( alphaWedgeCurrent < 1 )
								{
									context.globalAlpha = this.alphaOther.current();
									drawWedge
									(
										lastChildAngleEnd,
										angleEndCurrent,
										radiusOuter,
										gRadius,
										colorUnclassified,
										0
									);
									context.globalAlpha = alphaWedgeCurrent;
								}
								
								drawWedge
								(
									lastChildAngleEnd,
									angleEndCurrent,
									radiusOuter,
									gRadius,//this.radiusOuter.current() * gRadius,
									//'rgba(200, 0, 0, .1)',
									fill,
									this.alphaPattern.current()
								);
							}
						}
						
						if ( radiusOuter < gRadius )
						{
							// patch up the seam
							//
							context.beginPath();
							context.arc(0, 0, radiusOuter, angleStartCurrent/*lastChildAngleEnd*/, angleEndCurrent, false);
							context.strokeStyle = fill;
							context.lineWidth = 1;
							context.stroke();
						}
					}
					
					if ( this.keyed && selected && showKeys )//&& progress == 1 )
					{
						this.drawKey
						(
							(angleStartCurrent + angleEndCurrent) / 2,
							(
								this == highlightedNode ||
								this == focusNode ||
								this.searchResults
							),
							this == highlightedNode || this == focusNode
						);
					}
				}
			}
		}
		
		this.hiddenLabels = Array();
		
		if ( drawChildren )
		{
			// draw children
			//
			for ( var i = 0; i < this.children.length; i++ )
			{
				if ( this.drawHiddenChildren(i, selected, labelMode, searchHighlighted) )
				{
					i = this.children[i].hiddenEnd;
				}
				else
				{
					this.children[i].draw(labelMode, selected, searchHighlighted);
				}
			}
		}
	};
	
	this.drawHiddenChildren = function
	(
		firstHiddenChild,
		selected,
		labelMode,
		searchHighlighted
	)
	{
		var firstChild = this.children[firstHiddenChild];
		
		if ( firstChild.hiddenEnd == null || firstChild.radiusInner.current() == 1 )
		{
			return false;
		}
		
		for ( var i = firstHiddenChild; i < firstChild.hiddenEnd; i++ )
		{
			if ( ! this.children[i].hide || ! this.children[i].hidePrev && progress < 1 )
			{
				return false;
			}
		}
		
		var angleStart = firstChild.angleStart.current() + rotationOffset;
		var lastChild = this.children[firstChild.hiddenEnd];
		var angleEnd = lastChild.angleEnd.current() + rotationOffset;
		var radiusInner = gRadius * firstChild.radiusInner.current();
		var hiddenChildren = firstChild.hiddenEnd - firstHiddenChild + 1;
		
		if ( labelMode )
		{
			var hiddenSearchResults = 0;
			
			for ( var i = firstHiddenChild; i <= firstChild.hiddenEnd; i++ )
			{
				hiddenSearchResults += this.children[i].searchResults;
				
				if ( this.children[i].magnitude == 0 )
				{
					hiddenChildren--;
				}
			}
			
			if
			(
				selected &&
				(angleEnd - angleStart) * 
				(gRadius + gRadius) >=
				minWidth() ||
				this == highlightedNode &&
				hiddenChildren ||
				hiddenSearchResults
			)
			{
				context.globalAlpha = this.alphaWedge.current();
				
				this.drawHiddenLabel
				(
					angleStart,
					angleEnd,
					hiddenChildren,
					hiddenSearchResults
				);
			}
		}
		
		var drawWedges = true;
		
		for ( var i = firstHiddenChild; i <= firstChild.hiddenEnd; i++ )
		{
			// all hidden children must be completely hidden to draw together
			
			if ( this.children[i].alphaPattern.current() != this.children[i].alphaWedge.current() )
			{
				drawWedges = false;
				break;
			}
		}
		
		if ( labelMode )
		{
			if ( drawWedges )
			{
				var drawRadial = (angleEnd < this.angleEnd.current() + rotationOffset);
				this.drawLines(angleStart, angleEnd, radiusInner, drawRadial);
			}
			
			if ( hiddenSearchResults && ! searchHighlighted )
			{
				drawWedge
				(
					angleStart,
					angleEnd,
					radiusInner,
					gRadius,//this.radiusOuter.current() * gRadius,
					highlightFill,
					0,
					true
				);
			}
		}
		else if ( drawWedges )
		{
			context.globalAlpha = this.alphaWedge.current();
			
			var fill = rgbText
			(
				firstChild.r.current(),
				firstChild.g.current(),
				firstChild.b.current()
			);
			
			drawWedge
			(
				angleStart,
				angleEnd,
				radiusInner,
				gRadius,//this.radiusOuter.current() * gRadius,
				fill,
				context.globalAlpha,
				false
			);
		}
		
		return drawWedges;
	}
	
	this.drawHiddenLabel = function(angleStart, angleEnd, value, hiddenSearchResults)
	{
		var textAngle = (angleStart + angleEnd) / 2;
		var labelRadius = gRadius + fontSize;//(radiusInner + radius) / 2;
		
		var hiddenLabel = Array();
		
		hiddenLabel.value = value;
		hiddenLabel.angle = textAngle;
		hiddenLabel.search = hiddenSearchResults;
		
		this.hiddenLabels.push(hiddenLabel);
		
		drawTick(gRadius - fontSize * .75, fontSize * 1.5, textAngle);
		drawTextPolar
		(
			value.toString() + ' more',
			0, // inner text
			textAngle,
			labelRadius,
			true, // radial
			hiddenSearchResults, // bubble
			this == highlightedNode || this == focusNode, // bold
			false,
			hiddenSearchResults
		);
	}
	
	this.drawHighlight = function(bold)
	{
		var angleStartCurrent = this.angleStart.current() + rotationOffset;
		var angleEndCurrent = this.angleEnd.current() + rotationOffset;
		var radiusInner = this.radiusInner.current() * gRadius;
		
		//this.setHighlightStyle();
		
		if ( this == focusNode && this == highlightedNode && this.hasChildren() )
		{
//			context.fillStyle = "rgba(255, 255, 255, .3)";
			arrow
			(
				angleStartCurrent,
				angleEndCurrent,
				radiusInner
			);
		}
		else
		{
			drawWedge
			(
				angleStartCurrent,
				angleEndCurrent,
				radiusInner,
				gRadius,
				highlightFill,
				0,
				true
			);
		}
		
		// check if hidden children should be highlighted
		//
		for ( var i = 0; i < this.children.length; i++ )
		{
			if
			(
				this.children[i].getDepth() - selectedNode.getDepth() + 1 <=
				maxDisplayDepth &&
				this.children[i].hiddenEnd != null
			)
			{
				var firstChild = this.children[i];
				var lastChild = this.children[firstChild.hiddenEnd];
				var hiddenAngleStart = firstChild.angleStart.current() + rotationOffset;
				var hiddenAngleEnd = lastChild.angleEnd.current() + rotationOffset;
				var hiddenRadiusInner = gRadius * firstChild.radiusInner.current();
				
				drawWedge
				(
					hiddenAngleStart,
					hiddenAngleEnd,
					hiddenRadiusInner,
					gRadius,
					'rgba(255, 255, 255, .3)',
					0,
					true
				);
				
				if ( false && ! this.searchResults )
				{
					this.drawHiddenLabel
					(
						hiddenAngleStart,
						hiddenAngleEnd,
						firstChild.hiddenEnd - i + 1
					);
				}
				
				i = firstChild.hiddenEnd;
			}
		}
		
//			context.strokeStyle = 'black';
		context.fillStyle = 'black';
		
		var highlight = ! ( progress < 1 && zoomOut && this == selectedNodeLast );
		
		var angle = (angleEndCurrent + angleStartCurrent) / 2;
		
		if ( ! (this.keyed && showKeys) )
		{
			this.drawLabel(angle, true, bold, true, this.radial);
		}
	}
	
	this.drawHighlightCenter = function()
	{
		if ( ! this.canDisplayHistory() )
		{
			return;
		}
		
		context.lineWidth = highlightLineWidth;
		context.strokeStyle = 'black';
		context.fillStyle = "rgba(255, 255, 255, .6)";
		
		context.fillStyle = 'black';
		this.drawLabel(3 * Math.PI / 2, true, true, false);
		context.font = fontNormal;
	}
	
	this.drawKey = function(angle, highlight, bold)
	{
		var offset = keyOffset();
		var color;
		var colorText = this.magnitude == 0 ? 'gray' : 'black';
		var patternAlpha = this.alphaPattern.end;
		var boxLeft = imageWidth - keySize - margin;
		var textY = offset + keySize / 2;
		
		var label;
		var keyNameWidth;
		
		if ( this == selectedNode )
		{
			color = colorUnclassified;
			label =
				this.getUnclassifiedText() +
				'   ' +
				this.getUnclassifiedPercentage();
			keyNameWidth = measureText(label, false);
		}
		else
		{
			label = this.keyLabel;
			color = rgbText(this.r.end, this.g.end, this.b.end);
			
			if ( highlight )
			{
				if ( this.searchResultChildren() )
				{
					label = label + searchResultString(this.searchResultChildren());
				}
				
				keyNameWidth = measureText(label, bold);
			}
			else
			{
				keyNameWidth = this.keyNameWidth;
			}
		}
		
		var textLeft = boxLeft - keyBuffer - keyNameWidth - fontSize / 2;
		var labelLeft = textLeft;
		
		if ( labelLeft > keyMinTextLeft - fontSize / 2 )
		{
			keyMinTextLeft -= fontSize / 2;
			
			if ( keyMinTextLeft < centerX - gRadius + fontSize / 2 )
			{
				keyMinTextLeft = centerX - gRadius + fontSize / 2;
			}
			
			labelLeft = keyMinTextLeft;
		}
		
		var lineX = new Array();
		var lineY = new Array();
		
		var bendRadius;
		var keyAngle = Math.atan((textY - centerY) / (labelLeft - centerX));
		var arcAngle;
		
		if ( keyAngle < 0 )
		{
			keyAngle += Math.PI;
		}
		
		if ( keyMinAngle == 0 || angle < keyMinAngle )
		{
			keyMinAngle = angle;
		}
		
		if ( angle > Math.PI && keyMinAngle > Math.PI )
		{
			// allow lines to come underneath the chart
			
			angle -= Math.PI * 2;
		}
		
		lineX.push(Math.cos(angle) * gRadius);
		lineY.push(Math.sin(angle) * gRadius);
		
		if ( angle < keyAngle && textY > centerY + Math.sin(angle) * (gRadius + buffer * (currentKey - 1) / (keys + 1) / 2 + buffer / 2) )
		{
			bendRadius = gRadius + buffer - buffer * currentKey / (keys + 1) / 2;
		}
		else
		{
			bendRadius = gRadius + buffer * currentKey / (keys + 1) / 2 + buffer / 2;
		}
		
		var outside =
			Math.sqrt
			(
				Math.pow(labelLeft - centerX, 2) +
				Math.pow(textY - centerY, 2)
			) > bendRadius;
		
		if ( ! outside )
		{
			arcAngle = Math.asin((textY - centerY) / bendRadius);
			
			keyMinTextLeft = min(keyMinTextLeft, centerX + bendRadius * Math.cos(arcAngle) - fontSize / 2);
			
			if ( labelLeft < textLeft && textLeft > centerX + bendRadius * Math.cos(arcAngle) )
			{
				lineX.push(textLeft - centerX);
				lineY.push(textY - centerY);
			}
		}
		else
		{
			keyMinTextLeft = min(keyMinTextLeft, labelLeft - fontSize / 2);
			
			if ( angle < keyAngle )
			{
				// flip everything over y = x
				//
				arcAngle = Math.PI / 2 - keyLineAngle
				(
					Math.PI / 2 - angle,
					Math.PI / 2 - keyAngle,
					bendRadius,
					textY - centerY,
					labelLeft - centerX,
					lineY,
					lineX
				);
				
			}
			else
			{
				arcAngle = keyLineAngle
				(
					angle,
					keyAngle,
					bendRadius,
					labelLeft - centerX,
					textY - centerY,
					lineX,
					lineY
				);
			}
		}
		
		if ( labelLeft > centerX + bendRadius * Math.cos(arcAngle) ||
		textY > centerY + bendRadius * Math.sin(arcAngle) + .01)
//		if ( outside ||  )
		{
			lineX.push(labelLeft - centerX);
			lineY.push(textY - centerY);
			
			if ( textLeft != labelLeft )
			{
				lineX.push(textLeft - centerX);
				lineY.push(textY - centerY);
			}
		}
		
		context.globalAlpha = this.alphaWedge.current();
		
		if ( snapshotMode )
		{
			var labelSVG;
			
			if ( this == selectedNode )
			{
				labelSVG =
					this.getUnclassifiedText() +
					spacer() +
					this.getUnclassifiedPercentage();
			}
			else
			{
				labelSVG = this.name + spacer() + this.getPercentage() + '%';
			}
			
			svg +=
				'<rect fill="' + color + '" ' +
				'x="' + boxLeft + '" y="' + offset +
				'" width="' + keySize + '" height="' + keySize + '"/>';
			
			if ( patternAlpha )
			{
				svg +=
					'<rect fill="url(#hiddenPattern)" style="stroke:none" ' +
					'x="' + boxLeft + '" y="' + offset +
					'" width="' + keySize + '" height="' + keySize + '"/>';
			}
			
			svg +=
				'<path class="line' +
				(highlight ? ' highlight' : '') +
				'" d="M ' + (lineX[0] + centerX) + ',' +
				(lineY[0] + centerY);
			
			if ( angle != arcAngle )
			{
				svg +=
					' L ' + (centerX + bendRadius * Math.cos(angle)) + ',' +
					(centerY + bendRadius * Math.sin(angle)) +
					' A ' + bendRadius + ',' + bendRadius + ' 0 ' +
					'0,' + (angle > arcAngle ? '0' : '1') + ' ' +
					(centerX + bendRadius * Math.cos(arcAngle)) + ',' +
					(centerY + bendRadius * Math.sin(arcAngle));
			}
			
			for ( var i = 1; i < lineX.length; i++ )
			{
				svg +=
					' L ' + (centerX + lineX[i]) + ',' +
					(centerY + lineY[i]);
			}
			
			svg += '"/>';
			
			if ( highlight )
			{
				if ( this.searchResultChildren() )
				{
					labelSVG = labelSVG + searchResultString(this.searchResultChildren());
				}
				
				drawBubbleSVG
				(
					boxLeft - keyBuffer - keyNameWidth - fontSize / 2,
					textY - fontSize,
					keyNameWidth + fontSize,
					fontSize * 2,
					fontSize,
					0
				);
				
				if ( this.isSearchResult )
				{
					drawSearchHighlights
					(
						label,
						boxLeft - keyBuffer - keyNameWidth,
						textY,
						0
					)
				}
			}
			
			svg += svgText(labelSVG, boxLeft - keyBuffer, textY, 'end', bold, colorText);
		}
		else
		{
			context.fillStyle = color;
			context.translate(-centerX, -centerY);
			context.strokeStyle = 'black';
				context.globalAlpha = 1;//this.alphaWedge.current();
			
			context.fillRect(boxLeft, offset, keySize, keySize);
			
			if ( patternAlpha )
			{
				context.globalAlpha = patternAlpha;
				context.fillStyle = hiddenPattern;
				
				// make clipping box for Firefox performance
				context.beginPath();
				context.moveTo(boxLeft, offset);
				context.lineTo(boxLeft + keySize, offset);
				context.lineTo(boxLeft + keySize, offset + keySize);
				context.lineTo(boxLeft, offset + keySize);
				context.closePath();
				context.save();
				context.clip();
				
				context.fillRect(boxLeft, offset, keySize, keySize);
				context.fillRect(boxLeft, offset, keySize, keySize);
				
				context.restore(); // remove clipping region
			}
			
			if ( highlight )
			{
				this.setHighlightStyle();
				context.fillRect(boxLeft, offset, keySize, keySize);
			}
			else
			{
				context.lineWidth = thinLineWidth;
			}
			
			context.strokeRect(boxLeft, offset, keySize, keySize);
			
			if ( lineX.length )
			{
				context.beginPath();
				context.moveTo(lineX[0] + centerX, lineY[0] + centerY);
				
				context.arc(centerX, centerY, bendRadius, angle, arcAngle, angle > arcAngle);
				
				for ( var i = 1; i < lineX.length; i++ )
				{
					context.lineTo(lineX[i] + centerX, lineY[i] + centerY);
				}
				
				context.globalAlpha = this == selectedNode ?
					this.children[0].alphaWedge.current() :
					this.alphaWedge.current();
				context.lineWidth = highlight ? highlightLineWidth : thinLineWidth;
				context.stroke();
				context.globalAlpha = 1;
			}
			
			if ( highlight )
			{
				drawBubbleCanvas
				(
					boxLeft - keyBuffer - keyNameWidth - fontSize / 2,
					textY - fontSize,
					keyNameWidth + fontSize,
					fontSize * 2,
					fontSize,
					0
				);
				
				if ( this.isSearchResult )
				{
					drawSearchHighlights
					(
						label,
						boxLeft - keyBuffer - keyNameWidth,
						textY,
						0
					)
				}
			}
			
			drawText(label, boxLeft - keyBuffer, offset + keySize / 2, 0, 'end', bold, colorText);
			
			context.translate(centerX, centerY);
		}
		
		currentKey++;
	}
	
	this.drawLabel = function(angle, bubble, bold, selected, radial)
	{
		if ( context.globalAlpha == 0 )
		{
			return;
		}
		
		var innerText;
		var label;
		var radius;
		
		if ( radial )
		{
			radius = (this.radiusInner.current() + 1) * gRadius / 2;
		}
		else
		{
			radius = this.labelRadius.current() * gRadius;
		}
		
		if ( radial && (selected || bubble ) )
		{
			var percentage = this.getPercentage();
			innerText = percentage + '%';
		}
		
		if
		(
			! radial &&
			this != selectedNode &&
			! bubble &&
			( !zoomOut || this != selectedNodeLast)
		)
		{
			label = this.shortenLabel();
		}
		else
		{
			label = this.name;
		}
		
		var flipped = drawTextPolar
		(
			label,
			innerText,
			angle,
			radius,
			radial,
			bubble,
			bold,
//			this.isSearchResult && this.shouldAddSearchResultsString() && (!selected || this == selectedNode || highlight),
			this.isSearchResult && (!selected || this == selectedNode || bubble),
			(this.hideAlone || !selected || this == selectedNode ) ? this.searchResultChildren() : 0
		);
		
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
		
		if
		(
			! radial &&
			! bubble &&
			this != selectedNode &&
			this.angleEnd.end != this.angleStart.end &&
			nLabelOffsets[depth - 2] > 2 &&
			this.labelWidth.current() > (this.angleEnd.end - this.angleStart.end) * Math.abs(radius) &&
			! ( zoomOut && this == selectedNodeLast ) &&
			this.labelRadius.end > 0
		)
		{
			// name extends beyond wedge; draw tick mark towards the central
			// radius for easier identification
			
			var radiusCenter = compress ?
				(compressedRadii[depth - 1] + compressedRadii[depth - 2]) / 2 :
				(depth - .5) * nodeRadius;
			
			if ( this.labelRadius.end > radiusCenter )
			{
				if ( flipped )
				{
					drawTick(radius - tickLength * 1.4 , tickLength, angle);
				}
				else
				{
					drawTick(radius - tickLength * 1.7, tickLength, angle);
				}
			}
			else
			{
				if ( flipped )
				{
					drawTick(radius + tickLength * .7, tickLength, angle);
				}
				else
				{
					drawTick(radius + tickLength * .4, tickLength, angle);
				}
			}
		}
	}
	
	this.drawLines = function(angleStart, angleEnd, radiusInner, drawRadial, selected)
	{
		if ( snapshotMode )
		{
			if ( this != selectedNode)
			{
				if ( angleEnd == angleStart + Math.PI * 2 )
				{
					// fudge to prevent overlap, which causes arc ambiguity
					//
					angleEnd -= .1 / gRadius;
				}
				
				var longArc = angleEnd - angleStart > Math.PI ? 1 : 0;
				
				var x1 = centerX + radiusInner * Math.cos(angleStart);
				var y1 = centerY + radiusInner * Math.sin(angleStart);
				
				var x2 = centerX + gRadius * Math.cos(angleStart);
				var y2 = centerY + gRadius * Math.sin(angleStart);
				
				var x3 = centerX + gRadius * Math.cos(angleEnd);
				var y3 = centerY + gRadius * Math.sin(angleEnd);
				
				var x4 = centerX + radiusInner * Math.cos(angleEnd);
				var y4 = centerY + radiusInner * Math.sin(angleEnd);
				
				if ( this.alphaArc.end )
				{
					var dArray =
					[
						" M ", x4, ",", y4,
						" A ", radiusInner, ",", radiusInner, " 0 ", longArc,
							" 0 ", x1, ",", y1
					];
					
					svg += '<path class="line" d="' + dArray.join('') + '"/>';
				}
				
				if ( drawRadial && this.alphaLine.end )
				{
					svg += '<line x1="' + x3 + '" y1="' + y3 + '" x2="' + x4 + '" y2="' + y4 + '"/>';
				}
			}
		}
		else
		{
			context.lineWidth = thinLineWidth;
			context.strokeStyle = 'black';
			context.beginPath();
			context.arc(0, 0, radiusInner, angleStart, angleEnd, false);
			context.globalAlpha = this.alphaArc.current();
			context.stroke();
			
			if ( drawRadial )
			{
				var x1 = radiusInner * Math.cos(angleEnd);
				var y1 = radiusInner * Math.sin(angleEnd);
				var x2 = gRadius * Math.cos(angleEnd);
				var y2 = gRadius * Math.sin(angleEnd);
				
				context.beginPath();
				context.moveTo(x1, y1);
				context.lineTo(x2, y2);
				
//				if ( this.getCollapse() )//( selected && this != selectedNode )
				{
					context.globalAlpha = this.alphaLine.current();
				}
				
				context.stroke();
			}
		}
	}
	
	this.drawMap = function(child)
	{
		if ( this.parent )
		{
			this.parent.drawMap(child);
		}
		
		if ( this.getCollapse() && this != child || this == focusNode )
		{
			return;
		}
		
		var angleStart =
			(child.baseMagnitude - this.baseMagnitude) / this.magnitude * Math.PI * 2 +
			rotationOffset;
		var angleEnd =
			(child.baseMagnitude - this.baseMagnitude + child.magnitude) /
			this.magnitude * Math.PI * 2 +
			rotationOffset;
		
		var box = this.getMapPosition();
		
		context.save();
		context.fillStyle = 'black';
		context.textAlign = 'end';
		context.textBaseline = 'middle';
		
		var textX = box.x - mapRadius - mapBuffer;
		var percentage = getPercentage(child.magnitude / this.magnitude);
		
		var highlight = this == selectedNode || this == highlightedNode;
		
		if ( highlight )
		{
			context.font = fontBold;
		}
		else
		{
			context.font = fontNormal;
		}
		
		context.fillText(percentage + '% of', textX, box.y - mapRadius / 3);
		context.fillText(this.name, textX, box.y + mapRadius / 3);
		
		if ( highlight )
		{
			context.font = fontNormal;
		}
		
		if ( this == highlightedNode && this != selectedNode )
		{
			context.fillStyle = 'rgb(245, 245, 245)';
//			context.fillStyle = 'rgb(200, 200, 200)';
		}
		else
		{
			context.fillStyle = 'rgb(255, 255, 255)';
		}
		
		context.beginPath();
		context.arc(box.x, box.y, mapRadius, 0, Math.PI * 2, true);
		context.closePath();
		context.fill();
		
		if ( this == selectedNode )
		{
			context.lineWidth = 1;
			context.fillStyle = 'rgb(100, 100, 100)';
		}
		else
		{
			if ( this == highlightedNode )
			{
				context.lineWidth = .2;
				context.fillStyle = 'rgb(190, 190, 190)';
			}
			else
			{
				context.lineWidth = .2;
				context.fillStyle = 'rgb(200, 200, 200)';
			}
		}
		
		var maxDepth = this.getMaxDepth();
		
		if ( ! compress && maxDepth > maxPossibleDepth + this.getDepth() - 1 )
		{
			maxDepth = maxPossibleDepth + this.getDepth() - 1;
		}
		
		if ( this.getDepth() < selectedNode.getDepth() )
		{
			if ( child.getDepth() - 1 >= maxDepth )
			{
				maxDepth = child.getDepth();
			}
		}
		
		var radiusInner;
		
		if ( compress )
		{
			radiusInner = 0;
//				Math.atan(child.getDepth() - this.getDepth()) /
//				Math.PI * 2 * .9;
		}
		else
		{
			radiusInner =
				(child.getDepth() - this.getDepth()) /
				(maxDepth - this.getDepth() + 1);
		}
		
		context.stroke();
		context.beginPath();
		
		if ( radiusInner == 0 )
		{
			context.moveTo(box.x, box.y);
		}
		else
		{
			context.arc(box.x, box.y, mapRadius * radiusInner, angleEnd, angleStart, true);
		}
		
		context.arc(box.x, box.y, mapRadius, angleStart, angleEnd, false);
		context.closePath();
		context.fill();
		
		if ( this == highlightedNode && this != selectedNode )
		{
			context.lineWidth = 1;
			context.stroke();
		}
		
		context.restore();
	}
	
	this.drawReferenceRings = function(childRadiusInner)
	{
		if ( snapshotMode )
		{
			svg +=
				'<circle cx="' + centerX + '" cy="' + centerY +
				'" r="' + childRadiusInner + '"/>';
			svg +=
				'<circle cx="' + centerX + '" cy="' + centerY +
				'" r="' + gRadius + '"/>';
		}
		else
		{
			context.globalAlpha = 1 - this.alphaLine.current();//this.getUncollapsed().alphaLine.current();
			context.beginPath();
			context.arc(0, 0, childRadiusInner, 0, Math.PI * 2, false);
			context.stroke();
			context.beginPath();
			context.arc(0, 0, gRadius, 0, Math.PI * 2, false);
			context.stroke();
		}
	}
	
	this.getCollapse = function()
	{
		return (
			collapse &&
			this.collapse &&
			this.depth != maxAbsoluteDepth
			);
	}
	
	this.getDepth = function()
	{
		if ( collapse )
		{
			return this.depthCollapsed;
		}
		else
		{
			return this.depth;
		}
	}
	
	this.getMagnitude = function()
	{
		return this.attributes[magnitudeIndex][currentDataset];
	}
	
	this.getMapPosition = function()
	{
		return {
			x : (details.offsetLeft + details.clientWidth - mapRadius),
			y : ((focusNode.getDepth() - this.getDepth()) *
				(mapBuffer + mapRadius * 2) - mapRadius) +
				details.clientHeight + details.offsetTop
		};
	}
	
	this.getMaxDepth = function(limit)
	{
		var max;
		
		if ( collapse )
		{
			return this.maxDepthCollapsed;
		}
		else
		{
			if ( this.maxDepth > maxAbsoluteDepth )
			{
				return maxAbsoluteDepth;
			}
			else
			{
				return this.maxDepth;
			}
		}
	}
	
	this.getData = function(index, summary)
	{
		var files = new Array();
		
		if
		(
			this.attributes[index] != null &&
			this.attributes[index][currentDataset] != null &&
			this.attributes[index][currentDataset] != ''
		)
		{
			files.push
			(
				document.location +
				'.files/' +
				this.attributes[index][currentDataset]
			);
		}
		
		if ( summary )
		{
			for ( var i = 0; i < this.children.length; i++ )
			{
				files = files.concat(this.children[i].getData(index, true));
			}
		}
		
		return files;
	}
	
	this.getList = function(index, summary)
	{
		var list;
		
		if
		(
			this.attributes[index] != null &&
			this.attributes[index][currentDataset] != null
		)
		{
			list = this.attributes[index][currentDataset];
		}
		else
		{
			list = new Array();
		}
		
		if ( summary )
		{
			for ( var i = 0; i < this.children.length; i++ )
			{
				list = list.concat(this.children[i].getList(index, true));
			}
		}
		
		return list;
	}
	
	this.getParent = function()
	{
		// returns parent, accounting for collapsing or 0 if doesn't exist
		
		var parent = this.parent;
		
		while ( parent != 0 && parent.getCollapse() )
		{
			parent = parent.parent;
		}
		
		return parent;
	}
	
	this.getPercentage = function()
	{
		return getPercentage(this.magnitude / selectedNode.magnitude);
	}
	
	this.getUnclassifiedPercentage = function()
	{
		var lastChild = this.children[this.children.length - 1];
		
		return getPercentage
		(
			(
				this.baseMagnitude +
				this.magnitude -
				lastChild.magnitude -
				lastChild.baseMagnitude
			) / this.magnitude
		) + '%';
	}
	
	this.getUnclassifiedText = function()
	{
		return '[other '+ this.name + ']';
	}
	
	this.getUncollapsed = function()
	{
		// recurse through collapsed children until uncollapsed node is found
		
		if ( this.getCollapse() )
		{
			return this.children[0].getUncollapsed();
		}
		else
		{
			return this;
		}
	}
	
	this.hasChildren = function()
	{
		return this.children.length && this.depth < maxAbsoluteDepth && this.magnitude;
	}
	
	this.hasParent = function(parent)
	{
		if ( this.parent )
		{
			if ( this.parent == parent )
			{
				return true;
			}
			else
			{
				return this.parent.hasParent(parent);
			}
		}
		else
		{
			return false;
		}
	}
	
	this.maxVisibleDepth = function(maxDepth)
	{
		var childInnerRadius;
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
		var currentMaxDepth = depth;
		
		if ( this.hasChildren() && depth < maxDepth)
		{
			var lastChild = this.children[this.children.length - 1];
			
			if ( this.name == 'Pseudomonadaceae' )
			{
				var x = 3;
			}
			
			if
			(
				lastChild.baseMagnitude + lastChild.magnitude <
				this.baseMagnitude + this.magnitude
			)
			{
				currentMaxDepth++;
			}
			
			if ( compress )
			{
				childInnerRadius = compressedRadii[depth - 1];
			}
			else
			{
				childInnerRadius = (depth) / maxDepth;
			}
			
			for ( var i = 0; i < this.children.length; i++ )
			{
				if
				(//true ||
					this.children[i].magnitude *
					angleFactor *
					(childInnerRadius + 1) *
					gRadius >=
					minWidth()
				)
				{
					var childMaxDepth = this.children[i].maxVisibleDepth(maxDepth);
					
					if ( childMaxDepth > currentMaxDepth )
					{
						currentMaxDepth = childMaxDepth;
					}
				}
			}
		}
		
		return currentMaxDepth;
	}
	
	this.resetLabelWidth = function()
	{
		var nameWidthOld = this.nameWidth;
		
		if ( true || ! this.radial )//&& fontSize != fontSizeLast )
		{
			var dim = context.measureText(this.name);
			this.nameWidth = dim.width;
		}
		
		if ( fontSize != fontSizeLast && this.labelWidth.end == nameWidthOld * labelWidthFudge )
		{
			// font size changed; adjust start of tween to match
			
			this.labelWidth.start = this.nameWidth * labelWidthFudge;
		}
		else
		{
			this.labelWidth.start = this.labelWidth.current();
		}
		
		this.labelWidth.end = this.nameWidth * labelWidthFudge;
	}
	
	this.restrictLabelWidth = function(width)
	{
		if ( width < this.labelWidth.end )
		{
			this.labelWidth.end = width;
		}
	}
	
	this.search = function()
	{
		this.isSearchResult = false;
		this.searchResults = 0;
		
		if
		(
			! this.getCollapse() &&
			search.value != '' &&
			this.name.toLowerCase().indexOf(search.value.toLowerCase()) != -1
		)
		{
			this.isSearchResult = true;
			this.searchResults = 1;
			nSearchResults++;
		}
		
		for ( var i = 0; i < this.children.length; i++ )
		{
			this.searchResults += this.children[i].search();
		}
		
		return this.searchResults;
	}
	
	this.searchResultChildren = function()
	{
		if ( this.isSearchResult )
		{
			return this.searchResults - 1;
		}
		else
		{
			return this.searchResults;
		}
	}
	
	this.setDepth = function(depth, depthCollapsed)
	{
		this.depth = depth;
		this.depthCollapsed = depthCollapsed;
		
		if
		(
			this.children.length == 1 &&
//			this.magnitude > 0 &&
			this.children[0].magnitude == this.magnitude &&
			( head.children.length > 1 || this.children[0].children.length )
		)
		{
			this.collapse = true;
		}
		else
		{
			this.collapse = false;
			depthCollapsed++;
		}
		
		for ( var i = 0; i < this.children.length; i++ )
		{
			this.children[i].setDepth(depth + 1, depthCollapsed);
		}
	}
	
	this.setHighlightStyle = function()
	{
		context.lineWidth = highlightLineWidth;
		
		if ( this.hasChildren() || this != focusNode || this != highlightedNode )
		{
			context.strokeStyle = 'black';
			context.fillStyle = "rgba(255, 255, 255, .3)";
		}
		else
		{
			context.strokeStyle = 'rgb(90,90,90)';
			context.fillStyle = "rgba(155, 155, 155, .3)";
		}
	}
	
	this.setLabelWidth = function(node)
	{
		if ( ! shorten || this.radial )
		{
			return; // don't need to set width
		}
		
		if ( node.hide )
		{
			alert('wtf');
			return;
		}
		
		var angle = (this.angleStart.end + this.angleEnd.end) / 2;
		var a; // angle difference
		
		if ( node == selectedNode )
		{
			a = Math.abs(angle - node.angleOther);
		}
		else
		{
			a = Math.abs(angle - (node.angleStart.end + node.angleEnd.end) / 2);
		}
		
		if ( a == 0 )
		{
			return;
		}
		
		if ( a > Math.PI )
		{
			a = 2 * Math.PI - a;
		}
		
		if ( node.radial || node == selectedNode )
		{
			var nodeLabelRadius;
			
			if ( node == selectedNode )
			{
				// radial 'other' label
				
				nodeLabelRadius = (node.children[0].radiusInner.end + 1) / 2;
			}
			else
			{
				nodeLabelRadius = (node.radiusInner.end + 1) / 2;
			}
			
			if ( a < Math.PI / 2 )
			{
				var r = this.labelRadius.end * gRadius + .5 * fontSize
				var hypotenuse = r / Math.cos(a);
				var opposite = r * Math.tan(a);
				var fontRadius = .8 * fontSize;
				
				if
				(
					nodeLabelRadius * gRadius < hypotenuse &&
					this.labelWidth.end / 2 + fontRadius > opposite
				)
				{
					this.labelWidth.end = 2 * (opposite - fontRadius);
				}
			}
		}
		else if
		(
			this.labelRadius.end == node.labelRadius.end &&
			a < Math.PI / 4
		)
		{
			// same radius with small angle; use circumferential approximation
			
			var dist = a * this.labelRadius.end * gRadius - fontSize * (1 - a * 4 / Math.PI) * 1.3;
			
			if ( this.labelWidth.end < dist )
			{
				node.restrictLabelWidth((dist - this.labelWidth.end / 2) * 2);
			}
			else if ( node.labelWidth.end < dist )
			{
				this.restrictLabelWidth((dist - node.labelWidth.end / 2) * 2);
			}
			else
			{
				// both labels reach halfway point; restrict both
				
				this.labelWidth.end = dist;
				node.labelWidth.end = dist
			}
		}
		else
		{
			var r1 = this.labelRadius.end * gRadius;
			var r2 = node.labelRadius.end * gRadius;
			
			// first adjust the radii to account for the height of the font by shifting them
			// toward each other
			//
			var fontFudge = .35 * fontSize;
			//
			if ( this.labelRadius.end < node.labelRadius.end )
			{
				r1 += fontFudge;
				r2 -= fontFudge;
			}
			else if ( this.labelRadius.end > node.labelRadius.end )
			{
				r1 -= fontFudge;
				r2 += fontFudge;
			}
			else
			{
				r1 -= fontFudge;
				r2 -= fontFudge;
			}
			
			var r1s = r1 * r1;
			var r2s = r2 * r2;
			
			// distance between the centers of the two labels
			//
			var dist = Math.sqrt(r1s + r2s - 2 * r1 * r2 * Math.cos(a));
			
			// angle at our label center between our radius and the line to the other label center
			//
			var b = Math.acos((r1s + dist * dist - r2s) / (2 * r1 * dist));
			
			// distance from our label center to the intersection of the two tangents
			//
			var l1 = Math.sin(a + b - Math.PI / 2) * dist / Math.sin(Math.PI - a);
			
			// distance from other label center the the intersection of the two tangents
			//
			var l2 = Math.sin(Math.PI / 2 - b) * dist / Math.sin(Math.PI - a);
			
			l1 = Math.abs(l1) - .4 * fontSize;
			l2 = Math.abs(l2) - .4 * fontSize;
/*			
			// amount to shorten the distances because of the height of the font
			//
			var l3 = 0;
			var fontRadius = fontSize * .55;
			//
			if ( l1 < 0 || l2 < 0 )
			{
				var l4 = fontRadius / Math.tan(a);
			l1 = Math.abs(l1);
			l2 = Math.abs(l2);
			
				l1 -= l4;
				l2 -= l4;
			}
			else
			{
				var c = Math.PI - a;
				
				l3 = fontRadius * Math.tan(c / 2);
			}
*/			
			if ( this.labelWidth.end / 2 > l1 && node.labelWidth.end / 2 > l2 )
			{
				// shorten the farthest one from the intersection
				
				if ( l1 > l2 )
				{
					this.restrictLabelWidth(2 * (l1));// - l3 - fontRadius));
				}
				else
				{
					node.restrictLabelWidth(2 * (l2));// - l3 - fontRadius));
				}
			}/*
			else if ( this.labelWidth.end / 2 > l1 + l3 && node.labelWidth.end / 2 > l2 - l3 )
			{
				node.restrictLabelWidth(2 * (l2 - l3));
			}
			else if ( this.labelWidth.end / 2 > l1 - l3 && node.labelWidth.end / 2 > l2 + l3 )
			{
				this.restrictLabelWidth(2 * (l1 - l3));
			}*/
		}
	}
	
	this.setMagnitudes = function(baseMagnitude)
	{
		this.magnitude = this.getMagnitude();
		this.baseMagnitude = baseMagnitude;
		
		for ( var i = 0; i < this.children.length; i++ )
		{
			this.children[i].setMagnitudes(baseMagnitude);
			baseMagnitude += this.children[i].magnitude;
		}
		
		this.maxChildMagnitude = baseMagnitude;
	}
	
	this.setMaxDepths = function()
	{
		this.maxDepth = this.depth;
		this.maxDepthCollapsed = this.depthCollapsed;
		
		for ( i in this.children )
		{
			var child = this.children[i];
			
			child.setMaxDepths();
			
			if ( child.maxDepth > this.maxDepth )
			{
				this.maxDepth = child.maxDepth;
			}
			
			if
			(
				child.maxDepthCollapsed > this.maxDepthCollapsed &&
				(child.depth <= maxAbsoluteDepth || maxAbsoluteDepth == 0)
			)
			{
				this.maxDepthCollapsed = child.maxDepthCollapsed;
			}
		}
	}
	
	this.setTargetLabelRadius = function()
	{
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
		var index = depth - 2;
		var labelOffset = labelOffsets[index];
		
		if ( this.radial )
		{
			//this.labelRadius.setTarget((this.radiusInner.end + 1) / 2);
			var max =
				depth == maxDisplayDepth ?
				1 :
				compressedRadii[index + 1];
			
			this.labelRadius.setTarget((compressedRadii[index] + max) / 2);
		}
		else
		{
			var radiusCenter;
			var width;
			
			if ( compress )
			{
				if ( nLabelOffsets[index] > 1 )
				{
					this.labelRadius.setTarget
					(
						lerp
						(
							labelOffset + .75,
							0,
							nLabelOffsets[index] + .5,
							compressedRadii[index],
							compressedRadii[index + 1]
						)
					);
				}
				else
				{
					this.labelRadius.setTarget((compressedRadii[index] + compressedRadii[index + 1]) / 2);
				}
			}
			else
			{
				radiusCenter =
					nodeRadius * (depth - 1) +
					nodeRadius / 2;
				width = nodeRadius;
				
				this.labelRadius.setTarget
				(
					radiusCenter + width * ((labelOffset + 1) / (nLabelOffsets[index] + 1) - .5)
				);
			}
		}
		
		if ( ! this.hide && ! this.keyed && nLabelOffsets[index] )
		{
			// check last and first labels in each track for overlap
			
			for ( var i = 0; i < maxDisplayDepth - 1; i++ )
			{
				for ( var j = 0; j <= nLabelOffsets[i]; j++ )
				{
					var last = labelLastNodes[i][j];
					var first = labelFirstNodes[i][j];
					
					if ( last )
					{
						if ( j == nLabelOffsets[i] )
						{
							// last is radial
							this.setLabelWidth(last);
						}
						else
						{
							last.setLabelWidth(this);
						}
					}
					
					if ( first )
					{
						if ( j == nLabelOffsets[i] )
						{
							this.setLabelWidth(first);
						}
						else
						{
							first.setLabelWidth(this);
						}
					}
				}
			}
			
			if ( selectedNode.canDisplayLabelOther )
			{
				this.setLabelWidth(selectedNode); // in case there is an 'other' label
			}
			
			if ( this.radial )
			{
				// use the last 'track' of this depth for radial
				
				labelLastNodes[index][nLabelOffsets[index]] = this;
				
				if ( labelFirstNodes[index][nLabelOffsets[index]] == 0 )
				{
					labelFirstNodes[index][nLabelOffsets[index]] = this;
				}
			}
			else
			{
				labelLastNodes[index][labelOffset] = this;
				
				// update offset
				
				labelOffsets[index] += 1;
				
				if ( labelOffsets[index] > nLabelOffsets[index] )
				{
					labelOffsets[index] -= nLabelOffsets[index];
					
					if ( !(nLabelOffsets[index] & 1) )
					{
						labelOffsets[index]--;
					}
				}
				else if ( labelOffsets[index] == nLabelOffsets[index] )
				{
					labelOffsets[index] -= nLabelOffsets[index];
					
					if ( false && !(nLabelOffsets[index] & 1) )
					{
						labelOffsets[index]++;
					}
				}
				
				if ( labelFirstNodes[index][labelOffset] == 0 )
				{
					labelFirstNodes[index][labelOffset] = this;
				}
			}
		}
		else if ( this.hide )
		{
			this.labelWidth.end = 0;
		}
	}
	
	this.setTargets = function()
	{
		if ( this == selectedNode )
		{
			this.setTargetsSelected
			(
				0,
				1,
				lightnessBase,
				false,
				false
			);
			return;
		}
		
		var depthRelative = this.getDepth() - selectedNode.getDepth();
		
		var parentOfSelected = selectedNode.hasParent(this);
/*		(
//			! this.getCollapse() &&
			this.baseMagnitude <= selectedNode.baseMagnitude &&
			this.baseMagnitude + this.magnitude >=
			selectedNode.baseMagnitude + selectedNode.magnitude
		);
*/		
		if ( parentOfSelected )
		{
			this.resetLabelWidth();
		}
		else
		{
			//context.font = fontNormal;
			var dim = context.measureText(this.name);
			this.nameWidth = dim.width;
			//this.labelWidth.setTarget(this.labelWidth.end);
			this.labelWidth.setTarget(0);
		}
		
		// set angles
		//
		if ( this.baseMagnitude <= selectedNode.baseMagnitude )
		{
			this.angleStart.setTarget(0);
		}
		else
		{
			this.angleStart.setTarget(Math.PI * 2);
		}
		//
		if
		(
			parentOfSelected ||
			this.baseMagnitude + this.magnitude >=
			selectedNode.baseMagnitude + selectedNode.magnitude
		)
		{
			this.angleEnd.setTarget(Math.PI * 2);
		}
		else
		{
			this.angleEnd.setTarget(0);
		}
		
		// children
		//
		for ( var i = 0; i < this.children.length; i++ )
		{
			this.children[i].setTargets();
		}
		
		if ( this.getDepth() <= selectedNode.getDepth() )
		{
			// collapse in
			
			this.radiusInner.setTarget(0);
			
			if ( parentOfSelected )
			{
				this.labelRadius.setTarget
				(
					(depthRelative) *
					historySpacingFactor * fontSize / gRadius
				);
				//this.scale.setTarget(1 - (selectedNode.getDepth() - this.getDepth()) / 18); // TEMP
			}
			else
			{
				this.labelRadius.setTarget(0);
				//this.scale.setTarget(1); // TEMP
			}
		}
		else if ( depthRelative + 1 > maxDisplayDepth )
		{
			// collapse out
			
			this.radiusInner.setTarget(1);
			this.labelRadius.setTarget(1);
			//this.scale.setTarget(1); // TEMP
		}
		else
		{
			// don't collapse
			
			if ( compress )
			{
				this.radiusInner.setTarget(compressedRadii[depthRelative - 1]);
			}
			else
			{
				this.radiusInner.setTarget(nodeRadius * (depthRelative));
			}
			
			//this.scale.setTarget(1); // TEMP
			
			if ( this == selectedNode )
			{
				this.labelRadius.setTarget(0);
			}
			else
			{
				if ( compress )
				{
					this.labelRadius.setTarget
					(
						(compressedRadii[depthRelative - 1] + compressedRadii[depthRelative]) / 2
					);
				}
				else
				{
					this.labelRadius.setTarget(nodeRadius * (depthRelative) + nodeRadius / 2);
				}
			}
		}
		
//		this.r.start = this.r.end;
//		this.g.start = this.g.end;
//		this.b.start = this.b.end;
		
		this.r.setTarget(255);
		this.g.setTarget(255);
		this.b.setTarget(255);

		this.alphaLine.setTarget(0);
		this.alphaArc.setTarget(0);
		this.alphaWedge.setTarget(0);
		this.alphaPattern.setTarget(0);
		this.alphaOther.setTarget(0);
		
		if ( parentOfSelected && ! this.getCollapse() )
		{
			var alpha =
			(
				1 -
				(selectedNode.getDepth() - this.getDepth()) /
				(Math.floor((compress ? compressedRadii[0] : nodeRadius) * gRadius / (historySpacingFactor * fontSize) - .5) + 1)
			);
			
			if ( alpha < 0 )
			{
				alpha = 0;
			}
			
			this.alphaLabel.setTarget(alpha);
			this.radial = false;
		}
		else
		{
			this.alphaLabel.setTarget(0);
		}
		
		this.hideAlonePrev = this.hideAlone;
		this.hidePrev = this.hide;
		
		if ( parentOfSelected )
		{
			this.hideAlone = false;
			this.hide = false;
		}
		
		if ( this.getParent() == selectedNode.getParent() )
		{
			this.hiddenEnd = null;
		}
		
		this.radialPrev = this.radial;
	}
	
	this.setTargetsSelected = function(hueMin, hueMax, lightness, hide, nextSiblingHidden)
	{
		var collapse = this.getCollapse();
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
		var canDisplayChildLabels = false;
		var lastChild;
		
		if ( this.hasChildren() )//&& ! hide )
		{
			lastChild = this.children[this.children.length - 1];
			this.hideAlone = true;
		}
		else
		{
			this.hideAlone = false;
		}
		
		// set child wedges
		//
		for ( var i = 0; i < this.children.length; i++ )
		{
			this.children[i].setTargetWedge();
			
			if
			(
				! this.children[i].hide &&
				( collapse || depth < maxDisplayDepth ) &&
				this.depth < maxAbsoluteDepth
			)
			{
				canDisplayChildLabels = true;
				this.hideAlone = false;
			}
		}
		
		if ( this == selectedNode || lastChild && lastChild.angleEnd.end < this.angleEnd.end - .01)
		{
			this.hideAlone = false;
		}
		
		if ( this.hideAlonePrev == undefined )
		{
			this.hideAlonePrev = this.hideAlone;
		}
		
		if ( this == selectedNode )
		{
			var otherArc = 
				angleFactor *
				(
					this.baseMagnitude + this.magnitude -
					lastChild.baseMagnitude - lastChild.magnitude
				);
			this.canDisplayLabelOther =
				otherArc *
				(this.children[0].radiusInner.end + 1) * gRadius >=
				minWidth();
			
			this.keyUnclassified = false;
			
			if ( this.canDisplayLabelOther )
			{
				this.angleOther = Math.PI * 2 - otherArc / 2;
			}
			else if ( otherArc > 0.0000000001 )
			{
				this.keyUnclassified = true;
				keys++;
			}
			
			this.angleStart.setTarget(0);
			this.angleEnd.setTarget(Math.PI * 2);
			this.radiusInner.setTarget(0);
			this.hidePrev = this.hide;
			this.hide = false;
			this.hideAlonePrev = this.hideAlone;
			this.hideAlone = false;
			this.keyed = false;
		}
		
		if ( hueMax - hueMin > 1 / 12 )
		{
			hueMax = hueMin + 1 / 12;
		}
		
		// set lightness
		//
		if ( ! ( hide || this.hideAlone ) )
		{
			if ( useHue() )
			{
				lightness = (lightnessBase + lightnessMax) / 2;
			}
			else
			{
				lightness = lightnessBase + (depth - 1) * lightnessFactor;
				
				if ( lightness > lightnessMax )
				{
					lightness = lightnessMax;
				}
			}
		}
		
		if ( hide )
		{
			this.hide = true;
		}
		
		if ( this.hidePrev == undefined )
		{
			this.hidePrev = this.hide;
		}
		
		var hiddenStart = -1;
		var hiddenHueNumer = 0;
		var hiddenHueDenom = 0;
		var i = 0;
		
		if ( ! this.hide )
		{
			this.hiddenEnd = null;
		}
		
		while ( true )
		{
			if ( ! this.hideAlone && ! hide && ( i == this.children.length || ! this.children[i].hide ) )
			{
				// reached a non-hidden child or the end; set targets for
				// previous group of hidden children (if any) using their
				// average hue
				
				if ( hiddenStart != -1 )
				{
					var hiddenHue = hiddenHueDenom ? hiddenHueNumer / hiddenHueDenom : hueMin;
					
					for ( var j = hiddenStart; j < i; j++ )
					{
						this.children[j].setTargetsSelected
						(
							hiddenHue,
							null,
							lightness,
							false,
							j < i - 1
						);
						
						this.children[j].hiddenEnd = null;
					}
					
					this.children[hiddenStart].hiddenEnd = i - 1;
				}
			}
			
			if ( i == this.children.length )
			{
				break;
			}
			
			var child = this.children[i];
			var childHueMin;
			var childHueMax;
			
			if ( this.magnitude > 0 && ! this.hide && ! this.hideAlone )
			{
				if ( useHue() )
				{
					childHueMin = child.hues[currentDataset];
				}
				else if ( this == selectedNode )
				{
					var min = 0.0;
					var max = 1.0;
					
					if ( this.children.length > 6 )
					{
						childHueMin = lerp((1 - Math.pow(1 - i / this.children.length, 1.4)) * .95, 0, 1, min, max);
						childHueMax = lerp((1 - Math.pow(1 - (i + .55) / this.children.length, 1.4)) * .95, 0, 1, min, max);
					}
					else
					{
						childHueMin = lerp(i / this.children.length, 0, 1, min, max);
						childHueMax = lerp((i + .55) / this.children.length, 0, 1, min, max);
					}
				}
				else
				{
					childHueMin = lerp
					(
						child.baseMagnitude,
						this.baseMagnitude, 
						this.baseMagnitude + this.magnitude,
						hueMin,
						hueMax
					);
					childHueMax = lerp
					(
						child.baseMagnitude + child.magnitude * .99,
						this.baseMagnitude,
						this.baseMagnitude + this.magnitude,
						hueMin,
						hueMax
					);
				}
			}
			else
			{
				childHueMin = hueMin;
				childHueMax = hueMax;
			}
			
			if ( ! this.hideAlone && ! hide && ! this.hide && child.hide )
			{
				if ( hiddenStart == -1 )
				{
					hiddenStart = i;
				}
				
				if ( useHue() )
				{
					hiddenHueNumer += childHueMin * child.magnitude;
					hiddenHueDenom += child.magnitude;
				}
				else
				{
					hiddenHueNumer += childHueMin;
					hiddenHueDenom++;
				}
			}
			else
			{
				hiddenStart = -1;
				
				this.children[i].setTargetsSelected
				(
					childHueMin,
					childHueMax,
					lightness,
					hide || this.keyed || this.hideAlone || this.hide && ! collapse,
					false
				);
			}
			
			i++;
		}
		
	 	if ( this.hue && this.magnitude )
	 	{
		 	this.hue.setTarget(this.hues[currentDataset]);
			
			if ( this.attributes[magnitudeIndex][lastDataset] == 0 )
			{
				this.hue.start = this.hue.end;
			}
		}
	 	
		this.radialPrev = this.radial;
		
		if ( this == selectedNode )
		{
			this.resetLabelWidth();
			this.labelWidth.setTarget(this.nameWidth * labelWidthFudge);
			this.alphaWedge.setTarget(0);
			this.alphaLabel.setTarget(1);
			this.alphaOther.setTarget(1);
			this.alphaArc.setTarget(0);
			this.alphaLine.setTarget(0);
			this.alphaPattern.setTarget(0);
			this.r.setTarget(255);
			this.g.setTarget(255);
			this.b.setTarget(255);
			this.radial = false;
			this.labelRadius.setTarget(0);
		}
		else
		{
			var rgb = hslToRgb
			(
				hueMin,
				saturation,
				lightness
			);
			
			this.r.setTarget(rgb.r);
			this.g.setTarget(rgb.g);
			this.b.setTarget(rgb.b);
			this.alphaOther.setTarget(0);
			
			this.alphaWedge.setTarget(1);
			
			if ( this.hide || this.hideAlone )
			{
				this.alphaPattern.setTarget(1);
			}
			else
			{
				this.alphaPattern.setTarget(0);
			}
			
			// set radial
			//
			if ( ! ( hide || this.hide ) )//&& ! this.keyed )
			{
				if ( this.hideAlone )
				{
					this.radial = true;
				}
				else if ( false && canDisplayChildLabels )
				{
					this.radial = false;
				}
				else
				{
					this.radial = true;
					
					if ( this.hasChildren() && depth < maxDisplayDepth )
					{
						var lastChild = this.children[this.children.length - 1];
						
						if
						(
							lastChild.angleEnd.end == this.angleEnd.end ||
							(
								(this.angleStart.end + this.angleEnd.end) / 2 -
								lastChild.angleEnd.end
							) * (this.radiusInner.end + 1) * gRadius * 2 <
							minWidth()
						)
						{
							this.radial = false;
						}
					}
				}
			}
			
			// set alphaLabel
			//
			if
			(
				collapse ||
				hide ||
				this.hide ||
				this.keyed ||
				depth > maxDisplayDepth ||
				! this.canDisplayDepth()
			)
			{
				this.alphaLabel.setTarget(0);
			}
			else
			{
				if
				(
					(this.radial || nLabelOffsets[depth - 2])
				)
				{
					this.alphaLabel.setTarget(1);
				}
				else
				{
					this.alphaLabel.setTarget(0);
					
					if ( this.radialPrev )
					{
						this.alphaLabel.start = 0;
					}
				}
			}
			
			// set alphaArc
			//
			if
			(
				collapse ||
				hide ||
				depth > maxDisplayDepth ||
				! this.canDisplayDepth()
			)
			{
				this.alphaArc.setTarget(0);
			}
			else
			{
				this.alphaArc.setTarget(1);
			}
			
			// set alphaLine
			//
			if
			(
				hide ||
				this.hide && nextSiblingHidden ||
				depth > maxDisplayDepth ||
				! this.canDisplayDepth()
			)
			{
				this.alphaLine.setTarget(0);
			}
			else
			{
				this.alphaLine.setTarget(1);
			}
			
			//if (  ! this.radial )
			{
				this.resetLabelWidth();
			}
			
			// set labelRadius target
			//
			if ( collapse )
			{
				this.labelRadius.setTarget(this.radiusInner.end);
			}
			else
			{
				if ( depth > maxDisplayDepth || ! this.canDisplayDepth() )
				{
					this.labelRadius.setTarget(1);
				}
				else
				{
					this.setTargetLabelRadius();
				}
			}
		}
	}
	
	this.setTargetWedge = function()
	{
		var depth = this.getDepth() - selectedNode.getDepth() + 1;
		
		// set angles
		//
		var baseMagnitudeRelative = this.baseMagnitude - selectedNode.baseMagnitude;
		//
		this.angleStart.setTarget(baseMagnitudeRelative * angleFactor);
		this.angleEnd.setTarget((baseMagnitudeRelative + this.magnitude) * angleFactor);
		
		// set radiusInner
		//
		if ( depth > maxDisplayDepth || ! this.canDisplayDepth() )
		{
			this.radiusInner.setTarget(1);
		}
		else
		{
			if ( compress )
			{
				this.radiusInner.setTarget(compressedRadii[depth - 2]);
			}
			else
			{
				this.radiusInner.setTarget(nodeRadius * (depth - 1));
			}
		}
		
		if ( this.hide != undefined )
		{
			this.hidePrev = this.hide;
		}
		
		if ( this.hideAlone != undefined )
		{
			this.hideAlonePrev = this.hideAlone;
		}
		
		// set hide
		//
		if
		(
			(this.angleEnd.end - this.angleStart.end) *
			(this.radiusInner.end * gRadius + gRadius) <
			minWidth()
		)
		{
			if ( depth == 2 && ! this.getCollapse() && this.depth <= maxAbsoluteDepth )
			{
				this.keyed = true;
				keys++;
				this.hide = false;
				
				var percentage = this.getPercentage();
				this.keyLabel = this.name + '   ' + percentage + '%';
				var dim = context.measureText(this.keyLabel);
				this.keyNameWidth = dim.width;
			}
			else
			{
				this.keyed = false;
				this.hide = depth > 2;
			}
		}
		else
		{
			this.keyed = false;
			this.hide = false;
		}
	}
	
	this.shortenLabel = function()
	{
		var label = this.name;
		
		var labelWidth = this.nameWidth;
		var maxWidth = this.labelWidth.current();
		var minEndLength = 0;
		
		if ( labelWidth > maxWidth && label.length > minEndLength * 2 )
		{
			var endLength =
				Math.floor((label.length - 1) * maxWidth / labelWidth / 2);
			
			if ( endLength < minEndLength )
			{
				endLength = minEndLength;
			}
			
			return (
				label.substring(0, endLength) +
				'...' +
				label.substring(label.length - endLength));
		}
		else
		{
			return label;
		}
	}
	
/*	this.shouldAddSearchResultsString = function()
	{
		if ( this.isSearchResult )
		{
			return this.searchResults > 1;
		}
		else
		{
			return this.searchResults > 0;
		}
	}
*/	
	this.sort = function()
	{
		this.children.sort(function(a, b){return b.getMagnitude() - a.getMagnitude()});
		
		for (var i = 0; i < this.children.length; i++)
		{
			this.children[i].sort();
		}
	}
}

var options;

function addOptionElement(position, innerHTML, title)
{
	var div = document.createElement("div");
//	div.style.position = 'absolute';
//	div.style.top = position + 'px';
	div.innerHTML = innerHTML;
//	div.style.display = 'block';
	div.style.padding = '2px';
	
	if ( title )
	{
		div.title = title;
	}
	
	options.appendChild(div);
	var height = 0;//div.clientHeight;
	return position + height;
}

function addOptionElements(hueName, hueDefault)
{
	options = document.createElement('div');
	options.style.position = 'absolute';
	options.style.top = '0px';
	options.addEventListener('mousedown', function(e) {mouseClick(e)}, false);
//	options.onmouseup = function(e) {mouseUp(e)}
	document.body.appendChild(options);
	
	document.body.style.font = '11px sans-serif';
	var position = 5;
	
	details = document.createElement('div');
	details.style.position = 'absolute';
	details.style.top = '1%';
	details.style.right = '2%';
	details.style.textAlign = 'right';
	document.body.insertBefore(details, canvas);
//		<div id="details" style="position:absolute;top:1%;right:2%;text-align:right;">

	details.innerHTML = '\
<span id="detailsName" style="font-weight:bold"></span>&nbsp;\
<input type="button" id="detailsExpand" onclick="expand(focusNode);"\
value="&harr;" title="Expand this wedge to become the new focus of the chart"/><br/>\
<div id="detailsInfo" style="float:right"></div>';

	keyControl = document.createElement('input');
	keyControl.type = 'button';
	keyControl.value = showKeys ? 'x' : '';
	keyControl.style.position = '';
	keyControl.style.position = 'fixed';
	keyControl.style.visibility = 'hidden';
	
	document.body.insertBefore(keyControl, canvas);
	
	var logoElement = document.getElementById('logo');
	
	if ( logoElement )
	{
		logoImage = logoElement.src;
	}
	else
	{
		logoImage = 'http://marbl.github.io/Krona/img/logo-med.png';
	}
	
//	document.getElementById('options').style.fontSize = '9pt';
	position = addOptionElement
	(
		position,
'<a style="margin:2px" target="_blank" href="https://github.com/marbl/Krona/wiki"><img style="vertical-align:middle;width:108px;height:30px;" src="' + logoImage + '"/></a><input type="button" id="back" value="&larr;" title="Go back (Shortcut: &larr;)"/>\
<input type="button" id="forward" value="&rarr;" title="Go forward (Shortcut: &rarr;)"/> \
&nbsp;Search: <input type="text" id="search"/>\
<input id="searchClear" type="button" value="x" onclick="clearSearch()"/> \
<span id="searchResults"></span>'
	);
	
	if ( datasets > 1 )
	{
		var size = datasets < datasetSelectSize ? datasets : datasetSelectSize;
		
		var select =
			'<table style="border-collapse:collapse;padding:0px"><tr><td style="padding:0px">' +
			'<select id="datasets" style="min-width:100px" size="' + size + '" onchange="onDatasetChange()">';
		
		for ( var i = 0; i < datasetNames.length; i++ )
		{
			select += '<option>' + datasetNames[i] + '</option>';
		}
		
		select +=
			'</select></td><td style="vertical-align:top;padding:1px;">' +
			'<input style="display:block" title="Previous dataset (Shortcut: &uarr;)" id="prevDataset" type="button" value="&uarr;" onclick="prevDataset()" disabled="true"/>' +
			'<input title="Next dataset (Shortcut: &darr;)" id="nextDataset" type="button" value="&darr;" onclick="nextDataset()"/><br/></td>' +
			'<td style="padding-top:1px;vertical-align:top"><input title="Switch to the last dataset that was viewed (Shortcut: TAB)" id="lastDataset" type="button" style="font:11px Times new roman" value="last" onclick="selectLastDataset()"/></td></tr></table>';
		
		position = addOptionElement(position + 5, select);
		
		datasetDropDown = document.getElementById('datasets');
		datasetButtonLast = document.getElementById('lastDataset');
		datasetButtonPrev = document.getElementById('prevDataset');
		datasetButtonNext = document.getElementById('nextDataset');
		
		position += datasetDropDown.clientHeight;
	}
	
	position = addOptionElement
	(
		position + 5,
'<input type="button" id="maxAbsoluteDepthDecrease" value="-"/>\
<span id="maxAbsoluteDepth"></span>\
&nbsp;<input type="button" id="maxAbsoluteDepthIncrease" value="+"/> Max depth',
'Maximum depth to display, counted from the top level \
and including collapsed wedges.'
	);
	
	position = addOptionElement
	(
		position,
'<input type="button" id="fontSizeDecrease" value="-"/>\
<span id="fontSize"></span>\
&nbsp;<input type="button" id="fontSizeIncrease" value="+"/> Font size'
	);
	
	position = addOptionElement
	(
		position,
'<input type="button" id="radiusDecrease" value="-"/>\
<input type="button" id="radiusIncrease" value="+"/> Chart size'
	);
	
	if ( hueName )
	{
		hueDisplayName = attributes[attributeIndex(hueName)].displayName;
		
		position = addOptionElement
		(
			position + 5,
			'<input type="checkbox" id="useHue" style="float:left" ' +
			'/><div>Color by<br/>' + hueDisplayName +
			'</div>'
		);
		
		useHueCheckBox = document.getElementById('useHue');
		useHueCheckBox.checked = hueDefault;
		useHueCheckBox.onclick = handleResize;
		useHueCheckBox.onmousedown = suppressEvent;
	}
	/*
	position = addOptionElement
	(
		position + 5,
		'&nbsp;<input type="checkbox" id="shorten" checked="checked" />Shorten labels</div>',
		'Prevent labels from overlapping by shortening them'
	);
	
	position = addOptionElement
	(
		position,
		'&nbsp;<input type="checkbox" id="compress" checked="checked" />Compress',
		'Compress wedges if needed to show the entire depth'
	);
	*/
	position = addOptionElement
	(
		position,
		'<input type="checkbox" id="collapse" checked="checked" />Collapse',
		'Collapse wedges that are redundant (entirely composed of another wedge)'
	);
	
	position = addOptionElement
	(
		position + 5,
		'<input type="button" id="snapshot" value="Snapshot"/>',
'Render the current view as SVG (Scalable Vector Graphics), a publication-\
quality format that can be printed and saved (see Help for browser compatibility)'
	);
	
	position = addOptionElement
	(
		position + 5,
'<input type="button" id="linkButton" value="Link"/>\
<input type="text" size="30" id="linkText"/>',
'Show a link to this view that can be copied for bookmarking or sharing'
	);
	
	position = addOptionElement
	(
		position + 5,
'<input type="button" id="help" value="?"\
onclick="window.open(\'https://github.com/marbl/Krona/wiki/Browsing%20Krona%20charts\', \'help\')"/>',
'Help'
	);
}

function arrow(angleStart, angleEnd, radiusInner)
{
	if ( context.globalAlpha == 0 )
	{
		return;
	}
	
	var angleCenter = (angleStart + angleEnd) / 2;
	var radiusArrowInner = radiusInner - gRadius / 10;//nodeRadius * gRadius;
	var radiusArrowOuter = gRadius * 1.1;//(1 + nodeRadius);
	var radiusArrowCenter = (radiusArrowInner + radiusArrowOuter) / 2;
	var pointLength = (radiusArrowOuter - radiusArrowInner) / 5;
	
	context.fillStyle = highlightFill;
	context.lineWidth = highlightLineWidth;
	
	// First, mask out the first half of the arrow.  This will prevent the tips
	// from superimposing if the arrow goes most of the way around the circle.
	// Masking is done by setting the clipping region to the inverse of the
	// half-arrow, which is defined by cutting the half-arrow out of a large
	// rectangle
	//
	context.beginPath();
	context.arc(0, 0, radiusInner, angleCenter, angleEnd, false);
	context.lineTo
	(
		radiusArrowInner * Math.cos(angleEnd),
		radiusArrowInner * Math.sin(angleEnd)
	);
	context.lineTo
	(
		radiusArrowCenter * Math.cos(angleEnd) - pointLength * Math.sin(angleEnd),
		radiusArrowCenter * Math.sin(angleEnd) + pointLength * Math.cos(angleEnd)
	);
	context.lineTo
	(
		radiusArrowOuter * Math.cos(angleEnd),
		radiusArrowOuter * Math.sin(angleEnd)
	);
	context.arc(0, 0, gRadius, angleEnd, angleCenter, true);
	context.closePath();
	context.moveTo(-imageWidth, -imageHeight);
	context.lineTo(imageWidth, -imageHeight);
	context.lineTo(imageWidth, imageHeight);
	context.lineTo(-imageWidth, imageHeight);
	context.closePath();
	context.save();
	context.clip();
	
	// Next, draw the other half-arrow with the first half masked out
	//
	context.beginPath();
	context.arc(0, 0, radiusInner, angleCenter, angleStart, true);
	context.lineTo
	(
		radiusArrowInner * Math.cos(angleStart),
		radiusArrowInner * Math.sin(angleStart)
	);
	context.lineTo
	(
		radiusArrowCenter * Math.cos(angleStart) + pointLength * Math.sin(angleStart),
		radiusArrowCenter * Math.sin(angleStart) - pointLength * Math.cos(angleStart)
	);
	context.lineTo
	(
		radiusArrowOuter * Math.cos(angleStart),
		radiusArrowOuter * Math.sin(angleStart)
	);
	context.arc(0, 0, gRadius, angleStart, angleCenter, false);
	context.fill();
	context.stroke();
	
	// Finally, remove the clipping region and draw the first half-arrow.  This
	// half is extended slightly to fill the seam.
	//
	context.restore();
	context.beginPath();
	context.arc(0, 0, radiusInner, angleCenter - 2 / (2 * Math.PI * radiusInner), angleEnd, false);
	context.lineTo
	(
		radiusArrowInner * Math.cos(angleEnd),
		radiusArrowInner * Math.sin(angleEnd)
	);
	context.lineTo
	(
		radiusArrowCenter * Math.cos(angleEnd) - pointLength * Math.sin(angleEnd),
		radiusArrowCenter * Math.sin(angleEnd) + pointLength * Math.cos(angleEnd)
	);
	context.lineTo
	(
		radiusArrowOuter * Math.cos(angleEnd),
		radiusArrowOuter * Math.sin(angleEnd)
	);
	context.arc(0, 0, gRadius, angleEnd, angleCenter - 2 / (2 * Math.PI * gRadius), true);
	context.fill();
	context.stroke();
}

function attributeIndex(aname)
{
	for ( var i = 0 ; i < attributes.length; i++ )
	{
		if ( aname == attributes[i].name )
		{
			return i;
		}
	}
	
	return null;
}

function checkHighlight()
{
	var lastHighlightedNode = highlightedNode;
	var lastHighlightingHidden = highlightingHidden;
	
	highlightedNode = selectedNode;
	resetKeyOffset();
	
	if ( progress == 1 )
	{
		selectedNode.checkHighlight();
		if ( selectedNode.getParent() )
		{
			selectedNode.getParent().checkHighlightCenter();
		}
		
		focusNode.checkHighlightMap();
	}
	
	if ( highlightedNode != selectedNode )
	{
		if ( highlightedNode == focusNode )
		{
//			canvas.style.display='none';
//			window.resizeBy(1,0);
//			canvas.style.cursor='ew-resize';
//			window.resizeBy(-1,0);
//			canvas.style.display='inline';
		}
		else
		{
//			canvas.style.cursor='pointer';
		}
	}
	else
	{
//		canvas.style.cursor='auto';
	}
	
	if
	(
		(
			true ||
			highlightedNode != lastHighlightedNode ||
			highlightingHidden != highlightingHiddenLast
		) &&
		progress == 1
	)
	{
		draw(); // TODO: handle in update()
	}
}

function checkSelectedCollapse()
{
	var newNode = selectedNode;
	
	while ( newNode.getCollapse() )
	{
		newNode = newNode.children[0];
	}
	
	if ( newNode.children.length == 0 )
	{
		newNode = newNode.getParent();
	}
	
	if ( newNode != selectedNode )
	{
		selectNode(newNode);
	}
}

function clearSearch()
{
	if ( search.value != '' )
	{
		search.value = '';
		onSearchChange();
	}
}

function createSVG()
{
	svgNS = "http://www.w3.org/2000/svg";
	var SVG = {};
	SVG.xlinkns = "http://www.w3.org/1999/xlink";
	
	var newSVG = document.createElementNS(svgNS, "svg:svg");
	
	newSVG.setAttribute("id", "canvas");
	// How big is the canvas in pixels
	newSVG.setAttribute("width", '100%');
	newSVG.setAttribute("height", '100%');
	// Set the coordinates used by drawings in the canvas
//	newSVG.setAttribute("viewBox", "0 0 " + imageWidth + " " + imageHeight);
	// Define the XLink namespace that SVG uses
	newSVG.setAttributeNS
	(
		"http://www.w3.org/2000/xmlns/",
		"xmlns:xlink",
		SVG.xlinkns
	);
	
	return newSVG;
}

function degrees(radians)
{
	return radians * 180 / Math.PI;
}

function draw()
{
	tweenFrames++;
	//resize();
//	context.fillRect(0, 0, imageWidth, imageHeight);
	context.clearRect(0, 0, imageWidth, imageHeight);
	
	context.font = fontNormal;
	context.textBaseline = 'middle';
	
	//context.strokeStyle = 'rgba(0, 0, 0, 0.3)';
	context.translate(centerX, centerY);
	
	resetKeyOffset();
	
	head.draw(false, false); // draw pie slices
	head.draw(true, false); // draw labels
	
	var pathRoot = selectedNode;
	
	if ( focusNode != 0 && focusNode != selectedNode )
	{
		context.globalAlpha = 1;
		focusNode.drawHighlight(true);
		pathRoot = focusNode;
	}
	
	if
	(
		highlightedNode &&
		highlightedNode.getDepth() >= selectedNode.getDepth() &&
		highlightedNode != focusNode
	)
	{
		if
		(
			progress == 1 &&
			highlightedNode != selectedNode &&
			(
				highlightedNode != focusNode ||
				focusNode.children.length > 0
			)
		)
		{
			context.globalAlpha = 1;
			highlightedNode.drawHighlight(true);
		}
		
		//pathRoot = highlightedNode;
	}
	else if
	(
		progress == 1 &&
		highlightedNode.getDepth() < selectedNode.getDepth()
	)
	{
		context.globalAlpha = 1;
		highlightedNode.drawHighlightCenter();
	}
	
	if ( quickLook && false) // TEMP
	{
		context.globalAlpha = 1 - progress / 2;
		selectedNode.drawHighlight(true);
	}
	else if ( progress < 1 )//&& zoomOut() )
	{
		if ( !zoomOut)//() )
		{
			context.globalAlpha = selectedNode.alphaLine.current();
			selectedNode.drawHighlight(true);
		}
		else if ( selectedNodeLast )
		{
			context.globalAlpha = 1 - 4 * Math.pow(progress - .5, 2);
			selectedNodeLast.drawHighlight(false);
		}
	}
	
	drawDatasetName();
	
	//drawHistory();
	
	context.translate(-centerX, -centerY);
	context.globalAlpha = 1;
	
	mapRadius =
		(imageHeight / 2 - details.clientHeight - details.offsetTop) /
		(pathRoot.getDepth() - 1) * 3 / 4 / 2;
	
	if ( mapRadius > maxMapRadius )
	{
		mapRadius = maxMapRadius;
	}
	
	mapBuffer = mapRadius / 2;
	
	//context.font = fontNormal;
	pathRoot.drawMap(pathRoot);
	
	if ( hueDisplayName && useHue() )
	{
		drawLegend();
	}
}

function drawBubble(angle, radius, width, radial, flip)
{
	var height = fontSize * 2;
	var x;
	var y;
	
	width = width + fontSize;
	
	if ( radial )
	{
		y = -fontSize;
		
		if ( flip )
		{
			x = radius - width + fontSize / 2;
		}
		else
		{
			x = radius - fontSize / 2;
		}
	}
	else
	{
		x = -width / 2;
		y = -radius - fontSize;
	}
	
	if ( snapshotMode )
	{
		drawBubbleSVG(x + centerX, y + centerY, width, height, fontSize, angle);
	}
	else
	{
		drawBubbleCanvas(x, y, width, height, fontSize, angle);
	}
}

function drawBubbleCanvas(x, y, width, height, radius, rotation)
{
	context.strokeStyle = 'black';
	context.lineWidth = highlightLineWidth;
	context.fillStyle = 'rgba(255, 255, 255, .75)';
	context.rotate(rotation);
	roundedRectangle(x, y, width, fontSize * 2, fontSize);
	context.fill();
	context.stroke();
	context.rotate(-rotation);
}

function drawBubbleSVG(x, y, width, height, radius, rotation)
{
	svg +=
		'<rect x="' + x + '" y="' + y +
		'" width="' + width +
		'" height="' + height +
		'" rx="' + radius +
		'" ry="' + radius +
		'" fill="rgba(255, 255, 255, .75)' +
		'" class="highlight" ' +
		'transform="rotate(' +
		degrees(rotation) + ',' + centerX + ',' + centerY +
		')"/>';
}

function drawDatasetName()
{
	var alpha = datasetAlpha.current();
	
	if ( alpha > 0 )
	{
		var radius = gRadius * compressedRadii[0] / -2;
		
		if ( alpha > 1 )
		{
			alpha = 1;
		}
		
		context.globalAlpha = alpha;
		
		drawBubble(0, -radius, datasetWidths[currentDataset], false, false);
		drawText(datasetNames[currentDataset], 0, radius, 0, 'center', true);
	}
}

function drawHistory()
{
	var alpha = 1;
	context.textAlign = 'center';
	
	for ( var i = 0; i < nodeHistoryPosition && alpha > 0; i++ )
	{
		
		context.globalAlpha = alpha - historyAlphaDelta * tweenFactor;
		context.fillText
		(
			nodeHistory[nodeHistoryPosition - i - 1].name,
			0,
			(i + tweenFactor) * historySpacingFactor * fontSize - 1
		);
		
		if ( alpha > 0 )
		{
			alpha -= historyAlphaDelta;
		}
	}
	
	context.globalAlpha = 1;
}

function drawLegend()
{
	var left = imageWidth * .01;
	var width = imageHeight * .0265;
	var height = imageHeight * .15;
	var top = imageHeight - fontSize * 3.5 - height;
	var textLeft = left + width + fontSize / 2;
	
	context.fillStyle = 'black';
	context.textAlign = 'start';
	context.font = fontNormal;
//	context.fillText(valueStartText, textLeft, top + height);
//	context.fillText(valueEndText, textLeft, top);
	context.fillText(hueDisplayName, left, imageHeight - fontSize * 1.5);
	
	var gradient = context.createLinearGradient(0, top + height, 0, top);
	
	for ( var i = 0; i < hueStopPositions.length; i++ )
	{
		gradient.addColorStop(hueStopPositions[i], hueStopHsl[i]);
		
		var textY = top + (1 - hueStopPositions[i]) * height;
		
		if
		(
			i == 0 ||
			i == hueStopPositions.length - 1 ||
			textY > top + fontSize && textY < top + height - fontSize
		)
		{
			context.fillText(hueStopText[i], textLeft, textY);
		}
	}
	
	context.fillStyle = gradient;
	context.fillRect(left, top, width, height);
	context.lineWidth = thinLineWidth;
	context.strokeRect(left, top, width, height);
}

function drawLegendSVG()
{
	var left = imageWidth * .01;
	var width = imageHeight * .0265;
	var height = imageHeight * .15;
	var top = imageHeight - fontSize * 3.5 - height;
	var textLeft = left + width + fontSize / 2;

	var text = '';
	
	text += svgText(hueDisplayName, left, imageHeight - fontSize * 1.5);
	
	var svgtest = '<linearGradient id="gradient" x1="0%" y1="100%" x2="0%" y2="0%">';
	
	for ( var i = 0; i < hueStopPositions.length; i++ )
	{
		svgtest +=
			'<stop offset="' + round(hueStopPositions[i] * 100) +
			'%" style="stop-color:' + hueStopHsl[i] + '"/>';
		
		var textY = top + (1 - hueStopPositions[i]) * height;
		
		if
		(
			i == 0 ||
			i == hueStopPositions.length - 1 ||
			textY > top + fontSize && textY < top + height - fontSize
		)
		{
			text += svgText(hueStopText[i], textLeft, textY);
		}
	}
	
	svgtest += '</linearGradient>';
	//alert(svgtest);
	svg += svgtest;
	svg +=
		'<rect style="fill:url(#gradient)" x="' + left + '" y="' + top +
		'" width="' + width + '" height="' + height + '"/>';
	
	svg += text;
}

function drawSearchHighlights(label, bubbleX, bubbleY, rotation, center)
{
	var index = -1;
	var labelLength = label.length;
	
	bubbleX -= fontSize / 4;
	
	do
	{
		index = label.toLowerCase().indexOf(search.value.toLowerCase(), index + 1);
		
		if ( index != -1 && index < labelLength )
		{
			var dim = context.measureText(label.substr(0, index));
			var x = bubbleX + dim.width;
			
			dim = context.measureText(label.substr(index, search.value.length));
			
			var y = bubbleY - fontSize * 3 / 4;
			var width = dim.width + fontSize / 2;
			var height = fontSize * 3 / 2;
			var radius = fontSize / 2;
			
			if ( snapshotMode )
			{
				if ( center )
				{
					x += centerX;
					y += centerY;
				}
				
				svg +=
					'<rect x="' + x + '" y="' + y +
					'" width="' + width +
					'" height="' + height +
					'" rx="' + radius +
					'" ry="' + radius +
					'" class="searchHighlight' +
					'" transform="rotate(' +
					degrees(rotation) + ',' + centerX + ',' + centerY +
					')"/>';
			}
			else
			{
				context.fillStyle = 'rgb(255, 255, 100)';
				context.rotate(rotation);
				roundedRectangle(x, y, width, height, radius);
				context.fill();
				context.rotate(-rotation);
			}
		}
	}
	while ( index != -1 && index < labelLength );
}

function drawText(text, x, y, angle, anchor, bold, color)
{
	if ( color == undefined )
	{
		color = 'black';
	}
	
	if ( snapshotMode )
	{
		svg +=
			'<text x="' + (centerX + x) + '" y="' + (centerY + y) +
			'" text-anchor="' + anchor + '" style="font-color:' + color + ';font-weight:' + (bold ? 'bold' : 'normal') +
			'" transform="rotate(' + degrees(angle) + ',' + centerX + ',' + centerY + ')">' +
			text + '</text>';
	}
	else
	{
		context.fillStyle = color;
		context.textAlign = anchor;
		context.font = bold ? fontBold : fontNormal;
		context.rotate(angle);
		context.fillText(text, x, y);
		context.rotate(-angle);
	}
}

function drawTextPolar
(
	text,
	innerText,
	angle,
	radius,
	radial,
	bubble,
	bold, 
	searchResult,
	searchResults
)
{
	var anchor;
	var textX;
	var textY;
	var spacer;
	var totalText = text;
	var flip;
	
	if ( snapshotMode )
	{
		spacer = '&#160;&#160;&#160;';
	}
	else
	{
		spacer = '   ';
	}
	
	if ( radial )
	{
		flip = angle < 3 * Math.PI / 2;
		
		if ( flip )
		{
			angle -= Math.PI;
			radius = -radius;
			anchor = 'end';
			
			if ( innerText )
			{
				totalText = text + spacer + innerText;
			}
		}
		else
		{
			anchor = 'start';
			
			if ( innerText )
			{
				totalText = innerText + spacer + text;
			}
		}
		
		textX = radius;
		textY = 0;
	}
	else
	{
		flip = angle < Math.PI || angle > 2 * Math.PI;
		var label;
		
		anchor = snapshotMode ? 'middle' : 'center';
		
		if ( flip )
		{
			angle -= Math.PI;
			radius = -radius;
		}
		
		angle += Math.PI / 2;
		textX = 0;
		textY = -radius;
	}
	
	if ( bubble )
	{
		var textActual = totalText;
		
		if ( innerText && snapshotMode )
		{
			if ( flip )
			{
				textActual = text + '   ' + innerText;
			}
			else
			{
				textActual = innerText + '   ' + text;
			}
		}
		
		if ( searchResults )
		{
			textActual = textActual + searchResultString(searchResults);
		}
		
		var textWidth = measureText(textActual, bold);
		
		var x = textX;
		
		if ( anchor == 'end' )
		{
			x -= textWidth;
		}
		else if ( anchor != 'start' )
		{
			// centered
			x -= textWidth / 2;
		}
		
		drawBubble(angle, radius, textWidth, radial, flip);
		
		if ( searchResult )
		{
			drawSearchHighlights
			(
				textActual,
				x,
				textY,
				angle,
				true
			)
		}
	}
	
	if ( searchResults )
	{
		totalText = totalText + searchResultString(searchResults);
	}
	
	drawText(totalText, textX, textY, angle, anchor, bold);
	
	return flip;
}

function drawTick(start, length, angle)
{
	if ( snapshotMode )
	{
		svg +=
			'<line x1="' + (centerX + start) +
			'" y1="' + centerY +
			'" x2="' + (centerX + start + length) +
			'" y2="' + centerY +
			'" class="tick" transform="rotate(' +
			degrees(angle) + ',' + centerX + ',' + centerY +
			')"/>';
	}
	else
	{
		context.rotate(angle);
		context.beginPath();
		context.moveTo(start, 0);
		context.lineTo(start + length, 0);
		context.lineWidth = thinLineWidth * 2;
		context.stroke();
		context.rotate(-angle);
	}
}

function drawWedge
(
	angleStart,
	angleEnd,
	radiusInner,
	radiusOuter,
	color,
	patternAlpha,
	highlight
)
{
	if ( context.globalAlpha == 0 )
	{
		return;
	}
	
	if ( snapshotMode )
	{
		if ( angleEnd == angleStart + Math.PI * 2 )
		{
			// fudge to prevent overlap, which causes arc ambiguity
			//
			angleEnd -= .1 / gRadius;
		}
		
		var longArc = angleEnd - angleStart > Math.PI ? 1 : 0;
		
		var x1 = centerX + radiusInner * Math.cos(angleStart);
		var y1 = centerY + radiusInner * Math.sin(angleStart);
		
		var x2 = centerX + gRadius * Math.cos(angleStart);
		var y2 = centerY + gRadius * Math.sin(angleStart);
		
		var x3 = centerX + gRadius * Math.cos(angleEnd);
		var y3 = centerY + gRadius * Math.sin(angleEnd);
		
		var x4 = centerX + radiusInner * Math.cos(angleEnd);
		var y4 = centerY + radiusInner * Math.sin(angleEnd);
		
		var dArray =
		[
			" M ", x1, ",", y1,
			" L ", x2, ",", y2,
			" A ", gRadius, ",", gRadius, " 0 ", longArc, ",1 ", x3, ",", y3,
			" L ", x4, ",", y4,
			" A ", radiusInner, ",", radiusInner, " 0 ", longArc, " 0 ", x1, ",", y1,
			" Z "
		];
		
		svg +=
			'<path class="'+ (highlight ? 'highlight' : 'wedge') + '" fill="' + color +
			'" d="' + dArray.join('') + '"/>';
		
		if ( patternAlpha > 0 )
		{
			svg +=
				'<path class="wedge" fill="url(#hiddenPattern)" d="' +
				dArray.join('') + '"/>';
		}
	}
	else
	{
		// fudge to prevent seams during animation
		//
		angleEnd += 1 / gRadius;
		
		context.fillStyle = color;
		context.beginPath();
		context.arc(0, 0, radiusInner, angleStart, angleEnd, false);
		context.arc(0, 0, radiusOuter, angleEnd, angleStart, true);
		context.closePath();
		context.fill();
		
		if ( patternAlpha > 0 )
		{
			context.save();
			context.clip();
			context.globalAlpha = patternAlpha;
			context.fillStyle = hiddenPattern;
			context.fill();
			context.restore();
		}
		
		if ( highlight )
		{
			context.lineWidth = highlight ? highlightLineWidth : thinLineWidth;
			context.strokeStyle = 'black';
			context.stroke();
		}
	}
}

function expand(node)
{
	selectNode(node);
	updateView();
}

function focusLost()
{
	mouseX = -1;
	mouseY = -1;
	checkHighlight();
	document.body.style.cursor = 'auto';
}

function fontSizeDecrease()
{
	if ( fontSize > 1 )
	{
		fontSize--;
		updateViewNeeded = true;
	}
}

function fontSizeIncrease()
{
	fontSize++;
	updateViewNeeded = true;
}

function getGetString(name, value, bool)
{
	return name + '=' + (bool ? value ? 'true' : 'false' : value);
}

function hideLink()
{
	hide(linkText);
	show(linkButton);
}

function show(object)
{
	object.style.display = 'inline';
}

function hide(object)
{
	object.style.display = 'none';
}

function showLink()
{
	var urlHalves = String(document.location).split('?');
	var newGetVariables = new Array();
	
	newGetVariables.push
	(
		getGetString('dataset', currentDataset, false),
		getGetString('node', selectedNode.id, false),
		getGetString('collapse', collapse, true),
		getGetString('color', useHue(), true),
		getGetString('depth', maxAbsoluteDepth - 1, false),
		getGetString('font', fontSize, false),
		getGetString('key', showKeys, true)
	);
	
	hide(linkButton);
	show(linkText);
	linkText.value = urlHalves[0] + '?' + getVariables.concat(newGetVariables).join('&');
	//linkText.disabled = false;
	linkText.focus();
	linkText.select();
	//linkText.disabled = true;
//	document.location = urlHalves[0] + '?' + getVariables.join('&');
}

function getFirstChild(element)
{
	element = element.firstChild;
	
	if ( element && element.nodeType != 1 )
	{
		element = getNextSibling(element);
	}
	
	return element;
}

function getNextSibling(element)
{
	do
	{
		element = element.nextSibling;
	}
	while ( element && element.nodeType != 1 );
	
	return element;
}

function getPercentage(fraction)
{
	return round(fraction * 100);
}

function hslText(hue)
{
	if ( 1 || snapshotMode )
	{
		// Safari doesn't seem to allow hsl() in SVG
		
		var rgb = hslToRgb(hue, saturation, (lightnessBase + lightnessMax) / 2);
		
		return rgbText(rgb.r, rgb.g, rgb.b);
	}
	else
	{
		var hslArray =
		[
			'hsl(',
			Math.floor(hue * 360),
			',',
			Math.floor(saturation * 100),
			'%,',
			Math.floor((lightnessBase + lightnessMax) * 50),
			'%)'
		];
		
		return hslArray.join('');
	}
}

function hslToRgb(h, s, l)
{
	var m1, m2;
	var r, g, b;
	
	if (s == 0)
	{
		r = g = b = Math.floor((l * 255));
	}
	else
	{
		if (l <= 0.5)
		{
			m2 = l * (s + 1);
		}
		else
		{
			m2 = l + s - l * s;
		}
		
		m1 = l * 2 - m2;
		
		r = Math.floor(hueToRgb(m1, m2, h + 1 / 3));
		g = Math.floor(hueToRgb(m1, m2, h));
		b = Math.floor(hueToRgb(m1, m2, h - 1/3));
	}
	
	return {r: r, g: g, b: b};
}

function hueToRgb(m1, m2, hue)
{
	var v;
	
	while (hue < 0)
	{
		hue += 1;
	}
	
	while (hue > 1)
	{
		hue -= 1;
	}
	
	if (6 * hue < 1)
		v = m1 + (m2 - m1) * hue * 6;
	else if (2 * hue < 1)
		v = m2;
	else if (3 * hue < 2)
		v = m1 + (m2 - m1) * (2/3 - hue) * 6;
	else
		v = m1;

	return 255 * v;
}

function interpolateHue(hueStart, hueEnd, valueStart, valueEnd)
{
	// since the gradient will be RGB based, we need to add stops to hit all the
	// colors in the hue spectrum
	
	hueStopPositions = new Array();
	hueStopHsl = new Array();
	hueStopText = new Array();
	
	hueStopPositions.push(0);
	hueStopHsl.push(hslText(hueStart));
	hueStopText.push(round(valueStart));
	
	for
	(
		var i = (hueStart > hueEnd ? 5 / 6 : 1 / 6);
		(hueStart > hueEnd ? i > 0 : i < 1);
		i += (hueStart > hueEnd ? -1 : 1) / 6
	)
	{
		if
		(
			hueStart > hueEnd ?
				i > hueEnd && i < hueStart :
				i > hueStart && i < hueEnd
		)
		{
			hueStopPositions.push(lerp(i, hueStart, hueEnd, 0, 1));
			hueStopHsl.push(hslText(i));
			hueStopText.push(round(lerp
			(
				i,
				hueStart,
				hueEnd,
				valueStart,
				valueEnd
			)));
		}
	}
	
	hueStopPositions.push(1);
	hueStopHsl.push(hslText(hueEnd));
	hueStopText.push(round(valueEnd));
}

function keyLineAngle(angle, keyAngle, bendRadius, keyX, keyY, pointsX, pointsY)
{
	if ( angle < Math.PI / 2 && keyY < bendRadius * Math.sin(angle) 
	|| angle > Math.PI / 2 && keyY < bendRadius)
	{
		return Math.asin(keyY / bendRadius);
	}
	else
	{
		// find the angle of the normal to a tangent line that goes to
		// the label
		
		var textDist = Math.sqrt
		(
			Math.pow(keyX, 2) +
			Math.pow(keyY, 2)
		);
		
		var tanAngle = Math.acos(bendRadius / textDist) + keyAngle;
		
		if ( angle < tanAngle || angle < Math.PI / 2 )//|| labelLeft < centerX )
		{
			// angle doesn't reach far enough for tangent; collapse and
			// connect directly to label
			
			if ( keyY / Math.tan(angle) > 0 )
			{
				pointsX.push(keyY / Math.tan(angle));
				pointsY.push(keyY);
			}
			else
			{
				pointsX.push(bendRadius * Math.cos(angle));
				pointsY.push(bendRadius * Math.sin(angle));
			}
			
			return angle;
		}
		else
		{
			return tanAngle;
		}
	}
}

function keyOffset()
{
	return imageHeight - (keys - currentKey + 1) * (keySize + keyBuffer) + keyBuffer - margin;
}

function lerp(value, fromStart, fromEnd, toStart, toEnd)
{
	return (value - fromStart) *
		(toEnd - toStart) /
		(fromEnd - fromStart) +
		toStart;
}

function createCanvas()
{
	canvas = document.createElement('canvas');
	document.body.appendChild(canvas);
	context = canvas.getContext('2d');
}

function load()
{
	document.body.style.overflow = "hidden";
	document.body.style.margin = 0;
	
	createCanvas();
	
	if ( context == undefined )
	{
		document.body.innerHTML = '\
<br/>This browser does not support HTML5 (see \
<a href="https://github.com/marbl/Krona/wiki/Browser%20support">Browser support</a>).\
	';
		return;
	}

	if ( typeof context.fillText != 'function' )
	{
		document.body.innerHTML = '\
<br/>This browser does not support HTML5 canvas text (see \
<a href="https://github.com/marbl/Krona/wiki/Browser%20support">Browser support</a>).\
	';
		return;
	}
	
	resize();
	
	var kronaElement = document.getElementsByTagName('krona')[0];
	
	var magnitudeName;
	var hueName;
	var hueDefault;
	var hueStart;
	var hueEnd;
	var valueStart;
	var valueEnd;
	
	if ( kronaElement.getAttribute('collapse') != undefined )
	{
		collapse = kronaElement.getAttribute('collapse') == 'true';
	}
	
	if ( kronaElement.getAttribute('key') != undefined )
	{
		showKeys = kronaElement.getAttribute('key') == 'true';
	}
	
	for
	(
		var element = getFirstChild(kronaElement);
		element;
		element = getNextSibling(element)
	)
	{
		switch ( element.tagName.toLowerCase() )
		{
			case 'attributes':
				magnitudeName = element.getAttribute('magnitude');
				//
				for
				(
					var attributeElement = getFirstChild(element);
					attributeElement;
					attributeElement = getNextSibling(attributeElement)
				)
				{
					var tag = attributeElement.tagName.toLowerCase();
					
					if ( tag == 'attribute' )
					{
						var attribute = new Attribute();
						attribute.name = attributeElement.firstChild.nodeValue.toLowerCase();
						attribute.displayName = attributeElement.getAttribute('display');
						
						if ( attributeElement.getAttribute('hrefBase') )
						{
							attribute.hrefBase = attributeElement.getAttribute('hrefBase');
						}
						
						if ( attributeElement.getAttribute('target') )
						{
							attribute.target = attributeElement.getAttribute('target');
						}
						
						if ( attribute.name == magnitudeName )
						{
							magnitudeIndex = attributes.length;
						}
						
						if ( attributeElement.getAttribute('listAll') )
						{
							attribute.listAll = attributeElement.getAttribute('listAll').toLowerCase();
						}
						else if ( attributeElement.getAttribute('listNode') )
						{
							attribute.listNode = attributeElement.getAttribute('listNode').toLowerCase();
						}
						else if ( attributeElement.getAttribute('dataAll') )
						{
							attribute.dataAll = attributeElement.getAttribute('dataAll').toLowerCase();
						}
						else if ( attributeElement.getAttribute('dataNode') )
						{
							attribute.dataNode = attributeElement.getAttribute('dataNode').toLowerCase();
						}
						
						if ( attributeElement.getAttribute('postUrl') )
						{
							attribute.postUrl = attributeElement.getAttribute('postUrl');
						}
						
						if ( attributeElement.getAttribute('postVar') )
						{
							attribute.postVar = attributeElement.getAttribute('postVar');
						}
						
						if ( attributeElement.getAttribute('mono') )
						{
							attribute.mono = true;
						}
						
						attributes.push(attribute);
					}
					else if ( tag == 'list' )
					{
						var attribute = new Attribute();
						
						attribute.name = attributeElement.firstChild.nodeValue;
						attribute.list = true;
						attributes.push(attribute);
					}
					else if ( tag == 'data' )
					{
						var attribute = new Attribute();
						
						attribute.name = attributeElement.firstChild.nodeValue;
						attribute.data = true;
						attributes.push(attribute);
						
						var enableScript = document.createElement('script');
						var date = new Date();
						enableScript.src =
							attributeElement.getAttribute('enable') + '?' +
							date.getTime();
						document.body.appendChild(enableScript);
					}
				}
				break;
			
			case 'color':
				hueName = element.getAttribute('attribute');
				hueStart = Number(element.getAttribute('hueStart')) / 360;
				hueEnd = Number(element.getAttribute('hueEnd')) / 360;
				valueStart = Number(element.getAttribute('valueStart'));
				valueEnd = Number(element.getAttribute('valueEnd'));
				//
				interpolateHue(hueStart, hueEnd, valueStart, valueEnd);
				//
				if ( element.getAttribute('default') == 'true' )
				{
					hueDefault = true;
				}
				break;
			
			case 'datasets':
				datasetNames = new Array();
				//
				for ( j = getFirstChild(element); j; j = getNextSibling(j) )
				{
					datasetNames.push(j.firstChild.nodeValue);
				}
				datasets = datasetNames.length;
				break;
			
			case 'node':
				head = loadTreeDOM
				(
					element,
					magnitudeName,
					hueName,
					hueStart,
					hueEnd,
					valueStart,
					valueEnd
				);
				break;
		}
	}
	
	// get GET options
	//
	var urlHalves = String(document.location).split('?');
	var datasetDefault = 0;
	var maxDepthDefault;
	var nodeDefault = 0;
	//
	if ( urlHalves[1] )
	{
		var vars = urlHalves[1].split('&');
		
		for ( i = 0; i < vars.length; i++ )
		{
			var pair = vars[i].split('=');
			
			switch ( pair[0] )
			{
				case 'collapse':
					collapse = pair[1] == 'true';
					break;
				
				case 'color':
					hueDefault = pair[1] == 'true';
					break;
				
				case 'dataset':
					datasetDefault = Number(pair[1]);
					break;
					
				case 'depth':
					maxDepthDefault = Number(pair[1]) + 1;
					break;
				
				case 'key':
					showKeys = pair[1] == 'true';
					break;
				
				case 'font':
					fontSize = Number(pair[1]);
					break;
				
				case 'node':
					nodeDefault = Number(pair[1]);
					break;
				
				default:
					getVariables.push(pair[0] + '=' + pair[1]);
					break;
			}
		}
	}
	
	addOptionElements(hueName, hueDefault);
	setCallBacks();
	
	head.sort();
	maxAbsoluteDepth = 0;
	selectDataset(datasetDefault);
	
	if ( maxDepthDefault && maxDepthDefault < head.maxDepth )
	{
		maxAbsoluteDepth = maxDepthDefault;
	}
	else
	{
		maxAbsoluteDepth = head.maxDepth;
	}
	
	selectNode(nodes[nodeDefault]);
	
	setInterval(update, 20);
	
	window.onresize = handleResize;
	updateMaxAbsoluteDepth();
	updateViewNeeded = true;
}

function loadTreeDOM
(
	domNode,
	magnitudeName,
	hueName,
	hueStart,
	hueEnd,
	valueStart,
	valueEnd
)
{
	var newNode = new Node();
	
	newNode.name = domNode.getAttribute('name');
	
	if ( domNode.getAttribute('href') )
	{
		newNode.href = domNode.getAttribute('href');
	}
	
	if ( hueName )
	{
		newNode.hues = new Array();
	}
	
	for ( var i = getFirstChild(domNode); i; i = getNextSibling(i) )
	{
		switch ( i.tagName.toLowerCase() )
		{
		case 'node': 
			var newChild = loadTreeDOM
			(
				i,
				magnitudeName,
				hueName,
				hueStart,
				hueEnd,
				valueStart,
				valueEnd
			);
			newChild.parent = newNode;
			newNode.children.push(newChild);
			break;
			
		default:
			var attributeName = i.tagName.toLowerCase();
			var index = attributeIndex(attributeName);
			//
			newNode.attributes[index] = new Array();
			//
			for ( var j = getFirstChild(i); j; j = getNextSibling(j) )
			{
				if ( attributes[index] == undefined )
				{
					var x = 5;
				}
				if ( attributes[index].list )
				{
					newNode.attributes[index].push(new Array());
					
					for ( var k = getFirstChild(j); k; k = getNextSibling(k) )
					{
						newNode.attributes[index][newNode.attributes[index].length - 1].push(k.firstChild.nodeValue);
					}
				}
				else
				{
					var value = j.firstChild ? j.firstChild.nodeValue : '';
					
					if ( j.getAttribute('href') )
					{
						var target;
						
						if ( attributes[index].target )
						{
							target = ' target="' + attributes[index].target + '"';
						}
						
						value = '<a href="' + attributes[index].hrefBase + j.getAttribute('href') + '"' + target + '>' + value + '</a>';
					}
					
					newNode.attributes[index].push(value);
				}
			}
			//
			if ( attributeName == magnitudeName || attributeName == hueName )
			{
				for ( j = 0; j < datasets; j++ )
				{
					var value = newNode.attributes[index][j] == undefined ? 0 : Number(newNode.attributes[index][j]);
					
					newNode.attributes[index][j] = value;
					
					if ( attributeName == hueName )
					{
						var hue = lerp
						(
							value,
							valueStart,
							valueEnd,
							hueStart,
							hueEnd
						);
						
						if ( hue < hueStart == hueStart < hueEnd )
						{
							hue = hueStart;
						}
						else if ( hue > hueEnd == hueStart < hueEnd )
						{
							hue = hueEnd;
						}
						
						newNode.hues[j] = hue;
					}
				}
				
				if ( attributeName == hueName )
				{
					newNode.hue = new Tween(newNode.hues[0], newNode.hues[0]);
				}
			}
			break;
		}
	}
	
	return newNode;
}

function maxAbsoluteDepthDecrease()
{
	if ( maxAbsoluteDepth > 2 )
	{
		maxAbsoluteDepth--;
		head.setMaxDepths();
		handleResize();
	}
}

function maxAbsoluteDepthIncrease()
{
	if ( maxAbsoluteDepth < head.maxDepth )
	{
		maxAbsoluteDepth++;
		head.setMaxDepths();
		handleResize();
	}
}

function measureText(text, bold)
{
	context.font = bold ? fontBold : fontNormal;
	var dim = context.measureText(text);
	return dim.width;
}

function min(a, b)
{
	return a < b ? a : b;
}

function minWidth()
{
	// Min wedge width (at center) for displaying a node (or for displaying a
	// label if it's at the highest level being viewed, multiplied by 2 to make
	// further calculations simpler
	
	return (fontSize * 2.3);
}

function mouseMove(e)
{
	mouseX = e.pageX;
	mouseY = e.pageY - headerHeight;
	mouseXRel = (mouseX - centerX) * backingScale()
	mouseYRel = (mouseY - centerY) * backingScale()
	
	if ( head && ! quickLook )
	{
		checkHighlight();
	}
}

function mouseClick(e)
{
	if ( highlightedNode == focusNode && focusNode != selectedNode || selectedNode.hasParent(highlightedNode) )
	{
		if ( highlightedNode.hasChildren() )
		{
			expand(highlightedNode);
		}
	}
	else if ( progress == 1 )//( highlightedNode != selectedNode )
	{
		setFocus(highlightedNode);
//		document.body.style.cursor='ew-resize';
		draw();
		checkHighlight();
		var date = new Date();
		mouseDownTime = date.getTime();
		mouseDown = true;
	}
}

function mouseUp(e)
{
	if ( quickLook )
	{
		navigateBack();
		quickLook = false;
	}
	
	mouseDown = false;
}

function navigateBack()
{
	if ( nodeHistoryPosition > 0 )
	{
		nodeHistory[nodeHistoryPosition] = selectedNode;
		nodeHistoryPosition--;
		
		if ( nodeHistory[nodeHistoryPosition].collapse )
		{
			collapseCheckBox.checked = collapse = false;
		}
		
		setSelectedNode(nodeHistory[nodeHistoryPosition]);
		updateDatasetButtons();
		updateView();
	}
}

function navigateUp()
{
	if ( selectedNode.getParent() )
	{
		selectNode(selectedNode.getParent());
		updateView();
	}
}

function navigateForward()
{
	if ( nodeHistoryPosition < nodeHistory.length - 1 )
	{
		nodeHistoryPosition++;
		var newNode = nodeHistory[nodeHistoryPosition];
		
		if ( newNode.collapse )
		{
			collapseCheckBox.checked = collapse = false;
		}
		
		if ( nodeHistoryPosition == nodeHistory.length - 1 )
		{
			// this will ensure the forward button is disabled
			
			nodeHistory.length = nodeHistoryPosition;
		}
		
		setSelectedNode(newNode);
		updateDatasetButtons();
		updateView();
	}
}

function nextDataset()
{
	var newDataset = currentDataset;
	
	do
	{
		if ( newDataset == datasets - 1 )
		{
			newDataset = 0;
		}
		else
		{
			newDataset++;
		}
	}
	while ( datasetDropDown.options[newDataset].disabled )
	
	selectDataset(newDataset);
}

function onDatasetChange()
{
	selectDataset(datasetDropDown.selectedIndex);
}

function onKeyDown(event)
{
	if
	(
		event.keyCode == 37 &&
		document.activeElement.id != 'search' &&
		document.activeElement.id != 'linkText'
	)
	{
		navigateBack();
		event.preventDefault();
	}
	else if
	(
		event.keyCode == 39 &&
		document.activeElement.id != 'search' &&
		document.activeElement.id != 'linkText'
	)
	{
		navigateForward();
		event.preventDefault();
	}
	else if ( event.keyCode == 38 && datasets > 1 )
	{
		prevDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
	else if ( event.keyCode == 40 && datasets > 1 )
	{
		nextDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
	else if ( event.keyCode == 9 && datasets > 1 )
	{
		selectLastDataset();
		event.preventDefault();
	}
	else if ( event.keyCode == 83 )
	{
		progress += .2;
	}
	else if ( event.keyCode == 66 )
	{
		progress -= .2;
	}
	else if ( event.keyCode == 70 )
	{
		progress = 1;
	}
}

function onKeyPress(event)
{
	if ( event.keyCode == 38 && datasets > 1 )
	{
//		prevDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
	else if ( event.keyCode == 40 && datasets > 1 )
	{
//		nextDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
}

function onKeyUp(event)
{
	if ( event.keyCode == 27 && document.activeElement.id == 'search' )
	{
		search.value = '';
		onSearchChange();
	}
	else if ( event.keyCode == 38 && datasets > 1 )
	{
//		prevDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
	else if ( event.keyCode == 40 && datasets > 1 )
	{
//		nextDataset();
		
		//if ( document.activeElement.id == 'datasets' )
		{
			event.preventDefault();
		}
	}
}

function onSearchChange()
{
	nSearchResults = 0;
	head.search();
	
	if ( search.value == '' )
	{
		searchResults.innerHTML = '';
	}
	else
	{
		searchResults.innerHTML = nSearchResults + ' results';
	}
	
	setFocus(selectedNode);
	draw();
}

function post(url, variable, value, postWindow)
{
	var form = document.createElement('form');
	var input = document.createElement('input');
	var inputDataset = document.createElement('input');
	
	form.appendChild(input);
	form.appendChild(inputDataset);
	
	form.method = "POST";
	form.action = url;
	
	if ( postWindow == undefined )
	{
		form.target = '_blank';
		postWindow = window;
	}
	
	input.type = 'hidden';
	input.name = variable;
	input.value = value;
	
	inputDataset.type = 'hidden';
	inputDataset.name = 'dataset';
	inputDataset.value = currentDataset;
	
	postWindow.document.body.appendChild(form);
	form.submit();
}

function prevDataset()
{
	var newDataset = currentDataset;
	
	do
	{
		if ( newDataset == 0 )
		{
			newDataset = datasets - 1;
		}
		else
		{
			newDataset--;
		}
	}
	while ( datasetDropDown.options[newDataset].disabled );
	
	selectDataset(newDataset);
}

function radiusDecrease()
{
	if ( bufferFactor < .309 )
	{
		bufferFactor += .03;
		updateViewNeeded = true;
	}
}

function radiusIncrease()
{
	if ( bufferFactor > .041 )
	{
		bufferFactor -= .03;
		updateViewNeeded = true;
	}
}

function resetKeyOffset()
{
	currentKey = 1;
	keyMinTextLeft = centerX + gRadius + buffer - buffer / (keys + 1) / 2 + fontSize / 2;
	keyMinAngle = 0;
}

function rgbText(r, g, b)
{
	var rgbArray =
	[
		"rgb(",
		Math.floor(r),
		",",
		Math.floor(g),
		",",
		Math.floor(b),
		")"
	];
	
	return rgbArray.join('');
}

function round(number)
{
	if ( number >= 1 || number <= -1 )
	{
		return number.toFixed(0);
	}
	else
	{
		return number.toPrecision(1);
	}
}

function roundedRectangle(x, y, width, height, radius)
{
	if ( radius * 2 > width )
	{
		radius = width / 2;
	}
	
	if ( radius * 2 > height )
	{
		radius = height / 2;
	}
	
	context.beginPath();
	context.arc(x + radius, y + radius, radius, Math.PI, Math.PI * 3 / 2, false);
	context.lineTo(x + width - radius, y);
	context.arc(x + width - radius, y + radius, radius, Math.PI * 3 / 2, Math.PI * 2, false);
	context.lineTo(x + width, y + height - radius);
	context.arc(x + width - radius, y + height - radius, radius, 0, Math.PI / 2, false);
	context.lineTo(x + radius, y + height);
	context.arc(x + radius, y + height - radius, radius, Math.PI / 2, Math.PI, false);
	context.lineTo(x, y + radius);
}

function passClick(e)
{
	mouseClick(e);
}

function searchResultString(results)
{
	var searchResults = this.searchResults;
	
	if ( this.isSearchResult )
	{
		// don't count ourselves
		searchResults--;
	}
	
	return ' - ' + results + (results > 1 ? ' results' : ' result');
}

function setCallBacks()
{
	canvas.onselectstart = function(){return false;} // prevent unwanted highlighting
	options.onselectstart = function(){return false;} // prevent unwanted highlighting
	document.onmousemove = mouseMove;
	window.onblur = focusLost;
	window.onmouseout = focusLost;
	document.onkeyup = onKeyUp;
	document.onkeydown = onKeyDown;
	canvas.onmousedown = mouseClick;
	document.onmouseup = mouseUp;
	keyControl.onclick = toggleKeys;
	collapseCheckBox = document.getElementById('collapse');
	collapseCheckBox.checked = collapse;
	collapseCheckBox.onclick = handleResize;
	collapseCheckBox.onmousedown = suppressEvent;
	maxAbsoluteDepthText = document.getElementById('maxAbsoluteDepth');
	maxAbsoluteDepthButtonDecrease = document.getElementById('maxAbsoluteDepthDecrease');
	maxAbsoluteDepthButtonIncrease = document.getElementById('maxAbsoluteDepthIncrease');
	maxAbsoluteDepthButtonDecrease.onclick = maxAbsoluteDepthDecrease;
	maxAbsoluteDepthButtonIncrease.onclick = maxAbsoluteDepthIncrease;
	maxAbsoluteDepthButtonDecrease.onmousedown = suppressEvent;
	maxAbsoluteDepthButtonIncrease.onmousedown = suppressEvent;
	fontSizeText = document.getElementById('fontSize');
	fontSizeButtonDecrease = document.getElementById('fontSizeDecrease');
	fontSizeButtonIncrease = document.getElementById('fontSizeIncrease');
	fontSizeButtonDecrease.onclick = fontSizeDecrease;
	fontSizeButtonIncrease.onclick = fontSizeIncrease;
	fontSizeButtonDecrease.onmousedown = suppressEvent;
	fontSizeButtonIncrease.onmousedown = suppressEvent;
	radiusButtonDecrease = document.getElementById('radiusDecrease');
	radiusButtonIncrease = document.getElementById('radiusIncrease');
	radiusButtonDecrease.onclick = radiusDecrease;
	radiusButtonIncrease.onclick = radiusIncrease;
	radiusButtonDecrease.onmousedown = suppressEvent;
	radiusButtonIncrease.onmousedown = suppressEvent;
	maxAbsoluteDepth = 0;
	backButton = document.getElementById('back');
	backButton.onclick = navigateBack;
	backButton.onmousedown = suppressEvent;
	forwardButton = document.getElementById('forward');
	forwardButton.onclick = navigateForward;
	forwardButton.onmousedown = suppressEvent;
	snapshotButton = document.getElementById('snapshot');
	snapshotButton.onclick = snapshot;
	snapshotButton.onmousedown = suppressEvent;
	detailsName = document.getElementById('detailsName');
	detailsExpand = document.getElementById('detailsExpand');
	detailsInfo = document.getElementById('detailsInfo');
	search = document.getElementById('search');
	search.onkeyup = onSearchChange;
	search.onmousedown = suppressEvent;
	searchResults = document.getElementById('searchResults');
	useHueDiv = document.getElementById('useHueDiv');
	linkButton = document.getElementById('linkButton');
	linkButton.onclick = showLink;
	linkButton.onmousedown = suppressEvent;
	linkText = document.getElementById('linkText');
	linkText.onblur = hideLink;
	linkText.onmousedown = suppressEvent;
	hide(linkText);
	var helpButton = document.getElementById('help');
	helpButton.onmousedown = suppressEvent;
	var searchClear = document.getElementById('searchClear');
	searchClear.onmousedown = suppressEvent;
	if ( datasets > 1 )
	{
		datasetDropDown.onmousedown = suppressEvent;
		var prevDatasetButton = document.getElementById('prevDataset');
		prevDatasetButton.onmousedown = suppressEvent;
		var nextDatasetButton = document.getElementById('nextDataset');
		nextDatasetButton.onmousedown = suppressEvent;
		var lastDatasetButton = document.getElementById('lastDataset');
		lastDatasetButton.onmousedown = suppressEvent;
	}
	
	image = document.getElementById('hiddenImage');
	
	if ( image.complete )
	{
		hiddenPattern = context.createPattern(image, 'repeat');
	}
	else
	{
		image.onload = function()
		{
			hiddenPattern = context.createPattern(image, 'repeat');
		}
	}
	
	var loadingImageElement = document.getElementById('loadingImage');
	
	if ( loadingImageElement )
	{
		loadingImage = loadingImageElement.src;
	}
}

function selectDataset(newDataset)
{
	lastDataset = currentDataset;
	currentDataset = newDataset
	if ( datasets > 1 )
	{
		datasetDropDown.selectedIndex = currentDataset;
		updateDatasetButtons();
		datasetAlpha.start = 1.5;
		datasetChanged = true;
	}
	head.setMagnitudes(0);
	head.setDepth(1, 1);
	head.setMaxDepths();
	handleResize();
}

function selectLastDataset()
{
	selectDataset(lastDataset);
	handleResize();
}

function selectNode(newNode)
{
	if ( selectedNode != newNode )
	{
		// truncate history at current location to create a new branch
		//
		nodeHistory.length = nodeHistoryPosition;
		
		if ( selectedNode != 0 )
		{
			nodeHistory.push(selectedNode);
			nodeHistoryPosition++;
		}
		
		setSelectedNode(newNode);
		//updateView();
	}
	
	updateDatasetButtons();
}

function setFocus(node)
{
	if ( node == focusNode )
	{
//		return;
	}
	
	focusNode = node;
	
	if ( node.href )
	{
		detailsName.innerHTML =
			'<a target="_blank" href="' + node.href + '">' + node.name + '</a>';
	}
	else
	{
		detailsName.innerHTML = node.name;
	}
	
	var table = '<table>';
	
	table += '<tr><td></td></tr>';
	
	for ( var i = 0; i < node.attributes.length; i++ )
	{
		if ( attributes[i].displayName && node.attributes[i] != undefined )
		{
			var index = node.attributes[i].length == 1 && attributes[i].mono ? 0 : currentDataset;
			
			if ( typeof node.attributes[i][currentDataset] == 'number' || node.attributes[i][index] != undefined && node.attributes[i][currentDataset] != '' )
			{
				var value = node.attributes[i][index];
				
				if ( attributes[i].listNode != undefined )
				{
					value =
						'<a href="" onclick="showList(' +
						attributeIndex(attributes[i].listNode) + ',' + i +
						',false);return false;" title="Show list">' +
						value + '</a>';
				}
				else if ( attributes[i].listAll != undefined )
				{
					value =
						'<a href="" onclick="showList(' +
						attributeIndex(attributes[i].listAll) + ',' + i +
						',true);return false;" title="Show list">' +
						value + '</a>';
				}
				else if ( attributes[i].dataNode != undefined && dataEnabled )
				{
					value =
						'<a href="" onclick="showData(' +
						attributeIndex(attributes[i].dataNode) + ',' + i +
						',false);return false;" title="Show data">' +
						value + '</a>';
				}
				else if ( attributes[i].dataAll != undefined && dataEnabled )
				{
					value =
						'<a href="" onclick="showData(' +
						attributeIndex(attributes[i].dataAll) + ',' + i +
						',true);return false;" title="Show data">' +
						value + '</a>';
				}
				
				table +=
					'<tr><td><strong>' + attributes[i].displayName + ':</strong></td><td>' +
					value + '</td></tr>';
			}
		}
	}
	
	table += '</table>';
	detailsInfo.innerHTML = table;
	
	detailsExpand.disabled = !focusNode.hasChildren() || focusNode == selectedNode;
}

function setSelectedNode(newNode)
{
	if ( selectedNode && selectedNode.hasParent(newNode) )
	{
		zoomOut = true;
	}
	else
	{
		zoomOut = false;
	}
	
	selectedNodeLast = selectedNode;
	selectedNode = newNode;
	
	//if ( focusNode != selectedNode )
	{
		setFocus(selectedNode);
	}
}

function waitForData(dataWindow, target, title, time, postUrl, postVar)
{
	if ( nodeData.length == target )
	{
		if ( postUrl != undefined )
		{
			for ( var i = 0; i < nodeData.length; i++ )
			{
				nodeData[i] = nodeData[i].replace(/\n/g, ',');
			}
			
			var postString = nodeData.join('');
			postString = postString.slice(0, -1);
			
			dataWindow.document.body.removeChild(dataWindow.document.getElementById('loading'));
			document.body.removeChild(document.getElementById('data'));
			
			post(postUrl, postVar, postString, dataWindow);
		}
		else
		{
			//dataWindow.document.body.removeChild(dataWindow.document.getElementById('loading'));
			//document.body.removeChild(document.getElementById('data'));
			
			dataWindow.document.open();
			dataWindow.document.write('<pre>' + nodeData.join('') + '</pre>');
			dataWindow.document.close();
		}
		
		dataWindow.document.title = title; // replace after document.write()
	}
	else
	{
		var date = new Date();
		
		if ( date.getTime() - time > 10000 )
		{
			dataWindow.document.body.removeChild(dataWindow.document.getElementById('loading'));
			document.body.removeChild(document.getElementById('data'));
			dataWindow.document.body.innerHTML =
				'Timed out loading supplemental files for:<br/>' + document.location;
		}
		else
		{
			setTimeout(function() {waitForData(dataWindow, target, title, time, postUrl, postVar);}, 100);
		}
	}
}

function data(newData)
{
	nodeData.push(newData);
}

function enableData()
{
	dataEnabled = true;
}

function showData(indexData, indexAttribute, summary)
{
	var dataWindow = window.open('', '_blank');
	var title = 'Krona - ' + attributes[indexAttribute].displayName + ' - ' + focusNode.name;
	dataWindow.document.title = title;
	
	nodeData = new Array();
	
	if ( dataWindow && dataWindow.document && dataWindow.document.body != null )
	{
		//var loadImage = document.createElement('img');
		//loadImage.src = "file://localhost/Users/ondovb/Krona/KronaTools/img/loading.gif";
		//loadImage.id = "loading";
		//loadImage.alt = "Loading...";
		//dataWindow.document.body.appendChild(loadImage);
		dataWindow.document.body.innerHTML =
			'<img id="loading" src="' + loadingImage + '" alt="Loading..."></img>';
	}
	
	var scripts = document.createElement('div');
	scripts.id = 'data';
	document.body.appendChild(scripts);
	
	var files = focusNode.getData(indexData, summary);
	
	var date = new Date();
	var time = date.getTime();
	
	for ( var i = 0; i < files.length; i++ )
	{
		var script = document.createElement('script');
		script.src = files[i] + '?' + time;
		scripts.appendChild(script);
	}
	
	waitForData(dataWindow, files.length, title, time, attributes[indexAttribute].postUrl, attributes[indexAttribute].postVar);
	
	return false;
}

function showList(indexList, indexAttribute, summary)
{
	var list = focusNode.getList(indexList, summary);
	
	if ( attributes[indexAttribute].postUrl != undefined )
	{
		post(attributes[indexAttribute].postUrl, attributes[indexAttribute].postVar, list.join(','));
	}
	else
	{
		var dataWindow = window.open('', '_blank');
		
		if ( true || navigator.appName == 'Microsoft Internet Explorer' ) // :(
		{
			dataWindow.document.open();
			dataWindow.document.write('<pre>' + list.join('\n') + '</pre>');
			dataWindow.document.close();
		}
		else
		{
			var pre = document.createElement('pre');
			dataWindow.document.body.appendChild(pre);
			pre.innerHTML = list;
		}
		
		dataWindow.document.title = 'Krona - ' + attributes[indexAttribute].displayName + ' - ' + focusNode.name;
	}
}

function snapshot()
{
	svg = svgHeader();
	
	resetKeyOffset();
	
	snapshotMode = true;
	
	selectedNode.draw(false, true);
	selectedNode.draw(true, true);
	
	if ( focusNode != 0 && focusNode != selectedNode )
	{
		context.globalAlpha = 1;
		focusNode.drawHighlight(true);
	}
	
	if ( hueDisplayName && useHue() )
	{
		drawLegendSVG();
	}
	
	snapshotMode = false;
	
	svg += svgFooter();
	
	snapshotWindow = window.open
	(
		'data:image/svg+xml;charset=utf-8,' + encodeURIComponent(svg),
		'_blank'
	);
/*	var data = window.open('data:text/plain;charset=utf-8,hello', '_blank');
	var data = window.open('', '_blank');
	data.document.open('text/plain');
	data.document.write('hello');
	data.document.close();
	var button = document.createElement('input');
	button.type = 'button';
	button.value = 'save';
	button.onclick = save;
	data.document.body.appendChild(button);
//	snapshotWindow.document.write(svg);
//	snapshotWindow.document.close();
*/	
}

function save()
{
	alert(document.body.innerHTML);
}

function spacer()
{
	if ( snapshotMode )
	{
		return '&#160;&#160;&#160;';
	}
	else
	{
		return '   ';
	}
}

function suppressEvent(e)
{
	e.cancelBubble = true;
	if (e.stopPropagation) e.stopPropagation();
}

function svgFooter()
{
	return '</svg>';
}

function svgHeader()
{
	var patternWidth = fontSize * .6;//radius / 50;
	
	return '\
<?xml version="1.0" standalone="no"?>\
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" \
	"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">\
<svg width="' + imageWidth + '" height="' + imageHeight + '" version="1.1"\
	xmlns="http://www.w3.org/2000/svg">\
<title>Krona (snapshot) - ' +
(datasets > 1 ? datasetNames[currentDataset] + ' - ' : '') + selectedNode.name +
'</title>\
<defs>\
	<style type="text/css">\
	text {font-size: ' + fontSize + 'px; font-family: ' + fontFamily + '; dominant-baseline:central}\
	path {stroke-width:' + thinLineWidth * fontSize / 12 + ';}\
	path.wedge {stroke:none}\
	path.line {fill:none;stroke:black;}\
	line {stroke:black;stroke-width:' + thinLineWidth * fontSize / 12 + ';}\
	line.tick {stroke-width:' + thinLineWidth * fontSize / 6 + ';}\
	line.pattern {stroke-width:' + thinLineWidth * fontSize / 18 + ';}\
	circle {fill:none;stroke:black;stroke-width:' + thinLineWidth * fontSize / 12 + ';}\
	rect {stroke:black;stroke-width:' + thinLineWidth * fontSize / 12 + ';}\
	.highlight {stroke:black;stroke-width:'+ highlightLineWidth * fontSize / 12 + ';}\
	.searchHighlight {fill:rgb(255, 255, 100);stroke:none;}\
	</style>\
<pattern id="hiddenPattern" patternUnits="userSpaceOnUse" \
x="0" y="0" width="' + patternWidth + '" height="' + patternWidth + '">\
<line class="pattern" x1="0" y1="0" x2="' + patternWidth / 2 + '" y2="' + patternWidth / 2 + '"/>\
<line class="pattern" x1="' + patternWidth / 2 + '" y1="' + patternWidth +
'" x2="' + patternWidth + '" y2="' + patternWidth / 2 + '"/>\
</pattern>\
</defs>\
';
}

function svgText(text, x, y, anchor, bold, color)
{
	if ( typeof(anchor) == 'undefined' )
	{
		anchor = 'start';
	}
	
	if ( color == undefined )
	{
		color = 'black';
	}
	
	return '<text x="' + x + '" y="' + y +
		'" style="font-color:' + color + ';font-weight:' + (bold ? 'bold' : 'normal') +
		'" text-anchor="' + anchor + '">' + text + '</text>';
}

function toggleKeys()
{
	if ( showKeys )
	{
		keyControl.value = '';
		showKeys = false;
	}
	else
	{
		keyControl.value = 'x';
		showKeys = true;
	}
	
	updateKeyControl();
	
	if ( progress == 1 )
	{
		draw();
	}
}

function update()
{
	if ( ! head )
	{
		return;
	}
	
	if ( mouseDown && focusNode != selectedNode )
	{
		var date = new Date();
		
		if ( date.getTime() - mouseDownTime > quickLookHoldLength )
		{
			if ( focusNode.hasChildren() )
			{
				expand(focusNode);
				quickLook = true;
			}
		}
	}
	
	if ( updateViewNeeded )
	{
		resize();
		mouseX = -1;
		mouseY = -1;
		
		collapse = collapseCheckBox.checked;
		compress = true;//compressCheckBox.checked;
		shorten = true;//shortenCheckBox.checked;
		
		checkSelectedCollapse();
		updateMaxAbsoluteDepth();
		
		if ( focusNode.getCollapse() || focusNode.depth > maxAbsoluteDepth )
		{
			setFocus(selectedNode);
		}
		else
		{
			setFocus(focusNode);
		}
		
		updateView();
		
		updateViewNeeded = false;
	}
	
	var date = new Date();
	progress = (date.getTime() - tweenStartTime) / tweenLength;
//	progress += .01;
	
	if ( progress >= 1 )
	{
		progress = 1;
	}
	
	if ( progress != progressLast )
	{
		tweenFactor =// progress;
			(1 / (1 + Math.exp(-tweenCurvature * (progress - .5))) - .5) /
			(tweenMax - .5) / 2 + .5;
		
		if ( progress == 1 )
		{
			snapshotButton.disabled = false;
			zoomOut = false;
			
			//updateKeyControl();
			
			if ( ! quickLook )
			{
				//checkHighlight();
			}
			
			
			if ( fpsDisplay )
			{
				fpsDisplay.innerHTML = 'fps: ' + Math.round(tweenFrames * 1000 / tweenLength);
			}
		}
		
		draw();
	}
	
	progressLast = progress;
}

function updateDatasetButtons()
{
	if ( datasets == 1 )
	{
		return;
	}
	
	var node = selectedNode ? selectedNode : head;
	
	datasetButtonLast.disabled =
		node.attributes[magnitudeIndex][lastDataset] == 0;
	
	datasetButtonPrev.disabled = true;
	datasetButtonNext.disabled = true;
	
	for ( var i = 0; i < datasets; i++ )
	{
		var disable = node.attributes[magnitudeIndex][i] == 0;
		
		datasetDropDown.options[i].disabled = disable;
		
		if ( ! disable )
		{
			if ( i != currentDataset )
			{
				datasetButtonPrev.disabled = false;
				datasetButtonNext.disabled = false;
			}
		}
	}
}

function updateDatasetWidths()
{
	if ( datasets > 1 )
	{
		for ( var i = 0; i < datasets; i++ )
		{
			context.font = fontBold;
			var dim = context.measureText(datasetNames[i]);
			datasetWidths[i] = dim.width;
		}
	}
}

function updateKeyControl()
{
	if ( keys == 0 )//|| progress != 1 )
	{
		keyControl.style.visibility = 'hidden';
	}
	else
	{
		keyControl.style.visibility = 'visible';
		keyControl.style.right = margin + 'px';
		
		if ( showKeys )
		{
			keyControl.style.top =
				imageHeight -
				(
					keys * (keySize + keyBuffer) -
					keyBuffer +
					margin +
					keyControl.clientHeight * 1.5
				) + 'px';
		}
		else
		{
			keyControl.style.top =
				(imageHeight - margin - keyControl.clientHeight) + 'px';
		}
	}
}

function updateView()
{
	if ( selectedNode.depth > maxAbsoluteDepth - 1 )
	{
		maxAbsoluteDepth = selectedNode.depth + 1;
	}
	
	highlightedNode = selectedNode;
	
	angleFactor = 2 * Math.PI / (selectedNode.magnitude);
	
	maxPossibleDepth = Math.floor(gRadius / (fontSize * minRingWidthFactor));
	
	if ( maxPossibleDepth < 4 )
	{
		maxPossibleDepth = 4;
	}
	
	var minRadiusInner = fontSize * 8 / gRadius;
	var minRadiusFirst = fontSize * 6 / gRadius;
	var minRadiusOuter = fontSize * 5 / gRadius;
	
	if ( .25 < minRadiusInner )
	{
		minRadiusInner = .25;
	}
	
	if ( .15 < minRadiusFirst )
	{
		minRadiusFirst = .15;
	}
	
	if ( .15 < minRadiusOuter )
	{
		minRadiusOuter = .15;
	}
	
	// visibility of nodes depends on the depth they are displayed at,
	// so we need to set the max depth assuming they can all be displayed
	// and iterate it down based on the deepest child node we can display
	//
	var maxDepth;
	var newMaxDepth = selectedNode.getMaxDepth() - selectedNode.getDepth() + 1;
	//
	do
	{
		maxDepth = newMaxDepth;
		
		if ( ! compress && maxDepth > maxPossibleDepth )
		{
			maxDepth = maxPossibleDepth;
		}
		
		if ( compress )
		{
			compressedRadii = new Array(maxDepth);
			
			compressedRadii[0] = minRadiusInner;
			
			var offset = 0;
			
			while
			(
				lerp
				(
					Math.atan(offset + 2),
					Math.atan(offset + 1),
					Math.atan(maxDepth + offset - 1),
					minRadiusInner,
					1 - minRadiusOuter
				) - minRadiusInner > minRadiusFirst &&
				offset < 10
			)
			{
				offset++;
			}
			
			offset--;
			
			for ( var i = 1; i < maxDepth; i++ )
			{
				compressedRadii[i] = lerp
				(
					Math.atan(i + offset),
					Math.atan(offset),
					Math.atan(maxDepth + offset - 1),
					minRadiusInner,
					1 - minRadiusOuter
				)
			}
		}
		else
		{
			nodeRadius = 1 / maxDepth;
		}
		
		newMaxDepth = selectedNode.maxVisibleDepth(maxDepth);
		
		if ( compress )
		{
			if ( newMaxDepth <= maxPossibleDepth )
			{
//				compress
			}
		}
		else
		{
			if ( newMaxDepth > maxPossibleDepth )
			{
				newMaxDepth = maxPossibleDepth;
			}
		}
	}
	while ( newMaxDepth < maxDepth );
	
	maxDisplayDepth = maxDepth;
	
	lightnessFactor = (lightnessMax - lightnessBase) / (maxDepth > 8 ? 8 : maxDepth);
	keys = 0;
	
	nLabelOffsets = new Array(maxDisplayDepth - 1);
	labelOffsets = new Array(maxDisplayDepth - 1);
	labelLastNodes = new Array(maxDisplayDepth - 1);
	labelFirstNodes = new Array(maxDisplayDepth - 1);
	
	for ( var i = 0; i < maxDisplayDepth - 1; i++ )
	{
		if ( compress )
		{
			if ( i == maxDisplayDepth - 1 )
			{
				nLabelOffsets[i] = 0;
			}
			else
			{
				var width =
					(compressedRadii[i + 1] - compressedRadii[i]) *
					gRadius;
				
				nLabelOffsets[i] = Math.floor(width / fontSize / 1.2);
				
				if ( nLabelOffsets[i] > 2 )
				{
					nLabelOffsets[i] = min
					(
						Math.floor(width / fontSize / 1.75),
						5
					);
				}
			}
		}
		else
		{
			nLabelOffsets[i] = Math.max
			(
				Math.floor(Math.sqrt((nodeRadius * gRadius / fontSize)) * 1.5),
				3
			);
		}
		
		labelOffsets[i] = Math.floor((nLabelOffsets[i] - 1) / 2);
		labelLastNodes[i] = new Array(nLabelOffsets[i] + 1);
		labelFirstNodes[i] = new Array(nLabelOffsets[i] + 1);
		
		for ( var j = 0; j <= nLabelOffsets[i]; j++ )
		{
			// these arrays will allow nodes with neighboring labels to link to
			// each other to determine max label length
			
			labelLastNodes[i][j] = 0;
			labelFirstNodes[i][j] = 0;
		}
	}
	
	fontSizeText.innerHTML = fontSize;
	fontNormal = fontSize + 'px ' + fontFamily;
	context.font = fontNormal;
	fontBold = 'bold ' + fontSize + 'px ' + fontFamily;
	tickLength = fontSize * .7;
	
	head.setTargets(0);
	
	keySize = ((imageHeight - margin * 3) * 1 / 2) / keys * 3 / 4;
	
	if ( keySize > fontSize * maxKeySizeFactor )
	{
		keySize = fontSize * maxKeySizeFactor;
	}
	
	keyBuffer = keySize / 3;
	
	fontSizeLast = fontSize;
	
	if ( datasetChanged )
	{
		datasetChanged = false;
	}
	else
	{
		datasetAlpha.start = 0;
	}
	
	var date = new Date();
	tweenStartTime = date.getTime();
	progress = 0;
	tweenFrames = 0;
	
	updateKeyControl();
	updateDatasetWidths();
	
	document.title = 'Krona - ' + selectedNode.name;
	updateNavigationButtons();
	snapshotButton.disabled = true;
	
	maxAbsoluteDepthText.innerHTML = maxAbsoluteDepth - 1;
	
	maxAbsoluteDepthButtonDecrease.disabled = (maxAbsoluteDepth == 2);
	maxAbsoluteDepthButtonIncrease.disabled = (maxAbsoluteDepth == head.maxDepth);
	
	if ( collapse != collapseLast && search.value != '' )
	{
		onSearchChange();
		collapseLast = collapse;
	}
}

function updateMaxAbsoluteDepth()
{
	while ( selectedNode.depth > maxAbsoluteDepth - 1 )
	{
		selectedNode = selectedNode.getParent();
	}
}

function updateNavigationButtons()
{
	backButton.disabled = (nodeHistoryPosition == 0);
//	upButton.disabled = (selectedNode.getParent() == 0);
	forwardButton.disabled = (nodeHistoryPosition == nodeHistory.length);
}

function useHue()
{
	return useHueCheckBox && useHueCheckBox.checked;
}
/*
function zoomOut()
{
	return (
		selectedNodeLast != 0 &&
		selectedNodeLast.getDepth() < selectedNode.getDepth());
}
*/
</script>
    </head>
    <body>
        <img id="hiddenImage" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAgAAAAICAYAAADED76LAAAAAXNSR0IArs4c6QAAAAZiS0dEAP8A/wD/oL2nkwAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9oLCBQhNQwWVnsAAAAidEVYdENvbW1lbnQAQ3JlYXRlZCB3aXRoIEdJTVAgb24gYSBNYWOHqHdDAAABE0lEQVQYGQEIAff+AwAAABkAAAAAAAAA+gAAAAAAAAAAAAAAAAAAAAAAAAAMAwAAAAAAAAANAAAAAAAAAPoAAAAAAAAADAAAAAYAAAD0AwAAAPoAAAAAAAAAAAAAAPoAAAAMAAAADQAAAPoAAAD6AAAAAAAAAAAAAAAAAAAAAAwAAAAZAAAADAAAAAAAAAAAAAAAAAAAAAAAAAAADAAAABkAAAAMAAAAAAAAAAAAAAAAAAAAAAAAAAAMAAAAGQAAAAwAAAAAAAAADAAAAAwAAAAABAAAAAAAAAAAAAAA8wAAAPQAAAAAAAAAAAAAAA0AAAAAAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwAAAAZRssKC5OpXwYAAAAASUVORK5CYII="  style="display:none"/>
        <img id="loadingImage" src="data:image/gif;base64,R0lGODlhEAAQAPIAAP///wAAAMLCwkJCQgAAAGJiYoKCgpKSkiH/C05FVFNDQVBFMi4wAwEAAAAh/hpDcmVhdGVkIHdpdGggYWpheGxvYWQuaW5mbwAh+QQJCgAAACwAAAAAEAAQAAADMwi63P4wyklrE2MIOggZnAdOmGYJRbExwroUmcG2LmDEwnHQLVsYOd2mBzkYDAdKa+dIAAAh+QQJCgAAACwAAAAAEAAQAAADNAi63P5OjCEgG4QMu7DmikRxQlFUYDEZIGBMRVsaqHwctXXf7WEYB4Ag1xjihkMZsiUkKhIAIfkECQoAAAAsAAAAABAAEAAAAzYIujIjK8pByJDMlFYvBoVjHA70GU7xSUJhmKtwHPAKzLO9HMaoKwJZ7Rf8AYPDDzKpZBqfvwQAIfkECQoAAAAsAAAAABAAEAAAAzMIumIlK8oyhpHsnFZfhYumCYUhDAQxRIdhHBGqRoKw0R8DYlJd8z0fMDgsGo/IpHI5TAAAIfkECQoAAAAsAAAAABAAEAAAAzIIunInK0rnZBTwGPNMgQwmdsNgXGJUlIWEuR5oWUIpz8pAEAMe6TwfwyYsGo/IpFKSAAAh+QQJCgAAACwAAAAAEAAQAAADMwi6IMKQORfjdOe82p4wGccc4CEuQradylesojEMBgsUc2G7sDX3lQGBMLAJibufbSlKAAAh+QQJCgAAACwAAAAAEAAQAAADMgi63P7wCRHZnFVdmgHu2nFwlWCI3WGc3TSWhUFGxTAUkGCbtgENBMJAEJsxgMLWzpEAACH5BAkKAAAALAAAAAAQABAAAAMyCLrc/jDKSatlQtScKdceCAjDII7HcQ4EMTCpyrCuUBjCYRgHVtqlAiB1YhiCnlsRkAAAOwAAAAAAAAAAAA=="  style="display:none"/>
        <img id="logo" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAANgAAAA8CAYAAAAE9XR5AAAC0WlDQ1BJQ0MgUHJvZmlsZQAAOI2NlM9LFGEYx7+zjRgoQWBme4ihQ0ioTBZlROWuv9i0bVl/lBLE7Oy7u5Ozs9PM7JoiEV46ZtE9Kg8e+gM8eOiUl8LALALpblFEgpeS7Xlnxt0R7ccLM/N5nx/f53nf4X2BGlkxTT0kAXnDsZJ9Uen66JhU+xEhHEEdwqhTVNuMJBIDoMFjsWtsvofAvyute/v/OurStpoHhP1A6Eea2Sqw7xfZC1lqBBC5XsOEYzrE9zhbnv0x55TH8659KNlFvEh8QDUtHv+auEPNKWmgRiRuyQZiUgHO60XV7+cgPfXMGB6k73Hq6S6ze3wWZtJKdz9xG/HnNOvu4ZrE8xmtN0bcTM9axuod9lg4oTmxIY9DI4YeH/C5yUjFr/qaoulEk9v6dmmwZ9t+S7mcIA4TJ8cL/TymkXI7p3JD1zwW9KlcV9znd1Yxyeseo5g5U3f/F/UWeoVR6GDQYNDbgIQk+hBFK0xYKCBDHo0iNLIyN8YitjG+Z6SORIAl8q9TzrqbcxtFyuZZI4jGMdNSUZDkD/JXeVV+Ks/JX2bDxeaqZ8a6qanLD76TLq+8ret7/Z48fZXqRsirI0vWfGVNdqDTQHcZYzZcVeI12P34ZmCVLFCpFSlXadytVHJ9Nr0jgWp/2j2KXZpebKrWWhUXbqzUL03v2KvCrlWxyqp2zqtxwXwmHhVPijGxQzwHSbwkdooXxW6anRcHKhnDpKJhwlWyoVCWgUnymjv+mRcL76y5o6GPGczSVImf/4RVyGg6CxzRf7j/c/B7xaOxIvDCBg6frto2ku4dIjQuV23OFeDCN7oP3lZtzXQeDj0BFs6oRavkSwvCG4pmdxw+6SqYk5aWzTlSuyyflSJ0JTEpZqhtLZKi65LrsiWL2cwqsXQb7Mypdk+lnnal5lO5vEHnr/YRsPWwXP75rFzeek49rAEv9d/AvP1FSOihagAAAAZiS0dEAP8A/wD/oL2nkwAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB+AEEhYRJKJxkPUAACAASURBVHgBAGaAmX8B/////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAvUj3EAAAIABJREFUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////AAAAAAD+/v4AAAAAAP///wD///8AAAAAAP///wAAAAAAAQEBAAEBAQAAAAAAAQEBAAICAgAAAAAAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wAAAAAA/v7+AP7+/gD9/f0A/v7+AP39/QD+/v4A/f39AP39/QD9/f0AAQIBAAMCAwAEBAQAAwMDAAMDAwACAgIAAwMDAAEBAQABAQEAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////AP7+/gD///8A/v7+APv7+wD7+/sA+/v7AN7e3gDd3d0A5eXlAPLy8gDn5+cA////ABUVFQAQEBAAGRkZACMjIwAfHx8ABQUFAAYGBgAFBQUABAQEAAQEBAACAgIAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAf////8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////AP39/QD8/PwA+vr6APb29gDFxcUAxMPDAN3a3ADz9/UACQQEABwKDAAQBQQA0e7tAAgVBwAVHg0A9e76AO/o9wD59voACwwKACQjJQA1NTUAODg4AAsLCwAJCQkABgYGAAQEBAACAgIAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH/////AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAZ2dnAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJmZmQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANra2gCRkZEA/Pz8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA8PDwB1dXUAFRUVAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/v7+AP39/QD5+fkA6+vrALKysgC/u7sA9vf3ADgUFQA1ERIAMBEQABUHCAACAAEAAAAAAF3JxgAZPhMAQ2gmAAABAAAAAAAA8uj5AOTW8ADey+wA383uAAkICQA7PDwAS0tLABMTEwAKCgoABgYGAAUFBQADAwMAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAODg4ACUlJQA/Pz8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsLCwBwcHAAFRUVAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/f39APv7+wD39/cAr6+vAK+urgAI/v4AWhscAEYXGQAIAwMAAAAAAAAA/wAA//8A/gAAAAAAAAAAAP4A/wD/AP/+/wAAAP8AAAAAAA0UBQApPhUAKkEXAPjz/ADXwekAwa6+AMTEwABISEwASEhIAA4ODgAKCgoABgYGAAICAgACAgIAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAObm5gCRkZEA+fn5AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAcHBwBvb28AGhoaAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/Pz8APr6+gDg4OAAj4yMAOTj4wBbHh8ATxscAAUBAgAAAP8A///+AAAAAAABAQEAAAAAAAD//wD///8A/wAAAP79/gD9/f0AAQAAAAAAAAD/AAAAAAAAAAMBAQAHCwMAMUwbAC5GGgDNseMAsK2tAM/PzgBhYWIAICAgAAwMDAAJCQkAAAAAAAICAgABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAOvr6wCQkJAA9vb2AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAUFBQBtbW0AHx8fAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/Pz8APj4+ADGxsYAeHR0AAXv8ACZMTQAVBwdAAUBAQD///4A////AAD/AAAA/wAA/v7+AP7+/gD///8AAAAAAAAAAAD///8A+/r6AP39/QD9/f0A/v3+AP79/gD9/P0A/fz9AP/+/wACBAAAOFYfAF2PNwD1AO0AiYiGAMnJyQDw8PAA9PT0APv7+wD+/v4A////AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAO/v7wCPj48A8fHxAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMDAwBoaGgAJSUlAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/Pz8APf39wC5ubkAiYiIAD4SEgCLLC4AFQcHAP///wD+/v8A//7/AP/+/wAAAAAA/wD/AP///wD///8A/v/+AP7//gAAAAAA//7/AP7+/wD9/f0A/f39AP39/QD9/f0A/v7+AP7+/gD9/P0A/fz9AP39/gASGgkAV4cwAB42DgCWlpUAv7+/APDw8AD29vYA+/v7AP7+/gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABpHbeyAAAgAElEQVQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPPz8wCQkJAA7e3tAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQBkZGQALCwsAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/Pz8APj4+AC/v78Al5WWAFsdHQBYHiAA/wAAAP8AAAD//v4AAAAAAAAAAQAAAAAA/f79AAAAAAAAAQEAAAAAAAIAAAAAAAAAAAAAAP8A/wD9/P0AAAICAAAAAAD/AAAAAAAAAAAAAAAAAAAAAAAAAAIDAgD9/f0AAAAAAAQEAQBIbCgAChAGAKSkowAEBAQAOjo6AAICAgACAgIAAAAAAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPf39wCRkZEA6OjoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABfX18AMjIyAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/f39APv7+wDR0dEAl5SUAFMaGwBXHh8A/v//AAEBAAAAAAAA/v8AAAAAAAAAAf8AAP//AAAAAAAAAAAAAP//AAD/AAD9AP8AAAAAAAD/AAD//v8A/f79AAAA/wAAAAAA/gAAAAQCAwD+/P0AAgQDAP39/QAAAAAA/gD/AAAAAAAAAAMAAwMBADxbIgD9+f8ApKWiAA0NDQAwMDAAAQEBAAoKCgACAgIAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPn5+QCTk5MA4+PjAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABXV1cAOTk5AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/v7+APz8/ADr6+sAkY2NADUREgBeICEA////AP7/AAAA//8AAP8AAAABAAAAAAAAAP4AAAAAAAD/AP8AAAAAAAD//wAAAAAAAAEBAAAAAAAAAAAA/wD/AP79/AACAAQAAAAAAAEAAAD+AAEAAAD/AP/8AAAAAAAAAAD/AAMDBAD+/f4AAQAAAAAAAwABAgEAQmMkAPDo9wCjo6IAGRkZAAQEBAAEBAQAAQEBAAMDAwABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPv7+wCYmJgA3d3dAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABPT08AQkJCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/v7+AMHBwQDBwcEA6urqAAcHBwAfHx8ANzc3ADY2NgADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////AP7+/gD5+fkAmpeXAAT6+gBuJykA/v/+AAIBAQAAAP8AAAAAAAABAAD///8A////AAEAAQAAAAAA///+AAAAAAAAAQEAAAAAAAD+/gAAAAAA/wD/AP/+/wD9/f4A/f35AAMCAwD/AAEAAwT/AAIAAQD+AAEAAAEAAAH+AQAAAPwAAQMDAAD9/QD/AAAAA/8CAP0DAgBLcioAnm3JAKqqqgBiYmIABgYGAAMDAwABAQEAAwMDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAOLi4gDIyMgA2traAOrq6gD7+/sADQ0NABISEgAqKioANDQ0ABoaGgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA+Pj4ANLS0gDV1dUA5eXlAO/v7wD4+PgA/f39AAUFBQALCwsAExMTAB8fHwAtLS0AKCgoAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP39/QCdnZ0A1tbWAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABGRkYASkpKAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABnZ2cAAAAAAAAAAAAAAAAAAAAAAAAAAACZmZkA/f39AJqamgDQ0NAAAAAAAAAAAAD29vYA19fXANfX1wDAwMAAeXl5AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP7+/gD7+/sAzc3NAMbJyQCFKy0ABwICAAAAAAD9/v4AAAAAAAD//wAAAAAAAAEBAAAAAAD///8AAAAAAP8AAAAAAAAAAP//AAAAAAAAAAAA/v//AAAAAAD///8A/v3+AAAAAAAAAP4AAAAAAAH8/wD+AAAAAgQAAAMCAgD9/PwAAAIDAAIAAQAAAwIAAAAAAAABAQD//v8ABAkCAEdqKQCwhc0A1zk5AAgICAAGBgYAAwMDAAICAgABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGdnZwAAAAAAAAAAAAAAAAAAAAAAAAAAAJmZmQAAAAAA8PDwAKioqADPz88AAAAAAAAAAAAAAAAAAAAAAPHx8QDx8fEA39/fALW1tQA4ODgAVFRUAA0NDQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwcHBALCwsAD29vYAAAAAAAAAAAAAAAAAAAAAAAAAAAD6+voA+vr6AO/v7wDc3NwAvb29ABwcHABfX18AHh4eAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wCjo6MAz8/PAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA+Pj4AU1NTAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKampgDQ0NAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAgIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wD+/v4A+Pj4AKejowA7FBUALA4OAP///wD///8AAAAAAAAAAAD///8A////AP7+/gD///8A/wD/AP3//gAAAAAAAAAAAP7//wD+/v4A/v/+AAAA/wAAAAAA////AP39/QD9/f4A/v3+APz8/gD9/v0A/f7+AP79/AD9/v0AAAAAAP3+/QD9/P0A/f3+AP79/QD+/f0AAAAAAAD/AAAjNBMAHzAUALa3tQDy8vIA9/f3APz8/AD+/v4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA4ODgAIWFhQDPz88AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAyMjIAI2NjQDx8fEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAp6enAKampgD29vYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADk5OQAiYmJANbW1gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACpqakAx8fHAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA2NjYAWlpaAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMrKygDExMQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACcnJwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD+/v4A+/v7AM/PzwDb398AWBsbAP///wAA//8AAP//AP///wD+/v4A/wAAAP8AAAAAAAAAAAAAAP7//wAAAAAA/v7+AP38/QD//f4A//7/AAD/AAAA/wAAAAAAAP7//wD9/f4A/f38AP39/QD//v0A/v3/AAEAAAD+/f4A/vz9AP3+/gAAAQEAAQAAAAAAAAAAAAEA+/39AP38/QD//f4A/P39AD1dIwDd3N4A2dnZAPX19QD7+/sA/v7+AP///wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA5OTkAJCQkADy8vIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADn5+cApKSkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwcHBAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/ACbm5sA9PT0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACwsLAAv7+/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAvLy8AYGBgAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACnp6cAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAoKCgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/v7+APv7+wC1tbUANBISAB0ICgAA/wAA/wAAAP7//wD///8A/v//AP7//wD+/v4A/v7+AP3+/QD//f4A//3+AP/+/wAAAAAAAAAAAAAAAAD//v8A9fz8AP3//wD79/0A+vX6AP79/gD9/f4AAAABAP39/gD7/P4A/wD/AP8AAQD9/P0A/f39AP3+/QD9/P0A/f38AAIAAAAAAQAAAAEBAAIBAQAXIg0AHSoTAMLEwgDz8/MA9/j3APz8/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJaWlgD39/cAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANLS0gDf398AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9PT0AK2trQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9vb2AMnJyQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAC5ubkAt7e3AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHBwcAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9vb2AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHBwcAHx8fAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP39/QDv7+8A1tPTAG8hIwD+//4AAAAAAAAAAAD+//8AAAAAAAAAAAAA//8A/wAAAAEAAAAA/gAAAAAAAP8A/wAAAAAAAAAAAP8AAADD7u4AuOzqANvz8wAAAAAA8NH1AOPQ9AAUHgoAIT0LAB8xDgAAAAAAAwD/AP79AwAAAP0AAwICAAIAAAAAAAEAAQAAAAAAAQAAAAAAAAAAAAAA/wD+//8AAgEAAC9IGgC7l9gA7u7uAA0NDQAJCAkABQUFAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADt7e0AAAAAAAAAAAAaGhoATExMACIiIgAICAgA3d3dAKysrADn5+cAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAy8vLACEhIQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMDAwAD5+fkAAAAAAAAAAAAAAAAAAAAAAAAAAABCQkIAQkJCABEREQD39/cA2dnZAKioqADz8/MAAAAAAAAAAAAAAAAAAAAAAAAAAADa2toAQ0NDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAArq6uAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALi4uAA1NTUAGRkZAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJiYmAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP7+/gD9/f0AzMzMAGfP6fwAACAASURBVP///wA0DxAAAAAAAAAAAAD+//8AAP//AAABAQAA//8A/v7+AAAAAAD/AAAAAAAAAP8AAAD+//8AAgAAAPP9/ACM3d8A3PbzACQnJwBNJCQAERERAAAAAADv7+8A3NzcAM2o2QDkzPUANlwVAAMMBwAAAAAA/Pz+AAAE/wAA/AAAAgQAAAAA/wD//AAAAAD/AP4AAAABAAAAAgABAOXV7wDcyesA+ff7AMbGxgATExMADw8PAAgICAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAtLS0AaGhoAAQEBAAAAAAAAAAAACwsLAAfHx8Ag4ODAPHx8QAAAAAAAAAAAAAAAAAAAAAAAAAAAN/f3wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADLy8sAAAAAAAAAAAAAAAAAAAAAAAAAAAA4ODgAV1dXAAAAAAAAAAAADQ0NADQ0NAABAQEAmpqaAAAAAAAAAAAAAAAAAAAAAAAAAAAA6enpAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD9/f0ApKSkAHR0dAABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQBHR0cAOzs7ABISEgD7+/sA8fHxAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wD///8A/Pz8ANLS0gApDQ4ABAIBAP7//wAA//8AAAAAAP8AAAD//f4AAAAAAAAAAAAAAAAA/wD/AAAAAAD+/wAAAAAAAPf+/QCC3dsA9QQFAFlcXAAaGhoACwsLAAYGBgABAQEA/v7+ADg4OADk5OQAqKmnAMSa6QA/bBoAAAYDAAMDAwAAAAEAAAAAAAAAAAAAAQEAAQIBAPPt+gDXveoA3s/vAPLr/AD++vsAAgkJAAMKCQDn5+cA9/f3APr6+gD9/f0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMDAwAZWVlAAQEBAAAAAAAAAAAAAAAAAAAAAAADQ0NAEJCQgCvr68AAAAAAAAAAAAAAAAAAAAAAAAAAADu7u4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAc3NzAOfn5wDm5uYA5+fnAObm5gDm5uYARkZGAAAAAAAAAAAAAAAAAAAAAAAAAAAAKCgoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPLy8gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANzc3ADDw8MAPj4+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABYWFgAQEBAAAAAAAAAAAAABAQEAAkJCQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////APz8/ADj398AOAsLAP7//wAA//8A/wAAAAEAAAAA/v8AAAD/AAAAAAD///8AAAAAAP7//wAAAAAAAAAAAAAAAACj5OQA8P8AAFZVVQAbHBsAEhESAA4ODgAGBgYAAQEBAAAAAAD7+/sAExMTABMTEwCRkY4A1bbuADBQFgD+/wAAAgD/AP37AADawewA4M7xAO3m+wD59fkAAgMDAA0oJgAXQS4ADS0rAAgfHQDknqMA8NbaAPr6+gD7+/sA////AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/AAcHBwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlJSUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9fX1AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA4ODgAODg4AJycnAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAoKCgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD5+fkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAArq6uAC8vLwAPDw8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAATExMAPj4+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP39/QD8/fwA6+vrABscHQAAAAAA/wD/AP/+AAD/AAAA/wD/AP7//wAAAAAAAAAAAAD//wAAAAAA////AAIBAADs+voAvu3sAEdISAAZGRkAFhUWABISEgAKCgoABQUFAAEBAQD///8A/v7+AAsLCwAFBQUASkpNAK2tqwAOGQYA6NX3AOvi+QD59PwA//7+AAkdHQARQy4ACy4rAAslJAACBwYAAAAAAAAAAAAAAAAACiAgAO7GygD8/PwA/f39AP39/QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACQkJAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAERERAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA6enpANLS0gDKysoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/Pz8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANDQ0AMzMzAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAPf39wCnp6cAaGhoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMzMzAAICAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP7+/gD///8A9PP0APX5+QAVCAgAAP//AAD+AAAAAP8AAAAAAAD//wAAAAAAAP//AAAAAAD///8AAAAAAP///gD9AAAAxu/vAA4MDAArKysAERERABAQEAANDQ0ABwcHAAMDAwAAAAAAAAAAAP7+/gADAwMA////ADs7OwCssK8AzKnrAAQQDgANMDYAFywqAAopJgAFDQ0AAAABAAEDAwAAAQAAAAAAAAAAAAAAAAAAAAAAAAYTEQD49/cA////AP7+/gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAASEhIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAkJCQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD9/f0A5eXlANzc3ADe3t4A29vbAPDw8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKCgoAcHBwABkZGQDc3NwA5OTkAAAAAAAAAAAAAAAAAAAAAAAAAAAAz8/PANHR0QAvLy8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAgIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////APX19QAAAAAACQABAP7+/gAAAAAA////AP4AAAAAAAAAAf/+AP7/AAAAAAAA/wD+AAEAAgAAAAAA////AN/39wAUFxcACgoKAAwMDAAKCgoABgYGAAMDAwABAQEAAAAAAAAAAAADAwMA////AAcHBwALCwsAHRscAO8TEQAdYl4AAQQDAAAAAAAAAQAAAP7+AP/+/gAAAP8AAAEBAAEAAAD/AgIAAAAAAAECAQADCgoA/v39APn5+QD///8A/v7+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFBQUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////ANXV1QDJyckA09PTAPf39wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGBgYAbm5uABwcHAAAAAAAXV1dAIiIiAD///8AAAAAAAAAAAAAAAAAAAAAAAAAAACmpqYAJycnAAgICAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAASEhIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD4+PgAAAAAAAcDAwAAAAAA/f//AAD//wAAAAAAAAAAAAAAAAAAAAAA///+AP8A/wD///0AAAAAAP///wDz//4ACQkJAAYGBgAGBgYABQUFAAEBAQABAQEAAAAAAAAAAAAAAAAAAAAAAAEBAQAEBAQACAgIABQSEgDw+/oA/v79AAD6/QACAgMAAAAAAP//AAABAAAAAPz+AP8AAAAA/wAAAQQAAAAEAwD//v8AAgcIAAAAAAD5+fkA////AP///wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAcHBwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwMDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA6enpAJ2dnQDh4eEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQAa2trACEhIQAAAAAAAAAAACAgIABCQkIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA8PDwAK+vrwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABwcHAAAAAAD5/fwA/wD/AAAAAAAAAAAA////AP///wD9/v4A/v/9AP///wAA/wAA//8AAP//AAAAAAAADAEBAPn5+QABAQEAAAAAAAAAAAABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAACAgIAAQEBAAICAgD2+PgAAAgHAAD+/QAA/P0A/v39AP7+/QAA/P0A//z9AP/+/QD//v0A//z9AP/8/QAA+/0AAf38AP34+AAAAAAABgYGAP7+/gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA4uLiAIWFhQDh4eEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAQEAZmZmACgoKAAAAAAAAAAAAAAAAAAAAAAATk5OABISEgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAwMAA39/fAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAkJCQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsLCwAAAAAA9vz9AP7//wAA//8A////AP///gD///4A/wD/AAD/AAD/AAAA/wAAAP7//wD+//4A/v//AB8HCADr6OgA+/z7APz8/AD9/f0A////AP///wAAAAAAAAAAAAAAAAAAAAAA/v7+AP7+/gD+/v4A4ePjAAkdHAAA/P4AAP79AAH//QAA/v0A/v79AP7+/QD+/P0AAQAAAAD+/QAA/v0AAP79AP8AAAD99fUAAQICAAcHBwACAgIAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgICAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA+fn5AJqamgD5+fkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQEBAAnJycAUFBQAH5+fgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAYmJiAC8vLwAAAAAAAAAAAAAAAAAAAAAAAAAAAAgICABbW1sAk5OTAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKKiogB8fHwAAwMDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGBgYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQAKCgoACwcHAOj5+QAgCQkAAQD/AP8AAAAAAAAAAAAAAP//AAAA/wAA/v//AAAAAAAAAP8A//8AAAEBAAA1Dg0AqunoANTT1AAVFBUA+fn5AAUFBQABAQEAAgICAAEBAQAAAAAA/f39AP39/QD4+PgAv7+/APj19QAbKSgA//z8AAAAAAD//wAAAQEBAAEEAAAAAAEAAAABAP/8AAAAAAAAAAAAAP/+AAABBAIA++zrAAoLCwACAgIAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMzMzADr6+sAAAAAAAAAAAAAAAAAAAAAgelArwAAIABJREFUAAAAAAAAAAAAMjIyADs7OwAnJycABQUFAAAAAAD///8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFWSqm1AgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANjY2AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsLCwANTU1AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAObm5gC5ubkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgICAAEBAQACAgIAFBQUAOX49wD///8A/f7+AP7//QD+//8A/v//AAAAAAAAAP8AAAAAAAAA/wD///8A///+AP7//wASBAUAPhARALy5uADo6OgA8vLyAPf39wD6+voA/Pz8APv7+wD7+/sA+/v7APb29gDc3NwApaioABA3NAABBwYA//79AP/+/gAA/f0A//z8AP78/QD//PwA//78AP7+/QAAAAAA//z9AP/+/QAA/P4A9eDhABISEgAFBQUABAQEAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMrKygAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABLS0sAZ2dnACwsLAAFBQUAAAAAAAAAAAD7+/sAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFZWVgAKCgoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAtbW1AOnp6QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgICAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAgIABAQEABoeHgDY9vUAAAAAAP8A/wAA/wAAAQAAAAAAAAAA//8A/v//AP///wD+//8AAAAAAAAAAAD9/v4A/v//AFcWFgAN/v4Am5eXANzc3ADv7+8A8vPyAPPz8wD19fUA9fX1APHy8QDU1NQAfn5/APf29gAPOjYAAf7/AAD8/gD//PwA//39AP/+/gAA/P4A//8AAP/7/QAA/v0A/v79AAAAAAABAAAAAAIAAPzf4gAWFBQABgYGAAMDAwADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADo6OgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQ0NDAAAAAAAAAAAAAAAAAAAAAAAAAAAA8/PzAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAODg4AVFRUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/AClpaUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQEBAAQEBAArKysA2fPzAPr9/AABAAEAAAEAAP///wD+/v8A/v8AAP//AAD///4A//7/AP7//wD9//8AAP//AAD//wAGAQEAdBocAA739gCLiYkAmpqaAMTDxADb29sA2traAL+/vwCPjo8Afn5/ABEFHQACAgQA5KKoAPz19AD+/fwA//39AP/9/QAA/P0AAP79AAD9/QAA/f0AAPz+AAH+/gD+/v0A//z9APvw8gDy3+AAHx8fAAYGBgAFBQUAAgICAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA8/PzAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/AAAAAAAAAAAAAAAAAAAAAAAAAAAANPT0wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/f39AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADs7OwDJyckA2dnZAAAAAAAAAAAAAAAAAAAAAAAAAAAA2dnZAMXFxQA7OzsAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQEBAAEBAQADAwMAMjIyAKKfnwDO8/IANQ4PAAD//wAAAAAAAAD/AAAAAAAAAP8A/wAAAAAA/wAA/wEAAAAAAAAAAAAAAAAA////AAoBAQB1Gx0AyfPzAL7V1ACTyckAAQEBAP///wACAQIAFQsfADwaYQAgDzAA8vvqALLeiQDinqQAHmNdAAIHBwABAQIA/v/9AAEA/wABAwMAAAMDAAEA/wD//PwAAAIDAP4EAADxzM0ABwkJACQkJAAHBwcABgYGAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAcHBwD29vYAAAAAAAAAAAAAAAAAAAAAAAAAAADLy8sAREREAAAAAAAAAAAAAAAAAOvr6wC0tLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/AADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAW1tbAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACsrKwA8vLyAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAQEABAQEAA8PDwAnKioAwe/vAAD/AAAAAAAA/v8AAP8AAAD+/v4A/v//AP7//wD+//8A/v/+AAD+/wD+/v4A/v7+AP///wD///8AAAAAADYMDAB2GRsAkCMkACwJCQANBhQAVSWJAEYecwAhEDIAAgIDABEJGwBSJXsA/O7xAOKdogD99fYA/v37AAD+/gAB/v8A//39AAD+/QD//P4AAAAAAAECAAAAAAAA8cjMACEdHgAMDAwACAgIAAUFBQABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAaGhoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAr6+vAMDAwAD09PQA9PT0AMTExACIiIgA8/PzAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD29vYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABcXFwBMTEwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9vb2AKqqqgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgICAAMDAwAHBwcAREVEANDx8QDj+PgA/v//AP///gD//v8AAAAAAAD//wAA//8AAAAAAAAAAAD/AP8AAAEAAP8AAAD/AP8A/gAAAP3//gD+//8AAP//AAgBAQAGAgIAAgEDAAUDCAABAQIAAQEBAAMCAwAEAgEAEgoaAFUofQD97vIA4qGlAP319QAAAAAAAAAAAP/9/QD//P0A//38AP/8/QD++/0A9tvcAPbe3wA2NjYACwsLAAkJCQAEBAQAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALy8vAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPz8/ACnp6cAc3NzAHNzcwCjo6MA9PT0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA7+/vAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAARkZGABsbGwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADLy8sA1NTUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQACAgIABgYGAC8uLwAgHR0AsOvqAP//AAD///8A////AP//AAD+AAAA////AP///wD///8A////AP/+/wD+/v4A/v7/AAD//wAB//8AAQD/AP///wD+/v4AAAAAAAABAAADAwMAAgIBAAECAgABAgAA/wAAAAEAAQAUCx0AVyuAAP7u8gDioacA/fX2AP79/AACAQIAAQABAAABAAAAAAAAAQEBAOy3uwAfISEAHx8fAAwMDAAHBwcABAQEAAICAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD4+PgANDQ0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQEBAAcHBwAAAAAAAAAAAAAAAAAAAAAAAAAAAOHh4QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMDAwBdXV0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKampgD5+fkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwMDAAUFBQAKCgoAUVRUAMrx8ADX9fQAAAEBAAABAAD+AP8AAP//AAEBAQAAAAAA////AP8A/wAAAAAAAAAAAP///wD///8A/v//AP7//wD+/v4A////AAD/AAABAAEABAMDAAMCAwAEAgEAAgIDAAICAwABAwIA/gAAABIKHABbLoQA/u7yAOOhpQD99vYA/v37AP79/AD//P0A//39APLP0QD1290AOzk5AA4ODgALCwsABwcHAAEBAQABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAICAgAZWVlAAICAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABgYGAGxsbAAyMjIAAAAAAAAAAAAAAAAAAAAAAAAAAADMzMwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAISEhAEJCQgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADt7e0AsrKyAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQACAgIADExMQAzMDAAnunoAPj9/QAAAAAAAAAAAAAAAAD///8A////AP///wD+/v8A/v//AAH//wD///8A////AP///wD///8AAAEAAAABAAAAAP8AAAAAAAECAQABAgAAAQIDAAIBAQACAQEAAQECAAMCAQACAgEAEgsaAFsvhQD97/IA46ClAP75+gACAQIAAAAAAP349gDqrrEALi0tACMjIwAPDw8ACQkJAAUFBQADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACcnJwBxcXEADw8PAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKCgoAHp6egAsLCwAIyMjAAAAAAAAAAAAAAAAAAAAAAAAAAAA3NzcAOPj4wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABPT08AERERAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAL29vQDh4eEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAd8jdrAAAgAElEQVQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAQEAAQEBAAcHBwALCwsAW15eAP0GBgCe5+cA/v7+AP/+/gD+/v4A////AP8AAAD+AP8AAAAAAAEA/wD+AAAAAAAAAAAAAAD///8A/v7/AP/+/wD//v8A/wAAAAEBAQACAgMABQQFAAMCAgABAwMAAQMDAAMCAQACAQIAAgEBAAIBAgAWDR4AYDKHAP3v8gDinaMA/Pf3AP/+/wDop60ACQUGAENEQwAREREADQ0NAAgICAADAwMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJiYmAISEhABZWVkAKysrAA8PDwADAwMACQkJABkZGQA+Pj4AcnJyAHFxcQAZGRkAAAAAACcnJwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACtra0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH/////AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wD9/f0A+vr6APHx8QDLzMsAlZGSAEQODwBSFhYA/v7+AP///wD+AP8AAP8AAP///wAAAP8A//8AAP///gD//wAAAAEAAP7//wAA/wAAAAD/AKDr6wDfAA8AWC2IAAD/AAAAAAAAAwIBAAICAgAAAQAAAgIBAAECAwADAgIAAQEAAAICAwDv9+UAncp1AAMSDwAJGhgA+NjbAFNRUQAsLCwAGhoaABEREQAKCgoABAQEAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAQEAAwMDAAUFBQALCwsALy4vAFhbWwC47+4As+rqAAAAAAAAAAAAAAABAAEA/wAAAAAA////AP///wD//wAA/v//AP7//wAA//8AAAD/AAAAAAD/AAAAAAAAAAQCAgADAwEABQUDAAIDAgACAwEAAgEBAAEBAwADAQIAAgEAAAQCAwADAgIAFQwfAGQ5jAD77O4A8tLVAEE/PwAiIiIAEBAQAAwMDAAHBwcAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBAQACAgIACAgIAA0NDQA/Pz8A+/v7AKqpqQDw/f0AVxUVAAIA/wAAAAAAAAAAAP//AAD///4AAAABAAIBAQD+//4A////AP///wAAAAAAAAAAAAEAAAABAQEAAwMCAAAAAAAA/wAAAgECAAIDAwAAAAAAAQIAAAECAwD/AgIAAgECAP0C/ADH3rAAAwECADMzMQAsLCwAExMTABEREQAKCgoABQUFAAMDAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwMDAAAAAAAKCgoA////AENDQwALCwsApaKiAOPm5QBiFRQAEgMFAP8A/wAAAAAAAAEBAP7+/wAAAP8A//8AAAAAAAAAAAAAAAAAAAD//wAAAgEAAwQEAAEA/wAAAAAAAgMCAAAAAAACAwIAAAAAAAMBAQABAgIAAgECAPD56ACu0IkA8vvqAEtLSQAnJycAFRUVABQUFAAMDAwABQUFAAICAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAgIABQUFAAsLCwABAQEAOTk5ABYWFgCinp4AguPkAEoREQBBDg4AAv8AAP4A/wAAAP8A////AAAAAAD///8A/v//AAD//wD/AAAAAQABAAMDAgAA/wAAAAAAAAICAgAAAAAAAQEBAAECAgD///8A/v/7AMTerwDG3qwAGBkXAFlKWQAbHBsAEhESABEREQAPDw8ACAgIAAQEBAABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQEBAAICAgAFBQUACgoKAAICAgAEBAQAdHh4ALi4uAC6v78AwPT0AEMMDAA3CwwABgD/AP///wAAAAAA////AAAAAAD+AAAAAAAAAAABAAACAgIAAQEBAAAAAAACAAAA/wICAAIAAAD5/fYAz+S6AMnhsAABA/4AOz46ADo5OgAREREAERERAA4PDgAKCQoACAgIAAICAgABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAQEAAgICAAUFBQACAgIAAgICAAUFBQBPT08ANzg3AL28vQC4zMwAsO7uAC0KCgAoBwcAJggIAA8BAgD+//4AAAAAAAD/AAAAAAAABAMDAAAAAAAAAAAA7/boAOLw1QDb68oA3uvPAAcHCAAtMCsANzc3ABUVFQAQEBAADg4OAAwMDAAJCQkABQUFAAMDAwABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB/////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA////APz8/AD7+/sA9/f3APX19QD09PQA7+/vAM7OzgDS0dEA5uTkAPb5+QAHAQEAFwcIAAsA/gDl+vsA+AEHABwRKQD2+e8A7vfoAPr89wALCgwAGRoZACYmJQArKysADg4OAA0NDQANDQ0ACwwLAAoJCgAHBwcAAwMDAAICAgABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAf////8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/v7+AP39/QD5+fkA+fn5APf39wD19fUA9/f3APb29gDi4uIA5OTkAOvr6wD29vYA7e3tAAAAAAAQEBAACAgIABYWFgAaGhoAGhoaAAgICAAICAgACgoKAAsLCwAJCgkACAcIAAYGBgAFBQUAAwMDAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAJItZQAAAArMSURBVAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAH/////AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/f39APz8/AD9/f0A+fn5APr6+gD6+voA+fn5APr6+gD6+voA+fn5APj5+AD9/P0AAgICAAYGBgAHBwcABgYGAAYGBgAICAgABgYGAAcHBwAGBgYABQUFAAMDAwACAgIAAgICAAEBAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB/////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP///wAAAAAA/v7+AP7+/gD8/PwA/v7+APz8/AD8/PwA/P38AP38/QD7+/sA/Pz8AAMDAwAFBQUAAgICAAUFBQAEBAQAAwMDAAQEBAADAwMAAgICAAEBAQACAgIAAQEBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAf////8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD///8A/v7+AAAAAAD///8A////AP///wD9/f0A/v7+AAAAAAAAAAAAAQEBAAEBAQADAwMAAQEBAAEBAQABAQEAAgICAAAAAAABAQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACanuzMvN2MUAAAAAElFTkSuQmCC"  style="display:none"/>
        <noscript>Javascript must be enabled to view this page.</noscript>
        <div style="display:none"><krona>
            <attributes magnitude="magnitude">
                <attribute display="grams">magnitude</attribute>
            </attributes>
<node name="COGs"><magnitude>
<val> 797362.631071001</val>
</magnitude>
<node name="CELLULAR PROCESSES AND SIGNALING"><magnitude>
<val> 185233.753382</val>
</magnitude>
<node name="Cell cycle control, cell division, chromosome partitioning"><magnitude>
<val> 10862.577659</val>
</magnitude>
<node name="ATPases involved in chromosome partitioning"><magnitude>
<val> 1297.970109</val>
</magnitude>
</node>
<node name="Actin-like ATPase involved in cell division"><magnitude>
<val> 276.692397</val>
</magnitude>
</node>
<node name="Actin-like ATPase involved in cell morphogenesis"><magnitude>
<val> 336.074129</val>
</magnitude>
</node>
<node name="Alpha-tubulin suppressor and related RCC1 domain-containing proteins"><magnitude>
<val> 122.46007</val>
</magnitude>
</node>
<node name="Anaphase-promoting complex (APC), subunit 10"><magnitude>
<val> 52.868164</val>
</magnitude>
</node>
<node name="Antitoxin of toxin-antitoxin stability system"><magnitude>
<val> 251.857519</val>
</magnitude>
</node>
<node name="Bacterial cell division membrane protein"><magnitude>
<val> 357.939589</val>
</magnitude>
</node>
<node name="CAS/CSE protein involved in chromosome segregation"><magnitude>
<val> 146.86713</val>
</magnitude>
</node>
<node name="CO dehydrogenase maturation factor"><magnitude>
<val> 330.367491</val>
</magnitude>
</node>
<node name="Cdk activating kinase (CAK)/RNA polymerase II transcription initiation/nucleotide excision repair factor TFIIH, subunit TFB3"><magnitude>
<val> 4.897582</val>
</magnitude>
</node>
<node name="Cdk activating kinase (CAK)/RNA polymerase II transcription initiation/nucleotide excision repair factor TFIIH/TFIIK, cyclin H subunit"><magnitude>
<val> 8.347093</val>
</magnitude>
</node>
<node name="Cell cycle control protein"><magnitude>
<val> 113.109572</val>
</magnitude>
</node>
<node name="Cell division GTPase"><magnitude>
<val> 354.907098</val>
</magnitude>
</node>
<node name="Cell division control protein, negative regulator of transcription"><magnitude>
<val> 291.04188</val>
</magnitude>
</node>
<node name="Cell division initiation protein"><magnitude>
<val> 194.036388</val>
</magnitude>
</node>
<node name="Cell division protein"><magnitude>
<val> 706.150983</val>
</magnitude>
</node>
<node name="Chromosome segregation ATPases"><magnitude>
<val> 1650.49033</val>
</magnitude>
</node>
<node name="Cohesin"><magnitude>
<val> 4.945362</val>
</magnitude>
</node>
<node name="Cyclin"><magnitude>
<val> 49.458886</val>
</magnitude>
</node>
<node name="DNA segregation ATPase FtsK/SpoIIIE and related proteins"><magnitude>
<val> 250.255752</val>
</magnitude>
</node>
<node name="Dynactin complex subunit involved in mitotic spindle partitioning in anaphase B"><magnitude>
<val> 80.254782</val>
</magnitude>
</node>
<node name="Integral membrane protein possibly involved in chromosome condensation"><magnitude>
<val> 129.031088</val>
</magnitude>
</node>
<node name="Intracellular septation protein A"><magnitude>
<val> 70.40629</val>
</magnitude>
</node>
<node name="Membrane-bound metallopeptidase"><magnitude>
<val> 280.872191</val>
</magnitude>
</node>
<node name="Microtubule-binding protein involved in cell cycle control"><magnitude>
<val> 8.900037</val>
</magnitude>
</node>
<node name="Mitotic inducer, protein phosphatase"><magnitude>
<val> 60.778756</val>
</magnitude>
</node>
<node name="NAD/FAD-utilizing enzyme apparently involved in cell division"><magnitude>
<val> 350.749422</val>
</magnitude>
</node>
<node name="Negative regulator of septation ring formation"><magnitude>
<val> 148.582809</val>
</magnitude>
</node>
<node name="Nucleotide-binding protein implicated in inhibition of septum formation"><magnitude>
<val> 229.035662</val>
</magnitude>
</node>
<node name="Phosphotyrosyl phosphatase activator"><magnitude>
<val> 12.619263</val>
</magnitude>
</node>
<node name="Predicted ATPase involved in cell division"><magnitude>
<val> 174.479568</val>
</magnitude>
</node>
<node name="Predicted ATPase of the PP-loop superfamily implicated in cell cycle control"><magnitude>
<val> 534.25139</val>
</magnitude>
</node>
<node name="Protein involved in cell division"><magnitude>
<val> 131.969591</val>
</magnitude>
</node>
<node name="Protein involved in chromosome segregation, interacts with SMC proteins"><magnitude>
<val> 137.610164</val>
</magnitude>
</node>
<node name="Protein involved in regulation of cellular morphogenesis/cytokinesis"><magnitude>
<val> 51.935486</val>
</magnitude>
</node>
<node name="Protein predicted to be involved in meiotic recombination"><magnitude>
<val> 43.181272</val>
</magnitude>
</node>
<node name="Protein required for the initiation of cell division"><magnitude>
<val> 19.784785</val>
</magnitude>
</node>
<node name="Regulator of cell morphogenesis and NO signaling"><magnitude>
<val> 176.777252</val>
</magnitude>
</node>
<node name="SOS-response cell division inhibitor, blocks FtsZ ring formation"><magnitude>
<val> 8.99137</val>
</magnitude>
</node>
<node name="Separase, a protease involved in sister chromatid separation"><magnitude>
<val> 94.536998</val>
</magnitude>
</node>
<node name="Septin family protein"><magnitude>
<val> 151.643009</val>
</magnitude>
</node>
<node name="Septum formation inhibitor"><magnitude>
<val> 172.943829</val>
</magnitude>
</node>
<node name="Septum formation inhibitor-activating ATPase"><magnitude>
<val> 91.943447</val>
</magnitude>
</node>
<node name="Septum formation initiator"><magnitude>
<val> 140.035003</val>
</magnitude>
</node>
<node name="Septum formation topological specificity factor"><magnitude>
<val> 60.722974</val>
</magnitude>
</node>
<node name="Serine/threonine kinase of the haspin family"><magnitude>
<val> 6.19236</val>
</magnitude>
</node>
<node name="Sporulation protein and related proteins"><magnitude>
<val> 206.104599</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in chromosome partitioning"><magnitude>
<val> 411.126985</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in cytokinesis, contains TGc (transglutaminase/protease-like) domain"><magnitude>
<val> 76.379753</val>
</magnitude>
</node>
</node>
<node name="Cell motility"><magnitude>
<val> 19595.321808</val>
</magnitude>
<node name="ABC-type uncharacterized transport system involved in gliding motility, auxiliary component"><magnitude>
<val> 140.946494</val>
</magnitude>
</node>
<node name="Archaeal flagella assembly protein J"><magnitude>
<val> 62.275069</val>
</magnitude>
</node>
<node name="Archaeal flagellins"><magnitude>
<val> 64.728381</val>
</magnitude>
</node>
<node name="Autotransporter protein or domain, integral membrane beta-barrel involved in protein secretion"><magnitude>
<val> 9.251402</val>
</magnitude>
</node>
<node name="Bacterial surface proteins containing Ig-like domains"><magnitude>
<val> 462.469023</val>
</magnitude>
</node>
<node name="Chemotaxis protein"><magnitude>
<val> 53.026297</val>
</magnitude>
</node>
<node name="Chemotaxis protein CheC, inhibitor of MCP methylation"><magnitude>
<val> 142.632853</val>
</magnitude>
</node>
<node name="Chemotaxis protein histidine kinase and related kinases"><magnitude>
<val> 608.175765</val>
</magnitude>
</node>
<node name="Chemotaxis protein; stimulates methylation of MCP proteins"><magnitude>
<val> 185.040893</val>
</magnitude>
</node>
<node name="Chemotaxis response regulator containing a CheY-like receiver domain and a methylesterase domain"><magnitude>
<val> 523.982553</val>
</magnitude>
</node>
<node name="Chemotaxis signal transduction protein"><magnitude>
<val> 766.818807</val>
</magnitude>
</node>
<node name="Conserved protein implicated in secretion"><magnitude>
<val> 24.381532</val>
</magnitude>
</node>
<node name="Cytolysin, a secreted calcineurin-like phosphatase"><magnitude>
<val> 23.681287</val>
</magnitude>
</node>
<node name="Flagellar GTP-binding protein"><magnitude>
<val> 108.735237</val>
</magnitude>
</node>
<node name="Flagellar basal body L-ring protein"><magnitude>
<val> 118.852043</val>
</magnitude>
</node>
<node name="Flagellar basal body P-ring biosynthesis protein"><magnitude>
<val> 97.661065</val>
</magnitude>
</node>
<node name="Flagellar basal body protein"><magnitude>
<val> 124.536723</val>
</magnitude>
</node>
<node name="Flagellar basal body rod protein"><magnitude>
<val> 428.676702</val>
</magnitude>
</node>
<node name="Flagellar basal body-associated protein"><magnitude>
<val> 186.743769</val>
</magnitude>
</node>
<node name="Flagellar basal-body P-ring protein"><magnitude>
<val> 107.418435</val>
</magnitude>
</node>
<node name="Flagellar biogenesis protein"><magnitude>
<val> 118.583927</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis chaperone"><magnitude>
<val> 97.177778</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis pathway, component FlhA"><magnitude>
<val> 266.901928</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis pathway, component FlhB"><magnitude>
<val> 327.207006</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis pathway, component FliP"><magnitude>
<val> 173.437992</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis pathway, component FliQ"><magnitude>
<val> 130.974168</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis pathway, component FliR"><magnitude>
<val> 196.656602</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis regulator FlaF"><magnitude>
<val> 7.549118</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis regulator FlbT"><magnitude>
<val> 34.380986</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis/type III secretory pathway ATPase"><magnitude>
<val> 193.183024</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis/type III secretory pathway chaperone"><magnitude>
<val> 38.689705</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis/type III secretory pathway lipoprotein"><magnitude>
<val> 267.525706</val>
</magnitude>
</node>
<node name="Flagellar biosynthesis/type III secretory pathway protein"><magnitude>
<val> 188.001154</val>
</magnitude>
</node>
<node name="Flagellar capping protein"><magnitude>
<val> 218.827862</val>
</magnitude>
</node>
<node name="Flagellar hook capping protein"><magnitude>
<val> 146.301992</val>
</magnitude>
</node>
<node name="Flagellar hook protein FlgE"><magnitude>
<val> 121.956333</val>
</magnitude>
</node>
<node name="Flagellar hook-associated protein"><magnitude>
<val> 182.873939</val>
</magnitude>
</node>
<node name="Flagellar hook-basal body protein"><magnitude>
<val> 158.295107</val>
</magnitude>
</node>
<node name="Flagellar hook-length control protein"><magnitude>
<val> 168.129614</val>
</magnitude>
</node>
<node name="Flagellar motor component"><magnitude>
<val> 311.011856</val>
</magnitude>
</node>
<node name="Flagellar motor protein"><magnitude>
<val> 344.827697</val>
</magnitude>
</node>
<node name="Flagellar motor switch protein"><magnitude>
<val> 260.225368</val>
</magnitude>
</node>
<node name="Flagellar motor switch/type III secretory pathway protein"><magnitude>
<val> 132.622249</val>
</magnitude>
</node>
<node name="Flagellin and related hook-associated proteins"><magnitude>
<val> 392.056295</val>
</magnitude>
</node>
<node name="Flagellin-specific chaperone FliS"><magnitude>
<val> 113.481555</val>
</magnitude>
</node>
<node name="Flp pilus assembly protein TadC"><magnitude>
<val> 102.201462</val>
</magnitude>
</node>
<node name="Methyl-accepting chemotaxis protein"><magnitude>
<val> 3335.091024</val>
</magnitude>
</node>
<node name="Methylase of chemotaxis methyl-accepting proteins"><magnitude>
<val> 371.609143</val>
</magnitude>
</node>
<node name="Muramidase (flagellum-specific)"><magnitude>
<val> 33.042823</val>
</magnitude>
</node>
<node name="P pilus assembly protein, chaperone PapD"><magnitude>
<val> 87.636497</val>
</magnitude>
</node>
<node name="P pilus assembly protein, pilin FimA"><magnitude>
<val> 53.631949</val>
</magnitude>
</node>
<node name="P pilus assembly protein, porin PapC"><magnitude>
<val> 245.490158</val>
</magnitude>
</node>
<node name="PPE-repeat proteins"><magnitude>
<val> 77.417813</val>
</magnitude>
</node>
<node name="Predicted ATPases involved in biogenesis of archaeal flagella"><magnitude>
<val> 57.992729</val>
</magnitude>
</node>
<node name="Predicted inhibitor of MCP methylation, homolog of CheC"><magnitude>
<val> 284.867519</val>
</magnitude>
</node>
<node name="Predicted periplasmic or secreted lipoprotein"><magnitude>
<val> 289.803174</val>
</magnitude>
</node>
<node name="Predicted small periplasmic lipoprotein"><magnitude>
<val> 80.442011</val>
</magnitude>
</node>
<node name="Putative archaeal flagellar protein C"><magnitude>
<val> 4.091755</val>
</magnitude>
</node>
<node name="Putative archaeal flagellar protein D/E"><magnitude>
<val> 5.870221</val>
</magnitude>
</node>
<node name="Putative archaeal flagellar protein F"><magnitude>
<val> 28.904122</val>
</magnitude>
</node>
<node name="Putative archaeal flagellar protein G"><magnitude>
<val> 86.83657</val>
</magnitude>
</node>
<node name="Subunit of TRAPP, an ER-Golgi tethering complex"><magnitude>
<val> 1.478448</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein FimT"><magnitude>
<val> 182.224528</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein FimV"><magnitude>
<val> 200.853074</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein PilE"><magnitude>
<val> 170.63081</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein PilF"><magnitude>
<val> 911.583404</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein PilN"><magnitude>
<val> 184.941414</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein PilO"><magnitude>
<val> 164.000474</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein PilP"><magnitude>
<val> 129.875409</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein PilV"><magnitude>
<val> 89.380991</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein PilW"><magnitude>
<val> 55.46348</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein PilX"><magnitude>
<val> 121.812593</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein PilZ"><magnitude>
<val> 36.985928</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein, ATPase PilM"><magnitude>
<val> 237.148663</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein, ATPase PilU"><magnitude>
<val> 17.588587</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein, major pilin PilA"><magnitude>
<val> 13.136097</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein, pilus retraction ATPase PilT"><magnitude>
<val> 495.407471</val>
</magnitude>
</node>
<node name="Tfp pilus assembly protein, tip-associated adhesin PilY1"><magnitude>
<val> 313.561455</val>
</magnitude>
</node>
<node name="Type II secretory pathway, ATPase PulE/Tfp pilus assembly pathway, ATPase PilB"><magnitude>
<val> 615.201138</val>
</magnitude>
</node>
<node name="Type II secretory pathway, component PulD"><magnitude>
<val> 243.464447</val>
</magnitude>
</node>
<node name="Type II secretory pathway, component PulF"><magnitude>
<val> 429.920408</val>
</magnitude>
</node>
<node name="Type II secretory pathway, prepilin signal peptidase PulO and related peptidases"><magnitude>
<val> 235.878942</val>
</magnitude>
</node>
<node name="Type II secretory pathway, pseudopilin PulG"><magnitude>
<val> 395.036683</val>
</magnitude>
</node>
<node name="Type IV pili component"><magnitude>
<val> 86.288652</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, TrbD component"><magnitude>
<val> 4.266154</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, VirB11 components, and related ATPases involved in archaeal flagella biosynthesis"><magnitude>
<val> 92.571423</val>
</magnitude>
</node>
<node name="Uncharacterized flagellar protein FlaG"><magnitude>
<val> 93.447371</val>
</magnitude>
</node>
<node name="Uncharacterized protein, possibly involved in motility"><magnitude>
<val> 97.331503</val>
</magnitude>
</node>
</node>
<node name="Cell wall/membrane/envelope biogenesis"><magnitude>
<val> 51425.473386</val>
</magnitude>
<node name="3-deoxy-D-manno-octulosonic acid (KDO) 8-phosphate synthase"><magnitude>
<val> 215.661459</val>
</magnitude>
</node>
<node name="3-deoxy-D-manno-octulosonic-acid transferase"><magnitude>
<val> 158.925576</val>
</magnitude>
</node>
<node name="4-amino-4-deoxy-L-arabinose transferase and related glycosyltransferases of PMT family"><magnitude>
<val> 840.030792</val>
</magnitude>
</node>
<node name="ABC-type transport system, involved in lipoprotein release, permease component"><magnitude>
<val> 1240.153652</val>
</magnitude>
</node>
<node name="ADP-heptose synthase, bifunctional sugar kinase/adenylyltransferase"><magnitude>
<val> 209.947625</val>
</magnitude>
</node>
<node name="ADP-heptose:LPS heptosyltransferase"><magnitude>
<val> 418.934596</val>
</magnitude>
</node>
<node name="Acyl-[acyl carrier protein]--UDP-N-acetylglucosamine O-acyltransferase"><magnitude>
<val> 198.724285</val>
</magnitude>
</node>
<node name="Alanine racemase"><magnitude>
<val> 297.806562</val>
</magnitude>
</node>
<node name="Alpha-N-acetylglucosamine transferase"><magnitude>
<val> 20.57763</val>
</magnitude>
</node>
<node name="Apolipoprotein N-acyltransferase"><magnitude>
<val> 232.126322</val>
</magnitude>
</node>
<node name="Aspartate racemase"><magnitude>
<val> 121.755663</val>
</magnitude>
</node>
<node name="Bacterial lipocalin"><magnitude>
<val> 123.446743</val>
</magnitude>
</node>
<node name="CMP-2-keto-3-deoxyoctulosonic acid synthetase"><magnitude>
<val> 140.60521</val>
</magnitude>
</node>
<node name="CMP-N-acetylneuraminic acid synthetase"><magnitude>
<val> 46.846682</val>
</magnitude>
</node>
<node name="CTP:phosphocholine cytidylyltransferase involved in choline phosphorylation for cell surface LPS epitopes"><magnitude>
<val> 122.002585</val>
</magnitude>
</node>
<node name="Capsular polysaccharide biosynthesis protein"><magnitude>
<val> 63.993896</val>
</magnitude>
</node>
<node name="Capsule polysaccharide export protein"><magnitude>
<val> 320.568308</val>
</magnitude>
</node>
<node name="Carbohydrate-selective porin"><magnitude>
<val> 119.478341</val>
</magnitude>
</node>
<node name="Cell division protein FtsI/penicillin-binding protein 2"><magnitude>
<val> 706.674594</val>
</magnitude>
</node>
<node name="Cell division septal protein"><magnitude>
<val> 101.286767</val>
</magnitude>
</node>
<node name="Cell envelope opacity-associated protein A"><magnitude>
<val> 4.447152</val>
</magnitude>
</node>
<node name="Cell shape-determining protein"><magnitude>
<val> 260.395777</val>
</magnitude>
</node>
<node name="Cell wall hydrolyses involved in spore germination"><magnitude>
<val> 166.98961</val>
</magnitude>
</node>
<node name="Cell wall-associated hydrolases (invasion-associated proteins)"><magnitude>
<val> 371.051375</val>
</magnitude>
</node>
<node name="Chain length determinant protein"><magnitude>
<val> 50.347685</val>
</magnitude>
</node>
<node name="Choline-glycine betaine transporter"><magnitude>
<val> 187.340885</val>
</magnitude>
</node>
<node name="Cyclopropane fatty acid synthase and related methyltransferases"><magnitude>
<val> 463.554956</val>
</magnitude>
</node>
<node name="Cytidylyltransferase"><magnitude>
<val> 156.019295</val>
</magnitude>
</node>
<node name="D-alanine-D-alanine ligase and related ATP-grasp enzymes"><magnitude>
<val> 294.795988</val>
</magnitude>
</node>
<node name="D-alanyl-D-alanine carboxypeptidase"><magnitude>
<val> 257.870131</val>
</magnitude>
</node>
<node name="D-alanyl-D-alanine carboxypeptidase (penicillin-binding protein 4)"><magnitude>
<val> 98.598995</val>
</magnitude>
</node>
<node name="D-alanyl-D-alanine dipeptidase"><magnitude>
<val> 81.966266</val>
</magnitude>
</node>
<node name="Diacylglycerol kinase"><magnitude>
<val> 131.598675</val>
</magnitude>
</node>
<node name="Endopolygalacturonase"><magnitude>
<val> 897.167191</val>
</magnitude>
</node>
<node name="Exopolysaccharide biosynthesis protein"><magnitude>
<val> 7.78921</val>
</magnitude>
</node>
<node name="FOG: LysM repeat"><magnitude>
<val> 229.950248</val>
</magnitude>
</node>
<node name="GDP-D-mannose dehydratase"><magnitude>
<val> 303.175691</val>
</magnitude>
</node>
<node name="Glucosamine 6-phosphate synthetase, contains amidotransferase and phosphosugar isomerase domains"><magnitude>
<val> 462.704821</val>
</magnitude>
</node>
<node name="Glutamate racemase"><magnitude>
<val> 168.566878</val>
</magnitude>
</node>
<node name="Glycosyltransferase"><magnitude>
<val> 4186.166509</val>
</magnitude>
</node>
<node name="Glycosyltransferase involved in LPS biosynthesis"><magnitude>
<val> 29.184043</val>
</magnitude>
</node>
<node name="Glycosyltransferases involved in cell wall biogenesis"><magnitude>
<val> 2346.29452</val>
</magnitude>
</node>
<node name="Glycosyltransferases, probably involved in cell wall biogenesis"><magnitude>
<val> 801.984564</val>
</magnitude>
</node>
<node name="Integral membrane protein CcmA involved in cell shape determination"><magnitude>
<val> 61.260569</val>
</magnitude>
</node>
<node name="L-alanine-DL-glutamate epimerase and related enzymes of enolase superfamily"><magnitude>
<val> 143.912663</val>
</magnitude>
</node>
<node name="LPS biosynthesis protein"><magnitude>
<val> 42.875283</val>
</magnitude>
</node>
<node name="Large-conductance mechanosensitive channel"><magnitude>
<val> 145.391237</val>
</magnitude>
</node>
<node name="Lauroyl/myristoyl acyltransferase"><magnitude>
<val> 223.764907</val>
</magnitude>
</node>
<node name="Lipid A core - O-antigen ligase and related enzymes"><magnitude>
<val> 367.510176</val>
</magnitude>
</node>
<node name="Lipid A disaccharide synthetase"><magnitude>
<val> 263.162582</val>
</magnitude>
</node>
<node name="Lipopolysaccharide biosynthesis protein"><magnitude>
<val> 95.405819</val>
</magnitude>
</node>
<node name="Lipopolysaccharide biosynthesis proteins, LPS:glycosyltransferases"><magnitude>
<val> 52.08471</val>
</magnitude>
</node>
<node name="Lipoprotein signal peptidase"><magnitude>
<val> 151.902268</val>
</magnitude>
</node>
<node name="Lipoproteins"><magnitude>
<val> 112.642162</val>
</magnitude>
</node>
<node name="Lyzozyme M1 (1,4-beta-N-acetylmuramidase)"><magnitude>
<val> 101.536513</val>
</magnitude>
</node>
<node name="Mannose-1-phosphate guanylyltransferase"><magnitude>
<val> 268.493213</val>
</magnitude>
</node>
<node name="Mannosyltransferase OCH1 and related enzymes"><magnitude>
<val> 96.650926</val>
</magnitude>
</node>
<node name="Membrane GTPase LepA"><magnitude>
<val> 224.611003</val>
</magnitude>
</node>
<node name="Membrane carboxypeptidase (penicillin-binding protein)"><magnitude>
<val> 424.368661</val>
</magnitude>
</node>
<node name="Membrane carboxypeptidase/penicillin-binding protein"><magnitude>
<val> 320.709379</val>
</magnitude>
</node>
<node name="Membrane carboxypeptidase/penicillin-binding protein PbpC"><magnitude>
<val> 81.642962</val>
</magnitude>
</node>
<node name="Membrane glycosyltransferase"><magnitude>
<val> 154.289703</val>
</magnitude>
</node>
<node name="Membrane protein involved in colicin uptake"><magnitude>
<val> 216.851844</val>
</magnitude>
</node>
<node name="Membrane proteins related to metalloendopeptidases"><magnitude>
<val> 490.362457</val>
</magnitude>
</node>
<node name="Membrane-bound lytic murein transglycosylase"><magnitude>
<val> 104.017552</val>
</magnitude>
</node>
<node name="Membrane-bound lytic murein transglycosylase B"><magnitude>
<val> 71.454994</val>
</magnitude>
</node>
<node name="Membrane-fusion protein"><magnitude>
<val> 1617.224032</val>
</magnitude>
</node>
<node name="Murein endopeptidase"><magnitude>
<val> 80.037939</val>
</magnitude>
</node>
<node name="Murein lipoprotein"><magnitude>
<val> 101.586853</val>
</magnitude>
</node>
<node name="N-acetylglucosamine-1-phosphate uridyltransferase (contains nucleotidyltransferase and I-patch acetyltransferase domains)"><magnitude>
<val> 211.238418</val>
</magnitude>
</node>
<node name="N-acetylmuramoyl-L-alanine amidase"><magnitude>
<val> 378.56076</val>
</magnitude>
</node>
<node name="Nucleoside-binding outer membrane protein"><magnitude>
<val> 17.388447</val>
</magnitude>
</node>
<node name="Nucleoside-diphosphate-sugar epimerases"><magnitude>
<val> 1515.395477</val>
</magnitude>
</node>
<node name="Nucleoside-diphosphate-sugar pyrophosphorylase involved in lipopolysaccharide biosynthesis/translation initiation factor 2B, gamma/epsilon subunits (eIF-2Bgamma/eIF-2Bepsilon)"><magnitude>
<val> 385.207548</val>
</magnitude>
</node>
<node name="O-Glycosyl hydrolase"><magnitude>
<val> 243.395006</val>
</magnitude>
</node>
<node name="Opacity protein and related surface antigens"><magnitude>
<val> 366.518382</val>
</magnitude>
</node>
<node name="Organic solvent tolerance protein OstA"><magnitude>
<val> 304.973893</val>
</magnitude>
</node>
<node name="Outer membrane lipoprotein"><magnitude>
<val> 217.388251</val>
</magnitude>
</node>
<node name="Outer membrane lipoprotein involved in outer membrane biogenesis"><magnitude>
<val> 105.852739</val>
</magnitude>
</node>
<node name="Outer membrane lipoprotein-sorting protein"><magnitude>
<val> 376.542898</val>
</magnitude>
</node>
<node name="Outer membrane protease"><magnitude>
<val> 72.470886</val>
</magnitude>
</node>
<node name="Outer membrane protein"><magnitude>
<val> 2292.074352</val>
</magnitude>
</node>
<node name="Outer membrane protein (porin)"><magnitude>
<val> 63.358257</val>
</magnitude>
</node>
<node name="Outer membrane protein V"><magnitude>
<val> 62.004336</val>
</magnitude>
</node>
<node name="Outer membrane protein W"><magnitude>
<val> 77.400912</val>
</magnitude>
</node>
<node name="Outer membrane protein and related peptidoglycan-associated (lipo)proteins"><magnitude>
<val> 799.736668</val>
</magnitude>
</node>
<node name="Outer membrane protein/protective antigen OMA87"><magnitude>
<val> 463.793783</val>
</magnitude>
</node>
<node name="Penicillin V acylase and related amidases"><magnitude>
<val> 120.979894</val>
</magnitude>
</node>
<node name="Periplasmic glycine betaine/choline-binding (lipo)protein of an ABC-type transport system (osmoprotectant binding protein)"><magnitude>
<val> 90.137452</val>
</magnitude>
</node>
<node name="Periplasmic protease"><magnitude>
<val> 442.509305</val>
</magnitude>
</node>
<node name="Periplasmic protein TonB, links inner and outer membranes"><magnitude>
<val> 990.728131</val>
</magnitude>
</node>
<node name="Periplasmic protein involved in polysaccharide export"><magnitude>
<val> 450.017048</val>
</magnitude>
</node>
<node name="Phosphoglycerol transferase and related proteins, alkaline phosphatase superfamily"><magnitude>
<val> 448.848296</val>
</magnitude>
</node>
<node name="Phospholipase C"><magnitude>
<val> 13.731358</val>
</magnitude>
</node>
<node name="Predicted S-adenosylmethionine-dependent methyltransferase involved in bacterial cell division"><magnitude>
<val> 147.970886</val>
</magnitude>
</node>
<node name="Predicted S-adenosylmethionine-dependent methyltransferase involved in cell envelope biogenesis"><magnitude>
<val> 264.645692</val>
</magnitude>
</node>
<node name="Predicted UDP-glucose 6-dehydrogenase"><magnitude>
<val> 301.749539</val>
</magnitude>
</node>
<node name="Predicted choline kinase involved in LPS biosynthesis"><magnitude>
<val> 47.020657</val>
</magnitude>
</node>
<node name="Predicted glycosyl hydrolase"><magnitude>
<val> 19.029171</val>
</magnitude>
</node>
<node name="Predicted glycosyltransferase"><magnitude>
<val> 233.345984</val>
</magnitude>
</node>
<node name="Predicted glycosyltransferase involved in capsule biosynthesis"><magnitude>
<val> 59.819149</val>
</magnitude>
</node>
<node name="Predicted membrane protein involved in D-alanine export"><magnitude>
<val> 214.842713</val>
</magnitude>
</node>
<node name="Predicted membrane-associated Zn-dependent proteases 1"><magnitude>
<val> 301.637214</val>
</magnitude>
</node>
<node name="Predicted nucleoside-diphosphate sugar epimerases"><magnitude>
<val> 517.923449</val>
</magnitude>
</node>
<node name="Predicted nucleoside-diphosphate-sugar epimerase"><magnitude>
<val> 69.721181</val>
</magnitude>
</node>
<node name="Predicted nucleoside-diphosphate-sugar epimerases"><magnitude>
<val> 188.589619</val>
</magnitude>
</node>
<node name="Predicted outer membrane protein"><magnitude>
<val> 320.364642</val>
</magnitude>
</node>
<node name="Predicted phosphosugar isomerases"><magnitude>
<val> 192.036004</val>
</magnitude>
</node>
<node name="Predicted pyridoxal phosphate-dependent enzyme apparently involved in regulation of cell wall biogenesis"><magnitude>
<val> 775.51932</val>
</magnitude>
</node>
<node name="Predicted soluble lytic transglycosylase fused to an ABC-type amino acid-binding protein"><magnitude>
<val> 137.847909</val>
</magnitude>
</node>
<node name="Predicted sugar isomerase"><magnitude>
<val> 41.698175</val>
</magnitude>
</node>
<node name="Predicted sugar nucleotidyltransferases"><magnitude>
<val> 45.104332</val>
</magnitude>
</node>
<node name="Predicted sugar phosphate isomerase involved in capsule formation"><magnitude>
<val> 259.416379</val>
</magnitude>
</node>
<node name="Prolipoprotein diacylglyceryltransferase"><magnitude>
<val> 201.794907</val>
</magnitude>
</node>
<node name="Protein involved in D-alanine esterification of lipoteichoic acid and wall teichoic acid (D-alanine transfer protein)"><magnitude>
<val> 48.612061</val>
</magnitude>
</node>
<node name="Protein involved in cellulose biosynthesis (CelD)"><magnitude>
<val> 53.539063</val>
</magnitude>
</node>
<node name="Putative cell wall-binding domain"><magnitude>
<val> 21.721669</val>
</magnitude>
</node>
<node name="Putative effector of murein hydrolase"><magnitude>
<val> 72.914685</val>
</magnitude>
</node>
<node name="Putative enzyme of poly-gamma-glutamate biosynthesis (capsule formation)"><magnitude>
<val> 238.247448</val>
</magnitude>
</node>
<node name="Putative exporter of polyketide antibiotics"><magnitude>
<val> 66.19553</val>
</magnitude>
</node>
<node name="Putative glycosyl/glycerophosphate transferases involved in teichoic acid biosynthesis TagF/TagB/EpsJ/RodC"><magnitude>
<val> 57.447669</val>
</magnitude>
</node>
<node name="Putative peptidoglycan-binding domain-containing protein"><magnitude>
<val> 248.397826</val>
</magnitude>
</node>
<node name="Putative salt-induced outer membrane protein"><magnitude>
<val> 75.962676</val>
</magnitude>
</node>
<node name="Rare lipoprotein B"><magnitude>
<val> 72.518643</val>
</magnitude>
</node>
<node name="Rhs family protein"><magnitude>
<val> 661.004993</val>
</magnitude>
</node>
<node name="Rod binding protein"><magnitude>
<val> 72.069589</val>
</magnitude>
</node>
<node name="S-layer domain"><magnitude>
<val> 295.820595</val>
</magnitude>
</node>
<node name="Secreted and surface protein containing fasciclin-like repeats"><magnitude>
<val> 61.901099</val>
</magnitude>
</node>
<node name="Short-chain dehydrogenase involved in D-alanine esterification of lipoteichoic acid and wall teichoic acid (D-alanine transfer protein)"><magnitude>
<val> 10.491402</val>
</magnitude>
</node>
<node name="Sialic acid synthase"><magnitude>
<val> 123.368364</val>
</magnitude>
</node>
<node name="Small-conductance mechanosensitive channel"><magnitude>
<val> 815.719189</val>
</magnitude>
</node>
<node name="Soluble lytic murein transglycosylase and related regulatory proteins (some contain LysM/invasin domains)"><magnitude>
<val> 547.585929</val>
</magnitude>
</node>
<node name="Sortase (surface protein transpeptidase)"><magnitude>
<val> 167.831514</val>
</magnitude>
</node>
<node name="Sortase and related acyltransferases"><magnitude>
<val> 219.062468</val>
</magnitude>
</node>
<node name="Spore coat assembly protein"><magnitude>
<val> 50.269331</val>
</magnitude>
</node>
<node name="Spore coat polysaccharide biosynthesis protein F, CMP-KDO synthetase homolog"><magnitude>
<val> 81.431707</val>
</magnitude>
</node>
<node name="Spore coat polysaccharide biosynthesis protein, predicted glycosyltransferase"><magnitude>
<val> 35.85626</val>
</magnitude>
</node>
<node name="Spore coat protein"><magnitude>
<val> 90.541434</val>
</magnitude>
</node>
<node name="Starvation-inducible outer membrane lipoprotein"><magnitude>
<val> 130.487609</val>
</magnitude>
</node>
<node name="Sugar transferases involved in lipopolysaccharide synthesis"><magnitude>
<val> 488.553394</val>
</magnitude>
</node>
<node name="Surface antigen"><magnitude>
<val> 31.414444</val>
</magnitude>
</node>
<node name="Surface lipoprotein"><magnitude>
<val> 177.700635</val>
</magnitude>
</node>
<node name="Teichoic acid biosynthesis proteins"><magnitude>
<val> 171.014741</val>
</magnitude>
</node>
<node name="Tetraacyldisaccharide-1-P 4'-kinase"><magnitude>
<val> 169.929346</val>
</magnitude>
</node>
<node name="Type V secretory pathway, adhesin AidA"><magnitude>
<val> 54.308324</val>
</magnitude>
</node>
<node name="UDP-3-O-[3-hydroxymyristoyl] glucosamine N-acyltransferase"><magnitude>
<val> 248.718376</val>
</magnitude>
</node>
<node name="UDP-3-O-acyl-N-acetylglucosamine deacetylase"><magnitude>
<val> 134.990283</val>
</magnitude>
</node>
<node name="UDP-N-acetyl-D-mannosaminuronate dehydrogenase"><magnitude>
<val> 673.79523</val>
</magnitude>
</node>
<node name="UDP-N-acetylglucosamine 2-epimerase"><magnitude>
<val> 553.641579</val>
</magnitude>
</node>
<node name="UDP-N-acetylglucosamine enolpyruvyl transferase"><magnitude>
<val> 233.602735</val>
</magnitude>
</node>
<node name="UDP-N-acetylglucosamine:LPS N-acetylglucosamine transferase"><magnitude>
<val> 370.114517</val>
</magnitude>
</node>
<node name="UDP-N-acetylmuramate dehydrogenase"><magnitude>
<val> 176.876744</val>
</magnitude>
</node>
<node name="UDP-N-acetylmuramate-alanine ligase"><magnitude>
<val> 301.199535</val>
</magnitude>
</node>
<node name="UDP-N-acetylmuramoylalanine-D-glutamate ligase"><magnitude>
<val> 341.735587</val>
</magnitude>
</node>
<node name="UDP-N-acetylmuramyl pentapeptide phosphotransferase/UDP-N-acetylglucosamine-1-phosphate transferase"><magnitude>
<val> 654.603685</val>
</magnitude>
</node>
<node name="UDP-N-acetylmuramyl pentapeptide synthase"><magnitude>
<val> 255.551377</val>
</magnitude>
</node>
<node name="UDP-N-acetylmuramyl tripeptide synthase"><magnitude>
<val> 379.782523</val>
</magnitude>
</node>
<node name="UDP-galactopyranose mutase"><magnitude>
<val> 163.458468</val>
</magnitude>
</node>
<node name="UDP-glucose 4-epimerase"><magnitude>
<val> 332.972373</val>
</magnitude>
</node>
<node name="UDP-glucose pyrophosphorylase"><magnitude>
<val> 215.715415</val>
</magnitude>
</node>
<node name="Uncharacterized lipoprotein"><magnitude>
<val> 152.861167</val>
</magnitude>
</node>
<node name="Uncharacterized lipoprotein NlpE involved in copper resistance"><magnitude>
<val> 43.102178</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in exopolysaccharide biosynthesis"><magnitude>
<val> 337.2299</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in formation of curli polymers"><magnitude>
<val> 76.464524</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in outer membrane biogenesis"><magnitude>
<val> 199.351158</val>
</magnitude>
</node>
<node name="Uncharacterized protein potentially involved in peptidoglycan biosynthesis"><magnitude>
<val> 57.586352</val>
</magnitude>
</node>
<node name="Uncharacterized protein, involved in the regulation of septum location"><magnitude>
<val> 83.655948</val>
</magnitude>
</node>
<node name="dTDP-4-dehydrorhamnose 3,5-epimerase and related enzymes"><magnitude>
<val> 455.83937</val>
</magnitude>
</node>
<node name="dTDP-4-dehydrorhamnose reductase"><magnitude>
<val> 302.990711</val>
</magnitude>
</node>
<node name="dTDP-D-glucose 4,6-dehydratase"><magnitude>
<val> 305.743927</val>
</magnitude>
</node>
<node name="dTDP-glucose pyrophosphorylase"><magnitude>
<val> 440.382508</val>
</magnitude>
</node>
</node>
<node name="Cytoskeleton"><magnitude>
<val> 730.886213</val>
</magnitude>
<node name="Actin and related proteins"><magnitude>
<val> 52.83534</val>
</magnitude>
</node>
<node name="Ca2+-binding actin-bundling protein fimbrin/plastin (EF-Hand superfamily)"><magnitude>
<val> 13.327302</val>
</magnitude>
</node>
<node name="Calponin"><magnitude>
<val> 20.914729</val>
</magnitude>
</node>
<node name="Dynein, heavy chain"><magnitude>
<val> 311.343108</val>
</magnitude>
</node>
<node name="Kinesin-like protein"><magnitude>
<val> 86.521553</val>
</magnitude>
</node>
<node name="Myosin heavy chain"><magnitude>
<val> 215.105256</val>
</magnitude>
</node>
<node name="Tubulin"><magnitude>
<val> 30.838925</val>
</magnitude>
</node>
</node>
<node name="Defense mechanisms"><magnitude>
<val> 16350.678158</val>
</magnitude>
<node name="ABC-type antimicrobial peptide transport system, ATPase component"><magnitude>
<val> 1732.596963</val>
</magnitude>
</node>
<node name="ABC-type antimicrobial peptide transport system, permease component"><magnitude>
<val> 957.201185</val>
</magnitude>
</node>
<node name="ABC-type bacteriocin/lantibiotic exporters, contain an N-terminal double-glycine peptidase domain"><magnitude>
<val> 503.841127</val>
</magnitude>
</node>
<node name="ABC-type multidrug transport system, ATPase and permease components"><magnitude>
<val> 1303.057136</val>
</magnitude>
</node>
<node name="ABC-type multidrug transport system, ATPase component"><magnitude>
<val> 1516.518351</val>
</magnitude>
</node>
<node name="ABC-type multidrug transport system, permease component"><magnitude>
<val> 926.56036</val>
</magnitude>
</node>
<node name="Abortive infection bacteriophage resistance protein"><magnitude>
<val> 20.561554</val>
</magnitude>
</node>
<node name="Aminoglycoside N3'-acetyltransferase"><magnitude>
<val> 8.478831</val>
</magnitude>
</node>
<node name="Beta-lactamase class A"><magnitude>
<val> 50.892744</val>
</magnitude>
</node>
<node name="Beta-lactamase class C and other penicillin binding proteins"><magnitude>
<val> 398.818007</val>
</magnitude>
</node>
<node name="Beta-lactamase class D"><magnitude>
<val> 35.435045</val>
</magnitude>
</node>
<node name="Cation/multidrug efflux pump"><magnitude>
<val> 1697.61618</val>
</magnitude>
</node>
<node name="Cephalosporin hydroxylase"><magnitude>
<val> 55.492069</val>
</magnitude>
</node>
<node name="Chloramphenicol 3-O-phosphotransferase"><magnitude>
<val> 24.223394</val>
</magnitude>
</node>
<node name="Chloramphenicol O-acetyltransferase"><magnitude>
<val> 30.456912</val>
</magnitude>
</node>
<node name="GTPase subunit of restriction endonuclease"><magnitude>
<val> 41.888475</val>
</magnitude>
</node>
<node name="Glycopeptide antibiotics resistance protein"><magnitude>
<val> 150.636952</val>
</magnitude>
</node>
<node name="Inner membrane protein involved in colicin E2 resistance"><magnitude>
<val> 98.901325</val>
</magnitude>
</node>
<node name="Lantibiotic modifying enzyme"><magnitude>
<val> 117.803776</val>
</magnitude>
</node>
<node name="McrBC 5-methylcytosine restriction system component"><magnitude>
<val> 65.420348</val>
</magnitude>
</node>
<node name="Membrane protein required for beta-lactamase induction"><magnitude>
<val> 5.105245</val>
</magnitude>
</node>
<node name="Multidrug resistance efflux pump"><magnitude>
<val> 1054.715526</val>
</magnitude>
</node>
<node name="Na+-driven multidrug efflux pump"><magnitude>
<val> 977.24597</val>
</magnitude>
</node>
<node name="Negative regulator of beta-lactamase expression"><magnitude>
<val> 366.012341</val>
</magnitude>
</node>
<node name="Predicted endonuclease distantly related to archaeal Holliday junction resolvase and Mrr-like restriction enzymes"><magnitude>
<val> 16.317736</val>
</magnitude>
</node>
<node name="Predicted restriction endonuclease"><magnitude>
<val> 234.724945</val>
</magnitude>
</node>
<node name="Predicted type IV restriction endonuclease"><magnitude>
<val> 78.480429</val>
</magnitude>
</node>
<node name="Restriction endonuclease"><magnitude>
<val> 844.803201</val>
</magnitude>
</node>
<node name="Restriction endonuclease S subunits"><magnitude>
<val> 432.448271</val>
</magnitude>
</node>
<node name="Streptogramin lyase"><magnitude>
<val> 108.91607</val>
</magnitude>
</node>
<node name="Streptomycin 6-kinase"><magnitude>
<val> 60.16923</val>
</magnitude>
</node>
<node name="Type I restriction-modification system methyltransferase subunit"><magnitude>
<val> 691.83706</val>
</magnitude>
</node>
<node name="Type I site-specific restriction-modification system, R (restriction) subunit and related helicases"><magnitude>
<val> 799.141761</val>
</magnitude>
</node>
<node name="Type II restriction enzyme, methylase subunits"><magnitude>
<val> 118.489492</val>
</magnitude>
</node>
<node name="Uncharacterized bacitracin resistance protein"><magnitude>
<val> 312.47141</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in methicillin resistance"><magnitude>
<val> 90.080979</val>
</magnitude>
</node>
<node name="Uncharacterized proteins, homologs of microcin C7 resistance protein MccF"><magnitude>
<val> 195.21407</val>
</magnitude>
</node>
<node name="Uncharacterized vancomycin resistance protein"><magnitude>
<val> 228.103688</val>
</magnitude>
</node>
</node>
<node name="Intracellular trafficking, secretion, and vesicular transport"><magnitude>
<val> 12474.203843</val>
</magnitude>
<node name="Autotransporter adhesin"><magnitude>
<val> 4.367186</val>
</magnitude>
</node>
<node name="Biopolymer transport protein"><magnitude>
<val> 366.873487</val>
</magnitude>
</node>
<node name="Biopolymer transport proteins"><magnitude>
<val> 405.43279</val>
</magnitude>
</node>
<node name="Clathrin adaptor complex, small subunit"><magnitude>
<val> 24.326065</val>
</magnitude>
</node>
<node name="Competence protein ComGC"><magnitude>
<val> 88.140517</val>
</magnitude>
</node>
<node name="Competence protein ComGF"><magnitude>
<val> 38.942186</val>
</magnitude>
</node>
<node name="Conjugal transfer/entry exclusion protein"><magnitude>
<val> 102.403675</val>
</magnitude>
</node>
<node name="Exocyst complex subunit SEC6"><magnitude>
<val> 87.402006</val>
</magnitude>
</node>
<node name="Flp pilus assembly protein CpaB"><magnitude>
<val> 73.182335</val>
</magnitude>
</node>
<node name="Flp pilus assembly protein TadB"><magnitude>
<val> 142.574177</val>
</magnitude>
</node>
<node name="Flp pilus assembly protein TadD, contains TPR repeats"><magnitude>
<val> 614.900778</val>
</magnitude>
</node>
<node name="Flp pilus assembly protein TadG"><magnitude>
<val> 104.39686</val>
</magnitude>
</node>
<node name="Flp pilus assembly protein, ATPase CpaE"><magnitude>
<val> 56.077747</val>
</magnitude>
</node>
<node name="Flp pilus assembly protein, ATPase CpaF"><magnitude>
<val> 134.731374</val>
</magnitude>
</node>
<node name="Flp pilus assembly protein, pilin Flp"><magnitude>
<val> 106.786134</val>
</magnitude>
</node>
<node name="Flp pilus assembly protein, secretin CpaC"><magnitude>
<val> 91.358878</val>
</magnitude>
</node>
<node name="GTPase-activating protein that regulates ARFs (ADP-ribosylation factors), involved in ARF-mediated vesicular transport"><magnitude>
<val> 96.313074</val>
</magnitude>
</node>
<node name="Golgi protein involved in Golgi-to-ER retrieval"><magnitude>
<val> 17.671124</val>
</magnitude>
</node>
<node name="Heme exporter protein D"><magnitude>
<val> 7.719251</val>
</magnitude>
</node>
<node name="Hemolysin activation/secretion protein"><magnitude>
<val> 159.745115</val>
</magnitude>
</node>
<node name="Karyopherin (importin) alpha"><magnitude>
<val> 83.898885</val>
</magnitude>
</node>
<node name="Karyopherin (importin) beta"><magnitude>
<val> 82.216923</val>
</magnitude>
</node>
<node name="Large exoproteins involved in heme utilization or adhesion"><magnitude>
<val> 43.613084</val>
</magnitude>
</node>
<node name="Membrane protein involved in ER to Golgi transport"><magnitude>
<val> 31.862241</val>
</magnitude>
</node>
<node name="Membrane protein involved in Golgi transport"><magnitude>
<val> 12.099507</val>
</magnitude>
</node>
<node name="Multiple antibiotic transporter"><magnitude>
<val> 149.035849</val>
</magnitude>
</node>
<node name="Nuclear pore complex subunit"><magnitude>
<val> 65.631474</val>
</magnitude>
</node>
<node name="Nuclear protein export factor"><magnitude>
<val> 65.252991</val>
</magnitude>
</node>
<node name="P pilus assembly/Cpx signaling pathway, periplasmic inhibitor/zinc-resistance associated protein"><magnitude>
<val> 147.992435</val>
</magnitude>
</node>
<node name="Peripheral Golgi membrane protein"><magnitude>
<val> 5.212729</val>
</magnitude>
</node>
<node name="Periplasmic component of the Tol biopolymer transport system"><magnitude>
<val> 368.014133</val>
</magnitude>
</node>
<node name="Phox homology (PX) domain protein"><magnitude>
<val> 11.948142</val>
</magnitude>
</node>
<node name="Predicted ABC-type exoprotein transport system, permease component"><magnitude>
<val> 20.843647</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit Sec61beta"><magnitude>
<val> 83.419408</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit Sec63"><magnitude>
<val> 25.245553</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit Sec66"><magnitude>
<val> 15.061275</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit SecA (ATPase, RNA helicase)"><magnitude>
<val> 325.725357</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit SecB"><magnitude>
<val> 23.798701</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit SecD"><magnitude>
<val> 486.401518</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit SecE"><magnitude>
<val> 110.390363</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit SecF"><magnitude>
<val> 245.580367</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit SecG"><magnitude>
<val> 148.894193</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit SecY"><magnitude>
<val> 236.219914</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit Sss1"><magnitude>
<val> 131.9731</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit YajC"><magnitude>
<val> 157.015582</val>
</magnitude>
</node>
<node name="Preprotein translocase subunit YidC"><magnitude>
<val> 261.185942</val>
</magnitude>
</node>
<node name="Protein involved in membrane traffic"><magnitude>
<val> 4.679193</val>
</magnitude>
</node>
<node name="Protein involved in vacuole import and degradation"><magnitude>
<val> 138.431128</val>
</magnitude>
</node>
<node name="Protein transporter of the TRAM (translocating chain-associating membrane) superfamily, longevity assurance factor"><magnitude>
<val> 58.987824</val>
</magnitude>
</node>
<node name="Proteins involved in synaptic transmission and general secretion, Sec1 family"><magnitude>
<val> 25.673892</val>
</magnitude>
</node>
<node name="Putative lipase essential for disintegration of autophagic bodies inside the vacuole"><magnitude>
<val> 21.176665</val>
</magnitude>
</node>
<node name="Rab GTPase interacting factor, Golgi membrane protein"><magnitude>
<val> 4.979823</val>
</magnitude>
</node>
<node name="Sec-independent protein secretion pathway component TatC"><magnitude>
<val> 233.840465</val>
</magnitude>
</node>
<node name="Sec-independent protein secretion pathway components"><magnitude>
<val> 536.877033</val>
</magnitude>
</node>
<node name="Signal peptidase I"><magnitude>
<val> 364.820441</val>
</magnitude>
</node>
<node name="Signal recognition particle 19 kDa protein"><magnitude>
<val> 113.268316</val>
</magnitude>
</node>
<node name="Signal recognition particle GTPase"><magnitude>
<val> 648.93475</val>
</magnitude>
</node>
<node name="Synaptobrevin/VAMP-like protein"><magnitude>
<val> 51.806547</val>
</magnitude>
</node>
<node name="Transport protein particle (TRAPP) complex subunit"><magnitude>
<val> 36.480241</val>
</magnitude>
</node>
<node name="Type II secretory pathway, component ExeA (predicted ATPase)"><magnitude>
<val> 369.203246</val>
</magnitude>
</node>
<node name="Type II secretory pathway, component HofQ"><magnitude>
<val> 166.473708</val>
</magnitude>
</node>
<node name="Type II secretory pathway, component PulC"><magnitude>
<val> 107.222636</val>
</magnitude>
</node>
<node name="Type II secretory pathway, component PulJ"><magnitude>
<val> 134.831019</val>
</magnitude>
</node>
<node name="Type II secretory pathway, component PulK"><magnitude>
<val> 179.536059</val>
</magnitude>
</node>
<node name="Type II secretory pathway, component PulL"><magnitude>
<val> 131.150983</val>
</magnitude>
</node>
<node name="Type II secretory pathway, component PulM"><magnitude>
<val> 24.628137</val>
</magnitude>
</node>
<node name="Type III secretory pathway, component EscR"><magnitude>
<val> 9.964451</val>
</magnitude>
</node>
<node name="Type III secretory pathway, component EscS"><magnitude>
<val> 5.277539</val>
</magnitude>
</node>
<node name="Type III secretory pathway, component EscT"><magnitude>
<val> 12.257614</val>
</magnitude>
</node>
<node name="Type III secretory pathway, component EscU"><magnitude>
<val> 32.087282</val>
</magnitude>
</node>
<node name="Type III secretory pathway, component EscV"><magnitude>
<val> 54.646497</val>
</magnitude>
</node>
<node name="Type III secretory pathway, lipoprotein EscJ"><magnitude>
<val> 76.238277</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, TrbF components"><magnitude>
<val> 62.646696</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, TrbL components"><magnitude>
<val> 224.134512</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, VirB10 components"><magnitude>
<val> 126.711175</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, VirB2 components (pilins)"><magnitude>
<val> 58.170577</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, VirB4 components"><magnitude>
<val> 298.395503</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, VirB6 components"><magnitude>
<val> 29.004861</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, VirB9 components"><magnitude>
<val> 12.698671</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, VirD2 components (relaxase)"><magnitude>
<val> 8.1657</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, VirD4 components"><magnitude>
<val> 284.326368</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, VirJ component"><magnitude>
<val> 81.936984</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, component VirB8"><magnitude>
<val> 14.102537</val>
</magnitude>
</node>
<node name="VAMP-associated protein involved in inositol metabolism"><magnitude>
<val> 30.455191</val>
</magnitude>
</node>
<node name="Vacuolar import and degradation protein"><magnitude>
<val> 12.210585</val>
</magnitude>
</node>
<node name="Vacuolar protein sorting-associated protein"><magnitude>
<val> 156.666678</val>
</magnitude>
</node>
<node name="Vesicle coat complex COPI, gamma subunit"><magnitude>
<val> 51.694293</val>
</magnitude>
</node>
<node name="Vesicle coat complex COPII, subunit SEC23"><magnitude>
<val> 36.078889</val>
</magnitude>
</node>
<node name="Vesicle coat complex COPII, subunit SEC24/subunit SFB2/subunit SFB3"><magnitude>
<val> 1258.521733</val>
</magnitude>
</node>
<node name="Vesicle coat complex, various subunits"><magnitude>
<val> 5.213397</val>
</magnitude>
</node>
<node name="t-SNARE complex subunit, syntaxin"><magnitude>
<val> 117.717655</val>
</magnitude>
</node>
</node>
<node name="Nuclear structure"><magnitude>
<val> 10.534003</val>
</magnitude>
<node name="Importin beta-related nuclear transport receptor"><magnitude>
<val> 9.174003</val>
</magnitude>
</node>
<node name="Nuclear pore protein"><magnitude>
<val> 1.36</val>
</magnitude>
</node>
</node>
<node name="Posttranslational modification, protein turnover, chaperones"><magnitude>
<val> 31823.992647</val>
</magnitude>
<node name="20S proteasome, alpha and beta subunits"><magnitude>
<val> 201.869084</val>
</magnitude>
</node>
<node name="26S proteasome regulatory complex component"><magnitude>
<val> 344.079932</val>
</magnitude>
</node>
<node name="26S proteasome regulatory complex component, contains PCI domain"><magnitude>
<val> 73.818183</val>
</magnitude>
</node>
<node name="26S proteasome regulatory complex, subunit RPN10/PSMD4"><magnitude>
<val> 49.084069</val>
</magnitude>
</node>
<node name="ABC-type sulfate transport system, permease component"><magnitude>
<val> 246.983083</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in Fe-S cluster assembly, ATPase component"><magnitude>
<val> 209.886052</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in Fe-S cluster assembly, permease and ATPase components"><magnitude>
<val> 77.323497</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in Fe-S cluster assembly, permease component"><magnitude>
<val> 226.191667</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in cytochrome c biogenesis, ATPase component"><magnitude>
<val> 37.623077</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in cytochrome c biogenesis, permease component"><magnitude>
<val> 368.314829</val>
</magnitude>
</node>
<node name="ACP:hemolysin acyltransferase (hemolysin-activating protein)"><magnitude>
<val> 24.455932</val>
</magnitude>
</node>
<node name="ADP-ribosylglycohydrolase"><magnitude>
<val> 423.498508</val>
</magnitude>
</node>
<node name="ATP-dependent 26S proteasome regulatory subunit"><magnitude>
<val> 135.06881</val>
</magnitude>
</node>
<node name="ATP-dependent Lon protease, bacterial type"><magnitude>
<val> 639.093082</val>
</magnitude>
</node>
<node name="ATP-dependent Zn proteases"><magnitude>
<val> 455.384087</val>
</magnitude>
</node>
<node name="ATP-dependent protease Clp, ATPase subunit"><magnitude>
<val> 223.846756</val>
</magnitude>
</node>
<node name="ATP-dependent protease HslVU (ClpYQ), ATPase subunit"><magnitude>
<val> 139.752578</val>
</magnitude>
</node>
<node name="ATP-dependent protease HslVU (ClpYQ), peptidase subunit"><magnitude>
<val> 102.240059</val>
</magnitude>
</node>
<node name="ATPases of the AAA+ class"><magnitude>
<val> 643.241929</val>
</magnitude>
</node>
<node name="ATPases with chaperone activity, ATP-binding subunit"><magnitude>
<val> 597.985272</val>
</magnitude>
</node>
<node name="Activator of HSP90 ATPase"><magnitude>
<val> 42.912323</val>
</magnitude>
</node>
<node name="Alkyl hydroperoxide reductase, large subunit"><magnitude>
<val> 70.542106</val>
</magnitude>
</node>
<node name="Arginine methyltransferase-interacting protein, contains RING Zn-finger"><magnitude>
<val> 11.91549</val>
</magnitude>
</node>
<node name="Aspartyl/asparaginyl beta-hydroxylase and related dioxygenases"><magnitude>
<val> 20.903661</val>
</magnitude>
</node>
<node name="Beta-tubulin folding cofactor D"><magnitude>
<val> 58.824112</val>
</magnitude>
</node>
<node name="Cbb3-type cytochrome oxidase, subunit 1"><magnitude>
<val> 73.178818</val>
</magnitude>
</node>
<node name="Cbb3-type cytochrome oxidase, subunit 3"><magnitude>
<val> 61.735102</val>
</magnitude>
</node>
<node name="Chaperone required for the assembly of the mitochondrial F1-ATPase"><magnitude>
<val> 25.056261</val>
</magnitude>
</node>
<node name="Chaperonin GroEL (HSP60 family)"><magnitude>
<val> 427.330641</val>
</magnitude>
</node>
<node name="Co-chaperonin GroES (HSP10)"><magnitude>
<val> 141.885539</val>
</magnitude>
</node>
<node name="Collagenase and related proteases"><magnitude>
<val> 596.943123</val>
</magnitude>
</node>
<node name="Component of SCF ubiquitin ligase and anaphase-promoting complex"><magnitude>
<val> 5.252974</val>
</magnitude>
</node>
<node name="Conserved domain frequently associated with peptide methionine sulfoxide reductase"><magnitude>
<val> 151.799069</val>
</magnitude>
</node>
<node name="Cullin, a subunit of E3 ubiquitin ligase"><magnitude>
<val> 39.045092</val>
</magnitude>
</node>
<node name="Cysteine protease"><magnitude>
<val> 205.936852</val>
</magnitude>
</node>
<node name="Cytochrome c biogenesis factor"><magnitude>
<val> 337.696931</val>
</magnitude>
</node>
<node name="Cytochrome c biogenesis protein"><magnitude>
<val> 293.918547</val>
</magnitude>
</node>
<node name="Cytochrome c-type biogenesis protein CcmE"><magnitude>
<val> 68.448677</val>
</magnitude>
</node>
<node name="Cytochrome oxidase assembly factor"><magnitude>
<val> 62.394929</val>
</magnitude>
</node>
<node name="Deoxyhypusine synthase"><magnitude>
<val> 122.531758</val>
</magnitude>
</node>
<node name="Disulfide bond chaperones of the HSP33 family"><magnitude>
<val> 140.717245</val>
</magnitude>
</node>
<node name="Disulfide bond formation protein DsbB"><magnitude>
<val> 22.15343</val>
</magnitude>
</node>
<node name="DnaJ-class molecular chaperone"><magnitude>
<val> 14.816991</val>
</magnitude>
</node>
<node name="DnaJ-class molecular chaperone with C-terminal Zn finger domain"><magnitude>
<val> 1072.021661</val>
</magnitude>
</node>
<node name="DnaJ-domain-containing proteins 1"><magnitude>
<val> 14.879602</val>
</magnitude>
</node>
<node name="Dolichyl-phosphate-mannose--protein O-mannosyl transferase"><magnitude>
<val> 150.653037</val>
</magnitude>
</node>
<node name="FKBP-type peptidyl-prolyl cis-trans isomerase (trigger factor)"><magnitude>
<val> 279.763105</val>
</magnitude>
</node>
<node name="FKBP-type peptidyl-prolyl cis-trans isomerases 1"><magnitude>
<val> 165.525643</val>
</magnitude>
</node>
<node name="FKBP-type peptidyl-prolyl cis-trans isomerases 2"><magnitude>
<val> 412.983305</val>
</magnitude>
</node>
<node name="Flp pilus assembly protein, protease CpaA"><magnitude>
<val> 156.781936</val>
</magnitude>
</node>
<node name="Glutamine cyclotransferase"><magnitude>
<val> 61.835172</val>
</magnitude>
</node>
<node name="Glutamine synthetase adenylyltransferase"><magnitude>
<val> 196.500247</val>
</magnitude>
</node>
<node name="Glutaredoxin 2"><magnitude>
<val> 37.962632</val>
</magnitude>
</node>
<node name="Glutaredoxin and related proteins"><magnitude>
<val> 321.905623</val>
</magnitude>
</node>
<node name="Glutaredoxin-related protein"><magnitude>
<val> 19.611011</val>
</magnitude>
</node>
<node name="Glutathione S-transferase"><magnitude>
<val> 116.114102</val>
</magnitude>
</node>
<node name="Glutathione peroxidase"><magnitude>
<val> 166.922082</val>
</magnitude>
</node>
<node name="Glycosylphosphatidylinositol transamidase (GPIT), subunit GPI8"><magnitude>
<val> 38.387824</val>
</magnitude>
</node>
<node name="HRD ubiquitin ligase complex, ER membrane component"><magnitude>
<val> 96.467798</val>
</magnitude>
</node>
<node name="Heat shock protein"><magnitude>
<val> 54.423718</val>
</magnitude>
</node>
<node name="Highly conserved protein containing a thioredoxin domain"><magnitude>
<val> 574.107604</val>
</magnitude>
</node>
<node name="Hydrogenase maturation factor"><magnitude>
<val> 1052.372528</val>
</magnitude>
</node>
<node name="Hydrogenase/urease accessory protein"><magnitude>
<val> 49.352529</val>
</magnitude>
</node>
<node name="Importin, protein involved in nuclear import"><magnitude>
<val> 150.058516</val>
</magnitude>
</node>
<node name="Inactive homolog of metal-dependent proteases, putative molecular chaperone"><magnitude>
<val> 169.0551</val>
</magnitude>
</node>
<node name="Isopeptidase T"><magnitude>
<val> 90.39686</val>
</magnitude>
</node>
<node name="La protein, small RNA-binding pol III transcript stabilizing protein and related La-motif-containing proteins involved in translation"><magnitude>
<val> 47.512683</val>
</magnitude>
</node>
<node name="Leu/Phe-tRNA-protein transferase"><magnitude>
<val> 113.714121</val>
</magnitude>
</node>
<node name="Lipase chaperone"><magnitude>
<val> 1.190217</val>
</magnitude>
</node>
<node name="Membrane protease subunits, stomatin/prohibitin homologs"><magnitude>
<val> 348.303951</val>
</magnitude>
</node>
<node name="Membrane protein implicated in regulation of membrane protease activity"><magnitude>
<val> 150.298046</val>
</magnitude>
</node>
<node name="Membrane protein putatively involved in post-translational modification of the autoinducing quorum-sensing peptide"><magnitude>
<val> 79.273973</val>
</magnitude>
</node>
<node name="Membrane-bound serine protease (ClpP class)"><magnitude>
<val> 335.248068</val>
</magnitude>
</node>
<node name="Metal-dependent proteases with possible chaperone activity"><magnitude>
<val> 262.610075</val>
</magnitude>
</node>
<node name="Mitochondrial import inner membrane translocase, subunit TIM22"><magnitude>
<val> 2.411955</val>
</magnitude>
</node>
<node name="Molecular chaperone"><magnitude>
<val> 584.856948</val>
</magnitude>
</node>
<node name="Molecular chaperone (small heat shock protein)"><magnitude>
<val> 921.656025</val>
</magnitude>
</node>
<node name="Molecular chaperone GrpE (heat shock protein)"><magnitude>
<val> 214.948</val>
</magnitude>
</node>
<node name="Molecular chaperone, HSP90 family"><magnitude>
<val> 428.99706</val>
</magnitude>
</node>
<node name="Negative regulator of genetic competence, sporulation and motility"><magnitude>
<val> 18.948822</val>
</magnitude>
</node>
<node name="Ni2+-binding GTPase involved in regulation of expression and maturation of urease and hydrogenase"><magnitude>
<val> 323.628866</val>
</magnitude>
</node>
<node name="Organic radical activating enzymes"><magnitude>
<val> 256.491343</val>
</magnitude>
</node>
<node name="Oxidoreductin, endoplasmic reticulum membrane-associated protein involved in disulfide bond formation"><magnitude>
<val> 11.248386</val>
</magnitude>
</node>
<node name="Parvulin-like peptidyl-prolyl isomerase"><magnitude>
<val> 604.392995</val>
</magnitude>
</node>
<node name="Peptide methionine sulfoxide reductase"><magnitude>
<val> 347.865863</val>
</magnitude>
</node>
<node name="Peptidyl-prolyl cis-trans isomerase (rotamase) - cyclophilin family"><magnitude>
<val> 271.461392</val>
</magnitude>
</node>
<node name="Periplasmic serine proteases (ClpP class)"><magnitude>
<val> 359.792325</val>
</magnitude>
</node>
<node name="Peroxiredoxin"><magnitude>
<val> 962.185267</val>
</magnitude>
</node>
<node name="Polyprenyltransferase (cytochrome oxidase assembly factor)"><magnitude>
<val> 124.883092</val>
</magnitude>
</node>
<node name="Predicted ATP-dependent Lon-type protease"><magnitude>
<val> 110.972324</val>
</magnitude>
</node>
<node name="Predicted ATP-dependent protease"><magnitude>
<val> 482.980498</val>
</magnitude>
</node>
<node name="Predicted ATP-dependent serine protease"><magnitude>
<val> 267.309724</val>
</magnitude>
</node>
<node name="Predicted ATPase with chaperone activity"><magnitude>
<val> 497.209222</val>
</magnitude>
</node>
<node name="Predicted O-linked N-acetylglucosamine transferase, SPINDLY family"><magnitude>
<val> 107.480275</val>
</magnitude>
</node>
<node name="Predicted Zn-dependent protease"><magnitude>
<val> 33.323897</val>
</magnitude>
</node>
<node name="Predicted carbamoyl transferase, NodU family"><magnitude>
<val> 80.270798</val>
</magnitude>
</node>
<node name="Predicted cysteine protease (OTU family)"><magnitude>
<val> 26.756492</val>
</magnitude>
</node>
<node name="Predicted glutathione S-transferase"><magnitude>
<val> 34.409639</val>
</magnitude>
</node>
<node name="Predicted membrane-bound dolichyl-phosphate-mannose-protein mannosyltransferase"><magnitude>
<val> 27.884559</val>
</magnitude>
</node>
<node name="Predicted membrane-bound mannosyltransferase"><magnitude>
<val> 46.892665</val>
</magnitude>
</node>
<node name="Predicted metalloendopeptidase"><magnitude>
<val> 148.911603</val>
</magnitude>
</node>
<node name="Predicted molecular chaperone distantly related to HSP70-fold metalloproteases"><magnitude>
<val> 82.090768</val>
</magnitude>
</node>
<node name="Predicted peptidyl-prolyl cis-trans isomerase (rotamase), cyclophilin family"><magnitude>
<val> 123.955943</val>
</magnitude>
</node>
<node name="Predicted prefoldin, molecular chaperone implicated in de novo protein folding"><magnitude>
<val> 64.282693</val>
</magnitude>
</node>
<node name="Predicted proline hydroxylase"><magnitude>
<val> 9.089924</val>
</magnitude>
</node>
<node name="Predicted protease"><magnitude>
<val> 279.207449</val>
</magnitude>
</node>
<node name="Predicted proteasome-type protease"><magnitude>
<val> 14.479829</val>
</magnitude>
</node>
<node name="Predicted protein-disulfide isomerase"><magnitude>
<val> 34.428235</val>
</magnitude>
</node>
<node name="Predicted redox protein, regulator of disulfide bond formation"><magnitude>
<val> 470.980926</val>
</magnitude>
</node>
<node name="Predicted secreted Zn-dependent protease"><magnitude>
<val> 47.89876</val>
</magnitude>
</node>
<node name="Predicted thioredoxin/glutaredoxin"><magnitude>
<val> 57.61224</val>
</magnitude>
</node>
<node name="Prefoldin, chaperonin cofactor"><magnitude>
<val> 137.918024</val>
</magnitude>
</node>
<node name="Prefoldin, molecular chaperone implicated in de novo protein folding, alpha subunit"><magnitude>
<val> 46.27034</val>
</magnitude>
</node>
<node name="Prenyltransferase, beta subunit"><magnitude>
<val> 83.916467</val>
</magnitude>
</node>
<node name="Protease subunit of ATP-dependent Clp proteases"><magnitude>
<val> 219.105708</val>
</magnitude>
</node>
<node name="Protease, Ulp1 family"><magnitude>
<val> 7.335246</val>
</magnitude>
</node>
<node name="Protein involved in Mod5 protein sorting"><magnitude>
<val> 16.880508</val>
</magnitude>
</node>
<node name="Protein prenyltransferase, alpha subunit"><magnitude>
<val> 48.190269</val>
</magnitude>
</node>
<node name="Protein-L-isoaspartate carboxylmethyltransferase"><magnitude>
<val> 237.63369</val>
</magnitude>
</node>
<node name="Protein-disulfide isomerase"><magnitude>
<val> 207.339689</val>
</magnitude>
</node>
<node name="Putative arginyl-tRNA:protein arginylyltransferase"><magnitude>
<val> 47.886594</val>
</magnitude>
</node>
<node name="Putative protein-S-isoprenylcysteine methyltransferase"><magnitude>
<val> 387.550956</val>
</magnitude>
</node>
<node name="Pyrrolidone-carboxylate peptidase (N-terminal pyroglutamyl peptidase)"><magnitude>
<val> 68.334555</val>
</magnitude>
</node>
<node name="Pyruvate-formate lyase-activating enzyme"><magnitude>
<val> 888.225337</val>
</magnitude>
</node>
<node name="RAB proteins geranylgeranyltransferase component A (RAB escort protein)"><magnitude>
<val> 82.312463</val>
</magnitude>
</node>
<node name="RING-finger-containing E3 ubiquitin ligase"><magnitude>
<val> 45.473908</val>
</magnitude>
</node>
<node name="RING-finger-containing ubiquitin ligase"><magnitude>
<val> 12.344059</val>
</magnitude>
</node>
<node name="Regulatory P domain of the subtilisin-like proprotein convertases and other proteases"><magnitude>
<val> 41.801088</val>
</magnitude>
</node>
<node name="ResB protein required for cytochrome c biosynthesis"><magnitude>
<val> 329.564385</val>
</magnitude>
</node>
<node name="Secreted trypsin-like serine protease"><magnitude>
<val> 5.270857</val>
</magnitude>
</node>
<node name="Secreted/periplasmic Zn-dependent peptidases, insulinase-like"><magnitude>
<val> 81.365696</val>
</magnitude>
</node>
<node name="Serine protease inhibitor"><magnitude>
<val> 93.09657</val>
</magnitude>
</node>
<node name="Subtilisin-like serine proteases"><magnitude>
<val> 632.830008</val>
</magnitude>
</node>
<node name="Thiol-disulfide isomerase and thioredoxins"><magnitude>
<val> 632.402142</val>
</magnitude>
</node>
<node name="Thiol:disulfide interchange protein"><magnitude>
<val> 285.620182</val>
</magnitude>
</node>
<node name="Thioredoxin domain-containing protein"><magnitude>
<val> 501.473577</val>
</magnitude>
</node>
<node name="Thioredoxin reductase"><magnitude>
<val> 429.048543</val>
</magnitude>
</node>
<node name="Thioredoxin-like proteins and domains"><magnitude>
<val> 157.441767</val>
</magnitude>
</node>
<node name="Thioredoxin-related protein"><magnitude>
<val> 9.489651</val>
</magnitude>
</node>
<node name="Trypsin-like serine proteases, typically periplasmic, contain C-terminal PDZ domain"><magnitude>
<val> 560.316251</val>
</magnitude>
</node>
<node name="Type IV secretory pathway, protease TraF"><magnitude>
<val> 72.363676</val>
</magnitude>
</node>
<node name="UTP:GlnB (protein PII) uridylyltransferase"><magnitude>
<val> 288.632702</val>
</magnitude>
</node>
<node name="Ubiquitin C-terminal hydrolase"><magnitude>
<val> 97.468112</val>
</magnitude>
</node>
<node name="Ubiquitin carboxyl-terminal hydrolase"><magnitude>
<val> 70.249465</val>
</magnitude>
</node>
<node name="Ubiquitin fusion degradation protein 2"><magnitude>
<val> 68.981336</val>
</magnitude>
</node>
<node name="Ubiquitin fusion-degradation protein"><magnitude>
<val> 4.271396</val>
</magnitude>
</node>
<node name="Ubiquitin-like protein"><magnitude>
<val> 7.103108</val>
</magnitude>
</node>
<node name="Ubiquitin-like protein (sentrin)"><magnitude>
<val> 7.349001</val>
</magnitude>
</node>
<node name="Ubiquitin-protein ligase"><magnitude>
<val> 39.84345</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in archaea"><magnitude>
<val> 31.037754</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria"><magnitude>
<val> 21.979733</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria containing thioredoxin-like domain"><magnitude>
<val> 83.533723</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in biosynthesis of c-type cytochromes"><magnitude>
<val> 62.152259</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in formate dehydrogenase formation"><magnitude>
<val> 46.336702</val>
</magnitude>
</node>
<node name="Uncharacterized protein predicted to be involved in C-type cytochrome biogenesis"><magnitude>
<val> 28.819082</val>
</magnitude>
</node>
<node name="Uncharacterized protein required for cytochrome oxidase assembly"><magnitude>
<val> 19.128633</val>
</magnitude>
</node>
<node name="Urease accessory protein UreE"><magnitude>
<val> 16.544772</val>
</magnitude>
</node>
<node name="Urease accessory protein UreF"><magnitude>
<val> 24.322063</val>
</magnitude>
</node>
<node name="Urease accessory protein UreH"><magnitude>
<val> 26.11424</val>
</magnitude>
</node>
<node name="Xanthine and CO dehydrogenases maturation factor, XdhC/CoxF family"><magnitude>
<val> 235.679532</val>
</magnitude>
</node>
<node name="Zn-dependent protease with chaperone function"><magnitude>
<val> 595.127317</val>
</magnitude>
</node>
<node name="tmRNA-binding protein"><magnitude>
<val> 196.406333</val>
</magnitude>
</node>
</node>
<node name="Signal transduction mechanisms"><magnitude>
<val> 41960.085665</val>
</magnitude>
<node name="14-3-3 family protein"><magnitude>
<val> 4.672872</val>
</magnitude>
</node>
<node name="ACT domain-containing protein"><magnitude>
<val> 31.106591</val>
</magnitude>
</node>
<node name="Activator of osmoprotectant transporter ProP"><magnitude>
<val> 65.037964</val>
</magnitude>
</node>
<node name="Adenylate cyclase, family 3 (some proteins contain HAMP domain)"><magnitude>
<val> 124.729708</val>
</magnitude>
</node>
<node name="Anti-anti-sigma regulatory factor (antagonist of anti-sigma factor)"><magnitude>
<val> 381.59785</val>
</magnitude>
</node>
<node name="Anti-sigma factor"><magnitude>
<val> 24.073144</val>
</magnitude>
</node>
<node name="Anti-sigma regulatory factor (Ser/Thr protein kinase)"><magnitude>
<val> 434.409904</val>
</magnitude>
</node>
<node name="Bacteriophytochrome (light-regulated signal transduction histidine kinase)"><magnitude>
<val> 1424.618955</val>
</magnitude>
</node>
<node name="CBS-domain-containing membrane protein"><magnitude>
<val> 282.936629</val>
</magnitude>
</node>
<node name="Ca2+-binding protein (EF-Hand superfamily)"><magnitude>
<val> 197.334132</val>
</magnitude>
</node>
<node name="Carbon starvation protein, predicted membrane protein"><magnitude>
<val> 268.829185</val>
</magnitude>
</node>
<node name="Carbon storage regulator (could also regulate swarming and quorum sensing)"><magnitude>
<val> 92.108799</val>
</magnitude>
</node>
<node name="Casein kinase II, beta subunit"><magnitude>
<val> 12.103135</val>
</magnitude>
</node>
<node name="DNA-binding transcriptional activator of the SARP family"><magnitude>
<val> 113.057533</val>
</magnitude>
</node>
<node name="Diadenosine tetraphosphatase and related serine/threonine protein phosphatases"><magnitude>
<val> 17.21044</val>
</magnitude>
</node>
<node name="DnaK suppressor protein"><magnitude>
<val> 305.704806</val>
</magnitude>
</node>
<node name="FOG: CheY-like receiver"><magnitude>
<val> 1786.150399</val>
</magnitude>
</node>
<node name="FOG: EAL domain"><magnitude>
<val> 612.70667</val>
</magnitude>
</node>
<node name="FOG: FHA domain"><magnitude>
<val> 123.562858</val>
</magnitude>
</node>
<node name="FOG: GAF domain"><magnitude>
<val> 241.554665</val>
</magnitude>
</node>
<node name="FOG: GGDEF domain"><magnitude>
<val> 1925.413046</val>
</magnitude>
</node>
<node name="FOG: HAMP domain"><magnitude>
<val> 49.215212</val>
</magnitude>
</node>
<node name="FOG: HPt domain"><magnitude>
<val> 129.386373</val>
</magnitude>
</node>
<node name="FOG: PAS/PAC domain"><magnitude>
<val> 1028.450118</val>
</magnitude>
</node>
<node name="GAF domain-containing protein"><magnitude>
<val> 33.641818</val>
</magnitude>
</node>
<node name="Growth inhibitor"><magnitude>
<val> 243.251606</val>
</magnitude>
</node>
<node name="Growth regulator"><magnitude>
<val> 34.075087</val>
</magnitude>
</node>
<node name="Guanosine polyphosphate pyrophosphohydrolases/synthetases"><magnitude>
<val> 507.209357</val>
</magnitude>
</node>
<node name="HD-GYP domain"><magnitude>
<val> 886.15629</val>
</magnitude>
</node>
<node name="Isocitrate dehydrogenase kinase/phosphatase"><magnitude>
<val> 52.270941</val>
</magnitude>
</node>
<node name="Low-affinity cAMP phosphodiesterase"><magnitude>
<val> 4.177931</val>
</magnitude>
</node>
<node name="LuxS protein involved in autoinducer AI2 synthesis"><magnitude>
<val> 11.089653</val>
</magnitude>
</node>
<node name="MHYT domain (predicted integral membrane sensor domain)"><magnitude>
<val> 21.770962</val>
</magnitude>
</node>
<node name="Mn2+-dependent serine/threonine protein kinase"><magnitude>
<val> 48.612932</val>
</magnitude>
</node>
<node name="N-acyl-L-homoserine lactone synthetase"><magnitude>
<val> 36.312145</val>
</magnitude>
</node>
<node name="NAD+--asparagine ADP-ribosyltransferase"><magnitude>
<val> 31.589723</val>
</magnitude>
</node>
<node name="Negative regulator of sigma E activity"><magnitude>
<val> 125.167531</val>
</magnitude>
</node>
<node name="Osmosensitive K+ channel histidine kinase"><magnitude>
<val> 426.327348</val>
</magnitude>
</node>
<node name="Phosphate starvation-inducible protein PhoH, predicted ATPase"><magnitude>
<val> 123.026759</val>
</magnitude>
</node>
<node name="Phosphatidylinositol 5-phosphate phosphatase"><magnitude>
<val> 54.389359</val>
</magnitude>
</node>
<node name="Phosphatidylinositol kinase and protein kinases of the PI-3 kinase family"><magnitude>
<val> 173.587126</val>
</magnitude>
</node>
<node name="Phosphatidylinositol-4-phosphate 5-kinase"><magnitude>
<val> 32.30421</val>
</magnitude>
</node>
<node name="Phosphohistidine phosphatase SixA"><magnitude>
<val> 132.385037</val>
</magnitude>
</node>
<node name="Phosphohistidine swiveling domain"><magnitude>
<val> 18.233463</val>
</magnitude>
</node>
<node name="Positive regulator of sigma E activity"><magnitude>
<val> 70.61143</val>
</magnitude>
</node>
<node name="Predicted ATPase related to phosphate starvation-inducible protein PhoH"><magnitude>
<val> 176.348822</val>
</magnitude>
</node>
<node name="Predicted NTPase (NACHT family)"><magnitude>
<val> 115.251927</val>
</magnitude>
</node>
<node name="Predicted Ser/Thr protein kinase"><magnitude>
<val> 64.91985</val>
</magnitude>
</node>
<node name="Predicted integral membrane sensor domain"><magnitude>
<val> 9.94161</val>
</magnitude>
</node>
<node name="Predicted membrane GTPase involved in stress response"><magnitude>
<val> 310.529525</val>
</magnitude>
</node>
<node name="Predicted periplasmic ligand-binding sensor domain"><magnitude>
<val> 726.980678</val>
</magnitude>
</node>
<node name="Predicted protein-tyrosine phosphatase"><magnitude>
<val> 61.394241</val>
</magnitude>
</node>
<node name="Predicted secreted protein containing a PDZ domain"><magnitude>
<val> 35.826065</val>
</magnitude>
</node>
<node name="Predicted sensor domain"><magnitude>
<val> 107.245554</val>
</magnitude>
</node>
<node name="Predicted sensor protein/domain"><magnitude>
<val> 4.156604</val>
</magnitude>
</node>
<node name="Predicted signal transduction protein"><magnitude>
<val> 526.971461</val>
</magnitude>
</node>
<node name="Predicted signal transduction protein containing EAL and modified HD-GYP domains"><magnitude>
<val> 76.462839</val>
</magnitude>
</node>
<node name="Predicted signal transduction protein containing a membrane domain, an EAL and a GGDEF domain"><magnitude>
<val> 323.669969</val>
</magnitude>
</node>
<node name="Predicted signal transduction protein containing sensor and EAL domains"><magnitude>
<val> 194.376975</val>
</magnitude>
</node>
<node name="Predicted signal transduction protein with a C-terminal ATPase domain"><magnitude>
<val> 162.108266</val>
</magnitude>
</node>
<node name="Predicted signal-transduction protein containing cAMP-binding and CBS domains"><magnitude>
<val> 366.549664</val>
</magnitude>
</node>
<node name="Predicted signaling protein consisting of a modified GGDEF domain and a DHH domain"><magnitude>
<val> 159.633141</val>
</magnitude>
</node>
<node name="Predicted transmembrane sensor domain"><magnitude>
<val> 128.786845</val>
</magnitude>
</node>
<node name="Protein tyrosine phosphatase"><magnitude>
<val> 39.952147</val>
</magnitude>
</node>
<node name="Protein tyrosine/serine phosphatase"><magnitude>
<val> 51.949879</val>
</magnitude>
</node>
<node name="Protein-tyrosine-phosphatase"><magnitude>
<val> 216.542483</val>
</magnitude>
</node>
<node name="Putative Ser protein kinase"><magnitude>
<val> 182.287758</val>
</magnitude>
</node>
<node name="Putative regulator of cell autolysis"><magnitude>
<val> 543.80967</val>
</magnitude>
</node>
<node name="RING-finger-containing E3 ubiquitin ligase"><magnitude>
<val> 8.532906</val>
</magnitude>
</node>
<node name="RIO-like serine/threonine protein kinase fused to N-terminal HTH domain"><magnitude>
<val> 72.975608</val>
</magnitude>
</node>
<node name="Ran GTPase-activating protein (RanGAP) involved in mRNA processing and transport"><magnitude>
<val> 50.966402</val>
</magnitude>
</node>
<node name="RecA-superfamily ATPases implicated in signal transduction"><magnitude>
<val> 266.274543</val>
</magnitude>
</node>
<node name="Regulator of polyketide synthase expression"><magnitude>
<val> 48.441226</val>
</magnitude>
</node>
<node name="Response regulator"><magnitude>
<val> 150.304301</val>
</magnitude>
</node>
<node name="Response regulator consisting of a CheY-like receiver domain and a Fis-type HTH domain"><magnitude>
<val> 26.823931</val>
</magnitude>
</node>
<node name="Response regulator containing CheY-like receiver and SARP domains"><magnitude>
<val> 69.126633</val>
</magnitude>
</node>
<node name="Response regulator containing CheY-like receiver domain and AraC-type DNA-binding domain"><magnitude>
<val> 203.025891</val>
</magnitude>
</node>
<node name="Response regulator containing CheY-like receiver, AAA-type ATPase, and DNA-binding domains"><magnitude>
<val> 3503.617713</val>
</magnitude>
</node>
<node name="Response regulator containing a CheY-like receiver domain and a GGDEF domain"><magnitude>
<val> 617.005358</val>
</magnitude>
</node>
<node name="Response regulator containing a CheY-like receiver domain and an HTH DNA-binding domain"><magnitude>
<val> 917.918385</val>
</magnitude>
</node>
<node name="Response regulator with putative antiterminator output domain"><magnitude>
<val> 97.924609</val>
</magnitude>
</node>
<node name="Response regulators consisting of a CheY-like receiver domain and a winged-helix DNA-binding domain"><magnitude>
<val> 2015.205497</val>
</magnitude>
</node>
<node name="RhoGEF, Guanine nucleotide exchange factor for Rho/Rac/Cdc42-like GTPases"><magnitude>
<val> 156.757793</val>
</magnitude>
</node>
<node name="SH3 domain protein"><magnitude>
<val> 49.177738</val>
</magnitude>
</node>
<node name="SPX domain-containing protein"><magnitude>
<val> .902062</val>
</magnitude>
</node>
<node name="Serine kinase of the HPr protein, regulates carbohydrate metabolism"><magnitude>
<val> 180.812444</val>
</magnitude>
</node>
<node name="Serine phosphatase RsbU, regulator of sigma subunit"><magnitude>
<val> 563.383831</val>
</magnitude>
</node>
<node name="Serine/threonine protein kinase involved in cell cycle control"><magnitude>
<val> 96.058047</val>
</magnitude>
</node>
<node name="Serine/threonine protein phosphatase"><magnitude>
<val> 170.95937</val>
</magnitude>
</node>
<node name="Serine/threonine protein phosphatase 2A, regulatory subunit"><magnitude>
<val> 39.49275</val>
</magnitude>
</node>
<node name="Signal transduction histidine kinase"><magnitude>
<val> 11093.032896</val>
</magnitude>
</node>
<node name="Signal transduction histidine kinase involved in nitrogen fixation and metabolism regulation"><magnitude>
<val> 422.097289</val>
</magnitude>
</node>
<node name="Signal transduction histidine kinase regulating C4-dicarboxylate transport system"><magnitude>
<val> 762.314456</val>
</magnitude>
</node>
<node name="Signal transduction histidine kinase regulating citrate/malate metabolism"><magnitude>
<val> 313.871714</val>
</magnitude>
</node>
<node name="Signal transduction histidine kinase regulating phosphoglycerate transport system"><magnitude>
<val> 165.805211</val>
</magnitude>
</node>
<node name="Signal transduction histidine kinase, glucose-6-phosphate specific"><magnitude>
<val> 49.282783</val>
</magnitude>
</node>
<node name="Signal transduction histidine kinase, nitrate/nitrite-specific"><magnitude>
<val> 161.032627</val>
</magnitude>
</node>
<node name="Signal transduction histidine kinase, nitrogen specific"><magnitude>
<val> 197.438039</val>
</magnitude>
</node>
<node name="Signal transduction protein containing GAF and PtsI domains"><magnitude>
<val> 106.160427</val>
</magnitude>
</node>
<node name="Stress-induced morphogen (activity unknown)"><magnitude>
<val> 4.850696</val>
</magnitude>
</node>
<node name="Transcriptional activator, adenine-specific DNA methyltransferase"><magnitude>
<val> 12.258676</val>
</magnitude>
</node>
<node name="Tryptophan-rich sensory protein (mitochondrial benzodiazepine receptor homolog)"><magnitude>
<val> 129.539639</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein, contains FHA domain"><magnitude>
<val> 81.7695</val>
</magnitude>
</node>
<node name="Uncharacterized domain of BarA-like signal transduction histidine kinases"><magnitude>
<val> 5.022817</val>
</magnitude>
</node>
<node name="Uncharacterized proteins involved in stress response, homologs of TerZ and putative cAMP-binding protein CABP1"><magnitude>
<val> 93.379688</val>
</magnitude>
</node>
<node name="Universal stress protein UspA and related nucleotide-binding proteins"><magnitude>
<val> 1056.294339</val>
</magnitude>
</node>
<node name="cAMP-binding proteins - catabolite gene activator and regulatory subunit of cAMP-dependent protein kinases"><magnitude>
<val> 672.390161</val>
</magnitude>
</node>
</node>
</node>
<node name="INFORMATION STORAGE AND PROCESSING"><magnitude>
<val> 140495.303842</val>
</magnitude>
<node name="Chromatin structure and dynamics"><magnitude>
<val> 1046.934241</val>
</magnitude>
<node name="Chromatin-associated proteins containing the HMG domain"><magnitude>
<val> 67.793353</val>
</magnitude>
</node>
<node name="Chromosome condensation complex Condensin, subunit D2"><magnitude>
<val> 78.27601</val>
</magnitude>
</node>
<node name="Chromosome condensation complex Condensin, subunit G"><magnitude>
<val> 76.661993</val>
</magnitude>
</node>
<node name="Chromosome condensation complex Condensin, subunit H"><magnitude>
<val> 76.424459</val>
</magnitude>
</node>
<node name="Deacetylases, including yeast histone deacetylase and acetoin utilization protein"><magnitude>
<val> 310.779181</val>
</magnitude>
</node>
<node name="Histone acetyltransferase (MYST family)"><magnitude>
<val> 16.571104</val>
</magnitude>
</node>
<node name="Histone acetyltransferase complex SAGA/ADA, subunit ADA2"><magnitude>
<val> 18.618489</val>
</magnitude>
</node>
<node name="Histone deacetylase complex, SIN3 component"><magnitude>
<val> 111.224535</val>
</magnitude>
</node>
<node name="Histones H3 and H4"><magnitude>
<val> 169.781217</val>
</magnitude>
</node>
<node name="RSC chromatin remodeling complex subunit RSC8"><magnitude>
<val> 85.477926</val>
</magnitude>
</node>
<node name="SWIB-domain-containing proteins implicated in chromatin remodeling"><magnitude>
<val> 5.221198</val>
</magnitude>
</node>
<node name="Transcription factor involved in chromatin remodeling, contains bromodomain"><magnitude>
<val> 30.104776</val>
</magnitude>
</node>
</node>
<node name="RNA processing and modification"><magnitude>
<val> 1516.157729</val>
</magnitude>
<node name="Oligoribonuclease (3'->5' exoribonuclease)"><magnitude>
<val> 162.836945</val>
</magnitude>
</node>
<node name="Poly(A) polymerase"><magnitude>
<val> 51.802867</val>
</magnitude>
</node>
<node name="Pre-mRNA 3'-end processing (cleavage and polyadenylation) factor"><magnitude>
<val> 5.697529</val>
</magnitude>
</node>
<node name="Pre-mRNA cleavage and polyadenylation factor IA/II complex, subunit CLP1"><magnitude>
<val> 27.528525</val>
</magnitude>
</node>
<node name="Pre-mRNA cleavage and polyadenylation specificity factor"><magnitude>
<val> 175.287493</val>
</magnitude>
</node>
<node name="Protein interacting with poly(A)-binding protein"><magnitude>
<val> 75.428445</val>
</magnitude>
</node>
<node name="Protein involved in mRNA turnover and stability"><magnitude>
<val> 182.213474</val>
</magnitude>
</node>
<node name="RNA 3'-terminal phosphate cyclase"><magnitude>
<val> 107.763636</val>
</magnitude>
</node>
<node name="Splicing factor"><magnitude>
<val> 24.381532</val>
</magnitude>
</node>
<node name="Splicing factor (branch point binding protein)"><magnitude>
<val> 9.725443</val>
</magnitude>
</node>
<node name="Splicing factor 3a, subunit 2"><magnitude>
<val> 12.258676</val>
</magnitude>
</node>
<node name="Splicing factor 3a, subunit 3"><magnitude>
<val> 65.100421</val>
</magnitude>
</node>
<node name="Splicing factor 3b, subunit 2"><magnitude>
<val> 23.379984</val>
</magnitude>
</node>
<node name="U1 snRNP component, mediates U1 snRNP association with cap-binding complex"><magnitude>
<val> 6.636741</val>
</magnitude>
</node>
<node name="U2 snRNP spliceosome subunit"><magnitude>
<val> 78.064484</val>
</magnitude>
</node>
<node name="U5 snRNP spliceosome subunit"><magnitude>
<val> 480.921764</val>
</magnitude>
</node>
<node name="mRNA capping enzyme, guanylyltransferase (alpha) subunit"><magnitude>
<val> 18.471767</val>
</magnitude>
</node>
<node name="mRNA deadenylase subunit"><magnitude>
<val> 8.658003</val>
</magnitude>
</node>
</node>
<node name="Replication, recombination and repair"><magnitude>
<val> 53309.669741</val>
</magnitude>
<node name="3-methyladenine DNA glycosylase"><magnitude>
<val> 573.768613</val>
</magnitude>
</node>
<node name="3-methyladenine DNA glycosylase/8-oxoguanine DNA glycosylase"><magnitude>
<val> 190.525452</val>
</magnitude>
</node>
<node name="5'-3' exonuclease"><magnitude>
<val> 90.108603</val>
</magnitude>
</node>
<node name="5'-3' exonuclease (including N-terminal domain of PolI)"><magnitude>
<val> 196.500376</val>
</magnitude>
</node>
<node name="A/G-specific DNA glycosylase"><magnitude>
<val> 139.569618</val>
</magnitude>
</node>
<node name="ATP-dependent DNA ligase"><magnitude>
<val> 235.936658</val>
</magnitude>
</node>
<node name="ATP-dependent DNA ligase, homolog of eukaryotic ligase III"><magnitude>
<val> 98.157112</val>
</magnitude>
</node>
<node name="ATP-dependent exoDNAse (exonuclease V) beta subunit (contains helicase and exonuclease domains)"><magnitude>
<val> 311.042086</val>
</magnitude>
</node>
<node name="ATP-dependent exoDNAse (exonuclease V), alpha subunit - helicase superfamily I member"><magnitude>
<val> 535.725593</val>
</magnitude>
</node>
<node name="ATP-dependent nuclease, subunit B"><magnitude>
<val> 215.650281</val>
</magnitude>
</node>
<node name="ATPase involved in DNA repair"><magnitude>
<val> 804.173403</val>
</magnitude>
</node>
<node name="ATPase involved in DNA replication"><magnitude>
<val> 38.414828</val>
</magnitude>
</node>
<node name="ATPase involved in DNA replication initiation"><magnitude>
<val> 300.043679</val>
</magnitude>
</node>
<node name="ATPase related to the helicase subunit of the Holliday junction resolvase"><magnitude>
<val> 290.326054</val>
</magnitude>
</node>
<node name="Adenine specific DNA methylase Mod"><magnitude>
<val> 211.798797</val>
</magnitude>
</node>
<node name="Adenine-specific DNA methylase"><magnitude>
<val> 97.001483</val>
</magnitude>
</node>
<node name="Adenine-specific DNA methylase containing a Zn-ribbon"><magnitude>
<val> 255.678184</val>
</magnitude>
</node>
<node name="Alkylated DNA repair protein"><magnitude>
<val> 27.995848</val>
</magnitude>
</node>
<node name="Antirestriction protein"><magnitude>
<val> 59.519431</val>
</magnitude>
</node>
<node name="Archaea-specific RecJ-like exonuclease, contains DnaJ-type Zn finger domain"><magnitude>
<val> 294.822838</val>
</magnitude>
</node>
<node name="Archaeal DNA polymerase II, large subunit"><magnitude>
<val> 146.515759</val>
</magnitude>
</node>
<node name="Archaeal DNA polymerase II, small subunit/DNA polymerase delta, subunit B"><magnitude>
<val> 108.671721</val>
</magnitude>
</node>
<node name="Bacterial nucleoid DNA-binding protein"><magnitude>
<val> 1082.536988</val>
</magnitude>
</node>
<node name="Cdc6-related protein, AAA superfamily ATPase"><magnitude>
<val> 201.614144</val>
</magnitude>
</node>
<node name="DNA G:T-mismatch repair endonuclease"><magnitude>
<val> 53.153727</val>
</magnitude>
</node>
<node name="DNA alkylation repair enzyme"><magnitude>
<val> 23.491665</val>
</magnitude>
</node>
<node name="DNA endonuclease related to intein-encoded endonucleases"><magnitude>
<val> 9.858758</val>
</magnitude>
</node>
<node name="DNA excision repair protein"><magnitude>
<val> 51.581144</val>
</magnitude>
</node>
<node name="DNA gyrase inhibitor"><magnitude>
<val> 56.131091</val>
</magnitude>
</node>
<node name="DNA mismatch repair enzyme (predicted ATPase)"><magnitude>
<val> 320.11208</val>
</magnitude>
</node>
<node name="DNA mismatch repair protein"><magnitude>
<val> 24.141628</val>
</magnitude>
</node>
<node name="DNA modification methylase"><magnitude>
<val> 333.354018</val>
</magnitude>
</node>
<node name="DNA polymerase I - 3'-5' exonuclease and polymerase domains"><magnitude>
<val> 280.285416</val>
</magnitude>
</node>
<node name="DNA polymerase III, alpha subunit"><magnitude>
<val> 417.227093</val>
</magnitude>
</node>
<node name="DNA polymerase III, alpha subunit (gram-positive type)"><magnitude>
<val> 325.238396</val>
</magnitude>
</node>
<node name="DNA polymerase III, chi subunit"><magnitude>
<val> 23.913793</val>
</magnitude>
</node>
<node name="DNA polymerase III, delta subunit"><magnitude>
<val> 205.431371</val>
</magnitude>
</node>
<node name="DNA polymerase III, epsilon subunit and related 3'-5' exonucleases"><magnitude>
<val> 149.582721</val>
</magnitude>
</node>
<node name="DNA polymerase III, gamma/tau subunits"><magnitude>
<val> 485.072418</val>
</magnitude>
</node>
<node name="DNA polymerase III, psi subunit"><magnitude>
<val> 9.095009</val>
</magnitude>
</node>
<node name="DNA polymerase IV (family X)"><magnitude>
<val> 134.463194</val>
</magnitude>
</node>
<node name="DNA polymerase alpha-primase complex, polymerase-associated subunit B"><magnitude>
<val> 10.858011</val>
</magnitude>
</node>
<node name="DNA polymerase elongation subunit (family B)"><magnitude>
<val> 333.380015</val>
</magnitude>
</node>
<node name="DNA polymerase sigma"><magnitude>
<val> 27.284186</val>
</magnitude>
</node>
<node name="DNA polymerase sliding clamp subunit (PCNA homolog)"><magnitude>
<val> 290.568654</val>
</magnitude>
</node>
<node name="DNA primase (bacterial type)"><magnitude>
<val> 502.399416</val>
</magnitude>
</node>
<node name="DNA recombination-dependent growth factor C"><magnitude>
<val> 147.835857</val>
</magnitude>
</node>
<node name="DNA repair exonuclease"><magnitude>
<val> 420.93196</val>
</magnitude>
</node>
<node name="DNA repair photolyase"><magnitude>
<val> 279.415296</val>
</magnitude>
</node>
<node name="DNA repair protein MMS21"><magnitude>
<val> 5.364664</val>
</magnitude>
</node>
<node name="DNA repair protein RAD4"><magnitude>
<val> 55.19044</val>
</magnitude>
</node>
<node name="DNA repair proteins"><magnitude>
<val> 382.391668</val>
</magnitude>
</node>
<node name="DNA replication protein"><magnitude>
<val> 1264.226534</val>
</magnitude>
</node>
<node name="DNA topoisomerase VI, subunit A"><magnitude>
<val> 109.833952</val>
</magnitude>
</node>
<node name="DNA topoisomerase VI, subunit B"><magnitude>
<val> 118.027452</val>
</magnitude>
</node>
<node name="DNA uptake protein and related DNA-binding proteins"><magnitude>
<val> 171.169808</val>
</magnitude>
</node>
<node name="DNA-damage-inducible protein J"><magnitude>
<val> 71.441246</val>
</magnitude>
</node>
<node name="Deoxyinosine 3'endonuclease (endonuclease V)"><magnitude>
<val> 57.658558</val>
</magnitude>
</node>
<node name="Deoxyribodipyrimidine photolyase"><magnitude>
<val> 203.648873</val>
</magnitude>
</node>
<node name="ERCC4-like helicases"><magnitude>
<val> 150.762488</val>
</magnitude>
</node>
<node name="ERCC4-type nuclease"><magnitude>
<val> 93.748674</val>
</magnitude>
</node>
<node name="Endonuclease I"><magnitude>
<val> 9.612851</val>
</magnitude>
</node>
<node name="Endonuclease IV"><magnitude>
<val> 273.297567</val>
</magnitude>
</node>
<node name="Eukaryotic-type DNA primase, catalytic (small) subunit"><magnitude>
<val> 71.564906</val>
</magnitude>
</node>
<node name="Eukaryotic-type DNA primase, large subunit"><magnitude>
<val> 87.493526</val>
</magnitude>
</node>
<node name="Excinuclease ATPase subunit"><magnitude>
<val> 610.830348</val>
</magnitude>
</node>
<node name="Exonuclease I"><magnitude>
<val> 100.927979</val>
</magnitude>
</node>
<node name="Exonuclease III"><magnitude>
<val> 210.76908</val>
</magnitude>
</node>
<node name="Exonuclease V gamma subunit"><magnitude>
<val> 247.817967</val>
</magnitude>
</node>
<node name="Exonuclease VII small subunit"><magnitude>
<val> 152.315798</val>
</magnitude>
</node>
<node name="Exonuclease VII, large subunit"><magnitude>
<val> 271.963478</val>
</magnitude>
</node>
<node name="FOG: Transposase"><magnitude>
<val> 159.279862</val>
</magnitude>
</node>
<node name="FOG: Transposase and inactivated derivatives"><magnitude>
<val> 1837.10914</val>
</magnitude>
</node>
<node name="Formamidopyrimidine-DNA glycosylase"><magnitude>
<val> 277.41864</val>
</magnitude>
</node>
<node name="G:T/U mismatch-specific DNA glycosylase"><magnitude>
<val> 38.816357</val>
</magnitude>
</node>
<node name="Helicase subunit of the DNA excision repair complex"><magnitude>
<val> 418.259538</val>
</magnitude>
</node>
<node name="Holliday junction resolvase"><magnitude>
<val> 9.358263</val>
</magnitude>
</node>
<node name="Holliday junction resolvase - archaeal type"><magnitude>
<val> 88.820682</val>
</magnitude>
</node>
<node name="Holliday junction resolvasome, DNA-binding subunit"><magnitude>
<val> 143.009817</val>
</magnitude>
</node>
<node name="Holliday junction resolvasome, endonuclease subunit"><magnitude>
<val> 180.176703</val>
</magnitude>
</node>
<node name="Holliday junction resolvasome, helicase subunit"><magnitude>
<val> 205.277907</val>
</magnitude>
</node>
<node name="HrpA-like helicases"><magnitude>
<val> 183.176285</val>
</magnitude>
</node>
<node name="Inactivated superfamily I helicase"><magnitude>
<val> 58.557751</val>
</magnitude>
</node>
<node name="Integrase"><magnitude>
<val> 471.30175</val>
</magnitude>
</node>
<node name="Intein/homing endonuclease"><magnitude>
<val> 44.947782</val>
</magnitude>
</node>
<node name="Methylated DNA-protein cysteine methyltransferase"><magnitude>
<val> 236.506961</val>
</magnitude>
</node>
<node name="Mg-dependent DNase"><magnitude>
<val> 331.706294</val>
</magnitude>
</node>
<node name="Micrococcal nuclease (thermonuclease) homologs"><magnitude>
<val> 196.968298</val>
</magnitude>
</node>
<node name="Mismatch repair ATPase (MutS family)"><magnitude>
<val> 868.109053</val>
</magnitude>
</node>
<node name="N6-adenine-specific methylase"><magnitude>
<val> 244.372255</val>
</magnitude>
</node>
<node name="NAD-dependent DNA ligase (contains BRCT domain type II)"><magnitude>
<val> 357.835349</val>
</magnitude>
</node>
<node name="NTP pyrophosphohydrolases containing a Zn-finger, probably nucleic-acid-binding"><magnitude>
<val> 279.130607</val>
</magnitude>
</node>
<node name="NTP pyrophosphohydrolases including oxidative damage repair enzymes"><magnitude>
<val> 296.141874</val>
</magnitude>
</node>
<node name="Negative regulator of replication initiationR"><magnitude>
<val> 27.458674</val>
</magnitude>
</node>
<node name="Nuclease subunit of the excinuclease complex"><magnitude>
<val> 323.531097</val>
</magnitude>
</node>
<node name="Nucleotide excision repair endonuclease NEF1, RAD10 subunit"><magnitude>
<val> 4.407939</val>
</magnitude>
</node>
<node name="Nucleotidyltransferase/DNA polymerase involved in DNA repair"><magnitude>
<val> 321.710161</val>
</magnitude>
</node>
<node name="Origin recognition complex, subunit 2"><magnitude>
<val> 58.46787</val>
</magnitude>
</node>
<node name="Oxidation resistance protein"><magnitude>
<val> 19.090411</val>
</magnitude>
</node>
<node name="Phage DNA packaging protein, Nu1 subunit of terminase"><magnitude>
<val> 35.345264</val>
</magnitude>
</node>
<node name="Phage terminase, small subunit"><magnitude>
<val> 196.483963</val>
</magnitude>
</node>
<node name="Phage-related protein, predicted endonuclease"><magnitude>
<val> 101.46346</val>
</magnitude>
</node>
<node name="Plasmid replication initiator protein"><magnitude>
<val> 28.163185</val>
</magnitude>
</node>
<node name="Plasmid rolling circle replication initiator protein and truncated derivatives"><magnitude>
<val> 14.750937</val>
</magnitude>
</node>
<node name="Predicted ATP-dependent endonuclease of the OLD family"><magnitude>
<val> 172.58254</val>
</magnitude>
</node>
<node name="Predicted ATPase involved in replication control, Cdc46/Mcm family"><magnitude>
<val> 261.566443</val>
</magnitude>
</node>
<node name="Predicted DNA alkylation repair enzyme"><magnitude>
<val> 163.819064</val>
</magnitude>
</node>
<node name="Predicted DNA modification methylase"><magnitude>
<val> 105.059611</val>
</magnitude>
</node>
<node name="Predicted EndoIII-related endonuclease"><magnitude>
<val> 314.739985</val>
</magnitude>
</node>
<node name="Predicted HD superfamily hydrolase, possibly a nuclease"><magnitude>
<val> 13.691454</val>
</magnitude>
</node>
<node name="Predicted HKD family nuclease"><magnitude>
<val> 30.68557</val>
</magnitude>
</node>
<node name="Predicted N6-adenine-specific DNA methylase"><magnitude>
<val> 140.692497</val>
</magnitude>
</node>
<node name="Predicted NTP pyrophosphohydrolase"><magnitude>
<val> 2.209559</val>
</magnitude>
</node>
<node name="Predicted Rossmann fold nucleotide-binding protein involved in DNA uptake"><magnitude>
<val> 252.982758</val>
</magnitude>
</node>
<node name="Predicted endonuclease (RecB family)"><magnitude>
<val> 13.606822</val>
</magnitude>
</node>
<node name="Predicted endonuclease containing a URI domain"><magnitude>
<val> 96.378482</val>
</magnitude>
</node>
<node name="Predicted endonuclease distantly related to archaeal Holliday junction resolvase"><magnitude>
<val> 157.843637</val>
</magnitude>
</node>
<node name="Predicted endonuclease involved in recombination (possible Holliday junction resolvase in Mycoplasmas and B. subtilis)"><magnitude>
<val> 150.113304</val>
</magnitude>
</node>
<node name="Predicted eukaryotic-type DNA primase"><magnitude>
<val> 63.696344</val>
</magnitude>
</node>
<node name="Predicted exonuclease"><magnitude>
<val> 79.066982</val>
</magnitude>
</node>
<node name="Predicted methylated DNA-protein cysteine methyltransferase"><magnitude>
<val> 28.724577</val>
</magnitude>
</node>
<node name="Predicted nuclease of the RecB family"><magnitude>
<val> 131.571793</val>
</magnitude>
</node>
<node name="Predicted polymerase, most proteins contain PALM domain, HD hydrolase domain and Zn-ribbon domain"><magnitude>
<val> 256.544522</val>
</magnitude>
</node>
<node name="Predicted site-specific integrase-resolvase"><magnitude>
<val> 387.954177</val>
</magnitude>
</node>
<node name="Primosomal protein N' (replication factor Y) - superfamily II helicase"><magnitude>
<val> 381.14728</val>
</magnitude>
</node>
<node name="Primosomal replication protein N"><magnitude>
<val> 59.46618</val>
</magnitude>
</node>
<node name="Primosomal replication protein N''"><magnitude>
<val> 33.787501</val>
</magnitude>
</node>
<node name="Protein involved in initiation of plasmid replication"><magnitude>
<val> 61.140407</val>
</magnitude>
</node>
<node name="Protein kinase essential for the initiation of DNA replication"><magnitude>
<val> 40.510754</val>
</magnitude>
</node>
<node name="Putative phage replication protein RstA"><magnitude>
<val> 31.969809</val>
</magnitude>
</node>
<node name="Putative primosome component and related proteins"><magnitude>
<val> 89.65087</val>
</magnitude>
</node>
<node name="RecA-family ATPase"><magnitude>
<val> 129.243444</val>
</magnitude>
</node>
<node name="RecA/RadA recombinase"><magnitude>
<val> 331.626784</val>
</magnitude>
</node>
<node name="RecB family exonuclease"><magnitude>
<val> 310.024112</val>
</magnitude>
</node>
<node name="RecB-family nuclease"><magnitude>
<val> 12.58856</val>
</magnitude>
</node>
<node name="RecG-like helicase"><magnitude>
<val> 349.191603</val>
</magnitude>
</node>
<node name="Recombination DNA repair protein (RAD52 pathway)"><magnitude>
<val> 31.448273</val>
</magnitude>
</node>
<node name="Recombinational DNA repair ATPase (RecF pathway)"><magnitude>
<val> 219.12725</val>
</magnitude>
</node>
<node name="Recombinational DNA repair protein (RecE pathway)"><magnitude>
<val> 57.730573</val>
</magnitude>
</node>
<node name="Recombinational DNA repair protein (RecF pathway)"><magnitude>
<val> 308.780301</val>
</magnitude>
</node>
<node name="Replication initiation/membrane attachment protein"><magnitude>
<val> 24.076477</val>
</magnitude>
</node>
<node name="Replicative DNA helicase"><magnitude>
<val> 345.505923</val>
</magnitude>
</node>
<node name="Retron-type reverse transcriptase"><magnitude>
<val> 430.604987</val>
</magnitude>
</node>
<node name="Reverse gyrase"><magnitude>
<val> 195.562896</val>
</magnitude>
</node>
<node name="Ribonuclease HI"><magnitude>
<val> 146.83829</val>
</magnitude>
</node>
<node name="Ribonuclease HII"><magnitude>
<val> 210.257601</val>
</magnitude>
</node>
<node name="Ribonuclease HIII"><magnitude>
<val> 15.771356</val>
</magnitude>
</node>
<node name="Single-stranded DNA-binding protein"><magnitude>
<val> 224.728946</val>
</magnitude>
</node>
<node name="Single-stranded DNA-binding replication protein A (RPA), large (70 kD) subunit and related ssDNA-binding proteins"><magnitude>
<val> 95.46961</val>
</magnitude>
</node>
<node name="Single-stranded DNA-binding replication protein A (RPA), medium (30 kD) subunit"><magnitude>
<val> 4.810896</val>
</magnitude>
</node>
<node name="Single-stranded DNA-specific exonuclease"><magnitude>
<val> 355.799835</val>
</magnitude>
</node>
<node name="Site-specific DNA methylase"><magnitude>
<val> 375.583515</val>
</magnitude>
</node>
<node name="Site-specific recombinase"><magnitude>
<val> 194.24872</val>
</magnitude>
</node>
<node name="Site-specific recombinase XerC"><magnitude>
<val> 162.5151</val>
</magnitude>
</node>
<node name="Site-specific recombinase XerD"><magnitude>
<val> 2010.954218</val>
</magnitude>
</node>
<node name="Site-specific recombinases, DNA invertase Pin homologs"><magnitude>
<val> 741.551007</val>
</magnitude>
</node>
<node name="Small primase-like proteins (Toprim domain)"><magnitude>
<val> 133.535692</val>
</magnitude>
</node>
<node name="Superfamily I DNA and RNA helicases"><magnitude>
<val> 652.995778</val>
</magnitude>
</node>
<node name="Superfamily I DNA and RNA helicases and helicase subunits"><magnitude>
<val> 272.433917</val>
</magnitude>
</node>
<node name="Superfamily II DNA and RNA helicases"><magnitude>
<val> 828.253655</val>
</magnitude>
</node>
<node name="Superfamily II DNA helicase"><magnitude>
<val> 453.428669</val>
</magnitude>
</node>
<node name="Superfamily II DNA/RNA helicase required for DNA uptake (late competence protein)"><magnitude>
<val> 100.594514</val>
</magnitude>
</node>
<node name="Superfamily II RNA helicase"><magnitude>
<val> 268.177829</val>
</magnitude>
</node>
<node name="Superfamily II helicase and inactivated derivatives"><magnitude>
<val> 109.022993</val>
</magnitude>
</node>
<node name="Thermostable 8-oxoguanine DNA glycosylase"><magnitude>
<val> 17.792463</val>
</magnitude>
</node>
<node name="Topoisomerase IA"><magnitude>
<val> 555.402706</val>
</magnitude>
</node>
<node name="Topoisomerase IB"><magnitude>
<val> 104.146333</val>
</magnitude>
</node>
<node name="Transcription-associated recombination protein"><magnitude>
<val> 76.521202</val>
</magnitude>
</node>
<node name="Transcription-repair coupling factor (superfamily II helicase)"><magnitude>
<val> 316.172493</val>
</magnitude>
</node>
<node name="Transposase"><magnitude>
<val> 1127.937579</val>
</magnitude>
</node>
<node name="Transposase and inactivated derivatives"><magnitude>
<val> 9921.83556599999</val>
</magnitude>
</node>
<node name="Transposase and inactivated derivatives, IS1 family"><magnitude>
<val> 53.755529</val>
</magnitude>
</node>
<node name="Transposase and inactivated derivatives, IS30 family"><magnitude>
<val> 131.219013</val>
</magnitude>
</node>
<node name="Transposase and inactivated derivatives, IS5 family"><magnitude>
<val> 417.594339</val>
</magnitude>
</node>
<node name="Transposase and inactivated derivatives, TnpA family"><magnitude>
<val> 113.391714</val>
</magnitude>
</node>
<node name="Type IIA topoisomerase (DNA gyrase/topo II, topoisomerase IV), A subunit"><magnitude>
<val> 400.721161</val>
</magnitude>
</node>
<node name="Type IIA topoisomerase (DNA gyrase/topo II, topoisomerase IV), B subunit"><magnitude>
<val> 317.648724</val>
</magnitude>
</node>
<node name="UV damage repair endonuclease"><magnitude>
<val> 20.361367</val>
</magnitude>
</node>
<node name="Uncharacterized protein predicted to be involved in DNA repair"><magnitude>
<val> 885.692744000001</val>
</magnitude>
</node>
<node name="Uncharacterized protein predicted to be involved in DNA repair (RAMP superfamily)"><magnitude>
<val> 477.99487</val>
</magnitude>
</node>
<node name="Uncharacterized protein related to Endonuclease III"><magnitude>
<val> 143.167764</val>
</magnitude>
</node>
<node name="Uracil DNA glycosylase"><magnitude>
<val> 160.69934</val>
</magnitude>
</node>
<node name="Uracil-DNA glycosylase"><magnitude>
<val> 174.521962</val>
</magnitude>
</node>
<node name="Zn-finger domain associated with topoisomerase type I"><magnitude>
<val> 71.290174</val>
</magnitude>
</node>
</node>
<node name="Transcription"><magnitude>
<val> 42937.518849</val>
</magnitude>
<node name="ATP-dependent transcriptional regulator"><magnitude>
<val> 635.190739</val>
</magnitude>
</node>
<node name="Antirepressor regulating drug resistance, predicted signal transduction N-terminal membrane component"><magnitude>
<val> 121.971165</val>
</magnitude>
</node>
<node name="AraC-type DNA-binding domain-containing proteins"><magnitude>
<val> 1997.753913</val>
</magnitude>
</node>
<node name="Archaeal DNA-binding protein"><magnitude>
<val> 82.342151</val>
</magnitude>
</node>
<node name="Arginine repressor"><magnitude>
<val> 45.079195</val>
</magnitude>
</node>
<node name="Barstar, RNAse (barnase) inhibitor"><magnitude>
<val> 4.316067</val>
</magnitude>
</node>
<node name="Biotin operon repressor"><magnitude>
<val> 48.608829</val>
</magnitude>
</node>
<node name="CCAAT-binding factor, subunit C"><magnitude>
<val> 24.076477</val>
</magnitude>
</node>
<node name="CCR4-NOT transcriptional regulation complex, NOT5 subunit"><magnitude>
<val> 32.194036</val>
</magnitude>
</node>
<node name="Cold shock proteins"><magnitude>
<val> 494.667561</val>
</magnitude>
</node>
<node name="DNA helicase TIP49, TBP-interacting protein"><magnitude>
<val> 101.235531</val>
</magnitude>
</node>
<node name="DNA methylase"><magnitude>
<val> 110.244282</val>
</magnitude>
</node>
<node name="DNA or RNA helicases of superfamily II"><magnitude>
<val> 276.270457</val>
</magnitude>
</node>
<node name="DNA-binding HTH domain-containing proteins"><magnitude>
<val> 461.358005</val>
</magnitude>
</node>
<node name="DNA-binding winged-HTH domains"><magnitude>
<val> 76.43874</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase specialized sigma subunit"><magnitude>
<val> 378.676268</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase specialized sigma subunit, sigma24 homolog"><magnitude>
<val> 1791.596827</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase specialized sigma subunit, sigma54 homolog"><magnitude>
<val> 235.86736</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, alpha subunit/40 kD subunit"><magnitude>
<val> 224.026208</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, beta subunit/140 kD subunit"><magnitude>
<val> 552.415162</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, beta' subunit/160 kD subunit"><magnitude>
<val> 451.907006</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, delta subunit"><magnitude>
<val> 51.581144</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, sigma subunit (sigma70/sigma32)"><magnitude>
<val> 543.75592</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, subunit E'"><magnitude>
<val> 64.304477</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, subunit E''"><magnitude>
<val> 8.182043</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, subunit H, RpoH/RPB5"><magnitude>
<val> 72.321529</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, subunit K/omega"><magnitude>
<val> 185.202896</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, subunit L"><magnitude>
<val> 67.252062</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, subunit M/Transcription elongation factor TFIIS"><magnitude>
<val> 116.947839</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, subunit N (RpoN/RPB10)"><magnitude>
<val> 61.37355</val>
</magnitude>
</node>
<node name="DNA-directed RNA polymerase, subunit RPC10 (contains C4-type Zn-finger)"><magnitude>
<val> 18.504197</val>
</magnitude>
</node>
<node name="Dopey and related predicted leucine zipper transcription factors"><magnitude>
<val> 165.279025</val>
</magnitude>
</node>
<node name="Exoribonuclease R"><magnitude>
<val> 256.775487</val>
</magnitude>
</node>
<node name="Factor for inversion stimulation Fis, transcriptional activator"><magnitude>
<val> 15.931613</val>
</magnitude>
</node>
<node name="GATA Zn-finger-containing transcription factor"><magnitude>
<val> 4.770285</val>
</magnitude>
</node>
<node name="General negative regulator of transcription subunit"><magnitude>
<val> 4.049443</val>
</magnitude>
</node>
<node name="Glucitol operon activator"><magnitude>
<val> 38.301638</val>
</magnitude>
</node>
<node name="Glycerol-3-phosphate responsive antiterminator (mRNA-binding)"><magnitude>
<val> 50.020105</val>
</magnitude>
</node>
<node name="Heat shock transcription factor"><magnitude>
<val> 27.710704</val>
</magnitude>
</node>
<node name="Histone acetyltransferase"><magnitude>
<val> 262.850526</val>
</magnitude>
</node>
<node name="IkappaB kinase complex, IKAP component"><magnitude>
<val> 177.787227</val>
</magnitude>
</node>
<node name="Mitochondrial DNA-directed RNA polymerase"><magnitude>
<val> 79.053163</val>
</magnitude>
</node>
<node name="Mn-dependent transcriptional regulator"><magnitude>
<val> 392.637553</val>
</magnitude>
</node>
<node name="Myb superfamily proteins, including transcription factors and mRNA splicing factors"><magnitude>
<val> 28.176116</val>
</magnitude>
</node>
<node name="NAD-dependent protein deacetylases, SIR2 family"><magnitude>
<val> 272.743314</val>
</magnitude>
</node>
<node name="Negative regulator of flagellin synthesis (anti-sigma28 factor)"><magnitude>
<val> 232.453429</val>
</magnitude>
</node>
<node name="Nucleosome binding factor SPN, SPT16 subunit"><magnitude>
<val> 90.054251</val>
</magnitude>
</node>
<node name="Nucleosome-binding factor SPN, POB3 subunit"><magnitude>
<val> 24.898529</val>
</magnitude>
</node>
<node name="Phage anti-repressor protein"><magnitude>
<val> 9.639552</val>
</magnitude>
</node>
<node name="Phage shock protein A (IM30), suppresses sigma54-dependent transcription"><magnitude>
<val> 156.122077</val>
</magnitude>
</node>
<node name="Phenylacetic acid-responsive transcriptional repressor"><magnitude>
<val> 103.527837</val>
</magnitude>
</node>
<node name="Pleiotropic transcriptional repressor"><magnitude>
<val> 53.184972</val>
</magnitude>
</node>
<node name="Predicted RNA polymerase sigma factor containing a TPR repeat domain"><magnitude>
<val> 73.843285</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein homologous to eukaryotic snRNP"><magnitude>
<val> 324.574325</val>
</magnitude>
</node>
<node name="Predicted metal-binding transcription factor"><magnitude>
<val> 64.709434</val>
</magnitude>
</node>
<node name="Predicted nucleic-acid-binding protein implicated in transcription termination"><magnitude>
<val> 106.770587</val>
</magnitude>
</node>
<node name="Predicted nucleotide-binding protein containing TIR -like domain"><magnitude>
<val> 93.440052</val>
</magnitude>
</node>
<node name="Predicted transcription factor, homolog of eukaryotic MBF1"><magnitude>
<val> 167.933261</val>
</magnitude>
</node>
<node name="Predicted transcription regulator containing HTH domain"><magnitude>
<val> 52.714918</val>
</magnitude>
</node>
<node name="Predicted transcription regulator, contains HTH domain (MarR family)"><magnitude>
<val> 17.929997</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator"><magnitude>
<val> 3842.864142</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator containing CBS domains"><magnitude>
<val> 47.857394</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator containing an HTH domain and an uncharacterized domain shared with the mammalian protein Schlafen"><magnitude>
<val> 515.018647</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator containing an HTH domain fused to a Zn-ribbon"><magnitude>
<val> 69.369575</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator containing the HTH domain"><magnitude>
<val> 116.384863</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator with C-terminal CBS domains"><magnitude>
<val> 230.996934</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator with an HTH domain"><magnitude>
<val> 27.335645</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator, BolA superfamily"><magnitude>
<val> 14.891183</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator, consists of a Zn-ribbon and ATP-cone domains"><magnitude>
<val> 148.786164</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator, contains C-terminal CBS domains"><magnitude>
<val> 360.11408</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator/sugar kinase"><magnitude>
<val> 130.0082</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulators"><magnitude>
<val> 4998.783316</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulators containing the CopG/Arc/MetJ DNA-binding domain"><magnitude>
<val> 56.051085</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulators containing the CopG/Arc/MetJ DNA-binding domain and a metal-binding domain"><magnitude>
<val> 422.922109</val>
</magnitude>
</node>
<node name="Predicted transmembrane transcriptional regulator (anti-sigma factor)"><magnitude>
<val> 163.35665</val>
</magnitude>
</node>
<node name="Prophage antirepressor"><magnitude>
<val> 134.217455</val>
</magnitude>
</node>
<node name="Putative stress-responsive transcriptional regulator"><magnitude>
<val> 69.850742</val>
</magnitude>
</node>
<node name="Putative transcription activator"><magnitude>
<val> 35.801906</val>
</magnitude>
</node>
<node name="Putative transcriptional regulator"><magnitude>
<val> 5.251993</val>
</magnitude>
</node>
<node name="Putative transcriptional regulator, homolog of Bvg accessory factor"><magnitude>
<val> 165.581112</val>
</magnitude>
</node>
<node name="Putative transcriptional repressor regulating G2/M transition"><magnitude>
<val> 75.476422</val>
</magnitude>
</node>
<node name="RNA polymerase II assessory factor"><magnitude>
<val> 25.511488</val>
</magnitude>
</node>
<node name="RNA polymerase II transcription initiation/nucleotide excision repair factor TFIIH, subunit SSL1"><magnitude>
<val> 31.122428</val>
</magnitude>
</node>
<node name="RNA polymerase II transcription initiation/nucleotide excision repair factor TFIIH, subunit TFB2"><magnitude>
<val> 32.661731</val>
</magnitude>
</node>
<node name="RNA polymerase II transcription initiation/nucleotide excision repair factor TFIIH, subunit TFB4"><magnitude>
<val> 9.129565</val>
</magnitude>
</node>
<node name="RNA polymerase II transcriptional regulation mediator"><magnitude>
<val> 18.47723</val>
</magnitude>
</node>
<node name="RNA polymerase II-interacting protein involved in transcription start site selection"><magnitude>
<val> 15.270669</val>
</magnitude>
</node>
<node name="Rad3-related DNA helicases"><magnitude>
<val> 400.102146</val>
</magnitude>
</node>
<node name="Regulator of arginine metabolism and related MADS box-containing transcription factors"><magnitude>
<val> 53.434198</val>
</magnitude>
</node>
<node name="Regulator of competence-specific genes"><magnitude>
<val> 17.994576</val>
</magnitude>
</node>
<node name="Regulator of nitric oxide reductase transcription"><magnitude>
<val> 89.540788</val>
</magnitude>
</node>
<node name="Regulator of sigma D"><magnitude>
<val> 23.713205</val>
</magnitude>
</node>
<node name="Regulators of stationary/sporulation gene expression"><magnitude>
<val> 376.744186</val>
</magnitude>
</node>
<node name="Response regulator containing a CheY-like receiver domain and an HD-GYP domain"><magnitude>
<val> 443.986337</val>
</magnitude>
</node>
<node name="Response regulator of citrate/malate metabolism"><magnitude>
<val> 136.86351</val>
</magnitude>
</node>
<node name="Response regulator of the LytR/AlgR family"><magnitude>
<val> 561.689792</val>
</magnitude>
</node>
<node name="SOS-response transcriptional repressors (RecA-mediated autopeptidases)"><magnitude>
<val> 382.714802</val>
</magnitude>
</node>
<node name="Sigma54-dependent transcription regulator containing an AAA-type ATPase domain and a DNA-binding domain"><magnitude>
<val> 37.293031</val>
</magnitude>
</node>
<node name="Small nuclear ribonucleoprotein (snRNP) homolog"><magnitude>
<val> 153.93923</val>
</magnitude>
</node>
<node name="Sugar diacid utilization regulator"><magnitude>
<val> 120.615365</val>
</magnitude>
</node>
<node name="Superfamily II DNA/RNA helicases, SNF2 family"><magnitude>
<val> 681.985018</val>
</magnitude>
</node>
<node name="TATA-box binding protein (TBP), component of TFIID and TFIIIB"><magnitude>
<val> 183.593523</val>
</magnitude>
</node>
<node name="TFIIF-interacting CTD phosphatases, including NLI-interacting factor"><magnitude>
<val> 7.868342</val>
</magnitude>
</node>
<node name="Transcription antiterminator"><magnitude>
<val> 512.735412</val>
</magnitude>
</node>
<node name="Transcription elongation factor"><magnitude>
<val> 665.369671</val>
</magnitude>
</node>
<node name="Transcription elongation factor SPT4"><magnitude>
<val> 2.718367</val>
</magnitude>
</node>
<node name="Transcription factor homologous to NACalpha-BTF3"><magnitude>
<val> 105.12096</val>
</magnitude>
</node>
<node name="Transcription factor of the Forkhead/HNF3 family"><magnitude>
<val> 4.142692</val>
</magnitude>
</node>
<node name="Transcription initiation factor IIA, gamma subunit"><magnitude>
<val> 7.475633</val>
</magnitude>
</node>
<node name="Transcription initiation factor IIA, large chain"><magnitude>
<val> 26.015311</val>
</magnitude>
</node>
<node name="Transcription initiation factor IIE, alpha subunit"><magnitude>
<val> 78.130468</val>
</magnitude>
</node>
<node name="Transcription initiation factor IIE, beta subunit"><magnitude>
<val> 29.772029</val>
</magnitude>
</node>
<node name="Transcription initiation factor IIF, auxiliary subunit"><magnitude>
<val> 48.695047</val>
</magnitude>
</node>
<node name="Transcription initiation factor IIF, small subunit (RAP30)"><magnitude>
<val> 36.52272</val>
</magnitude>
</node>
<node name="Transcription initiation factor TFIID, subunit TAF1"><magnitude>
<val> 108.486705</val>
</magnitude>
</node>
<node name="Transcription initiation factor TFIID, subunit TAF12 (also component of histone acetyltransferase SAGA)"><magnitude>
<val> 18.821031</val>
</magnitude>
</node>
<node name="Transcription initiation factor TFIID, subunit TAF13"><magnitude>
<val> 3.894118</val>
</magnitude>
</node>
<node name="Transcription initiation factor TFIID, subunit TAF6 (also component of histone acetyltransferase SAGA)"><magnitude>
<val> 62.739205</val>
</magnitude>
</node>
<node name="Transcription initiation factor TFIIIB, Brf1 subunit/Transcription initiation factor TFIIB"><magnitude>
<val> 177.710889</val>
</magnitude>
</node>
<node name="Transcription regulator of the Arc/MetJ class"><magnitude>
<val> 36.083526</val>
</magnitude>
</node>
<node name="Transcription termination factor"><magnitude>
<val> 362.782836</val>
</magnitude>
</node>
<node name="Transcriptional accessory protein"><magnitude>
<val> 276.748823</val>
</magnitude>
</node>
<node name="Transcriptional antiterminator"><magnitude>
<val> 249.45839</val>
</magnitude>
</node>
<node name="Transcriptional regulator"><magnitude>
<val> 4103.072104</val>
</magnitude>
</node>
<node name="Transcriptional regulator containing GAF, AAA-type ATPase, and DNA binding domains"><magnitude>
<val> 208.066276</val>
</magnitude>
</node>
<node name="Transcriptional regulator containing PAS, AAA-type ATPase, and DNA-binding domains"><magnitude>
<val> 1153.663717</val>
</magnitude>
</node>
<node name="Transcriptional regulator containing an amidase domain and an AraC-type DNA-binding HTH domain"><magnitude>
<val> 252.139391</val>
</magnitude>
</node>
<node name="Transcriptional regulator of a riboflavin/FAD biosynthetic operon"><magnitude>
<val> 113.88244</val>
</magnitude>
</node>
<node name="Transcriptional regulator of aromatic amino acids metabolism"><magnitude>
<val> 143.442868</val>
</magnitude>
</node>
<node name="Transcriptional regulator of heat shock gene"><magnitude>
<val> 165.871635</val>
</magnitude>
</node>
<node name="Transcriptional regulator, contains sigma factor-related N-terminal domain"><magnitude>
<val> 135.209822</val>
</magnitude>
</node>
<node name="Transcriptional regulator, effector-binding domain/component"><magnitude>
<val> 259.997671</val>
</magnitude>
</node>
<node name="Transcriptional regulator/sugar kinase"><magnitude>
<val> 386.764066</val>
</magnitude>
</node>
<node name="Transcriptional regulators"><magnitude>
<val> 3616.258816</val>
</magnitude>
</node>
<node name="Transcriptional regulators containing a DNA-binding HTH domain and an aminotransferase domain (MocR family) and their eukaryotic orthologs"><magnitude>
<val> 521.138677</val>
</magnitude>
</node>
<node name="Transcriptional regulators containing an AAA-type ATPase domain and a DNA-binding domain"><magnitude>
<val> 215.176742</val>
</magnitude>
</node>
<node name="Transcriptional regulators of sugar metabolism"><magnitude>
<val> 347.896755</val>
</magnitude>
</node>
<node name="Transcriptional regulators, similar to M. xanthus CarD"><magnitude>
<val> 62.082657</val>
</magnitude>
</node>
<node name="Transcriptional repressor"><magnitude>
<val> 16.10849</val>
</magnitude>
</node>
<node name="Transcriptional repressor of class III stress genes"><magnitude>
<val> 66.531884</val>
</magnitude>
</node>
<node name="Trp operon repressor"><magnitude>
<val> 4.848005</val>
</magnitude>
</node>
<node name="dsRNA-specific ribonuclease"><magnitude>
<val> 233.227882</val>
</magnitude>
</node>
</node>
<node name="Translation, ribosomal structure and biogenesis"><magnitude>
<val> 41685.023282</val>
</magnitude>
<node name="16S RNA G1207 methylase RsmC"><magnitude>
<val> 86.999254</val>
</magnitude>
</node>
<node name="16S rRNA uridine-516 pseudouridylate synthase and related pseudouridylate synthases"><magnitude>
<val> 255.353497</val>
</magnitude>
</node>
<node name="2'-5' RNA ligase"><magnitude>
<val> 185.697465</val>
</magnitude>
</node>
<node name="2-methylthioadenine synthetase"><magnitude>
<val> 600.770659</val>
</magnitude>
</node>
<node name="23S rRNA methylase"><magnitude>
<val> 174.181685</val>
</magnitude>
</node>
<node name="Acetyltransferases, including N-acetylases of ribosomal proteins"><magnitude>
<val> 495.939298</val>
</magnitude>
</node>
<node name="Alanyl-tRNA synthetase"><magnitude>
<val> 652.458298</val>
</magnitude>
</node>
<node name="Aminoglycoside phosphotransferase"><magnitude>
<val> 8.556764</val>
</magnitude>
</node>
<node name="Archaeal Glu-tRNAGln amidotransferase subunit E (contains GAD domain)"><magnitude>
<val> 184.525853</val>
</magnitude>
</node>
<node name="Arginyl-tRNA synthetase"><magnitude>
<val> 465.49008</val>
</magnitude>
</node>
<node name="Asp-tRNAAsn/Glu-tRNAGln amidotransferase A subunit and related amidases"><magnitude>
<val> 354.349338</val>
</magnitude>
</node>
<node name="Asp-tRNAAsn/Glu-tRNAGln amidotransferase B subunit (PET112 homolog)"><magnitude>
<val> 331.5781</val>
</magnitude>
</node>
<node name="Asp-tRNAAsn/Glu-tRNAGln amidotransferase C subunit"><magnitude>
<val> 171.826277</val>
</magnitude>
</node>
<node name="Aspartyl-tRNA synthetase"><magnitude>
<val> 266.142824</val>
</magnitude>
</node>
<node name="Aspartyl/asparaginyl-tRNA synthetases"><magnitude>
<val> 285.927081</val>
</magnitude>
</node>
<node name="Cysteinyl-tRNA synthetase"><magnitude>
<val> 305.531818</val>
</magnitude>
</node>
<node name="Cytotoxic translational repressor of toxin-antitoxin stability system"><magnitude>
<val> 332.406467</val>
</magnitude>
</node>
<node name="D-Tyr-tRNAtyr deacylase"><magnitude>
<val> 174.427247</val>
</magnitude>
</node>
<node name="Dimethyladenosine transferase (rRNA methylation)"><magnitude>
<val> 299.643001</val>
</magnitude>
</node>
<node name="Diphthamide biosynthesis methyltransferase"><magnitude>
<val> 121.100591</val>
</magnitude>
</node>
<node name="Diphthamide synthase subunit DPH2"><magnitude>
<val> 71.401892</val>
</magnitude>
</node>
<node name="Fibrillarin-like rRNA methylase"><magnitude>
<val> 82.86284</val>
</magnitude>
</node>
<node name="GTP-binding protein required for 40S ribosome biogenesis"><magnitude>
<val> 134.906976</val>
</magnitude>
</node>
<node name="GTPases - translation elongation factors"><magnitude>
<val> 269.930158</val>
</magnitude>
</node>
<node name="Glutamyl- and glutaminyl-tRNA synthetases"><magnitude>
<val> 618.004212</val>
</magnitude>
</node>
<node name="Glycyl-tRNA synthetase (class II)"><magnitude>
<val> 243.387151</val>
</magnitude>
</node>
<node name="Glycyl-tRNA synthetase, alpha subunit"><magnitude>
<val> 150.795613</val>
</magnitude>
</node>
<node name="Glycyl-tRNA synthetase, beta subunit"><magnitude>
<val> 328.080565</val>
</magnitude>
</node>
<node name="Histidyl-tRNA synthetase"><magnitude>
<val> 455.043802</val>
</magnitude>
</node>
<node name="Isoleucyl-tRNA synthetase"><magnitude>
<val> 359.707292</val>
</magnitude>
</node>
<node name="Leucyl-tRNA synthetase"><magnitude>
<val> 403.099368</val>
</magnitude>
</node>
<node name="Lysyl-tRNA synthetase (class I)"><magnitude>
<val> 186.346237</val>
</magnitude>
</node>
<node name="Lysyl-tRNA synthetase (class II)"><magnitude>
<val> 574.031491</val>
</magnitude>
</node>
<node name="Methionine aminopeptidase"><magnitude>
<val> 321.724782</val>
</magnitude>
</node>
<node name="Methionyl-tRNA formyltransferase"><magnitude>
<val> 222.066052</val>
</magnitude>
</node>
<node name="Methionyl-tRNA synthetase"><magnitude>
<val> 416.63553</val>
</magnitude>
</node>
<node name="Methylase of polypeptide chain release factors"><magnitude>
<val> 383.421363</val>
</magnitude>
</node>
<node name="N-formylmethionyl-tRNA deformylase"><magnitude>
<val> 254.64816</val>
</magnitude>
</node>
<node name="N2,N2-dimethylguanosine tRNA methyltransferase"><magnitude>
<val> 170.519725</val>
</magnitude>
</node>
<node name="NAD(FAD)-utilizing enzyme possibly involved in translation"><magnitude>
<val> 194.029484</val>
</magnitude>
</node>
<node name="NMD protein affecting ribosome stability and mRNA decay"><magnitude>
<val> 198.190495</val>
</magnitude>
</node>
<node name="Nucleic-acid-binding protein possibly involved in ribosomal biogenesis"><magnitude>
<val> 50.064731</val>
</magnitude>
</node>
<node name="PUA domain (predicted RNA-binding domain)"><magnitude>
<val> 21.512063</val>
</magnitude>
</node>
<node name="Peptide chain release factor 1 (eRF1)"><magnitude>
<val> 181.773756</val>
</magnitude>
</node>
<node name="Peptide chain release factor RF-3"><magnitude>
<val> 233.943587</val>
</magnitude>
</node>
<node name="Peptidyl-tRNA hydrolase"><magnitude>
<val> 188.292138</val>
</magnitude>
</node>
<node name="Phenylalanyl-tRNA synthetase alpha subunit"><magnitude>
<val> 301.449593</val>
</magnitude>
</node>
<node name="Phenylalanyl-tRNA synthetase alpha subunit (archaeal type)"><magnitude>
<val> 107.134008</val>
</magnitude>
</node>
<node name="Phenylalanyl-tRNA synthetase beta subunit"><magnitude>
<val> 346.819153</val>
</magnitude>
</node>
<node name="Polyribonucleotide nucleotidyltransferase (polynucleotide phosphorylase)"><magnitude>
<val> 258.259691</val>
</magnitude>
</node>
<node name="Predicted GTPase, probable translation factor"><magnitude>
<val> 255.022417</val>
</magnitude>
</node>
<node name="Predicted RNA binding protein (contains ribosomal protein S1 domain)"><magnitude>
<val> 61.519156</val>
</magnitude>
</node>
<node name="Predicted RNA methylase"><magnitude>
<val> 235.677605</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein"><magnitude>
<val> 90.002771</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein (consists of S1 domain and a Zn-ribbon domain)"><magnitude>
<val> 58.504151</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein (contains PUA domain)"><magnitude>
<val> 72.714254</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein containing KH domain, possibly ribosomal protein"><magnitude>
<val> 112.565701</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein of the translin family"><magnitude>
<val> 39.346105</val>
</magnitude>
</node>
<node name="Predicted Zn-ribbon RNA-binding protein with a function in translation"><magnitude>
<val> 8.913369</val>
</magnitude>
</node>
<node name="Predicted exonuclease of the beta-lactamase fold involved in RNA processing"><magnitude>
<val> 144.051071</val>
</magnitude>
</node>
<node name="Predicted exosome subunit"><magnitude>
<val> 197.999122</val>
</magnitude>
</node>
<node name="Predicted exosome subunit/U3 small nucleolar ribonucleoprotein (snoRNP) component, contains IMP4 domain"><magnitude>
<val> 60.316241</val>
</magnitude>
</node>
<node name="Predicted pseudouridylate synthase"><magnitude>
<val> 79.575074</val>
</magnitude>
</node>
<node name="Predicted rRNA methylase"><magnitude>
<val> 222.399562</val>
</magnitude>
</node>
<node name="Predicted rRNA methylase (SpoU class)"><magnitude>
<val> 153.226364</val>
</magnitude>
</node>
<node name="Predicted ribosomal protein"><magnitude>
<val> 48.275881</val>
</magnitude>
</node>
<node name="Predicted subunit of tRNA(5-methylaminomethyl-2-thiouridylate) methyltransferase, contains the PP-loop ATPase domain"><magnitude>
<val> 120.685492</val>
</magnitude>
</node>
<node name="Predicted tRNA(5-methylaminomethyl-2-thiouridylate) methyltransferase, contains the PP-loop ATPase domain"><magnitude>
<val> 497.947177</val>
</magnitude>
</node>
<node name="Predicted translation initiation factor 2B subunit, eIF-2B alpha/beta/delta family"><magnitude>
<val> 223.090472</val>
</magnitude>
</node>
<node name="Prolyl-tRNA synthetase"><magnitude>
<val> 324.512336</val>
</magnitude>
</node>
<node name="Protein chain release factor A"><magnitude>
<val> 306.534303</val>
</magnitude>
</node>
<node name="Protein chain release factor B"><magnitude>
<val> 269.195098</val>
</magnitude>
</node>
<node name="Protein implicated in ribosomal biogenesis, Nop56p homolog"><magnitude>
<val> 105.264747</val>
</magnitude>
</node>
<node name="Protein involved in the nuclear export of pre-ribosomes"><magnitude>
<val> 157.085003</val>
</magnitude>
</node>
<node name="Protein required for biogenesis of the 60S ribosomal subunit"><magnitude>
<val> 51.694116</val>
</magnitude>
</node>
<node name="Pseudouridine synthase"><magnitude>
<val> 284.770211</val>
</magnitude>
</node>
<node name="Pseudouridylate synthase"><magnitude>
<val> 235.448838</val>
</magnitude>
</node>
<node name="Pseudouridylate synthases, 23S RNA-specific"><magnitude>
<val> 471.315115</val>
</magnitude>
</node>
<node name="Putative translation factor (SUA5)"><magnitude>
<val> 242.835421</val>
</magnitude>
</node>
<node name="Putative translation initiation inhibitor, yjgF family"><magnitude>
<val> 286.255886</val>
</magnitude>
</node>
<node name="Queuine tRNA-ribosyltransferases, contain PUA domain"><magnitude>
<val> 94.931463</val>
</magnitude>
</node>
<node name="Queuine/archaeosine tRNA-ribosyltransferase"><magnitude>
<val> 510.192037</val>
</magnitude>
</node>
<node name="RNA-binding protein Rrp4 and related proteins (contain S1 domain and KH domain)"><magnitude>
<val> 58.742328</val>
</magnitude>
</node>
<node name="RNA-binding protein involved in rRNA processing"><magnitude>
<val> 10.965409</val>
</magnitude>
</node>
<node name="RNA-binding protein of the Puf family, translational repressor"><magnitude>
<val> 107.697784</val>
</magnitude>
</node>
<node name="RNA-binding protein required for 60S ribosomal subunit biogenesis"><magnitude>
<val> 5.35328</val>
</magnitude>
</node>
<node name="RNA:NAD 2'-phosphotransferase"><magnitude>
<val> 48.02119</val>
</magnitude>
</node>
<node name="RNase P protein component"><magnitude>
<val> 89.797817</val>
</magnitude>
</node>
<node name="RNase P subunit RPR2"><magnitude>
<val> 42.622295</val>
</magnitude>
</node>
<node name="RNase P/RNase MRP subunit POP5"><magnitude>
<val> 58.742328</val>
</magnitude>
</node>
<node name="RNase P/RNase MRP subunit p29"><magnitude>
<val> 61.878969</val>
</magnitude>
</node>
<node name="RNase P/RNase MRP subunit p30"><magnitude>
<val> 63.024227</val>
</magnitude>
</node>
<node name="RNase PH"><magnitude>
<val> 149.208909</val>
</magnitude>
</node>
<node name="RNase PH-related exoribonuclease"><magnitude>
<val> 58.742328</val>
</magnitude>
</node>
<node name="Ribonuclease D"><magnitude>
<val> 135.637226</val>
</magnitude>
</node>
<node name="Ribonuclease I"><magnitude>
<val> 4.904955</val>
</magnitude>
</node>
<node name="Ribonucleases G and E"><magnitude>
<val> 311.903053</val>
</magnitude>
</node>
<node name="Ribosomal protein HS6-type (S12/L30/L7a)"><magnitude>
<val> 111.439416</val>
</magnitude>
</node>
<node name="Ribosomal protein L1"><magnitude>
<val> 218.736928</val>
</magnitude>
</node>
<node name="Ribosomal protein L10"><magnitude>
<val> 197.712654</val>
</magnitude>
</node>
<node name="Ribosomal protein L11"><magnitude>
<val> 196.804403</val>
</magnitude>
</node>
<node name="Ribosomal protein L11 methylase"><magnitude>
<val> 239.26196</val>
</magnitude>
</node>
<node name="Ribosomal protein L12E/L44/L45/RPP1/RPP2"><magnitude>
<val> 80.768582</val>
</magnitude>
</node>
<node name="Ribosomal protein L13"><magnitude>
<val> 189.928295</val>
</magnitude>
</node>
<node name="Ribosomal protein L13E"><magnitude>
<val> 66.176089</val>
</magnitude>
</node>
<node name="Ribosomal protein L14"><magnitude>
<val> 191.337196</val>
</magnitude>
</node>
<node name="Ribosomal protein L14E/L6E/L27E"><magnitude>
<val> 63.578237</val>
</magnitude>
</node>
<node name="Ribosomal protein L15"><magnitude>
<val> 206.605055</val>
</magnitude>
</node>
<node name="Ribosomal protein L15E"><magnitude>
<val> 83.225533</val>
</magnitude>
</node>
<node name="Ribosomal protein L16/L10E"><magnitude>
<val> 186.25498</val>
</magnitude>
</node>
<node name="Ribosomal protein L17"><magnitude>
<val> 125.492546</val>
</magnitude>
</node>
<node name="Ribosomal protein L18"><magnitude>
<val> 200.119829</val>
</magnitude>
</node>
<node name="Ribosomal protein L18E"><magnitude>
<val> 83.961745</val>
</magnitude>
</node>
<node name="Ribosomal protein L19"><magnitude>
<val> 171.134442</val>
</magnitude>
</node>
<node name="Ribosomal protein L19E"><magnitude>
<val> 62.741365</val>
</magnitude>
</node>
<node name="Ribosomal protein L2"><magnitude>
<val> 202.791574</val>
</magnitude>
</node>
<node name="Ribosomal protein L20"><magnitude>
<val> 138.20343</val>
</magnitude>
</node>
<node name="Ribosomal protein L20A (L18A)"><magnitude>
<val> 61.563237</val>
</magnitude>
</node>
<node name="Ribosomal protein L21"><magnitude>
<val> 117.04975</val>
</magnitude>
</node>
<node name="Ribosomal protein L21E"><magnitude>
<val> 62.812433</val>
</magnitude>
</node>
<node name="Ribosomal protein L22"><magnitude>
<val> 175.751374</val>
</magnitude>
</node>
<node name="Ribosomal protein L23"><magnitude>
<val> 201.910927</val>
</magnitude>
</node>
<node name="Ribosomal protein L24"><magnitude>
<val> 191.034858</val>
</magnitude>
</node>
<node name="Ribosomal protein L24E"><magnitude>
<val> 79.822712</val>
</magnitude>
</node>
<node name="Ribosomal protein L25 (general stress protein Ctc)"><magnitude>
<val> 129.605447</val>
</magnitude>
</node>
<node name="Ribosomal protein L27"><magnitude>
<val> 119.052174</val>
</magnitude>
</node>
<node name="Ribosomal protein L28"><magnitude>
<val> 137.363427</val>
</magnitude>
</node>
<node name="Ribosomal protein L29"><magnitude>
<val> 176.102516</val>
</magnitude>
</node>
<node name="Ribosomal protein L3"><magnitude>
<val> 232.795331</val>
</magnitude>
</node>
<node name="Ribosomal protein L30/L7E"><magnitude>
<val> 172.409194</val>
</magnitude>
</node>
<node name="Ribosomal protein L30E"><magnitude>
<val> 66.596818</val>
</magnitude>
</node>
<node name="Ribosomal protein L31"><magnitude>
<val> 171.624464</val>
</magnitude>
</node>
<node name="Ribosomal protein L31E"><magnitude>
<val> 90.816251</val>
</magnitude>
</node>
<node name="Ribosomal protein L32"><magnitude>
<val> 129.548148</val>
</magnitude>
</node>
<node name="Ribosomal protein L32E"><magnitude>
<val> 68.746554</val>
</magnitude>
</node>
<node name="Ribosomal protein L33"><magnitude>
<val> 86.37021</val>
</magnitude>
</node>
<node name="Ribosomal protein L34"><magnitude>
<val> 34.054407</val>
</magnitude>
</node>
<node name="Ribosomal protein L34E"><magnitude>
<val> 26.052382</val>
</magnitude>
</node>
<node name="Ribosomal protein L35"><magnitude>
<val> 172.89309</val>
</magnitude>
</node>
<node name="Ribosomal protein L36"><magnitude>
<val> 4.609375</val>
</magnitude>
</node>
<node name="Ribosomal protein L37AE/L43A"><magnitude>
<val> 112.951706</val>
</magnitude>
</node>
<node name="Ribosomal protein L37E"><magnitude>
<val> 1.531579</val>
</magnitude>
</node>
<node name="Ribosomal protein L39E"><magnitude>
<val> 5.221374</val>
</magnitude>
</node>
<node name="Ribosomal protein L4"><magnitude>
<val> 263.298481</val>
</magnitude>
</node>
<node name="Ribosomal protein L40E"><magnitude>
<val> 194.634682</val>
</magnitude>
</node>
<node name="Ribosomal protein L44E"><magnitude>
<val> 76.566673</val>
</magnitude>
</node>
<node name="Ribosomal protein L5"><magnitude>
<val> 196.53957</val>
</magnitude>
</node>
<node name="Ribosomal protein L6P/L9E"><magnitude>
<val> 208.460267</val>
</magnitude>
</node>
<node name="Ribosomal protein L7/L12"><magnitude>
<val> 130.658279</val>
</magnitude>
</node>
<node name="Ribosomal protein L9"><magnitude>
<val> 177.625686</val>
</magnitude>
</node>
<node name="Ribosomal protein S1"><magnitude>
<val> 287.870269</val>
</magnitude>
</node>
<node name="Ribosomal protein S10"><magnitude>
<val> 200.068708</val>
</magnitude>
</node>
<node name="Ribosomal protein S10E"><magnitude>
<val> 3.46933</val>
</magnitude>
</node>
<node name="Ribosomal protein S11"><magnitude>
<val> 225.755198</val>
</magnitude>
</node>
<node name="Ribosomal protein S12"><magnitude>
<val> 212.330505</val>
</magnitude>
</node>
<node name="Ribosomal protein S13"><magnitude>
<val> 216.474377</val>
</magnitude>
</node>
<node name="Ribosomal protein S14"><magnitude>
<val> 114.896465</val>
</magnitude>
</node>
<node name="Ribosomal protein S15P/S13E"><magnitude>
<val> 184.553595</val>
</magnitude>
</node>
<node name="Ribosomal protein S16"><magnitude>
<val> 209.793278</val>
</magnitude>
</node>
<node name="Ribosomal protein S17"><magnitude>
<val> 171.550699</val>
</magnitude>
</node>
<node name="Ribosomal protein S17E"><magnitude>
<val> 6.358996</val>
</magnitude>
</node>
<node name="Ribosomal protein S18"><magnitude>
<val> 135.678956</val>
</magnitude>
</node>
<node name="Ribosomal protein S19"><magnitude>
<val> 194.30455</val>
</magnitude>
</node>
<node name="Ribosomal protein S19E (S16A)"><magnitude>
<val> 147.172839</val>
</magnitude>
</node>
<node name="Ribosomal protein S2"><magnitude>
<val> 229.413723</val>
</magnitude>
</node>
<node name="Ribosomal protein S20"><magnitude>
<val> 137.348234</val>
</magnitude>
</node>
<node name="Ribosomal protein S21"><magnitude>
<val> 152.853402</val>
</magnitude>
</node>
<node name="Ribosomal protein S24E"><magnitude>
<val> 68.426817</val>
</magnitude>
</node>
<node name="Ribosomal protein S26"><magnitude>
<val> 24.326065</val>
</magnitude>
</node>
<node name="Ribosomal protein S27AE"><magnitude>
<val> 63.860727</val>
</magnitude>
</node>
<node name="Ribosomal protein S27E"><magnitude>
<val> 96.530476</val>
</magnitude>
</node>
<node name="Ribosomal protein S28E/S33"><magnitude>
<val> 67.58428</val>
</magnitude>
</node>
<node name="Ribosomal protein S3"><magnitude>
<val> 207.594441</val>
</magnitude>
</node>
<node name="Ribosomal protein S3AE"><magnitude>
<val> 73.432193</val>
</magnitude>
</node>
<node name="Ribosomal protein S4 and related proteins"><magnitude>
<val> 218.000234</val>
</magnitude>
</node>
<node name="Ribosomal protein S4E"><magnitude>
<val> 130.521247</val>
</magnitude>
</node>
<node name="Ribosomal protein S5"><magnitude>
<val> 195.366049</val>
</magnitude>
</node>
<node name="Ribosomal protein S6"><magnitude>
<val> 116.896337</val>
</magnitude>
</node>
<node name="Ribosomal protein S6E (S10)"><magnitude>
<val> 106.034073</val>
</magnitude>
</node>
<node name="Ribosomal protein S7"><magnitude>
<val> 215.536753</val>
</magnitude>
</node>
<node name="Ribosomal protein S8"><magnitude>
<val> 189.483766</val>
</magnitude>
</node>
<node name="Ribosomal protein S8E"><magnitude>
<val> 65.408531</val>
</magnitude>
</node>
<node name="Ribosomal protein S9"><magnitude>
<val> 181.243149</val>
</magnitude>
</node>
<node name="Ribosome recycling factor"><magnitude>
<val> 178.537715</val>
</magnitude>
</node>
<node name="Ribosome-associated chaperone zuotin"><magnitude>
<val> 41.929199</val>
</magnitude>
</node>
<node name="Ribosome-associated heat shock protein implicated in the recycling of the 50S subunit (S4 paralog)"><magnitude>
<val> 52.260148</val>
</magnitude>
</node>
<node name="Ribosome-associated protein Y (PSrp-1)"><magnitude>
<val> 155.447799</val>
</magnitude>
</node>
<node name="Ribosome-binding factor A"><magnitude>
<val> 125.024713</val>
</magnitude>
</node>
<node name="RimM protein, required for 16S rRNA processing"><magnitude>
<val> 199.885759</val>
</magnitude>
</node>
<node name="S-adenosylmethionine:tRNA-ribosyltransferase-isomerase (queuine synthetase)"><magnitude>
<val> 324.867262</val>
</magnitude>
</node>
<node name="SAM-dependent methyltransferases related to tRNA (uracil-5-)-methyltransferase"><magnitude>
<val> 280.811868</val>
</magnitude>
</node>
<node name="Selenocysteine-specific translation elongation factor"><magnitude>
<val> 195.623725</val>
</magnitude>
</node>
<node name="Seryl-tRNA synthetase"><magnitude>
<val> 365.822479</val>
</magnitude>
</node>
<node name="Small protein A (tmRNA-binding)"><magnitude>
<val> 105.809783</val>
</magnitude>
</node>
<node name="Threonyl-tRNA synthetase"><magnitude>
<val> 309.553934</val>
</magnitude>
</node>
<node name="Translation elongation factor EF-1alpha (GTPase)"><magnitude>
<val> 141.355271</val>
</magnitude>
</node>
<node name="Translation elongation factor EF-1beta"><magnitude>
<val> 64.354486</val>
</magnitude>
</node>
<node name="Translation elongation factor P (EF-P)/translation initiation factor 5A (eIF-5A)"><magnitude>
<val> 278.732741</val>
</magnitude>
</node>
<node name="Translation elongation factor Ts"><magnitude>
<val> 173.453668</val>
</magnitude>
</node>
<node name="Translation elongation factors (GTPases)"><magnitude>
<val> 680.609431</val>
</magnitude>
</node>
<node name="Translation initiation factor 1 (IF-1)"><magnitude>
<val> 187.881883</val>
</magnitude>
</node>
<node name="Translation initiation factor 1 (eIF-1/SUI1) and related proteins"><magnitude>
<val> 164.805743</val>
</magnitude>
</node>
<node name="Translation initiation factor 2 (IF-2; GTPase)"><magnitude>
<val> 331.859321</val>
</magnitude>
</node>
<node name="Translation initiation factor 2, alpha subunit (eIF-2alpha)"><magnitude>
<val> 98.453472</val>
</magnitude>
</node>
<node name="Translation initiation factor 2, beta subunit (eIF-2beta)/eIF-5 N-terminal domain"><magnitude>
<val> 73.692809</val>
</magnitude>
</node>
<node name="Translation initiation factor 2, gamma subunit (eIF-2gamma; GTPase)"><magnitude>
<val> 82.923994</val>
</magnitude>
</node>
<node name="Translation initiation factor 2B subunit, eIF-2B alpha/beta/delta family"><magnitude>
<val> 62.651948</val>
</magnitude>
</node>
<node name="Translation initiation factor 3 (IF-3)"><magnitude>
<val> 181.975092</val>
</magnitude>
</node>
<node name="Translation initiation factor 4E (eIF-4E)"><magnitude>
<val> 137.477199</val>
</magnitude>
</node>
<node name="Translation initiation factor 6 (eIF-6)"><magnitude>
<val> 62.844491</val>
</magnitude>
</node>
<node name="Truncated, possibly inactive, lysyl-tRNA synthetase (class II)"><magnitude>
<val> 57.847129</val>
</magnitude>
</node>
<node name="Tryptophanyl-tRNA synthetase"><magnitude>
<val> 344.277602</val>
</magnitude>
</node>
<node name="Tyrosyl-tRNA synthetase"><magnitude>
<val> 289.679392</val>
</magnitude>
</node>
<node name="Uncharacterized domain/protein associated with RNAses G and E"><magnitude>
<val> 31.493828</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing piwi/argonaute domain"><magnitude>
<val> 46.333238</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in ribosome biogenesis"><magnitude>
<val> 29.187255</val>
</magnitude>
</node>
<node name="Valyl-tRNA synthetase"><magnitude>
<val> 675.482881</val>
</magnitude>
</node>
<node name="rRNA methylase"><magnitude>
<val> 244.84118</val>
</magnitude>
</node>
<node name="rRNA methylases"><magnitude>
<val> 248.339092</val>
</magnitude>
</node>
<node name="tRNA and rRNA cytosine-C5-methylases"><magnitude>
<val> 260.301009</val>
</magnitude>
</node>
<node name="tRNA delta(2)-isopentenylpyrophosphate transferase"><magnitude>
<val> 207.699331</val>
</magnitude>
</node>
<node name="tRNA nucleotidyltransferase (CCA-adding enzyme)"><magnitude>
<val> 190.606304</val>
</magnitude>
</node>
<node name="tRNA nucleotidyltransferase/poly(A) polymerase"><magnitude>
<val> 459.002084</val>
</magnitude>
</node>
<node name="tRNA splicing endonuclease"><magnitude>
<val> 82.142255</val>
</magnitude>
</node>
<node name="tRNA(1-methyladenosine) methyltransferase and related methyltransferases"><magnitude>
<val> 189.085205</val>
</magnitude>
</node>
<node name="tRNA-(guanine-N1)-methyltransferase"><magnitude>
<val> 331.343515</val>
</magnitude>
</node>
<node name="tRNA-dihydrouridine synthase"><magnitude>
<val> 286.13399</val>
</magnitude>
</node>
</node>
</node>
<node name="METABOLISM"><magnitude>
<val> 277177.397576</val>
</magnitude>
<node name="Amino acid transport and metabolism"><magnitude>
<val> 54950.071548</val>
</magnitude>
<node name="1-aminocyclopropane-1-carboxylate deaminase"><magnitude>
<val> 56.367804</val>
</magnitude>
</node>
<node name="3'-phosphoadenosine 5'-phosphosulfate sulfotransferase (PAPS reductase)/FAD synthetase and related enzymes"><magnitude>
<val> 210.785321</val>
</magnitude>
</node>
<node name="3-dehydroquinate dehydratase"><magnitude>
<val> 135.573483</val>
</magnitude>
</node>
<node name="3-dehydroquinate dehydratase II"><magnitude>
<val> 104.984885</val>
</magnitude>
</node>
<node name="3-dehydroquinate synthetase"><magnitude>
<val> 202.834278</val>
</magnitude>
</node>
<node name="3-deoxy-D-arabino-heptulosonate 7-phosphate (DAHP) synthase"><magnitude>
<val> 523.46774</val>
</magnitude>
</node>
<node name="3-isopropylmalate dehydratase large subunit"><magnitude>
<val> 461.079373</val>
</magnitude>
</node>
<node name="3-isopropylmalate dehydratase small subunit"><magnitude>
<val> 396.880023</val>
</magnitude>
</node>
<node name="4-aminobutyrate aminotransferase and related aminotransferases"><magnitude>
<val> 260.245819</val>
</magnitude>
</node>
<node name="4-hydroxyphenylpyruvate dioxygenase and related hemolysins"><magnitude>
<val> 65.511513</val>
</magnitude>
</node>
<node name="5,10-methylenetetrahydrofolate reductase"><magnitude>
<val> 240.158297</val>
</magnitude>
</node>
<node name="5-carboxymethyl-2-hydroxymuconate isomerase"><magnitude>
<val> 36.690605</val>
</magnitude>
</node>
<node name="5-enolpyruvylshikimate-3-phosphate synthase"><magnitude>
<val> 270.676082</val>
</magnitude>
</node>
<node name="ABC-type amino acid transport system, permease component"><magnitude>
<val> 524.650089</val>
</magnitude>
</node>
<node name="ABC-type amino acid transport/signal transduction systems, periplasmic component/domain"><magnitude>
<val> 821.525983</val>
</magnitude>
</node>
<node name="ABC-type arginine transport system, ATPase component"><magnitude>
<val> 4.806818</val>
</magnitude>
</node>
<node name="ABC-type arginine transport system, permease component"><magnitude>
<val> 35.632256</val>
</magnitude>
</node>
<node name="ABC-type arginine/histidine transport system, permease component"><magnitude>
<val> 34.669625</val>
</magnitude>
</node>
<node name="ABC-type branched-chain amino acid transport system, permease component"><magnitude>
<val> 541.68273</val>
</magnitude>
</node>
<node name="ABC-type branched-chain amino acid transport systems, ATPase component"><magnitude>
<val> 978.89075</val>
</magnitude>
</node>
<node name="ABC-type branched-chain amino acid transport systems, periplasmic component"><magnitude>
<val> 983.923983</val>
</magnitude>
</node>
<node name="ABC-type dipeptide transport system, periplasmic component"><magnitude>
<val> 663.491127</val>
</magnitude>
</node>
<node name="ABC-type dipeptide/oligopeptide/nickel transport system, ATPase component"><magnitude>
<val> 304.773878</val>
</magnitude>
</node>
<node name="ABC-type dipeptide/oligopeptide/nickel transport systems, permease components"><magnitude>
<val> 688.839566</val>
</magnitude>
</node>
<node name="ABC-type histidine transport system, ATPase component"><magnitude>
<val> 24.408444</val>
</magnitude>
</node>
<node name="ABC-type oligopeptide transport system, ATPase component"><magnitude>
<val> 154.516891</val>
</magnitude>
</node>
<node name="ABC-type oligopeptide transport system, periplasmic component"><magnitude>
<val> 206.590123</val>
</magnitude>
</node>
<node name="ABC-type polar amino acid transport system, ATPase component"><magnitude>
<val> 483.688597</val>
</magnitude>
</node>
<node name="ABC-type proline/glycine betaine transport system, ATPase component"><magnitude>
<val> 111.67952</val>
</magnitude>
</node>
<node name="ABC-type proline/glycine betaine transport system, permease component"><magnitude>
<val> 85.186887</val>
</magnitude>
</node>
<node name="ABC-type proline/glycine betaine transport systems, ATPase components"><magnitude>
<val> 130.666798</val>
</magnitude>
</node>
<node name="ABC-type proline/glycine betaine transport systems, periplasmic components"><magnitude>
<val> 124.373982</val>
</magnitude>
</node>
<node name="ABC-type proline/glycine betaine transport systems, permease component"><magnitude>
<val> 74.55737</val>
</magnitude>
</node>
<node name="ABC-type spermidine/putrescine transport system, permease component I"><magnitude>
<val> 76.733744</val>
</magnitude>
</node>
<node name="ABC-type spermidine/putrescine transport system, permease component II"><magnitude>
<val> 106.72889</val>
</magnitude>
</node>
<node name="ABC-type spermidine/putrescine transport systems, ATPase components"><magnitude>
<val> 333.754892</val>
</magnitude>
</node>
<node name="ACT-domain-containing protein, predicted allosteric regulator of homoserine dehydrogenase"><magnitude>
<val> 124.693297</val>
</magnitude>
</node>
<node name="ATP phosphoribosyltransferase"><magnitude>
<val> 321.756282</val>
</magnitude>
</node>
<node name="ATP phosphoribosyltransferase involved in histidine biosynthesis"><magnitude>
<val> 97.084559</val>
</magnitude>
</node>
<node name="Acetolactate synthase"><magnitude>
<val> 94.920194</val>
</magnitude>
</node>
<node name="Acetolactate synthase, small (regulatory) subunit"><magnitude>
<val> 277.359261</val>
</magnitude>
</node>
<node name="Acetylglutamate kinase"><magnitude>
<val> 250.404061</val>
</magnitude>
</node>
<node name="Acetylglutamate semialdehyde dehydrogenase"><magnitude>
<val> 253.901491</val>
</magnitude>
</node>
<node name="Acetylglutamate synthase"><magnitude>
<val> 40.550452</val>
</magnitude>
</node>
<node name="Acetylornithine deacetylase/Succinyl-diaminopimelate desuccinylase and related deacylases"><magnitude>
<val> 329.048885</val>
</magnitude>
</node>
<node name="Alanine dehydrogenase"><magnitude>
<val> 118.272465</val>
</magnitude>
</node>
<node name="Alanine-alpha-ketoisovalerate (or valine-pyruvate) aminotransferase"><magnitude>
<val> 125.865682</val>
</magnitude>
</node>
<node name="Allophanate hydrolase subunit 1"><magnitude>
<val> 106.596711</val>
</magnitude>
</node>
<node name="Allophanate hydrolase subunit 2"><magnitude>
<val> 98.026095</val>
</magnitude>
</node>
<node name="Amino acid permeases"><magnitude>
<val> 74.720159</val>
</magnitude>
</node>
<node name="Amino acid transporters"><magnitude>
<val> 572.356401</val>
</magnitude>
</node>
<node name="Aminopeptidase C"><magnitude>
<val> 75.203897</val>
</magnitude>
</node>
<node name="Aminopeptidase N"><magnitude>
<val> 238.168116</val>
</magnitude>
</node>
<node name="Anaerobic glycerol-3-phosphate dehydrogenase"><magnitude>
<val> 64.539427</val>
</magnitude>
</node>
<node name="Anthranilate phosphoribosyltransferase"><magnitude>
<val> 365.611308</val>
</magnitude>
</node>
<node name="Anthranilate/para-aminobenzoate synthases component I"><magnitude>
<val> 514.054421</val>
</magnitude>
</node>
<node name="Anthranilate/para-aminobenzoate synthases component II"><magnitude>
<val> 289.735251</val>
</magnitude>
</node>
<node name="Archaeal S-adenosylmethionine synthetase"><magnitude>
<val> 155.898645</val>
</magnitude>
</node>
<node name="Archaeal shikimate kinase"><magnitude>
<val> 77.261882</val>
</magnitude>
</node>
<node name="Arginase/agmatinase/formimionoglutamate hydrolase, arginase family"><magnitude>
<val> 147.425405</val>
</magnitude>
</node>
<node name="Arginine decarboxylase (spermidine biosynthesis)"><magnitude>
<val> 258.79559</val>
</magnitude>
</node>
<node name="Arginine degradation protein (predicted deacylase)"><magnitude>
<val> 73.779923</val>
</magnitude>
</node>
<node name="Arginine deiminase"><magnitude>
<val> 51.086638</val>
</magnitude>
</node>
<node name="Arginine kinase"><magnitude>
<val> 175.062275</val>
</magnitude>
</node>
<node name="Arginine/lysine/ornithine decarboxylases"><magnitude>
<val> 159.326749</val>
</magnitude>
</node>
<node name="Arginine/ornithine N-succinyltransferase beta subunit"><magnitude>
<val> 59.788985</val>
</magnitude>
</node>
<node name="Argininosuccinate lyase"><magnitude>
<val> 260.460201</val>
</magnitude>
</node>
<node name="Argininosuccinate synthase"><magnitude>
<val> 297.90531</val>
</magnitude>
</node>
<node name="Asparaginase"><magnitude>
<val> 105.048315</val>
</magnitude>
</node>
<node name="Asparagine synthase (glutamine-hydrolyzing)"><magnitude>
<val> 390.619107</val>
</magnitude>
</node>
<node name="Asparagine synthetase A"><magnitude>
<val> 101.230069</val>
</magnitude>
</node>
<node name="Aspartate ammonia-lyase"><magnitude>
<val> 193.070026</val>
</magnitude>
</node>
<node name="Aspartate-semialdehyde dehydrogenase"><magnitude>
<val> 259.257683</val>
</magnitude>
</node>
<node name="Aspartate/tyrosine/aromatic aminotransferase"><magnitude>
<val> 1383.935716</val>
</magnitude>
</node>
<node name="Aspartokinases"><magnitude>
<val> 307.493579</val>
</magnitude>
</node>
<node name="Aspartyl aminopeptidase"><magnitude>
<val> 96.346388</val>
</magnitude>
</node>
<node name="Benzoyl-CoA reductase/2-hydroxyglutaryl-CoA dehydratase subunit, BcrC/BadD/HgdB"><magnitude>
<val> 312.962361</val>
</magnitude>
</node>
<node name="Bifunctional PLP-dependent enzyme with beta-cystathionase and maltose regulon repressor activities"><magnitude>
<val> 229.76365</val>
</magnitude>
</node>
<node name="Branched-chain amino acid ABC-type transport system, permease components"><magnitude>
<val> 507.692695</val>
</magnitude>
</node>
<node name="Branched-chain amino acid aminotransferase/4-amino-4-deoxychorismate lyase"><magnitude>
<val> 319.392513</val>
</magnitude>
</node>
<node name="Branched-chain amino acid permeases"><magnitude>
<val> 89.753353</val>
</magnitude>
</node>
<node name="Carbamate kinase"><magnitude>
<val> 127.966399</val>
</magnitude>
</node>
<node name="Carbamoylphosphate synthase large subunit (split gene in MJ)"><magnitude>
<val> 528.904613</val>
</magnitude>
</node>
<node name="Carbamoylphosphate synthase small subunit"><magnitude>
<val> 277.854969</val>
</magnitude>
</node>
<node name="Carboxypeptidase C (cathepsin A)"><magnitude>
<val> 106.326876</val>
</magnitude>
</node>
<node name="Choline dehydrogenase and related flavoproteins"><magnitude>
<val> 290.662849</val>
</magnitude>
</node>
<node name="Chorismate mutase"><magnitude>
<val> 125.355086</val>
</magnitude>
</node>
<node name="Chorismate synthase"><magnitude>
<val> 245.159418</val>
</magnitude>
</node>
<node name="Cystathionine beta-lyases/cystathionine gamma-synthases"><magnitude>
<val> 265.289519</val>
</magnitude>
</node>
<node name="Cysteine sulfinate desulfinase/cysteine desulfurase and related enzymes"><magnitude>
<val> 531.465188</val>
</magnitude>
</node>
<node name="Cysteine synthase"><magnitude>
<val> 417.10824</val>
</magnitude>
</node>
<node name="D-aminopeptidase"><magnitude>
<val> 16.693055</val>
</magnitude>
</node>
<node name="D-serine dehydratase"><magnitude>
<val> 88.652418</val>
</magnitude>
</node>
<node name="Di- and tripeptidases"><magnitude>
<val> 183.875868</val>
</magnitude>
</node>
<node name="Diaminopimelate decarboxylase"><magnitude>
<val> 454.547669</val>
</magnitude>
</node>
<node name="Diaminopimelate epimerase"><magnitude>
<val> 289.944796</val>
</magnitude>
</node>
<node name="Dihydrodipicolinate reductase"><magnitude>
<val> 269.857086</val>
</magnitude>
</node>
<node name="Dihydrodipicolinate synthase/N-acetylneuraminate lyase"><magnitude>
<val> 275.790736</val>
</magnitude>
</node>
<node name="Dihydroxyacid dehydratase/phosphogluconate dehydratase"><magnitude>
<val> 383.822863</val>
</magnitude>
</node>
<node name="Dipeptidase"><magnitude>
<val> 99.458723</val>
</magnitude>
</node>
<node name="Dipeptide/tripeptide permease"><magnitude>
<val> 245.335528</val>
</magnitude>
</node>
<node name="Dipeptidyl aminopeptidases/acylaminoacyl-peptidases"><magnitude>
<val> 400.104501</val>
</magnitude>
</node>
<node name="Ethanolamine ammonia-lyase, large subunit"><magnitude>
<val> 49.673299</val>
</magnitude>
</node>
<node name="Ethanolamine ammonia-lyase, small subunit"><magnitude>
<val> 38.465726</val>
</magnitude>
</node>
<node name="Ethanolamine utilization cobalamin adenosyltransferase"><magnitude>
<val> 67.483374</val>
</magnitude>
</node>
<node name="Ethanolamine utilization protein"><magnitude>
<val> 165.94319</val>
</magnitude>
</node>
<node name="Ethanolamine utilization protein, possible chaperonin"><magnitude>
<val> 48.845912</val>
</magnitude>
</node>
<node name="Ethanolamine utilization protein, possible chaperonin protecting lyase from inhibition"><magnitude>
<val> 139.888713</val>
</magnitude>
</node>
<node name="Gamma-aminobutyrate permease and related permeases"><magnitude>
<val> 213.276964</val>
</magnitude>
</node>
<node name="Gamma-glutamyl phosphate reductase"><magnitude>
<val> 188.294069</val>
</magnitude>
</node>
<node name="Gamma-glutamyltransferase"><magnitude>
<val> 118.990565</val>
</magnitude>
</node>
<node name="Glutamate 5-kinase"><magnitude>
<val> 166.690444</val>
</magnitude>
</node>
<node name="Glutamate decarboxylase and related PLP-dependent proteins"><magnitude>
<val> 451.76554</val>
</magnitude>
</node>
<node name="Glutamate dehydrogenase/leucine dehydrogenase"><magnitude>
<val> 424.993652</val>
</magnitude>
</node>
<node name="Glutamate formiminotransferase"><magnitude>
<val> 51.272101</val>
</magnitude>
</node>
<node name="Glutamate mutase epsilon subunit"><magnitude>
<val> 11.950103</val>
</magnitude>
</node>
<node name="Glutamate synthase domain 1"><magnitude>
<val> 244.561878</val>
</magnitude>
</node>
<node name="Glutamate synthase domain 2"><magnitude>
<val> 555.194964</val>
</magnitude>
</node>
<node name="Glutamate synthase domain 3"><magnitude>
<val> 238.86556</val>
</magnitude>
</node>
<node name="Glutaminase"><magnitude>
<val> 91.007098</val>
</magnitude>
</node>
<node name="Glutamine amidotransferase"><magnitude>
<val> 207.489997</val>
</magnitude>
</node>
<node name="Glutamine synthetase"><magnitude>
<val> 616.60729</val>
</magnitude>
</node>
<node name="Glutathionylspermidine synthase"><magnitude>
<val> 5.836987</val>
</magnitude>
</node>
<node name="Glycine cleavage system H protein (lipoate-binding)"><magnitude>
<val> 162.13362</val>
</magnitude>
</node>
<node name="Glycine cleavage system T protein (aminomethyltransferase)"><magnitude>
<val> 175.011951</val>
</magnitude>
</node>
<node name="Glycine cleavage system protein P (pyridoxal-binding), C-terminal domain"><magnitude>
<val> 182.866271</val>
</magnitude>
</node>
<node name="Glycine cleavage system protein P (pyridoxal-binding), N-terminal domain"><magnitude>
<val> 111.529157</val>
</magnitude>
</node>
<node name="Glycine cleavage system regulatory protein"><magnitude>
<val> 80.378923</val>
</magnitude>
</node>
<node name="Glycine/D-amino acid oxidases (deaminating)"><magnitude>
<val> 198.723365</val>
</magnitude>
</node>
<node name="Glycine/serine hydroxymethyltransferase"><magnitude>
<val> 298.180299</val>
</magnitude>
</node>
<node name="Histidine ammonia-lyase"><magnitude>
<val> 144.890784</val>
</magnitude>
</node>
<node name="Histidinol dehydrogenase"><magnitude>
<val> 251.474084</val>
</magnitude>
</node>
<node name="Histidinol phosphatase and related hydrolases of the PHP family"><magnitude>
<val> 255.943241</val>
</magnitude>
</node>
<node name="Histidinol phosphatase and related phosphatases"><magnitude>
<val> 118.385267</val>
</magnitude>
</node>
<node name="Histidinol-phosphate/aromatic aminotransferase and cobyric acid decarboxylase"><magnitude>
<val> 643.309823</val>
</magnitude>
</node>
<node name="Homocysteine/selenocysteine methylase (S-methylmethionine-dependent)"><magnitude>
<val> 90.86319</val>
</magnitude>
</node>
<node name="Homoserine acetyltransferase"><magnitude>
<val> 218.761575</val>
</magnitude>
</node>
<node name="Homoserine dehydrogenase"><magnitude>
<val> 269.699407</val>
</magnitude>
</node>
<node name="Homoserine kinase"><magnitude>
<val> 82.870507</val>
</magnitude>
</node>
<node name="Homoserine trans-succinylase"><magnitude>
<val> 80.250367</val>
</magnitude>
</node>
<node name="Hydantoin racemase"><magnitude>
<val> 21.80557</val>
</magnitude>
</node>
<node name="Imidazoleglycerol-phosphate dehydratase"><magnitude>
<val> 246.345216</val>
</magnitude>
</node>
<node name="Imidazoleglycerol-phosphate synthase"><magnitude>
<val> 338.521625</val>
</magnitude>
</node>
<node name="Indole-3-glycerol phosphate synthase"><magnitude>
<val> 218.603968</val>
</magnitude>
</node>
<node name="Isopropylmalate/homocitrate/citramalate synthases"><magnitude>
<val> 758.77925</val>
</magnitude>
</node>
<node name="Ketol-acid reductoisomerase"><magnitude>
<val> 299.067074</val>
</magnitude>
</node>
<node name="Kynureninase"><magnitude>
<val> 38.946125</val>
</magnitude>
</node>
<node name="L-aminopeptidase/D-esterase"><magnitude>
<val> 22.951491</val>
</magnitude>
</node>
<node name="L-asparaginase II"><magnitude>
<val> 63.002015</val>
</magnitude>
</node>
<node name="L-asparaginase/archaeal Glu-tRNAGln amidotransferase subunit D"><magnitude>
<val> 219.566101</val>
</magnitude>
</node>
<node name="L-serine deaminase"><magnitude>
<val> 169.444028</val>
</magnitude>
</node>
<node name="Lactoylglutathione lyase and related lyases"><magnitude>
<val> 457.323811</val>
</magnitude>
</node>
<node name="Leucyl aminopeptidase"><magnitude>
<val> 156.5837</val>
</magnitude>
</node>
<node name="Leucyl aminopeptidase (aminopeptidase T)"><magnitude>
<val> 248.089464</val>
</magnitude>
</node>
<node name="Lysine 2,3-aminomutase"><magnitude>
<val> 246.897647</val>
</magnitude>
</node>
<node name="Lysophospholipase L1 and related esterases"><magnitude>
<val> 401.652838</val>
</magnitude>
</node>
<node name="Methenyl tetrahydrofolate cyclohydrolase"><magnitude>
<val> 82.523236</val>
</magnitude>
</node>
<node name="Methionine synthase I (cobalamin-dependent), methyltransferase domain"><magnitude>
<val> 216.738189</val>
</magnitude>
</node>
<node name="Methionine synthase I, cobalamin-binding domain"><magnitude>
<val> 234.594574</val>
</magnitude>
</node>
<node name="Methionine synthase II (cobalamin-independent)"><magnitude>
<val> 174.249914</val>
</magnitude>
</node>
<node name="Methylaspartate ammonia-lyase"><magnitude>
<val> 26.081954</val>
</magnitude>
</node>
<node name="Monoamine oxidase"><magnitude>
<val> 48.671383</val>
</magnitude>
</node>
<node name="N-Dimethylarginine dimethylaminohydrolase"><magnitude>
<val> 101.952031</val>
</magnitude>
</node>
<node name="N-acetylglutamate synthase (N-acetylornithine aminotransferase)"><magnitude>
<val> 229.663303</val>
</magnitude>
</node>
<node name="N-acetylglutamate synthase and related acetyltransferases"><magnitude>
<val> 189.988514</val>
</magnitude>
</node>
<node name="N-formylglutamate amidohydrolase"><magnitude>
<val> 26.175179</val>
</magnitude>
</node>
<node name="N-methylhydantoinase A/acetone carboxylase, beta subunit"><magnitude>
<val> 180.565337</val>
</magnitude>
</node>
<node name="N-methylhydantoinase B/acetone carboxylase, alpha subunit"><magnitude>
<val> 241.693366</val>
</magnitude>
</node>
<node name="NAD-specific glutamate dehydrogenase"><magnitude>
<val> 420.324499</val>
</magnitude>
</node>
<node name="NADPH-dependent glutamate synthase beta chain and related oxidoreductases"><magnitude>
<val> 536.461814</val>
</magnitude>
</node>
<node name="Na+/alanine symporter"><magnitude>
<val> 203.968209</val>
</magnitude>
</node>
<node name="Na+/glutamate symporter"><magnitude>
<val> 6.173816</val>
</magnitude>
</node>
<node name="Na+/proline symporter"><magnitude>
<val> 575.299285</val>
</magnitude>
</node>
<node name="Na+/serine symporter"><magnitude>
<val> 49.095946</val>
</magnitude>
</node>
<node name="Nitrogen regulatory protein PII"><magnitude>
<val> 760.722252</val>
</magnitude>
</node>
<node name="O-acetylhomoserine sulfhydrylase"><magnitude>
<val> 371.965409</val>
</magnitude>
</node>
<node name="Oligoendopeptidase F"><magnitude>
<val> 230.589352</val>
</magnitude>
</node>
<node name="Ornithine carbamoyltransferase"><magnitude>
<val> 237.488581</val>
</magnitude>
</node>
<node name="Ornithine/acetylornithine aminotransferase"><magnitude>
<val> 460.922377</val>
</magnitude>
</node>
<node name="Peptidase E"><magnitude>
<val> 25.308475</val>
</magnitude>
</node>
<node name="Peptidylarginine deiminase and related enzymes"><magnitude>
<val> 150.898293</val>
</magnitude>
</node>
<node name="Phenylalanine-4-hydroxylase"><magnitude>
<val> 8.919503</val>
</magnitude>
</node>
<node name="Phosphoribosyl-AMP cyclohydrolase"><magnitude>
<val> 201.743254</val>
</magnitude>
</node>
<node name="Phosphoribosyl-ATP pyrophosphohydrolase"><magnitude>
<val> 101.346517</val>
</magnitude>
</node>
<node name="Phosphoribosylanthranilate isomerase"><magnitude>
<val> 256.835829</val>
</magnitude>
</node>
<node name="Phosphoribosylformimino-5-aminoimidazole carboxamide ribonucleotide (ProFAR) isomerase"><magnitude>
<val> 210.488722</val>
</magnitude>
</node>
<node name="Phosphoserine phosphatase"><magnitude>
<val> 228.399251</val>
</magnitude>
</node>
<node name="Predicted Zn peptidase"><magnitude>
<val> 196.664252</val>
</magnitude>
</node>
<node name="Predicted alternative 3-dehydroquinate synthase"><magnitude>
<val> 93.731989</val>
</magnitude>
</node>
<node name="Predicted amino acid aldolase or racemase"><magnitude>
<val> 187.528769</val>
</magnitude>
</node>
<node name="Predicted amino acid racemase"><magnitude>
<val> 25.965529</val>
</magnitude>
</node>
<node name="Predicted branched-chain amino acid permease (azaleucine resistance)"><magnitude>
<val> 125.068399</val>
</magnitude>
</node>
<node name="Predicted branched-chain amino acid permeases (azaleucine resistance)"><magnitude>
<val> 28.919331</val>
</magnitude>
</node>
<node name="Predicted carboxypeptidase"><magnitude>
<val> 14.602036</val>
</magnitude>
</node>
<node name="Predicted homoserine dehydrogenase"><magnitude>
<val> 80.862205</val>
</magnitude>
</node>
<node name="Predicted ornithine cyclodeaminase, mu-crystallin homolog"><magnitude>
<val> 205.082795</val>
</magnitude>
</node>
<node name="Prephenate dehydratase"><magnitude>
<val> 228.827456</val>
</magnitude>
</node>
<node name="Prephenate dehydrogenase"><magnitude>
<val> 315.965413</val>
</magnitude>
</node>
<node name="Proline dehydrogenase"><magnitude>
<val> 60.531222</val>
</magnitude>
</node>
<node name="Proline racemase"><magnitude>
<val> 93.837748</val>
</magnitude>
</node>
<node name="Protease II"><magnitude>
<val> 81.73598</val>
</magnitude>
</node>
<node name="Putative periplasmic protein kinase ArgK and related GTPases of G3E family"><magnitude>
<val> 182.475318</val>
</magnitude>
</node>
<node name="Putative threonine efflux protein"><magnitude>
<val> 364.55231</val>
</magnitude>
</node>
<node name="Pyrroline-5-carboxylate reductase"><magnitude>
<val> 225.31059</val>
</magnitude>
</node>
<node name="S-adenosylmethionine decarboxylase"><magnitude>
<val> 116.246877</val>
</magnitude>
</node>
<node name="Saccharopine dehydrogenase and related proteins"><magnitude>
<val> 268.201047</val>
</magnitude>
</node>
<node name="Sarcosine oxidase delta subunit"><magnitude>
<val> 20.883549</val>
</magnitude>
</node>
<node name="Sarcosine oxidase gamma subunit"><magnitude>
<val> 24.372691</val>
</magnitude>
</node>
<node name="Selenocysteine lyase"><magnitude>
<val> 417.329397</val>
</magnitude>
</node>
<node name="Selenocysteine synthase [seryl-tRNASer selenium transferase]"><magnitude>
<val> 200.368169</val>
</magnitude>
</node>
<node name="Selenophosphate synthase"><magnitude>
<val> 144.685452</val>
</magnitude>
</node>
<node name="Serine acetyltransferase"><magnitude>
<val> 334.108489</val>
</magnitude>
</node>
<node name="Serine proteases of the peptidase family S9A"><magnitude>
<val> 82.125059</val>
</magnitude>
</node>
<node name="Serine-pyruvate aminotransferase/archaeal aspartate aminotransferase"><magnitude>
<val> 325.812271</val>
</magnitude>
</node>
<node name="Shikimate 5-dehydrogenase"><magnitude>
<val> 291.159691</val>
</magnitude>
</node>
<node name="Shikimate kinase"><magnitude>
<val> 205.923398</val>
</magnitude>
</node>
<node name="Spermidine synthase"><magnitude>
<val> 201.254489</val>
</magnitude>
</node>
<node name="Spermidine/putrescine-binding periplasmic protein"><magnitude>
<val> 139.626756</val>
</magnitude>
</node>
<node name="Succinylarginine dihydrolase"><magnitude>
<val> 34.851673</val>
</magnitude>
</node>
<node name="Succinylglutamate desuccinylase"><magnitude>
<val> 35.558116</val>
</magnitude>
</node>
<node name="Tetrahydrodipicolinate N-succinyltransferase"><magnitude>
<val> 41.157199</val>
</magnitude>
</node>
<node name="Thiamine pyrophosphate-requiring enzymes [acetolactate synthase, pyruvate dehydrogenase (cytochrome), glyoxylate carboligase, phosphonopyruvate decarboxylase]"><magnitude>
<val> 706.08171</val>
</magnitude>
</node>
<node name="Threonine aldolase"><magnitude>
<val> 151.733005</val>
</magnitude>
</node>
<node name="Threonine dehydratase"><magnitude>
<val> 224.965374</val>
</magnitude>
</node>
<node name="Threonine dehydrogenase and related Zn-dependent dehydrogenases"><magnitude>
<val> 368.645199</val>
</magnitude>
</node>
<node name="Threonine synthase"><magnitude>
<val> 300.143164</val>
</magnitude>
</node>
<node name="Transglutaminase-like enzymes, putative cysteine proteases"><magnitude>
<val> 795.451903</val>
</magnitude>
</node>
<node name="Tryptophan 2,3-dioxygenase (vermilion)"><magnitude>
<val> 20.629363</val>
</magnitude>
</node>
<node name="Tryptophan synthase alpha chain"><magnitude>
<val> 260.661712</val>
</magnitude>
</node>
<node name="Tryptophan synthase beta chain"><magnitude>
<val> 332.75287</val>
</magnitude>
</node>
<node name="Tryptophanase"><magnitude>
<val> 74.361081</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein, possibly involved in methylthioadenosine recycling"><magnitude>
<val> 67.232067</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in cysteine biosynthesis"><magnitude>
<val> 53.580841</val>
</magnitude>
</node>
<node name="Urea amidohydrolase (urease) alpha subunit"><magnitude>
<val> 118.238421</val>
</magnitude>
</node>
<node name="Urea amidohydrolase (urease) beta subunit"><magnitude>
<val> 13.59318</val>
</magnitude>
</node>
<node name="Urea amidohydrolase (urease) gamma subunit"><magnitude>
<val> 54.587669</val>
</magnitude>
</node>
<node name="Urea transporter"><magnitude>
<val> 103.116848</val>
</magnitude>
</node>
<node name="Urocanate hydratase"><magnitude>
<val> 164.264006</val>
</magnitude>
</node>
<node name="V8-like Glu-specific endopeptidase"><magnitude>
<val> 118.15754</val>
</magnitude>
</node>
<node name="Xaa-Pro aminopeptidase"><magnitude>
<val> 508.19445</val>
</magnitude>
</node>
<node name="Zinc metalloprotease (elastase)"><magnitude>
<val> 88.172005</val>
</magnitude>
</node>
<node name="Zn-dependent carboxypeptidase"><magnitude>
<val> 46.089675</val>
</magnitude>
</node>
<node name="Zn-dependent dipeptidase, microsomal dipeptidase homolog"><magnitude>
<val> 126.9922</val>
</magnitude>
</node>
<node name="Zn-dependent oligopeptidases"><magnitude>
<val> 165.874017</val>
</magnitude>
</node>
</node>
<node name="Carbohydrate transport and metabolism"><magnitude>
<val> 42188.431572</val>
</magnitude>
<node name="1,4-alpha-glucan branching enzyme"><magnitude>
<val> 402.790457</val>
</magnitude>
</node>
<node name="2,4-dihydroxyhept-2-ene-1,7-dioic acid aldolase"><magnitude>
<val> 37.193669</val>
</magnitude>
</node>
<node name="2-keto-3-deoxy-6-phosphogluconate aldolase"><magnitude>
<val> 134.01102</val>
</magnitude>
</node>
<node name="2-keto-3-deoxy-galactonokinase"><magnitude>
<val> 12.937804</val>
</magnitude>
</node>
<node name="2-phosphoglycerate kinase"><magnitude>
<val> 108.315389</val>
</magnitude>
</node>
<node name="3-carboxymuconate cyclase"><magnitude>
<val> 159.140823</val>
</magnitude>
</node>
<node name="3-hexulose-6-phosphate synthase and related proteins"><magnitude>
<val> 61.510128</val>
</magnitude>
</node>
<node name="3-phosphoglycerate kinase"><magnitude>
<val> 265.081851</val>
</magnitude>
</node>
<node name="4-alpha-glucanotransferase"><magnitude>
<val> 326.46259</val>
</magnitude>
</node>
<node name="5-keto 4-deoxyuronate isomerase"><magnitude>
<val> 89.469586</val>
</magnitude>
</node>
<node name="6-phosphofructokinase"><magnitude>
<val> 391.457042</val>
</magnitude>
</node>
<node name="6-phosphogluconate dehydrogenase"><magnitude>
<val> 98.5946</val>
</magnitude>
</node>
<node name="6-phosphogluconolactonase/Glucosamine-6-phosphate isomerase/deaminase"><magnitude>
<val> 189.942834</val>
</magnitude>
</node>
<node name="ABC-type glucose/galactose transport system, permease component"><magnitude>
<val> 4.376179</val>
</magnitude>
</node>
<node name="ABC-type maltose transport systems, permease component"><magnitude>
<val> 10.072508</val>
</magnitude>
</node>
<node name="ABC-type polysaccharide transport system, permease component"><magnitude>
<val> 20.809074</val>
</magnitude>
</node>
<node name="ABC-type polysaccharide/polyol phosphate export systems, permease component"><magnitude>
<val> 243.066182</val>
</magnitude>
</node>
<node name="ABC-type polysaccharide/polyol phosphate transport system, ATPase component"><magnitude>
<val> 178.324221</val>
</magnitude>
</node>
<node name="ABC-type ribose transport system, auxiliary component"><magnitude>
<val> 25.75728</val>
</magnitude>
</node>
<node name="ABC-type sugar transport system, ATPase component"><magnitude>
<val> 302.874418</val>
</magnitude>
</node>
<node name="ABC-type sugar transport system, periplasmic component"><magnitude>
<val> 731.369728</val>
</magnitude>
</node>
<node name="ABC-type sugar transport system, permease component"><magnitude>
<val> 271.45102</val>
</magnitude>
</node>
<node name="ABC-type sugar transport systems, ATPase components"><magnitude>
<val> 131.173607</val>
</magnitude>
</node>
<node name="ABC-type sugar transport systems, permease components"><magnitude>
<val> 267.841824</val>
</magnitude>
</node>
<node name="ABC-type xylose transport system, periplasmic component"><magnitude>
<val> 42.525192</val>
</magnitude>
</node>
<node name="ABC-type xylose transport system, permease component"><magnitude>
<val> 80.658649</val>
</magnitude>
</node>
<node name="ADP-glucose pyrophosphorylase"><magnitude>
<val> 133.899072</val>
</magnitude>
</node>
<node name="Alpha-L-arabinofuranosidase"><magnitude>
<val> 249.879068</val>
</magnitude>
</node>
<node name="Alpha-L-fucosidase"><magnitude>
<val> 217.142425</val>
</magnitude>
</node>
<node name="Alpha-amylase/alpha-mannosidase"><magnitude>
<val> 225.848791</val>
</magnitude>
</node>
<node name="Alpha-galactosidase"><magnitude>
<val> 337.007303</val>
</magnitude>
</node>
<node name="Alpha-galactosidases/6-phospho-beta-glucosidases, family 4 of glycosyl hydrolases"><magnitude>
<val> 151.12099</val>
</magnitude>
</node>
<node name="Alpha-glucosidases, family 31 of glycosyl hydrolases"><magnitude>
<val> 410.073696</val>
</magnitude>
</node>
<node name="Alpha-glucuronidase"><magnitude>
<val> 143.217747</val>
</magnitude>
</node>
<node name="Alpha-mannosidase"><magnitude>
<val> 158.004502</val>
</magnitude>
</node>
<node name="Altronate dehydratase"><magnitude>
<val> 87.875297</val>
</magnitude>
</node>
<node name="Arabinogalactan endo-1,4-beta-galactosidase"><magnitude>
<val> 138.065003</val>
</magnitude>
</node>
<node name="Arabinose efflux permease"><magnitude>
<val> 2499.566901</val>
</magnitude>
</node>
<node name="Archaeal ADP-dependent phosphofructokinase/glucokinase"><magnitude>
<val> 72.597104</val>
</magnitude>
</node>
<node name="Archaeal fructose 1,6-bisphosphatase"><magnitude>
<val> 100.613182</val>
</magnitude>
</node>
<node name="Archaeal fructose-1,6-bisphosphatase and related enzymes of inositol monophosphatase family"><magnitude>
<val> 190.054503</val>
</magnitude>
</node>
<node name="Beta- N-acetylglucosaminidase"><magnitude>
<val> 19.245811</val>
</magnitude>
</node>
<node name="Beta-1,4-xylanase"><magnitude>
<val> 167.150638</val>
</magnitude>
</node>
<node name="Beta-fructosidases (levanase/invertase)"><magnitude>
<val> 280.139349</val>
</magnitude>
</node>
<node name="Beta-galactosidase"><magnitude>
<val> 328.351689</val>
</magnitude>
</node>
<node name="Beta-galactosidase, beta subunit"><magnitude>
<val> 93.524333</val>
</magnitude>
</node>
<node name="Beta-galactosidase/beta-glucuronidase"><magnitude>
<val> 1276.335578</val>
</magnitude>
</node>
<node name="Beta-glucanase/Beta-glucan synthetase"><magnitude>
<val> 120.760819</val>
</magnitude>
</node>
<node name="Beta-glucosidase-related glycosidases"><magnitude>
<val> 887.209046</val>
</magnitude>
</node>
<node name="Beta-glucosidase/6-phospho-beta-glucosidase/beta-galactosidase"><magnitude>
<val> 143.503099</val>
</magnitude>
</node>
<node name="Beta-mannanase"><magnitude>
<val> 254.711421</val>
</magnitude>
</node>
<node name="Beta-xylosidase"><magnitude>
<val> 1188.962593</val>
</magnitude>
</node>
<node name="Capsular polysaccharide biosynthesis protein"><magnitude>
<val> 144.823219</val>
</magnitude>
</node>
<node name="Cellobiohydrolase A (1,4-beta-cellobiosidase A)"><magnitude>
<val> 54.710536</val>
</magnitude>
</node>
<node name="Cellobiose phosphorylase"><magnitude>
<val> 280.37208</val>
</magnitude>
</node>
<node name="Cellulase M and related proteins"><magnitude>
<val> 214.902049</val>
</magnitude>
</node>
<node name="Chitinase"><magnitude>
<val> 171.850567</val>
</magnitude>
</node>
<node name="Citrate lyase beta subunit"><magnitude>
<val> 74.441295</val>
</magnitude>
</node>
<node name="D-mannonate dehydratase"><magnitude>
<val> 210.73525</val>
</magnitude>
</node>
<node name="DhnA-type fructose-1,6-bisphosphate aldolase and related enzymes"><magnitude>
<val> 249.757128</val>
</magnitude>
</node>
<node name="Dihydroxyacetone kinase"><magnitude>
<val> 128.885088</val>
</magnitude>
</node>
<node name="Endo-beta-N-acetylglucosaminidase D"><magnitude>
<val> 150.037582</val>
</magnitude>
</node>
<node name="Endo-beta-mannanase"><magnitude>
<val> 93.755512</val>
</magnitude>
</node>
<node name="Endoglucanase"><magnitude>
<val> 130.81193</val>
</magnitude>
</node>
<node name="Endoglucanase C-terminal domain/subunit and related proteins"><magnitude>
<val> 31.06681</val>
</magnitude>
</node>
<node name="Endoglucanase Y"><magnitude>
<val> 55.740197</val>
</magnitude>
</node>
<node name="Enolase"><magnitude>
<val> 359.054457</val>
</magnitude>
</node>
<node name="Exo-beta-1,3-glucanase"><magnitude>
<val> 20.772059</val>
</magnitude>
</node>
<node name="Exopolysaccharide biosynthesis protein"><magnitude>
<val> 87.806144</val>
</magnitude>
</node>
<node name="Exopolysaccharide biosynthesis protein related to N-acetylglucosamine-1-phosphodiester alpha-N-acetylglucosaminidase"><magnitude>
<val> 184.336077</val>
</magnitude>
</node>
<node name="Fructosamine-3-kinase"><magnitude>
<val> 30.342928</val>
</magnitude>
</node>
<node name="Fructose-1,6-bisphosphatase"><magnitude>
<val> 170.15166</val>
</magnitude>
</node>
<node name="Fructose-1,6-bisphosphatase/sedoheptulose 1,7-bisphosphatase and related proteins"><magnitude>
<val> 91.388922</val>
</magnitude>
</node>
<node name="Fructose-1,6-bisphosphate aldolase"><magnitude>
<val> 102.739349</val>
</magnitude>
</node>
<node name="Fructose-1-phosphate kinase and related fructose-6-phosphate kinase (PfkB)"><magnitude>
<val> 55.031454</val>
</magnitude>
</node>
<node name="Fructose-2,6-bisphosphatase"><magnitude>
<val> 306.089527</val>
</magnitude>
</node>
<node name="Fructose/tagatose bisphosphate aldolase"><magnitude>
<val> 342.287381</val>
</magnitude>
</node>
<node name="Fucose 4-O-acetylase and related acetyltransferases"><magnitude>
<val> 150.059199</val>
</magnitude>
</node>
<node name="Fucose dissimilation pathway protein FucU"><magnitude>
<val> 12.348885</val>
</magnitude>
</node>
<node name="Fucose permease"><magnitude>
<val> 299.218653</val>
</magnitude>
</node>
<node name="Galactokinase"><magnitude>
<val> 48.486317</val>
</magnitude>
</node>
<node name="Galactose mutarotase and related enzymes"><magnitude>
<val> 182.042215</val>
</magnitude>
</node>
<node name="Galactose-1-phosphate uridyltransferase"><magnitude>
<val> 67.716722</val>
</magnitude>
</node>
<node name="Glucan phosphorylase"><magnitude>
<val> 748.588199</val>
</magnitude>
</node>
<node name="Glucoamylase and related glycosyl hydrolases"><magnitude>
<val> 27.568273</val>
</magnitude>
</node>
<node name="Glucokinase"><magnitude>
<val> 50.800766</val>
</magnitude>
</node>
<node name="Gluconate kinase"><magnitude>
<val> 9.975152</val>
</magnitude>
</node>
<node name="Gluconolactonase"><magnitude>
<val> 166.364002</val>
</magnitude>
</node>
<node name="Glucose dehydrogenase"><magnitude>
<val> 158.71176</val>
</magnitude>
</node>
<node name="Glucose-6-P dehydrogenase subunit"><magnitude>
<val> 41.188399</val>
</magnitude>
</node>
<node name="Glucose-6-phosphate 1-dehydrogenase"><magnitude>
<val> 98.627343</val>
</magnitude>
</node>
<node name="Glucose-6-phosphate isomerase"><magnitude>
<val> 214.425386</val>
</magnitude>
</node>
<node name="Glucose/sorbosone dehydrogenases"><magnitude>
<val> 228.587197</val>
</magnitude>
</node>
<node name="Glucuronate isomerase"><magnitude>
<val> 103.476982</val>
</magnitude>
</node>
<node name="Glyceraldehyde-3-phosphate dehydrogenase/erythrose-4-phosphate dehydrogenase"><magnitude>
<val> 353.365184</val>
</magnitude>
</node>
<node name="Glycerate kinase"><magnitude>
<val> 84.833934</val>
</magnitude>
</node>
<node name="Glycerol uptake facilitator and related permeases (Major Intrinsic Protein Family)"><magnitude>
<val> 188.888109</val>
</magnitude>
</node>
<node name="Glycogen debranching enzyme"><magnitude>
<val> 474.681788</val>
</magnitude>
</node>
<node name="Glycogen synthase"><magnitude>
<val> 253.284881</val>
</magnitude>
</node>
<node name="Glycosidases"><magnitude>
<val> 567.671153</val>
</magnitude>
</node>
<node name="Glycosyl transferases, related to UDP-glucuronosyltransferase"><magnitude>
<val> 257.69066</val>
</magnitude>
</node>
<node name="Golgi nucleoside diphosphatase"><magnitude>
<val> 14.713871</val>
</magnitude>
</node>
<node name="H+/gluconate symporter and related permeases"><magnitude>
<val> 113.249605</val>
</magnitude>
</node>
<node name="Hexokinase"><magnitude>
<val> 35.100801</val>
</magnitude>
</node>
<node name="Hydroxypyruvate isomerase"><magnitude>
<val> 40.984118</val>
</magnitude>
</node>
<node name="L-arabinose isomerase"><magnitude>
<val> 208.77834</val>
</magnitude>
</node>
<node name="L-fucose isomerase and related proteins"><magnitude>
<val> 153.719419</val>
</magnitude>
</node>
<node name="L-rhamnose isomerase"><magnitude>
<val> 55.739468</val>
</magnitude>
</node>
<node name="Maltooligosyl trehalose synthase"><magnitude>
<val> 211.50461</val>
</magnitude>
</node>
<node name="Maltoporin (phage lambda and maltose receptor)"><magnitude>
<val> 67.996723</val>
</magnitude>
</node>
<node name="Maltose-binding periplasmic proteins/domains"><magnitude>
<val> 138.884413</val>
</magnitude>
</node>
<node name="Mannitol-1-phosphate/altronate dehydrogenases"><magnitude>
<val> 116.78031</val>
</magnitude>
</node>
<node name="Mannitol/fructose-specific phosphotransferase system, IIA domain"><magnitude>
<val> 83.097779</val>
</magnitude>
</node>
<node name="Mannose-6-phosphate isomerase"><magnitude>
<val> 328.35562</val>
</magnitude>
</node>
<node name="Mannosyltransferase"><magnitude>
<val> 44.841046</val>
</magnitude>
</node>
<node name="Membrane-anchored protein predicted to be involved in regulation of amylopullulanase"><magnitude>
<val> 20.685086</val>
</magnitude>
</node>
<node name="Methylglyoxal synthase"><magnitude>
<val> 71.194704</val>
</magnitude>
</node>
<node name="Muramidase (phage lambda lysozyme)"><magnitude>
<val> 2.559701</val>
</magnitude>
</node>
<node name="N-acetyl-beta-hexosaminidase"><magnitude>
<val> 207.925133</val>
</magnitude>
</node>
<node name="N-acetylglucosamine-6-phosphate deacetylase"><magnitude>
<val> 85.916592</val>
</magnitude>
</node>
<node name="N-acyl-D-glucosamine 2-epimerase"><magnitude>
<val> 166.809007</val>
</magnitude>
</node>
<node name="N-terminal domain of the phosphotransferase system fructose-specific component IIB"><magnitude>
<val> 24.001351</val>
</magnitude>
</node>
<node name="Na+/melibiose symporter and related transporters"><magnitude>
<val> 431.177774</val>
</magnitude>
</node>
<node name="Neuraminidase (sialidase)"><magnitude>
<val> 63.797959</val>
</magnitude>
</node>
<node name="Neutral trehalase"><magnitude>
<val> 82.818156</val>
</magnitude>
</node>
<node name="PEP phosphonomutase and related enzymes"><magnitude>
<val> 160.174058</val>
</magnitude>
</node>
<node name="Pectate lyase"><magnitude>
<val> 192.427197</val>
</magnitude>
</node>
<node name="Pectin methylesterase"><magnitude>
<val> 280.382083</val>
</magnitude>
</node>
<node name="Pentose-5-phosphate-3-epimerase"><magnitude>
<val> 247.800495</val>
</magnitude>
</node>
<node name="Permeases of the drug/metabolite transporter (DMT) superfamily"><magnitude>
<val> 993.3483</val>
</magnitude>
</node>
<node name="Permeases of the major facilitator superfamily"><magnitude>
<val> 9.410017</val>
</magnitude>
</node>
<node name="Phosphoenolpyruvate synthase/pyruvate phosphate dikinase"><magnitude>
<val> 1005.140567</val>
</magnitude>
</node>
<node name="Phosphoenolpyruvate-protein kinase (PTS system EI component in bacteria)"><magnitude>
<val> 272.901576</val>
</magnitude>
</node>
<node name="Phosphoglucomutase"><magnitude>
<val> 179.694715</val>
</magnitude>
</node>
<node name="Phosphoglycerate mutase 1"><magnitude>
<val> 166.488004</val>
</magnitude>
</node>
<node name="Phosphoglyceromutase"><magnitude>
<val> 240.096914</val>
</magnitude>
</node>
<node name="Phosphoheptose isomerase"><magnitude>
<val> 76.464369</val>
</magnitude>
</node>
<node name="Phosphoketolase"><magnitude>
<val> 105.576012</val>
</magnitude>
</node>
<node name="Phosphomannomutase"><magnitude>
<val> 603.345205</val>
</magnitude>
</node>
<node name="Phosphomannose isomerase"><magnitude>
<val> 72.742014</val>
</magnitude>
</node>
<node name="Phosphopentomutase"><magnitude>
<val> 88.283403</val>
</magnitude>
</node>
<node name="Phosphotransferase system IIA components"><magnitude>
<val> 75.471584</val>
</magnitude>
</node>
<node name="Phosphotransferase system IIB components"><magnitude>
<val> 57.948261</val>
</magnitude>
</node>
<node name="Phosphotransferase system IIC components, glucose/maltose/N-acetylglucosamine-specific"><magnitude>
<val> 193.461715</val>
</magnitude>
</node>
<node name="Phosphotransferase system cellobiose-specific component IIA"><magnitude>
<val> 47.175074</val>
</magnitude>
</node>
<node name="Phosphotransferase system cellobiose-specific component IIB"><magnitude>
<val> 78.298852</val>
</magnitude>
</node>
<node name="Phosphotransferase system cellobiose-specific component IIC"><magnitude>
<val> 201.711786</val>
</magnitude>
</node>
<node name="Phosphotransferase system fructose-specific component IIB"><magnitude>
<val> 17.579378</val>
</magnitude>
</node>
<node name="Phosphotransferase system mannitol/fructose-specific IIA domain (Ntr-type)"><magnitude>
<val> 131.045135</val>
</magnitude>
</node>
<node name="Phosphotransferase system sorbitol-specific component IIA"><magnitude>
<val> 38.31116</val>
</magnitude>
</node>
<node name="Phosphotransferase system sorbitol-specific component IIBC"><magnitude>
<val> 13.920424</val>
</magnitude>
</node>
<node name="Phosphotransferase system sorbitol-specific component IIC"><magnitude>
<val> 5.201777</val>
</magnitude>
</node>
<node name="Phosphotransferase system, HPr-related proteins"><magnitude>
<val> 130.016392</val>
</magnitude>
</node>
<node name="Phosphotransferase system, fructose-specific IIC component"><magnitude>
<val> 69.334988</val>
</magnitude>
</node>
<node name="Phosphotransferase system, galactitol-specific IIB component"><magnitude>
<val> 86.743511</val>
</magnitude>
</node>
<node name="Phosphotransferase system, galactitol-specific IIC component"><magnitude>
<val> 80.659716</val>
</magnitude>
</node>
<node name="Phosphotransferase system, mannitol-specific IIBC component"><magnitude>
<val> 117.1594</val>
</magnitude>
</node>
<node name="Phosphotransferase system, mannose/fructose-specific component IIA"><magnitude>
<val> 169.822122</val>
</magnitude>
</node>
<node name="Phosphotransferase system, mannose/fructose/N-acetylgalactosamine-specific component IIB"><magnitude>
<val> 99.066117</val>
</magnitude>
</node>
<node name="Phosphotransferase system, mannose/fructose/N-acetylgalactosamine-specific component IIC"><magnitude>
<val> 129.793723</val>
</magnitude>
</node>
<node name="Phosphotransferase system, mannose/fructose/N-acetylgalactosamine-specific component IID"><magnitude>
<val> 242.847127</val>
</magnitude>
</node>
<node name="Predicted 6-phosphogluconate dehydrogenase"><magnitude>
<val> 62.758011</val>
</magnitude>
</node>
<node name="Predicted N-acetylglucosamine kinase"><magnitude>
<val> 85.523776</val>
</magnitude>
</node>
<node name="Predicted N-acetylglucosaminyl transferase"><magnitude>
<val> 205.049494</val>
</magnitude>
</node>
<node name="Predicted bile acid beta-glucosidase"><magnitude>
<val> 163.215574</val>
</magnitude>
</node>
<node name="Predicted extracellular endo alpha-1,4 polygalactosaminidase or related polysaccharide hydrolase"><magnitude>
<val> 13.292035</val>
</magnitude>
</node>
<node name="Predicted glycosyl hydrolase"><magnitude>
<val> 52.846281</val>
</magnitude>
</node>
<node name="Predicted glycosylase"><magnitude>
<val> 168.989831</val>
</magnitude>
</node>
<node name="Predicted neuraminidase (sialidase)"><magnitude>
<val> 92.369722</val>
</magnitude>
</node>
<node name="Predicted phosphoglycerate mutase, AP superfamily"><magnitude>
<val> 286.631461</val>
</magnitude>
</node>
<node name="Predicted sugar epimerase"><magnitude>
<val> 33.769855</val>
</magnitude>
</node>
<node name="Predicted sugar kinase"><magnitude>
<val> 482.53709</val>
</magnitude>
</node>
<node name="Predicted sugar phosphatases of the HAD superfamily"><magnitude>
<val> 43.365356</val>
</magnitude>
</node>
<node name="Predicted tagatose 6-phosphate kinase"><magnitude>
<val> 59.704103</val>
</magnitude>
</node>
<node name="Predicted xylanase/chitin deacetylase"><magnitude>
<val> 779.485911</val>
</magnitude>
</node>
<node name="Protein involved in polysaccharide intercellular adhesin (PIA) synthesis/biofilm formation"><magnitude>
<val> 15.020946</val>
</magnitude>
</node>
<node name="Putative L-xylulose-5-phosphate 3-epimerase"><magnitude>
<val> 40.724247</val>
</magnitude>
</node>
<node name="Putative N-acetylmannosamine-6-phosphate epimerase"><magnitude>
<val> 75.728196</val>
</magnitude>
</node>
<node name="Putative alpha-1,2-mannosidase"><magnitude>
<val> 464.078527</val>
</magnitude>
</node>
<node name="Putative glucose uptake permease"><magnitude>
<val> 133.387392</val>
</magnitude>
</node>
<node name="Putative glycerate kinase"><magnitude>
<val> 49.372522</val>
</magnitude>
</node>
<node name="Pyruvate decarboxylase and related thiamine pyrophosphate-requiring enzymes"><magnitude>
<val> 138.720069</val>
</magnitude>
</node>
<node name="Pyruvate kinase"><magnitude>
<val> 366.286431</val>
</magnitude>
</node>
<node name="Ribose 5-phosphate isomerase"><magnitude>
<val> 195.775679</val>
</magnitude>
</node>
<node name="Ribose 5-phosphate isomerase RpiB"><magnitude>
<val> 205.659423</val>
</magnitude>
</node>
<node name="Ribose/xylose/arabinose/galactoside ABC-type transport systems, permease components"><magnitude>
<val> 181.220993</val>
</magnitude>
</node>
<node name="Ribulose 1,5-bisphosphate carboxylase, large subunit"><magnitude>
<val> 89.804943</val>
</magnitude>
</node>
<node name="Ribulose-5-phosphate 4-epimerase and related epimerases and aldolases"><magnitude>
<val> 295.675494</val>
</magnitude>
</node>
<node name="Sugar (pentulose and hexulose) kinases"><magnitude>
<val> 198.989984</val>
</magnitude>
</node>
<node name="Sugar kinases, ribokinase family"><magnitude>
<val> 655.194889</val>
</magnitude>
</node>
<node name="Sugar phosphate isomerases/epimerases"><magnitude>
<val> 480.668556</val>
</magnitude>
</node>
<node name="Sugar phosphate permease"><magnitude>
<val> 428.196105</val>
</magnitude>
</node>
<node name="TRAP-type C4-dicarboxylate transport system, large permease component"><magnitude>
<val> 223.389973</val>
</magnitude>
</node>
<node name="TRAP-type C4-dicarboxylate transport system, periplasmic component"><magnitude>
<val> 271.813006</val>
</magnitude>
</node>
<node name="TRAP-type C4-dicarboxylate transport system, small permease component"><magnitude>
<val> 136.379167</val>
</magnitude>
</node>
<node name="Tagatose-1,6-bisphosphate aldolase"><magnitude>
<val> 106.02272</val>
</magnitude>
</node>
<node name="Thermophilic glucose-6-phosphate isomerase and related metalloenzymes"><magnitude>
<val> 34.450109</val>
</magnitude>
</node>
<node name="Transaldolase"><magnitude>
<val> 272.705703</val>
</magnitude>
</node>
<node name="Transketolase"><magnitude>
<val> 353.163895</val>
</magnitude>
</node>
<node name="Transketolase, C-terminal subunit"><magnitude>
<val> 166.577491</val>
</magnitude>
</node>
<node name="Transketolase, N-terminal subunit"><magnitude>
<val> 159.010658</val>
</magnitude>
</node>
<node name="Trehalose and maltose hydrolases (possible phosphorylases)"><magnitude>
<val> 310.210678</val>
</magnitude>
</node>
<node name="Trehalose utilization protein"><magnitude>
<val> 18.514714</val>
</magnitude>
</node>
<node name="Trehalose-6-phosphatase"><magnitude>
<val> 142.167992</val>
</magnitude>
</node>
<node name="Trehalose-6-phosphate synthase"><magnitude>
<val> 138.653086</val>
</magnitude>
</node>
<node name="Triosephosphate isomerase"><magnitude>
<val> 250.734985</val>
</magnitude>
</node>
<node name="Type II secretory pathway, pullulanase PulA and related glycosidases"><magnitude>
<val> 137.879</val>
</magnitude>
</node>
<node name="UDP-glucose pyrophosphorylase"><magnitude>
<val> 41.066028</val>
</magnitude>
</node>
<node name="Uncharacterized enzyme involved in inositol metabolism"><magnitude>
<val> 30.941789</val>
</magnitude>
</node>
<node name="Uncharacterized enzymes related to aldose 1-epimerase"><magnitude>
<val> 45.01643</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria"><magnitude>
<val> 176.686331</val>
</magnitude>
</node>
<node name="Uncharacterized protein, probably involved in trehalose biosynthesis"><magnitude>
<val> 117.961837</val>
</magnitude>
</node>
<node name="Xylose isomerase"><magnitude>
<val> 68.010332</val>
</magnitude>
</node>
</node>
<node name="Coenzyme transport and metabolism"><magnitude>
<val> 34199.830041</val>
</magnitude>
<node name="1,4-dihydroxy-2-naphthoate octaprenyltransferase"><magnitude>
<val> 158.824855</val>
</magnitude>
</node>
<node name="2-polyprenyl-3-methyl-5-hydroxy-6-metoxy-1,4-benzoquinol methylase"><magnitude>
<val> 892.432537</val>
</magnitude>
</node>
<node name="2-polyprenyl-6-methoxyphenol hydroxylase and related FAD-dependent oxidoreductases"><magnitude>
<val> 30.539825</val>
</magnitude>
</node>
<node name="2-polyprenylphenol hydroxylase and related flavodoxin oxidoreductases"><magnitude>
<val> 594.00927</val>
</magnitude>
</node>
<node name="2-succinyl-6-hydroxy-2,4-cyclohexadiene-1-carboxylate synthase"><magnitude>
<val> 74.268119</val>
</magnitude>
</node>
<node name="3,4-dihydroxy-2-butanone 4-phosphate synthase"><magnitude>
<val> 309.668697</val>
</magnitude>
</node>
<node name="3-polyprenyl-4-hydroxybenzoate decarboxylase"><magnitude>
<val> 168.481476</val>
</magnitude>
</node>
<node name="3-polyprenyl-4-hydroxybenzoate decarboxylase and related decarboxylases"><magnitude>
<val> 256.178089</val>
</magnitude>
</node>
<node name="4-hydroxybenzoate polyprenyltransferase and related prenyltransferases"><magnitude>
<val> 427.907809</val>
</magnitude>
</node>
<node name="4-hydroxybenzoate synthetase (chorismate lyase)"><magnitude>
<val> 66.492367</val>
</magnitude>
</node>
<node name="5,10-methylene-tetrahydrofolate dehydrogenase/Methenyl tetrahydrofolate cyclohydrolase"><magnitude>
<val> 278.180469</val>
</magnitude>
</node>
<node name="5-formyltetrahydrofolate cyclo-ligase"><magnitude>
<val> 199.579927</val>
</magnitude>
</node>
<node name="6-pyruvoyl-tetrahydropterin synthase"><magnitude>
<val> 201.374551</val>
</magnitude>
</node>
<node name="7,8-dihydro-6-hydroxymethylpterin-pyrophosphokinase"><magnitude>
<val> 159.026582</val>
</magnitude>
</node>
<node name="7-keto-8-aminopelargonate synthetase and related enzymes"><magnitude>
<val> 243.493107</val>
</magnitude>
</node>
<node name="ABC-type cobalamin transport system, ATPase component"><magnitude>
<val> 5.195585</val>
</magnitude>
</node>
<node name="ABC-type cobalamin transport system, permease component"><magnitude>
<val> 16.82396</val>
</magnitude>
</node>
<node name="ABC-type thiamine transport system, ATPase component"><magnitude>
<val> 10.392738</val>
</magnitude>
</node>
<node name="ABC-type thiamine transport system, periplasmic component"><magnitude>
<val> 53.731831</val>
</magnitude>
</node>
<node name="ABC-type tungstate transport system, periplasmic component"><magnitude>
<val> 128.386094</val>
</magnitude>
</node>
<node name="ABC-type tungstate transport system, permease component"><magnitude>
<val> 254.879684</val>
</magnitude>
</node>
<node name="ATP:corrinoid adenosyltransferase"><magnitude>
<val> 230.821249</val>
</magnitude>
</node>
<node name="Adenosyl cobinamide kinase/adenosyl cobinamide phosphate guanylyltransferase"><magnitude>
<val> 174.054375</val>
</magnitude>
</node>
<node name="Adenosylmethionine-8-amino-7-oxononanoate aminotransferase"><magnitude>
<val> 317.143451</val>
</magnitude>
</node>
<node name="Archaeal riboflavin synthase"><magnitude>
<val> 69.984837</val>
</magnitude>
</node>
<node name="Aspartate 1-decarboxylase"><magnitude>
<val> 129.898504</val>
</magnitude>
</node>
<node name="Aspartate oxidase"><magnitude>
<val> 285.950069</val>
</magnitude>
</node>
<node name="Biotin synthase and related enzymes"><magnitude>
<val> 401.844398</val>
</magnitude>
</node>
<node name="Biotin-(acetyl-CoA carboxylase) ligase"><magnitude>
<val> 273.793483</val>
</magnitude>
</node>
<node name="Cobalamin biosynthesis protein CbiD"><magnitude>
<val> 184.644369</val>
</magnitude>
</node>
<node name="Cobalamin biosynthesis protein CbiG"><magnitude>
<val> 206.765557</val>
</magnitude>
</node>
<node name="Cobalamin biosynthesis protein CbiK, Co2+ chelatase"><magnitude>
<val> 218.992672</val>
</magnitude>
</node>
<node name="Cobalamin biosynthesis protein CobD/CbiB"><magnitude>
<val> 307.522885</val>
</magnitude>
</node>
<node name="Cobalamin biosynthesis protein CobN and related Mg-chelatases"><magnitude>
<val> 635.025402</val>
</magnitude>
</node>
<node name="Cobalamin biosynthesis protein CobT (nicotinate-mononucleotide:5, 6-dimethylbenzimidazole phosphoribosyltransferase)"><magnitude>
<val> 190.525584</val>
</magnitude>
</node>
<node name="Cobalamin-5-phosphate synthase"><magnitude>
<val> 215.26658</val>
</magnitude>
</node>
<node name="Cobyric acid synthase"><magnitude>
<val> 274.192641</val>
</magnitude>
</node>
<node name="Cobyrinic acid a,c-diamide synthase"><magnitude>
<val> 285.470665</val>
</magnitude>
</node>
<node name="Coenzyme F390 synthetase"><magnitude>
<val> 963.048297</val>
</magnitude>
</node>
<node name="Coproporphyrinogen III oxidase"><magnitude>
<val> 29.508988</val>
</magnitude>
</node>
<node name="Coproporphyrinogen III oxidase and related Fe-S oxidoreductases"><magnitude>
<val> 319.884592</val>
</magnitude>
</node>
<node name="Delta-aminolevulinic acid dehydratase"><magnitude>
<val> 239.729226</val>
</magnitude>
</node>
<node name="Demethylmenaquinone methyltransferase"><magnitude>
<val> 81.345863</val>
</magnitude>
</node>
<node name="Deoxyxylulose-5-phosphate synthase"><magnitude>
<val> 288.437104</val>
</magnitude>
</node>
<node name="Dephospho-CoA kinase"><magnitude>
<val> 221.84968</val>
</magnitude>
</node>
<node name="Dethiobiotin synthetase"><magnitude>
<val> 134.196939</val>
</magnitude>
</node>
<node name="Dihydrofolate reductase"><magnitude>
<val> 300.220577</val>
</magnitude>
</node>
<node name="Dihydroneopterin aldolase"><magnitude>
<val> 147.453791</val>
</magnitude>
</node>
<node name="Dihydropteroate synthase and related enzymes"><magnitude>
<val> 273.188361</val>
</magnitude>
</node>
<node name="Dihydroxynaphthoic acid synthase"><magnitude>
<val> 86.611266</val>
</magnitude>
</node>
<node name="Dinucleotide-utilizing enzymes involved in molybdopterin and thiamine biosynthesis family 1"><magnitude>
<val> 218.58212</val>
</magnitude>
</node>
<node name="Dinucleotide-utilizing enzymes involved in molybdopterin and thiamine biosynthesis family 2"><magnitude>
<val> 407.220245</val>
</magnitude>
</node>
<node name="FAD synthase"><magnitude>
<val> 269.161214</val>
</magnitude>
</node>
<node name="Flavoprotein involved in thiazole biosynthesis"><magnitude>
<val> 133.64548</val>
</magnitude>
</node>
<node name="Folylpolyglutamate synthase"><magnitude>
<val> 257.375776</val>
</magnitude>
</node>
<node name="GTP cyclohydrolase I"><magnitude>
<val> 143.639573</val>
</magnitude>
</node>
<node name="GTP cyclohydrolase II"><magnitude>
<val> 82.400041</val>
</magnitude>
</node>
<node name="GTP:adenosylcobinamide-phosphate guanylyltransferase"><magnitude>
<val> 185.011451</val>
</magnitude>
</node>
<node name="Gamma-glutamylcysteine synthetase"><magnitude>
<val> 160.422629</val>
</magnitude>
</node>
<node name="Geranylgeranyl pyrophosphate synthase"><magnitude>
<val> 369.608078</val>
</magnitude>
</node>
<node name="Glutamate-1-semialdehyde aminotransferase"><magnitude>
<val> 288.116599</val>
</magnitude>
</node>
<node name="Glutamyl-tRNA reductase"><magnitude>
<val> 311.417358</val>
</magnitude>
</node>
<node name="Glutathione synthase/Ribosomal protein S6 modification enzyme (glutaminyl transferase)"><magnitude>
<val> 276.607455</val>
</magnitude>
</node>
<node name="Hydroxyethylthiazole kinase, sugar kinase family"><magnitude>
<val> 93.23894</val>
</magnitude>
</node>
<node name="Hydroxymethylpyrimidine/phosphomethylpyrimidine kinase"><magnitude>
<val> 245.275647</val>
</magnitude>
</node>
<node name="Isochorismate synthase"><magnitude>
<val> 37.159594</val>
</magnitude>
</node>
<node name="Ketopantoate hydroxymethyltransferase"><magnitude>
<val> 180.549751</val>
</magnitude>
</node>
<node name="Ketopantoate reductase"><magnitude>
<val> 275.938603</val>
</magnitude>
</node>
<node name="Lipoate synthase"><magnitude>
<val> 160.099871</val>
</magnitude>
</node>
<node name="Lipoate-protein ligase A"><magnitude>
<val> 177.54081</val>
</magnitude>
</node>
<node name="Lipoate-protein ligase B"><magnitude>
<val> 128.335302</val>
</magnitude>
</node>
<node name="Membrane-associated lipoprotein involved in thiamine biosynthesis"><magnitude>
<val> 170.849238</val>
</magnitude>
</node>
<node name="Methenyltetrahydromethanopterin cyclohydrolase"><magnitude>
<val> 92.436395</val>
</magnitude>
</node>
<node name="Methyl coenzyme M reductase, alpha subunit"><magnitude>
<val> 237.624906</val>
</magnitude>
</node>
<node name="Methyl coenzyme M reductase, beta subunit"><magnitude>
<val> 123.086626</val>
</magnitude>
</node>
<node name="Methyl coenzyme M reductase, gamma subunit"><magnitude>
<val> 109.937178</val>
</magnitude>
</node>
<node name="Methyl coenzyme M reductase, subunit C"><magnitude>
<val> 62.750632</val>
</magnitude>
</node>
<node name="Methyl coenzyme M reductase, subunit D"><magnitude>
<val> 110.613158</val>
</magnitude>
</node>
<node name="Methylase involved in ubiquinone/menaquinone biosynthesis"><magnitude>
<val> 1988.523165</val>
</magnitude>
</node>
<node name="Mg-chelatase subunit ChlD"><magnitude>
<val> 240.338992</val>
</magnitude>
</node>
<node name="Mg-chelatase subunit ChlI"><magnitude>
<val> 177.464757</val>
</magnitude>
</node>
<node name="Molybdenum cofactor biosynthesis enzyme"><magnitude>
<val> 483.823469</val>
</magnitude>
</node>
<node name="Molybdopterin biosynthesis enzyme"><magnitude>
<val> 686.85579</val>
</magnitude>
</node>
<node name="Molybdopterin biosynthesis enzymes"><magnitude>
<val> 220.350289</val>
</magnitude>
</node>
<node name="Molybdopterin converting factor, large subunit"><magnitude>
<val> 108.524613</val>
</magnitude>
</node>
<node name="Molybdopterin converting factor, small subunit"><magnitude>
<val> 219.468311</val>
</magnitude>
</node>
<node name="Molybdopterin-binding protein"><magnitude>
<val> 94.842593</val>
</magnitude>
</node>
<node name="Molybdopterin-guanine dinucleotide biosynthesis protein"><magnitude>
<val> 165.835212</val>
</magnitude>
</node>
<node name="Molybdopterin-guanine dinucleotide biosynthesis protein A"><magnitude>
<val> 165.865854</val>
</magnitude>
</node>
<node name="NAD synthase"><magnitude>
<val> 268.606649</val>
</magnitude>
</node>
<node name="Na+/panthothenate symporter"><magnitude>
<val> 79.217128</val>
</magnitude>
</node>
<node name="NaMN:DMB phosphoribosyltransferase"><magnitude>
<val> 215.372473</val>
</magnitude>
</node>
<node name="Nicotinamide mononucleotide adenylyltransferase"><magnitude>
<val> 78.310252</val>
</magnitude>
</node>
<node name="Nicotinamide mononucleotide transporter"><magnitude>
<val> 37.105637</val>
</magnitude>
</node>
<node name="Nicotinate-nucleotide pyrophosphorylase"><magnitude>
<val> 292.991337</val>
</magnitude>
</node>
<node name="Nicotinic acid mononucleotide adenylyltransferase"><magnitude>
<val> 131.173842</val>
</magnitude>
</node>
<node name="Nicotinic acid phosphoribosyltransferase"><magnitude>
<val> 155.086122</val>
</magnitude>
</node>
<node name="O-succinylbenzoate synthase"><magnitude>
<val> 8.177463</val>
</magnitude>
</node>
<node name="Outer membrane cobalamin receptor protein"><magnitude>
<val> 524.767582</val>
</magnitude>
</node>
<node name="Panthothenate kinase"><magnitude>
<val> 59.103545</val>
</magnitude>
</node>
<node name="Panthothenate synthetase"><magnitude>
<val> 252.594571</val>
</magnitude>
</node>
<node name="Pantothenate kinase, acetyl-CoA regulated"><magnitude>
<val> 40.59289</val>
</magnitude>
</node>
<node name="Phosphoglycerate dehydrogenase and related dehydrogenases"><magnitude>
<val> 323.461827</val>
</magnitude>
</node>
<node name="Phosphopantetheine adenylyltransferase"><magnitude>
<val> 153.386116</val>
</magnitude>
</node>
<node name="Phosphopantetheinyl transferase"><magnitude>
<val> 55.465842</val>
</magnitude>
</node>
<node name="Phosphopantothenoylcysteine synthetase/decarboxylase"><magnitude>
<val> 233.310737</val>
</magnitude>
</node>
<node name="Phosphoribosyl-dephospho-CoA transferase (holo-ACP synthetase)"><magnitude>
<val> 97.797535</val>
</magnitude>
</node>
<node name="Phosphoserine aminotransferase"><magnitude>
<val> 131.147702</val>
</magnitude>
</node>
<node name="Phosphosulfolactate phosphohydrolase and related enzymes"><magnitude>
<val> 80.276347</val>
</magnitude>
</node>
<node name="Pimeloyl-CoA synthetase"><magnitude>
<val> 20.381119</val>
</magnitude>
</node>
<node name="Porphobilinogen deaminase"><magnitude>
<val> 201.071042</val>
</magnitude>
</node>
<node name="Precorrin isomerase"><magnitude>
<val> 185.694098</val>
</magnitude>
</node>
<node name="Precorrin-2 methylase"><magnitude>
<val> 208.44607</val>
</magnitude>
</node>
<node name="Precorrin-3B methylase"><magnitude>
<val> 220.619063</val>
</magnitude>
</node>
<node name="Precorrin-4 methylase"><magnitude>
<val> 241.639068</val>
</magnitude>
</node>
<node name="Precorrin-6B methylase 1"><magnitude>
<val> 169.516177</val>
</magnitude>
</node>
<node name="Precorrin-6B methylase 2"><magnitude>
<val> 267.831197</val>
</magnitude>
</node>
<node name="Precorrin-6x reductase"><magnitude>
<val> 113.826676</val>
</magnitude>
</node>
<node name="Predicted ATPase/kinase involved in NAD metabolism"><magnitude>
<val> 11.470752</val>
</magnitude>
</node>
<node name="Predicted HD superfamily hydrolase involved in NAD metabolism"><magnitude>
<val> 58.422793</val>
</magnitude>
</node>
<node name="Predicted glutamine amidotransferase involved in pyridoxine biosynthesis"><magnitude>
<val> 108.858578</val>
</magnitude>
</node>
<node name="Protoheme ferro-lyase (ferrochelatase)"><magnitude>
<val> 109.161396</val>
</magnitude>
</node>
<node name="Protoporphyrinogen oxidase"><magnitude>
<val> 243.536378</val>
</magnitude>
</node>
<node name="Pterin-4a-carbinolamine dehydratase"><magnitude>
<val> 47.664732</val>
</magnitude>
</node>
<node name="Putative p-aminobenzoyl-glutamate transporter"><magnitude>
<val> 110.953219</val>
</magnitude>
</node>
<node name="Pyridoxal phosphate biosynthesis protein"><magnitude>
<val> 324.465899</val>
</magnitude>
</node>
<node name="Pyridoxal/pyridoxine/pyridoxamine kinase"><magnitude>
<val> 75.391687</val>
</magnitude>
</node>
<node name="Pyridoxamine-phosphate oxidase"><magnitude>
<val> 81.228625</val>
</magnitude>
</node>
<node name="Pyridoxine biosynthesis enzyme"><magnitude>
<val> 175.231679</val>
</magnitude>
</node>
<node name="Pyrimidine deaminase"><magnitude>
<val> 103.753052</val>
</magnitude>
</node>
<node name="Pyrimidine reductase, riboflavin biosynthesis"><magnitude>
<val> 244.445338</val>
</magnitude>
</node>
<node name="Pyrroloquinoline quinone (Coenzyme PQQ) biosynthesis protein C"><magnitude>
<val> 58.730169</val>
</magnitude>
</node>
<node name="Quinolinate synthase"><magnitude>
<val> 253.432696</val>
</magnitude>
</node>
<node name="Riboflavin synthase alpha chain"><magnitude>
<val> 154.863484</val>
</magnitude>
</node>
<node name="Riboflavin synthase beta-chain"><magnitude>
<val> 228.142353</val>
</magnitude>
</node>
<node name="S-adenosylhomocysteine hydrolase"><magnitude>
<val> 279.223747</val>
</magnitude>
</node>
<node name="S-adenosylmethionine synthetase"><magnitude>
<val> 239.456603</val>
</magnitude>
</node>
<node name="Siroheme synthase (precorrin-2 oxidase/ferrochelatase domain)"><magnitude>
<val> 216.776625</val>
</magnitude>
</node>
<node name="Sulfur transfer protein involved in thiamine biosynthesis"><magnitude>
<val> 251.77234</val>
</magnitude>
</node>
<node name="Tetrahydromethanopterin S-methyltransferase, subunit A"><magnitude>
<val> 163.357433</val>
</magnitude>
</node>
<node name="Tetrahydromethanopterin S-methyltransferase, subunit B"><magnitude>
<val> 94.154932</val>
</magnitude>
</node>
<node name="Tetrahydromethanopterin S-methyltransferase, subunit C"><magnitude>
<val> 67.10211</val>
</magnitude>
</node>
<node name="Tetrahydromethanopterin S-methyltransferase, subunit D"><magnitude>
<val> 68.977867</val>
</magnitude>
</node>
<node name="Tetrahydromethanopterin S-methyltransferase, subunit E"><magnitude>
<val> 86.240446</val>
</magnitude>
</node>
<node name="Tetrahydromethanopterin S-methyltransferase, subunit F"><magnitude>
<val> 59.74163</val>
</magnitude>
</node>
<node name="Tetrahydromethanopterin S-methyltransferase, subunit G"><magnitude>
<val> 67.167563</val>
</magnitude>
</node>
<node name="Tetrahydromethanopterin S-methyltransferase, subunit H"><magnitude>
<val> 137.752044</val>
</magnitude>
</node>
<node name="Thiamine biosynthesis ATP pyrophosphatase"><magnitude>
<val> 187.395887</val>
</magnitude>
</node>
<node name="Thiamine biosynthesis enzyme ThiH and related uncharacterized enzymes"><magnitude>
<val> 749.530395</val>
</magnitude>
</node>
<node name="Thiamine biosynthesis protein ThiC"><magnitude>
<val> 365.640019</val>
</magnitude>
</node>
<node name="Thiamine monophosphate kinase"><magnitude>
<val> 243.597575</val>
</magnitude>
</node>
<node name="Thiamine monophosphate synthase"><magnitude>
<val> 299.319151</val>
</magnitude>
</node>
<node name="Thiamine pyrophosphokinase"><magnitude>
<val> 81.894014</val>
</magnitude>
</node>
<node name="Trimethylamine:corrinoid methyltransferase"><magnitude>
<val> 128.641857</val>
</magnitude>
</node>
<node name="Triphosphoribosyl-dephospho-CoA synthetase"><magnitude>
<val> 123.296373</val>
</magnitude>
</node>
<node name="Ubiquinone biosynthesis protein COQ7"><magnitude>
<val> 37.868251</val>
</magnitude>
</node>
<node name="Uncharacterized enzyme of heme biosynthesis"><magnitude>
<val> 383.534269</val>
</magnitude>
</node>
<node name="Uncharacterized enzyme of thiazole biosynthesis"><magnitude>
<val> 186.033962</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in ubiquinone biosynthesis"><magnitude>
<val> 9.517414</val>
</magnitude>
</node>
<node name="Uroporphyrinogen-III decarboxylase"><magnitude>
<val> 216.164798</val>
</magnitude>
</node>
<node name="Uroporphyrinogen-III methylase"><magnitude>
<val> 213.325717</val>
</magnitude>
</node>
<node name="Uroporphyrinogen-III synthase"><magnitude>
<val> 127.84739</val>
</magnitude>
</node>
</node>
<node name="Energy production and conversion"><magnitude>
<val> 59538.782679</val>
</magnitude>
<node name="2-oxoglutarate dehydrogenase complex, dehydrogenase (E1) component, and related enzymes"><magnitude>
<val> 253.571904</val>
</magnitude>
</node>
<node name="6Fe-6S prismane cluster-containing protein"><magnitude>
<val> 591.595195</val>
</magnitude>
</node>
<node name="ABC-type Na+ efflux pump, permease component"><magnitude>
<val> 212.10329</val>
</magnitude>
</node>
<node name="ABC-type Na+ transport system, ATPase component"><magnitude>
<val> 11.170951</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in cytochrome bd biosynthesis, ATPase and permease components"><magnitude>
<val> 218.738884</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in cytochrome bd biosynthesis, fused ATPase and permease components"><magnitude>
<val> 271.959444</val>
</magnitude>
</node>
<node name="ATP/ADP translocase"><magnitude>
<val> 76.269682</val>
</magnitude>
</node>
<node name="Acetate kinase"><magnitude>
<val> 238.841212</val>
</magnitude>
</node>
<node name="Acetyl-CoA hydrolase"><magnitude>
<val> 469.056931</val>
</magnitude>
</node>
<node name="Aconitase A"><magnitude>
<val> 360.806999</val>
</magnitude>
</node>
<node name="Aconitase B"><magnitude>
<val> 132.708257</val>
</magnitude>
</node>
<node name="Acyl-CoA synthetase (NDP forming)"><magnitude>
<val> 387.246774</val>
</magnitude>
</node>
<node name="Acylphosphatases"><magnitude>
<val> 322.966299</val>
</magnitude>
</node>
<node name="Aerobic-type carbon monoxide dehydrogenase, large subunit CoxL/CutL homologs"><magnitude>
<val> 425.997512</val>
</magnitude>
</node>
<node name="Aerobic-type carbon monoxide dehydrogenase, middle subunit CoxM/CutM homologs"><magnitude>
<val> 160.899136</val>
</magnitude>
</node>
<node name="Aerobic-type carbon monoxide dehydrogenase, small subunit CoxS/CutS homologs"><magnitude>
<val> 264.299128</val>
</magnitude>
</node>
<node name="Alcohol dehydrogenase, class IV"><magnitude>
<val> 662.871084</val>
</magnitude>
</node>
<node name="Aldehyde:ferredoxin oxidoreductase"><magnitude>
<val> 624.732927</val>
</magnitude>
</node>
<node name="Anaerobic dehydrogenases, typically selenocysteine-containing"><magnitude>
<val> 659.891906</val>
</magnitude>
</node>
<node name="Archaeal flavoproteins"><magnitude>
<val> 163.681689</val>
</magnitude>
</node>
<node name="Archaeal/vacuolar-type H+-ATPase subunit A"><magnitude>
<val> 339.024083</val>
</magnitude>
</node>
<node name="Archaeal/vacuolar-type H+-ATPase subunit B"><magnitude>
<val> 126.148283</val>
</magnitude>
</node>
<node name="Archaeal/vacuolar-type H+-ATPase subunit C"><magnitude>
<val> 144.918799</val>
</magnitude>
</node>
<node name="Archaeal/vacuolar-type H+-ATPase subunit D"><magnitude>
<val> 175.252583</val>
</magnitude>
</node>
<node name="Archaeal/vacuolar-type H+-ATPase subunit E"><magnitude>
<val> 124.198255</val>
</magnitude>
</node>
<node name="Archaeal/vacuolar-type H+-ATPase subunit F"><magnitude>
<val> 94.481073</val>
</magnitude>
</node>
<node name="Archaeal/vacuolar-type H+-ATPase subunit H"><magnitude>
<val> 28.516448</val>
</magnitude>
</node>
<node name="Archaeal/vacuolar-type H+-ATPase subunit I"><magnitude>
<val> 418.184599</val>
</magnitude>
</node>
<node name="Azurin"><magnitude>
<val> 10.50753</val>
</magnitude>
</node>
<node name="Butyrate kinase"><magnitude>
<val> 164.298876</val>
</magnitude>
</node>
<node name="C4-dicarboxylate transporter"><magnitude>
<val> 23.657113</val>
</magnitude>
</node>
<node name="CO dehydrogenase/acetyl-CoA synthase alpha subunit"><magnitude>
<val> 250.663389</val>
</magnitude>
</node>
<node name="CO dehydrogenase/acetyl-CoA synthase beta subunit"><magnitude>
<val> 281.017248</val>
</magnitude>
</node>
<node name="CO dehydrogenase/acetyl-CoA synthase delta subunit (corrinoid Fe-S protein)"><magnitude>
<val> 148.382609</val>
</magnitude>
</node>
<node name="CO dehydrogenase/acetyl-CoA synthase epsilon subunit"><magnitude>
<val> 59.938007</val>
</magnitude>
</node>
<node name="CO dehydrogenase/acetyl-CoA synthase gamma subunit (corrinoid Fe-S protein)"><magnitude>
<val> 203.176768</val>
</magnitude>
</node>
<node name="Cbb3-type cytochrome oxidase, cytochrome c subunit"><magnitude>
<val> 46.334765</val>
</magnitude>
</node>
<node name="Citrate lyase synthetase"><magnitude>
<val> 121.396526</val>
</magnitude>
</node>
<node name="Citrate lyase, alpha subunit"><magnitude>
<val> 156.496564</val>
</magnitude>
</node>
<node name="Citrate lyase, gamma subunit"><magnitude>
<val> 30.753353</val>
</magnitude>
</node>
<node name="Citrate synthase"><magnitude>
<val> 266.567495</val>
</magnitude>
</node>
<node name="Coenzyme F420-dependent N(5),N(10)-methenyltetrahydromethanopterin dehydrogenase"><magnitude>
<val> 76.25329</val>
</magnitude>
</node>
<node name="Coenzyme F420-dependent N5,N10-methylene tetrahydromethanopterin reductase and related flavin-dependent oxidoreductases"><magnitude>
<val> 136.568344</val>
</magnitude>
</node>
<node name="Coenzyme F420-reducing hydrogenase, alpha subunit"><magnitude>
<val> 425.493812</val>
</magnitude>
</node>
<node name="Coenzyme F420-reducing hydrogenase, beta subunit"><magnitude>
<val> 297.622373</val>
</magnitude>
</node>
<node name="Coenzyme F420-reducing hydrogenase, delta subunit"><magnitude>
<val> 391.778901</val>
</magnitude>
</node>
<node name="Coenzyme F420-reducing hydrogenase, gamma subunit"><magnitude>
<val> 354.617609</val>
</magnitude>
</node>
<node name="Cytochrome B561"><magnitude>
<val> 30.606875</val>
</magnitude>
</node>
<node name="Cytochrome b"><magnitude>
<val> 41.862512</val>
</magnitude>
</node>
<node name="Cytochrome b involved in lipid metabolism"><magnitude>
<val> 8.232364</val>
</magnitude>
</node>
<node name="Cytochrome b subunit of formate dehydrogenase"><magnitude>
<val> 112.717305</val>
</magnitude>
</node>
<node name="Cytochrome b subunit of the bc complex"><magnitude>
<val> 221.778103</val>
</magnitude>
</node>
<node name="Cytochrome bd-type quinol oxidase, subunit 1"><magnitude>
<val> 268.133263</val>
</magnitude>
</node>
<node name="Cytochrome bd-type quinol oxidase, subunit 2"><magnitude>
<val> 207.827778</val>
</magnitude>
</node>
<node name="Cytochrome c"><magnitude>
<val> 102.755735</val>
</magnitude>
</node>
<node name="Cytochrome c, mono- and diheme variants"><magnitude>
<val> 174.782517</val>
</magnitude>
</node>
<node name="Cytochrome c1"><magnitude>
<val> 65.363906</val>
</magnitude>
</node>
<node name="Cytochrome c2"><magnitude>
<val> 96.935982</val>
</magnitude>
</node>
<node name="Cytochrome c5"><magnitude>
<val> 20.449931</val>
</magnitude>
</node>
<node name="Cytochrome c553"><magnitude>
<val> 185.517778</val>
</magnitude>
</node>
<node name="Cytochrome c556"><magnitude>
<val> 132.317893</val>
</magnitude>
</node>
<node name="Dehydrogenases (flavoproteins)"><magnitude>
<val> 446.856639</val>
</magnitude>
</node>
<node name="Delta 1-pyrroline-5-carboxylate dehydrogenase"><magnitude>
<val> 54.173139</val>
</magnitude>
</node>
<node name="Desulfoferrodoxin"><magnitude>
<val> 130.838039</val>
</magnitude>
</node>
<node name="Dissimilatory sulfite reductase (desulfoviridin), alpha and beta subunits"><magnitude>
<val> 513.8688</val>
</magnitude>
</node>
<node name="Electron transfer flavoprotein, alpha subunit"><magnitude>
<val> 293.4494</val>
</magnitude>
</node>
<node name="Electron transfer flavoprotein, beta subunit"><magnitude>
<val> 331.023162</val>
</magnitude>
</node>
<node name="F0F1-type ATP synthase, alpha subunit"><magnitude>
<val> 203.599909</val>
</magnitude>
</node>
<node name="F0F1-type ATP synthase, beta subunit"><magnitude>
<val> 257.534829</val>
</magnitude>
</node>
<node name="F0F1-type ATP synthase, delta subunit (mitochondrial oligomycin sensitivity protein)"><magnitude>
<val> 179.393439</val>
</magnitude>
</node>
<node name="F0F1-type ATP synthase, epsilon subunit (mitochondrial delta subunit)"><magnitude>
<val> 155.624746</val>
</magnitude>
</node>
<node name="F0F1-type ATP synthase, gamma subunit"><magnitude>
<val> 172.87262</val>
</magnitude>
</node>
<node name="F0F1-type ATP synthase, subunit I"><magnitude>
<val> 9.113166</val>
</magnitude>
</node>
<node name="F0F1-type ATP synthase, subunit a"><magnitude>
<val> 168.427605</val>
</magnitude>
</node>
<node name="F0F1-type ATP synthase, subunit b"><magnitude>
<val> 317.801119</val>
</magnitude>
</node>
<node name="F0F1-type ATP synthase, subunit c/Archaeal/vacuolar-type H+-ATPase, subunit K"><magnitude>
<val> 272.761941</val>
</magnitude>
</node>
<node name="FAD/FMN-containing dehydrogenases"><magnitude>
<val> 461.65944</val>
</magnitude>
</node>
<node name="FOG: HEAT repeat"><magnitude>
<val> 843.444891</val>
</magnitude>
</node>
<node name="Fe-S oxidoreductase"><magnitude>
<val> 2475.584342</val>
</magnitude>
</node>
<node name="Fe-S oxidoreductase, related to NifB/MoaA family"><magnitude>
<val> 202.912776</val>
</magnitude>
</node>
<node name="Fe-S oxidoreductases"><magnitude>
<val> 290.863006</val>
</magnitude>
</node>
<node name="Fe-S-cluster-containing hydrogenase components 1"><magnitude>
<val> 539.650084</val>
</magnitude>
</node>
<node name="Fe-S-cluster-containing hydrogenase components 2"><magnitude>
<val> 175.084419</val>
</magnitude>
</node>
<node name="Ferredoxin"><magnitude>
<val> 900.81742</val>
</magnitude>
</node>
<node name="Ferredoxin-like protein"><magnitude>
<val> 37.529242</val>
</magnitude>
</node>
<node name="Ferredoxin-thioredoxin reductase, catalytic subunit"><magnitude>
<val> 87.01819</val>
</magnitude>
</node>
<node name="Flavodoxin"><magnitude>
<val> 95.05621</val>
</magnitude>
</node>
<node name="Flavodoxin reductases (ferredoxin-NADPH reductases) family 1"><magnitude>
<val> 72.633664</val>
</magnitude>
</node>
<node name="Flavodoxins"><magnitude>
<val> 353.697875</val>
</magnitude>
</node>
<node name="Formate hydrogenlyase subunit 3/Multisubunit Na+/H+ antiporter, MnhD subunit"><magnitude>
<val> 400.75671</val>
</magnitude>
</node>
<node name="Formate hydrogenlyase subunit 4"><magnitude>
<val> 251.727942</val>
</magnitude>
</node>
<node name="Formate hydrogenlyase subunit 6/NADH:ubiquinone oxidoreductase 23 kD subunit (chain I)"><magnitude>
<val> 671.384943</val>
</magnitude>
</node>
<node name="Formylmethanofuran dehydrogenase subunit A"><magnitude>
<val> 136.061612</val>
</magnitude>
</node>
<node name="Formylmethanofuran dehydrogenase subunit B"><magnitude>
<val> 181.509421</val>
</magnitude>
</node>
<node name="Formylmethanofuran dehydrogenase subunit C"><magnitude>
<val> 168.115116</val>
</magnitude>
</node>
<node name="Formylmethanofuran dehydrogenase subunit D"><magnitude>
<val> 118.928995</val>
</magnitude>
</node>
<node name="Formylmethanofuran dehydrogenase subunit E"><magnitude>
<val> 371.815804</val>
</magnitude>
</node>
<node name="Formylmethanofuran:tetrahydromethanopterin formyltransferase"><magnitude>
<val> 188.214536</val>
</magnitude>
</node>
<node name="Fumarase"><magnitude>
<val> 116.434901</val>
</magnitude>
</node>
<node name="Fumarate reductase subunit C"><magnitude>
<val> 27.634389</val>
</magnitude>
</node>
<node name="Fumarate reductase subunit D"><magnitude>
<val> 7.062226</val>
</magnitude>
</node>
<node name="Galactose-1-phosphate uridylyltransferase"><magnitude>
<val> 149.262293</val>
</magnitude>
</node>
<node name="Glycerol dehydrogenase and related enzymes"><magnitude>
<val> 149.885928</val>
</magnitude>
</node>
<node name="Glycerol kinase"><magnitude>
<val> 121.739375</val>
</magnitude>
</node>
<node name="Glycerol-3-phosphate dehydrogenase"><magnitude>
<val> 459.703137</val>
</magnitude>
</node>
<node name="Glycerophosphoryl diester phosphodiesterase"><magnitude>
<val> 334.615874</val>
</magnitude>
</node>
<node name="H+/citrate symporter"><magnitude>
<val> 43.316223</val>
</magnitude>
</node>
<node name="H2-forming N5,N10-methylenetetrahydromethanopterin dehydrogenase"><magnitude>
<val> 66.78419</val>
</magnitude>
</node>
<node name="Heme/copper-type cytochrome/quinol oxidase, subunit 3"><magnitude>
<val> 84.061163</val>
</magnitude>
</node>
<node name="Heme/copper-type cytochrome/quinol oxidase, subunit 4"><magnitude>
<val> 4.805612</val>
</magnitude>
</node>
<node name="Heme/copper-type cytochrome/quinol oxidases, subunit 1"><magnitude>
<val> 98.266309</val>
</magnitude>
</node>
<node name="Heme/copper-type cytochrome/quinol oxidases, subunit 2"><magnitude>
<val> 169.749048</val>
</magnitude>
</node>
<node name="Hemoglobin-like flavoprotein"><magnitude>
<val> 20.386164</val>
</magnitude>
</node>
<node name="Heterodisulfide reductase, subunit A and related polyferredoxins"><magnitude>
<val> 929.458945</val>
</magnitude>
</node>
<node name="Heterodisulfide reductase, subunit B"><magnitude>
<val> 316.974015</val>
</magnitude>
</node>
<node name="Heterodisulfide reductase, subunit C"><magnitude>
<val> 313.179985</val>
</magnitude>
</node>
<node name="Hydrogenase 4 membrane component (E)"><magnitude>
<val> 94.543545</val>
</magnitude>
</node>
<node name="Indolepyruvate ferredoxin oxidoreductase, alpha and beta subunits"><magnitude>
<val> 627.069238</val>
</magnitude>
</node>
<node name="Inorganic pyrophosphatase"><magnitude>
<val> 339.631311</val>
</magnitude>
</node>
<node name="Inorganic pyrophosphatase/exopolyphosphatase"><magnitude>
<val> 183.162513</val>
</magnitude>
</node>
<node name="Isocitrate dehydrogenases"><magnitude>
<val> 150.911142</val>
</magnitude>
</node>
<node name="Isocitrate lyase"><magnitude>
<val> 66.761668</val>
</magnitude>
</node>
<node name="Isocitrate/isopropylmalate dehydrogenase"><magnitude>
<val> 427.541199</val>
</magnitude>
</node>
<node name="L-lactate dehydrogenase (FMN-dependent) and related alpha-hydroxy acid dehydrogenases"><magnitude>
<val> 187.059815</val>
</magnitude>
</node>
<node name="L-lactate permease"><magnitude>
<val> 311.586857</val>
</magnitude>
</node>
<node name="Lactate dehydrogenase and related dehydrogenases"><magnitude>
<val> 336.937013</val>
</magnitude>
</node>
<node name="Malate synthase"><magnitude>
<val> 54.180174</val>
</magnitude>
</node>
<node name="Malate/L-lactate dehydrogenases"><magnitude>
<val> 134.019087</val>
</magnitude>
</node>
<node name="Malate/lactate dehydrogenases"><magnitude>
<val> 244.628709</val>
</magnitude>
</node>
<node name="Malic enzyme"><magnitude>
<val> 405.278062</val>
</magnitude>
</node>
<node name="Membrane domain of membrane-anchored glycerophosphoryl diester phosphodiesterase"><magnitude>
<val> 41.901372</val>
</magnitude>
</node>
<node name="MinD superfamily P-loop ATPase containing an inserted ferredoxin domain"><magnitude>
<val> 875.225669</val>
</magnitude>
</node>
<node name="Monomeric isocitrate dehydrogenase"><magnitude>
<val> 203.787806</val>
</magnitude>
</node>
<node name="NAD(P)H-nitrite reductase"><magnitude>
<val> 464.421358</val>
</magnitude>
</node>
<node name="NAD-dependent aldehyde dehydrogenases"><magnitude>
<val> 830.164007</val>
</magnitude>
</node>
<node name="NAD/NADP transhydrogenase alpha subunit"><magnitude>
<val> 26.374851</val>
</magnitude>
</node>
<node name="NAD/NADP transhydrogenase beta subunit"><magnitude>
<val> 93.174281</val>
</magnitude>
</node>
<node name="NADH dehydrogenase, FAD-containing subunit"><magnitude>
<val> 312.114275</val>
</magnitude>
</node>
<node name="NADH dehydrogenase/NADH:ubiquinone oxidoreductase 75 kD subunit (chain G)"><magnitude>
<val> 333.745559</val>
</magnitude>
</node>
<node name="NADH:flavin oxidoreductases, Old Yellow Enzyme family"><magnitude>
<val> 583.835312</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase 17.2 kD subunit"><magnitude>
<val> 2.877458</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase 20 kD subunit and related Fe-S oxidoreductases"><magnitude>
<val> 147.814065</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase 24 kD subunit"><magnitude>
<val> 474.682597</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase 27 kD subunit"><magnitude>
<val> 125.046397</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase 49 kD subunit 7"><magnitude>
<val> 227.146941</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase subunit 1 (chain H)"><magnitude>
<val> 230.389874</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase subunit 11 or 4L (chain K)"><magnitude>
<val> 213.899611</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase subunit 2 (chain N)"><magnitude>
<val> 242.850298</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase subunit 3 (chain A)"><magnitude>
<val> 239.230016</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase subunit 4 (chain M)"><magnitude>
<val> 289.690335</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase subunit 5 (chain L)/Multisubunit Na+/H+ antiporter, MnhA subunit"><magnitude>
<val> 442.260208</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase subunit 6 (chain J)"><magnitude>
<val> 208.536379</val>
</magnitude>
</node>
<node name="NADH:ubiquinone oxidoreductase, NADH-binding (51 kD) subunit"><magnitude>
<val> 570.743733</val>
</magnitude>
</node>
<node name="NADPH:quinone reductase and related Zn-dependent oxidoreductases"><magnitude>
<val> 383.567337</val>
</magnitude>
</node>
<node name="Na+-transporting NADH:ubiquinone oxidoreductase, subunit NqrA"><magnitude>
<val> 65.766142</val>
</magnitude>
</node>
<node name="Na+-transporting NADH:ubiquinone oxidoreductase, subunit NqrB"><magnitude>
<val> 85.133136</val>
</magnitude>
</node>
<node name="Na+-transporting NADH:ubiquinone oxidoreductase, subunit NqrC"><magnitude>
<val> 37.552385</val>
</magnitude>
</node>
<node name="Na+-transporting NADH:ubiquinone oxidoreductase, subunit NqrD"><magnitude>
<val> 38.050996</val>
</magnitude>
</node>
<node name="Na+-transporting NADH:ubiquinone oxidoreductase, subunit NqrE"><magnitude>
<val> 40.02217</val>
</magnitude>
</node>
<node name="Na+-transporting NADH:ubiquinone oxidoreductase, subunit NqrF"><magnitude>
<val> 183.549182</val>
</magnitude>
</node>
<node name="Na+-transporting methylmalonyl-CoA/oxaloacetate decarboxylase, beta subunit"><magnitude>
<val> 124.031351</val>
</magnitude>
</node>
<node name="Na+-transporting methylmalonyl-CoA/oxaloacetate decarboxylase, gamma subunit"><magnitude>
<val> 65.875162</val>
</magnitude>
</node>
<node name="Na+/H+ antiporter"><magnitude>
<val> 150.800611</val>
</magnitude>
</node>
<node name="Na+/H+-dicarboxylate symporters"><magnitude>
<val> 241.819802</val>
</magnitude>
</node>
<node name="Na+/citrate symporter"><magnitude>
<val> 46.850762</val>
</magnitude>
</node>
<node name="Ni,Fe-hydrogenase I cytochrome b subunit"><magnitude>
<val> 43.818868</val>
</magnitude>
</node>
<node name="Ni,Fe-hydrogenase I large subunit"><magnitude>
<val> 197.372421</val>
</magnitude>
</node>
<node name="Ni,Fe-hydrogenase I small subunit"><magnitude>
<val> 221.035337</val>
</magnitude>
</node>
<node name="Ni,Fe-hydrogenase III component G"><magnitude>
<val> 64.574576</val>
</magnitude>
</node>
<node name="Ni,Fe-hydrogenase III large subunit"><magnitude>
<val> 260.188153</val>
</magnitude>
</node>
<node name="Ni,Fe-hydrogenase III small subunit"><magnitude>
<val> 220.789372</val>
</magnitude>
</node>
<node name="Ni,Fe-hydrogenase maturation factor"><magnitude>
<val> 381.64353</val>
</magnitude>
</node>
<node name="NifU homolog involved in Fe-S cluster formation"><magnitude>
<val> 377.747424</val>
</magnitude>
</node>
<node name="Nitrate reductase alpha subunit"><magnitude>
<val> 153.564777</val>
</magnitude>
</node>
<node name="Nitrate reductase beta subunit"><magnitude>
<val> 59.745783</val>
</magnitude>
</node>
<node name="Nitrate reductase cytochrome c-type subunit"><magnitude>
<val> 37.118219</val>
</magnitude>
</node>
<node name="Nitrate reductase delta subunit"><magnitude>
<val> 9.525939</val>
</magnitude>
</node>
<node name="Nitrate reductase gamma subunit"><magnitude>
<val> 69.003103</val>
</magnitude>
</node>
<node name="Nitrate/TMAO reductases, membrane-bound tetraheme cytochrome c subunit"><magnitude>
<val> 307.809357</val>
</magnitude>
</node>
<node name="Nitrogenase molybdenum-iron protein, alpha and beta chains"><magnitude>
<val> 1200.587467</val>
</magnitude>
</node>
<node name="Nitroreductase"><magnitude>
<val> 1494.245679</val>
</magnitude>
</node>
<node name="Nitrous oxide reductase"><magnitude>
<val> 56.901754</val>
</magnitude>
</node>
<node name="Phosphoenolpyruvate carboxykinase (ATP)"><magnitude>
<val> 121.749491</val>
</magnitude>
</node>
<node name="Phosphoenolpyruvate carboxykinase (GTP)"><magnitude>
<val> 54.589499</val>
</magnitude>
</node>
<node name="Phosphoenolpyruvate carboxylase"><magnitude>
<val> 134.903768</val>
</magnitude>
</node>
<node name="Phosphoribulokinase"><magnitude>
<val> 6.82185</val>
</magnitude>
</node>
<node name="Phosphotransacetylase"><magnitude>
<val> 253.25031</val>
</magnitude>
</node>
<node name="Plastocyanin"><magnitude>
<val> 29.965972</val>
</magnitude>
</node>
<node name="Polyferredoxin"><magnitude>
<val> 424.293204</val>
</magnitude>
</node>
<node name="Polysulphide reductase"><magnitude>
<val> 161.702094</val>
</magnitude>
</node>
<node name="Predicted NADH:ubiquinone oxidoreductase, subunit RnfA"><magnitude>
<val> 71.96665</val>
</magnitude>
</node>
<node name="Predicted NADH:ubiquinone oxidoreductase, subunit RnfB"><magnitude>
<val> 68.4283</val>
</magnitude>
</node>
<node name="Predicted NADH:ubiquinone oxidoreductase, subunit RnfC"><magnitude>
<val> 264.026964</val>
</magnitude>
</node>
<node name="Predicted NADH:ubiquinone oxidoreductase, subunit RnfD"><magnitude>
<val> 154.13088</val>
</magnitude>
</node>
<node name="Predicted NADH:ubiquinone oxidoreductase, subunit RnfE"><magnitude>
<val> 88.930235</val>
</magnitude>
</node>
<node name="Predicted NADH:ubiquinone oxidoreductase, subunit RnfG"><magnitude>
<val> 86.751737</val>
</magnitude>
</node>
<node name="Predicted acetamidase/formamidase"><magnitude>
<val> 75.690645</val>
</magnitude>
</node>
<node name="Predicted acyl-CoA transferases/carnitine dehydratase"><magnitude>
<val> 243.453171</val>
</magnitude>
</node>
<node name="Predicted butyrate kinase"><magnitude>
<val> 24.381532</val>
</magnitude>
</node>
<node name="Predicted enolase-phosphatase"><magnitude>
<val> 61.780525</val>
</magnitude>
</node>
<node name="Predicted lipoprotein involved in nitrous oxide reduction"><magnitude>
<val> 180.386255</val>
</magnitude>
</node>
<node name="Predicted oxidoreductases (related to aryl-alcohol dehydrogenases)"><magnitude>
<val> 418.246887</val>
</magnitude>
</node>
<node name="Predicted thiol oxidoreductase"><magnitude>
<val> 69.103742</val>
</magnitude>
</node>
<node name="Protein involved in meta-pathway of phenol degradation"><magnitude>
<val> 84.072416</val>
</magnitude>
</node>
<node name="Pyruvate carboxylase"><magnitude>
<val> 416.933784</val>
</magnitude>
</node>
<node name="Pyruvate dehydrogenase complex, dehydrogenase (E1) component"><magnitude>
<val> 195.355038</val>
</magnitude>
</node>
<node name="Pyruvate-formate lyase"><magnitude>
<val> 330.351753</val>
</magnitude>
</node>
<node name="Pyruvate/2-oxoglutarate dehydrogenase complex, dehydrogenase (E1) component, eukaryotic type, alpha subunit"><magnitude>
<val> 231.553988</val>
</magnitude>
</node>
<node name="Pyruvate/2-oxoglutarate dehydrogenase complex, dehydrogenase (E1) component, eukaryotic type, beta subunit"><magnitude>
<val> 199.556424</val>
</magnitude>
</node>
<node name="Pyruvate/2-oxoglutarate dehydrogenase complex, dihydrolipoamide acyltransferase (E2) component, and related enzymes"><magnitude>
<val> 372.778584</val>
</magnitude>
</node>
<node name="Pyruvate/2-oxoglutarate dehydrogenase complex, dihydrolipoamide dehydrogenase (E3) component, and related enzymes"><magnitude>
<val> 611.952355</val>
</magnitude>
</node>
<node name="Pyruvate/oxaloacetate carboxyltransferase"><magnitude>
<val> 275.403195</val>
</magnitude>
</node>
<node name="Pyruvate:ferredoxin oxidoreductase and related 2-oxoacid:ferredoxin oxidoreductases, alpha subunit"><magnitude>
<val> 960.636083</val>
</magnitude>
</node>
<node name="Pyruvate:ferredoxin oxidoreductase and related 2-oxoacid:ferredoxin oxidoreductases, beta subunit"><magnitude>
<val> 754.620082</val>
</magnitude>
</node>
<node name="Pyruvate:ferredoxin oxidoreductase and related 2-oxoacid:ferredoxin oxidoreductases, delta subunit"><magnitude>
<val> 105.506881</val>
</magnitude>
</node>
<node name="Pyruvate:ferredoxin oxidoreductase and related 2-oxoacid:ferredoxin oxidoreductases, gamma subunit"><magnitude>
<val> 702.130453</val>
</magnitude>
</node>
<node name="Ribulose bisphosphate carboxylase small subunit"><magnitude>
<val> 24.141628</val>
</magnitude>
</node>
<node name="Ribulose kinase"><magnitude>
<val> 17.881358</val>
</magnitude>
</node>
<node name="Rieske Fe-S protein"><magnitude>
<val> 158.417303</val>
</magnitude>
</node>
<node name="Rubredoxin"><magnitude>
<val> 343.979643</val>
</magnitude>
</node>
<node name="Rubrerythrin"><magnitude>
<val> 547.127246</val>
</magnitude>
</node>
<node name="Succinate dehydrogenase, hydrophobic anchor subunit"><magnitude>
<val> 50.286469</val>
</magnitude>
</node>
<node name="Succinate dehydrogenase/fumarate reductase, Fe-S protein subunit"><magnitude>
<val> 164.012658</val>
</magnitude>
</node>
<node name="Succinate dehydrogenase/fumarate reductase, cytochrome b subunit"><magnitude>
<val> 51.141066</val>
</magnitude>
</node>
<node name="Succinate dehydrogenase/fumarate reductase, flavoprotein subunit"><magnitude>
<val> 473.752038</val>
</magnitude>
</node>
<node name="Succinyl-CoA synthetase, alpha subunit"><magnitude>
<val> 207.166813</val>
</magnitude>
</node>
<node name="Succinyl-CoA synthetase, beta subunit"><magnitude>
<val> 274.921538</val>
</magnitude>
</node>
<node name="Tartrate dehydratase alpha subunit/Fumarate hydratase class I, N-terminal domain"><magnitude>
<val> 261.104291</val>
</magnitude>
</node>
<node name="Tartrate dehydratase beta subunit/Fumarate hydratase class I, C-terminal domain"><magnitude>
<val> 166.236535</val>
</magnitude>
</node>
<node name="Thiosulfate reductase cytochrome B subunit (membrane anchoring protein)"><magnitude>
<val> 71.75412</val>
</magnitude>
</node>
<node name="Uncharacterized Fe-S oxidoreductase"><magnitude>
<val> 268.713274</val>
</magnitude>
</node>
<node name="Uncharacterized Fe-S protein"><magnitude>
<val> 346.587035</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein containing a ferredoxin-like domain"><magnitude>
<val> 366.963341</val>
</magnitude>
</node>
<node name="Uncharacterized flavoproteins"><magnitude>
<val> 490.36271</val>
</magnitude>
</node>
<node name="Uncharacterized oxidoreductases, Fe-dependent alcohol dehydrogenase family"><magnitude>
<val> 263.039155</val>
</magnitude>
</node>
<node name="Uncharacterized protein required for formate dehydrogenase activity"><magnitude>
<val> 236.300754</val>
</magnitude>
</node>
<node name="Vacuolar H+-ATPase V1 sector, subunit C"><magnitude>
<val> 17.172233</val>
</magnitude>
</node>
<node name="Vacuolar H+-ATPase V1 sector, subunit H"><magnitude>
<val> 26.477421</val>
</magnitude>
</node>
<node name="Zn-dependent alcohol dehydrogenases, class III"><magnitude>
<val> 93.860964</val>
</magnitude>
</node>
</node>
<node name="Inorganic ion transport and metabolism"><magnitude>
<val> 43634.079351</val>
</magnitude>
<node name="3'-Phosphoadenosine 5'-phosphosulfate (PAPS) 3'-phosphatase"><magnitude>
<val> 82.364936</val>
</magnitude>
</node>
<node name="ABC-type Co2+ transport system, periplasmic component"><magnitude>
<val> 211.622554</val>
</magnitude>
</node>
<node name="ABC-type Co2+ transport system, permease component"><magnitude>
<val> 439.859181</val>
</magnitude>
</node>
<node name="ABC-type Fe3+ transport system, periplasmic component"><magnitude>
<val> 140.577873</val>
</magnitude>
</node>
<node name="ABC-type Fe3+ transport system, permease component"><magnitude>
<val> 244.086774</val>
</magnitude>
</node>
<node name="ABC-type Fe3+-citrate transport system, periplasmic component"><magnitude>
<val> 65.221131</val>
</magnitude>
</node>
<node name="ABC-type Fe3+-hydroxamate transport system, periplasmic component"><magnitude>
<val> 1299.920846</val>
</magnitude>
</node>
<node name="ABC-type Fe3+-siderophore transport system, permease component"><magnitude>
<val> 722.775397</val>
</magnitude>
</node>
<node name="ABC-type Mn/Zn transport systems, ATPase component"><magnitude>
<val> 250.523604</val>
</magnitude>
</node>
<node name="ABC-type Mn2+/Zn2+ transport systems, permease components"><magnitude>
<val> 328.100337</val>
</magnitude>
</node>
<node name="ABC-type Zn2+ transport system, periplasmic component/surface adhesin"><magnitude>
<val> 55.711251</val>
</magnitude>
</node>
<node name="ABC-type anion transport system, duplicated permease component"><magnitude>
<val> 78.106212</val>
</magnitude>
</node>
<node name="ABC-type cobalamin/Fe3+-siderophores transport systems, ATPase components"><magnitude>
<val> 635.309889</val>
</magnitude>
</node>
<node name="ABC-type cobalt transport system, ATPase component"><magnitude>
<val> 780.095125</val>
</magnitude>
</node>
<node name="ABC-type cobalt transport system, periplasmic component"><magnitude>
<val> 134.062559</val>
</magnitude>
</node>
<node name="ABC-type cobalt transport system, permease component CbiQ and related transporters"><magnitude>
<val> 493.174691</val>
</magnitude>
</node>
<node name="ABC-type enterobactin transport system, permease component"><magnitude>
<val> 10.85307</val>
</magnitude>
</node>
<node name="ABC-type enterochelin transport system, ATPase component"><magnitude>
<val> 5.373247</val>
</magnitude>
</node>
<node name="ABC-type enterochelin transport system, periplasmic component"><magnitude>
<val> 16.680138</val>
</magnitude>
</node>
<node name="ABC-type enterochelin transport system, permease component"><magnitude>
<val> 56.562502</val>
</magnitude>
</node>
<node name="ABC-type hemin transport system, ATPase component"><magnitude>
<val> 49.640031</val>
</magnitude>
</node>
<node name="ABC-type hemin transport system, periplasmic component"><magnitude>
<val> 55.618772</val>
</magnitude>
</node>
<node name="ABC-type metal ion transport system, ATPase component"><magnitude>
<val> 128.234755</val>
</magnitude>
</node>
<node name="ABC-type metal ion transport system, periplasmic component/surface adhesin"><magnitude>
<val> 427.441285</val>
</magnitude>
</node>
<node name="ABC-type metal ion transport system, periplasmic component/surface antigen"><magnitude>
<val> 112.83444</val>
</magnitude>
</node>
<node name="ABC-type metal ion transport system, permease component"><magnitude>
<val> 93.544377</val>
</magnitude>
</node>
<node name="ABC-type molybdate transport system, ATPase component"><magnitude>
<val> 160.461602</val>
</magnitude>
</node>
<node name="ABC-type molybdate transport system, periplasmic component"><magnitude>
<val> 338.978582</val>
</magnitude>
</node>
<node name="ABC-type molybdate transport system, permease component"><magnitude>
<val> 215.047723</val>
</magnitude>
</node>
<node name="ABC-type molybdenum transport system, ATPase component/photorepair protein PhrA"><magnitude>
<val> 105.542202</val>
</magnitude>
</node>
<node name="ABC-type nitrate/sulfonate/bicarbonate transport system, ATPase component"><magnitude>
<val> 442.973306</val>
</magnitude>
</node>
<node name="ABC-type nitrate/sulfonate/bicarbonate transport system, permease component"><magnitude>
<val> 438.19968</val>
</magnitude>
</node>
<node name="ABC-type nitrate/sulfonate/bicarbonate transport systems, periplasmic components"><magnitude>
<val> 617.646568</val>
</magnitude>
</node>
<node name="ABC-type phosphate transport system, ATPase component"><magnitude>
<val> 352.070438</val>
</magnitude>
</node>
<node name="ABC-type phosphate transport system, auxiliary component"><magnitude>
<val> 26.166585</val>
</magnitude>
</node>
<node name="ABC-type phosphate transport system, periplasmic component"><magnitude>
<val> 577.416938</val>
</magnitude>
</node>
<node name="ABC-type phosphate transport system, permease component"><magnitude>
<val> 612.017218</val>
</magnitude>
</node>
<node name="ABC-type phosphate/phosphonate transport system, ATPase component"><magnitude>
<val> 87.748055</val>
</magnitude>
</node>
<node name="ABC-type phosphate/phosphonate transport system, periplasmic component"><magnitude>
<val> 131.061954</val>
</magnitude>
</node>
<node name="ABC-type phosphate/phosphonate transport system, permease component"><magnitude>
<val> 52.05179</val>
</magnitude>
</node>
<node name="ABC-type phosphonate transport system, ATPase component"><magnitude>
<val> 39.46905</val>
</magnitude>
</node>
<node name="ABC-type sulfate transport system, periplasmic component"><magnitude>
<val> 115.154874</val>
</magnitude>
</node>
<node name="ABC-type sulfate transport system, permease component"><magnitude>
<val> 116.85614</val>
</magnitude>
</node>
<node name="ABC-type sulfate/molybdate transport systems, ATPase component"><magnitude>
<val> 136.422985</val>
</magnitude>
</node>
<node name="ABC-type taurine transport system, ATPase component"><magnitude>
<val> 7.941933</val>
</magnitude>
</node>
<node name="ABC-type taurine transport system, periplasmic component"><magnitude>
<val> 45.818613</val>
</magnitude>
</node>
<node name="ATP sulfurylase (sulfate adenylyltransferase)"><magnitude>
<val> 153.034365</val>
</magnitude>
</node>
<node name="Adenylylsulfate kinase and related kinases"><magnitude>
<val> 93.572173</val>
</magnitude>
</node>
<node name="Alkaline phosphatase"><magnitude>
<val> 126.271427</val>
</magnitude>
</node>
<node name="Ammonia permease"><magnitude>
<val> 447.10308</val>
</magnitude>
</node>
<node name="Arsenate reductase and related proteins, glutaredoxin family"><magnitude>
<val> 61.000957</val>
</magnitude>
</node>
<node name="Arsenite efflux pump ACR3 and related permeases"><magnitude>
<val> 253.229404</val>
</magnitude>
</node>
<node name="Arylsulfatase A and related enzymes"><magnitude>
<val> 841.41646</val>
</magnitude>
</node>
<node name="Bacterioferritin (cytochrome b1)"><magnitude>
<val> 112.596221</val>
</magnitude>
</node>
<node name="Ca2+/H+ antiporter"><magnitude>
<val> 44.99878</val>
</magnitude>
</node>
<node name="Ca2+/Na+ antiporter"><magnitude>
<val> 303.374118</val>
</magnitude>
</node>
<node name="Carbonic anhydrase"><magnitude>
<val> 369.500789</val>
</magnitude>
</node>
<node name="Catalase"><magnitude>
<val> 141.566757</val>
</magnitude>
</node>
<node name="Catalase (peroxidase I)"><magnitude>
<val> 161.526101</val>
</magnitude>
</node>
<node name="Cation transport ATPase"><magnitude>
<val> 2720.213241</val>
</magnitude>
</node>
<node name="Chloride channel protein EriC"><magnitude>
<val> 227.404734</val>
</magnitude>
</node>
<node name="Chromate transport protein ChrA"><magnitude>
<val> 132.109521</val>
</magnitude>
</node>
<node name="Co/Zn/Cd efflux system component"><magnitude>
<val> 245.03549</val>
</magnitude>
</node>
<node name="Copper chaperone"><magnitude>
<val> 251.828673</val>
</magnitude>
</node>
<node name="Cu/Zn superoxide dismutase"><magnitude>
<val> 21.82707</val>
</magnitude>
</node>
<node name="Cyanate permease"><magnitude>
<val> 141.098925</val>
</magnitude>
</node>
<node name="Cystathionine beta-lyase family protein involved in aluminum resistance"><magnitude>
<val> 47.157068</val>
</magnitude>
</node>
<node name="Cytochrome c peroxidase"><magnitude>
<val> 358.981299</val>
</magnitude>
</node>
<node name="DNA-binding ferritin-like protein (oxidative damage protectant)"><magnitude>
<val> 34.186888</val>
</magnitude>
</node>
<node name="Di- and tricarboxylate transporters"><magnitude>
<val> 374.370023</val>
</magnitude>
</node>
<node name="Dissimilatory sulfite reductase (desulfoviridin), gamma subunit"><magnitude>
<val> 74.472535</val>
</magnitude>
</node>
<node name="Enterochelin esterase and related enzymes"><magnitude>
<val> 121.813719</val>
</magnitude>
</node>
<node name="Fe2+ transport system protein A"><magnitude>
<val> 349.257775</val>
</magnitude>
</node>
<node name="Fe2+ transport system protein B"><magnitude>
<val> 809.827603</val>
</magnitude>
</node>
<node name="Fe2+-dicitrate sensor, membrane component"><magnitude>
<val> 562.883969</val>
</magnitude>
</node>
<node name="Fe2+/Zn2+ uptake regulation proteins"><magnitude>
<val> 528.448121</val>
</magnitude>
</node>
<node name="Ferredoxin subunits of nitrite reductase and ring-hydroxylating dioxygenases"><magnitude>
<val> 108.962162</val>
</magnitude>
</node>
<node name="Ferritin-like protein"><magnitude>
<val> 171.07356</val>
</magnitude>
</node>
<node name="Formate-dependent nitrite reductase, membrane component"><magnitude>
<val> 19.996076</val>
</magnitude>
</node>
<node name="Formate-dependent nitrite reductase, periplasmic cytochrome c552 subunit"><magnitude>
<val> 702.185813</val>
</magnitude>
</node>
<node name="Formate/nitrite family of transporters"><magnitude>
<val> 253.923374</val>
</magnitude>
</node>
<node name="GTPases - Sulfate adenylate transferase subunit 1"><magnitude>
<val> 298.094597</val>
</magnitude>
</node>
<node name="Heme oxygenase"><magnitude>
<val> 10.421168</val>
</magnitude>
</node>
<node name="Hemerythrin"><magnitude>
<val> 340.625644</val>
</magnitude>
</node>
<node name="Hemin uptake protein"><magnitude>
<val> 32.311994</val>
</magnitude>
</node>
<node name="High-affinity Fe2+/Pb2+ permease"><magnitude>
<val> 21.651259</val>
</magnitude>
</node>
<node name="High-affinity K+ transport system, ATPase chain B"><magnitude>
<val> 418.811325</val>
</magnitude>
</node>
<node name="High-affinity nickel permease"><magnitude>
<val> 58.141758</val>
</magnitude>
</node>
<node name="K+ transport systems, NAD-binding component"><magnitude>
<val> 435.967331</val>
</magnitude>
</node>
<node name="K+ transporter"><magnitude>
<val> 312.971752</val>
</magnitude>
</node>
<node name="K+-transporting ATPase, A chain"><magnitude>
<val> 198.122689</val>
</magnitude>
</node>
<node name="K+-transporting ATPase, c chain"><magnitude>
<val> 120.982698</val>
</magnitude>
</node>
<node name="Kef-type K+ transport system, predicted NAD-binding component"><magnitude>
<val> 185.406431</val>
</magnitude>
</node>
<node name="Kef-type K+ transport systems, membrane components"><magnitude>
<val> 339.587895</val>
</magnitude>
</node>
<node name="Kef-type K+ transport systems, predicted NAD-binding component"><magnitude>
<val> 112.814956</val>
</magnitude>
</node>
<node name="Membrane protein TerC, possibly involved in tellurium resistance"><magnitude>
<val> 260.178723</val>
</magnitude>
</node>
<node name="Membrane transporters of cations and cationic drugs"><magnitude>
<val> 243.379749</val>
</magnitude>
</node>
<node name="Metal-dependent hydrolase involved in phosphonate metabolism"><magnitude>
<val> 70.910906</val>
</magnitude>
</node>
<node name="Mg/Co/Ni transporter MgtE (contains CBS domain)"><magnitude>
<val> 318.922518</val>
</magnitude>
</node>
<node name="Mg2+ and Co2+ transporters"><magnitude>
<val> 408.260083</val>
</magnitude>
</node>
<node name="Mn-containing catalase"><magnitude>
<val> 42.199676</val>
</magnitude>
</node>
<node name="Mn2+ and Fe2+ transporters of the NRAMP family"><magnitude>
<val> 291.897905</val>
</magnitude>
</node>
<node name="Multisubunit Na+/H+ antiporter, MnhB subunit"><magnitude>
<val> 213.421022</val>
</magnitude>
</node>
<node name="Multisubunit Na+/H+ antiporter, MnhC subunit"><magnitude>
<val> 178.400339</val>
</magnitude>
</node>
<node name="Multisubunit Na+/H+ antiporter, MnhE subunit"><magnitude>
<val> 200.255739</val>
</magnitude>
</node>
<node name="Multisubunit Na+/H+ antiporter, MnhF subunit"><magnitude>
<val> 169.67594</val>
</magnitude>
</node>
<node name="Multisubunit Na+/H+ antiporter, MnhG subunit"><magnitude>
<val> 141.410366</val>
</magnitude>
</node>
<node name="Na+/H+ antiporter"><magnitude>
<val> 286.374075</val>
</magnitude>
</node>
<node name="Na+/H+ antiporter NhaD and related arsenite permeases"><magnitude>
<val> 268.122079</val>
</magnitude>
</node>
<node name="Na+/phosphate symporter"><magnitude>
<val> 361.96796</val>
</magnitude>
</node>
<node name="NhaP-type Na+/H+ and K+/H+ antiporters"><magnitude>
<val> 180.536432</val>
</magnitude>
</node>
<node name="NhaP-type Na+/H+ and K+/H+ antiporters with a unique C-terminal domain"><magnitude>
<val> 199.140871</val>
</magnitude>
</node>
<node name="Nitrate/nitrite transporter"><magnitude>
<val> 356.084371</val>
</magnitude>
</node>
<node name="Nitric oxide reductase activation protein"><magnitude>
<val> 189.200742</val>
</magnitude>
</node>
<node name="Nitric oxide reductase large subunit"><magnitude>
<val> 157.190878</val>
</magnitude>
</node>
<node name="Nitric oxide synthase, oxygenase domain"><magnitude>
<val> 50.929715</val>
</magnitude>
</node>
<node name="Nitrogenase subunit NifH (ATPase)"><magnitude>
<val> 347.024598</val>
</magnitude>
</node>
<node name="Nitrous oxidase accessory protein"><magnitude>
<val> 520.120701</val>
</magnitude>
</node>
<node name="Outer membrane receptor for Fe3+-dicitrate"><magnitude>
<val> 133.472269</val>
</magnitude>
</node>
<node name="Outer membrane receptor for ferric coprogen and ferric-rhodotorulic acid"><magnitude>
<val> 218.58229</val>
</magnitude>
</node>
<node name="Outer membrane receptor for ferrienterochelin and colicins"><magnitude>
<val> 826.92645</val>
</magnitude>
</node>
<node name="Outer membrane receptor for monomeric catechols"><magnitude>
<val> 128.285617</val>
</magnitude>
</node>
<node name="Outer membrane receptor proteins, mostly Fe transport"><magnitude>
<val> 3651.834896</val>
</magnitude>
</node>
<node name="Oxyanion-translocating ATPase"><magnitude>
<val> 164.208475</val>
</magnitude>
</node>
<node name="Periplasmic glucans biosynthesis protein"><magnitude>
<val> 73.970715</val>
</magnitude>
</node>
<node name="Periplasmic molybdate-binding protein/domain"><magnitude>
<val> 166.04081</val>
</magnitude>
</node>
<node name="Permease, similar to cation transporters"><magnitude>
<val> 56.473775</val>
</magnitude>
</node>
<node name="Phenylpropionate dioxygenase and related ring-hydroxylating dioxygenases, large terminal subunit"><magnitude>
<val> 8.741923</val>
</magnitude>
</node>
<node name="Phosphate transport regulator (distant homolog of PhoU)"><magnitude>
<val> 231.420945</val>
</magnitude>
</node>
<node name="Phosphate uptake regulator"><magnitude>
<val> 461.774938</val>
</magnitude>
</node>
<node name="Phosphate-selective porin"><magnitude>
<val> 164.01697</val>
</magnitude>
</node>
<node name="Phosphate/sulphate permeases"><magnitude>
<val> 266.156649</val>
</magnitude>
</node>
<node name="Phosphodiesterase/alkaline phosphatase D"><magnitude>
<val> 166.126106</val>
</magnitude>
</node>
<node name="Polyphosphate kinase"><magnitude>
<val> 187.275229</val>
</magnitude>
</node>
<node name="Predicted Co/Zn/Cd cation transporters"><magnitude>
<val> 609.833509</val>
</magnitude>
</node>
<node name="Predicted divalent heavy-metal cations transporter"><magnitude>
<val> 259.654482</val>
</magnitude>
</node>
<node name="Predicted ferric reductase"><magnitude>
<val> 34.855982</val>
</magnitude>
</node>
<node name="Predicted flavoprotein involved in K+ transport"><magnitude>
<val> 90.774273</val>
</magnitude>
</node>
<node name="Predicted integral membrane protein linked to a cation pump"><magnitude>
<val> 16.947743</val>
</magnitude>
</node>
<node name="Predicted iron-dependent peroxidase"><magnitude>
<val> 101.99141</val>
</magnitude>
</node>
<node name="Predicted periplasmic lipoprotein involved in iron transport"><magnitude>
<val> 42.653735</val>
</magnitude>
</node>
<node name="Predicted permease, cadmium resistance protein"><magnitude>
<val> 55.261139</val>
</magnitude>
</node>
<node name="Predicted subunit of the Multisubunit Na+/H+ antiporter"><magnitude>
<val> 72.646008</val>
</magnitude>
</node>
<node name="Protein implicated in iron transport, frataxin homolog"><magnitude>
<val> 28.945776</val>
</magnitude>
</node>
<node name="Putative Mg2+ and Co2+ transporter CorB"><magnitude>
<val> 37.43305</val>
</magnitude>
</node>
<node name="Putative Mg2+ and Co2+ transporter CorC"><magnitude>
<val> 27.51284</val>
</magnitude>
</node>
<node name="Putative copper export protein"><magnitude>
<val> 72.146595</val>
</magnitude>
</node>
<node name="Putative heme degradation protein"><magnitude>
<val> 10.005087</val>
</magnitude>
</node>
<node name="Putative heme iron utilization protein"><magnitude>
<val> 26.770652</val>
</magnitude>
</node>
<node name="Putative regulatory, ligand-binding protein related to C-terminal domains of K+ channels"><magnitude>
<val> 115.506172</val>
</magnitude>
</node>
<node name="Putative silver efflux pump"><magnitude>
<val> 441.881249</val>
</magnitude>
</node>
<node name="Rhodanese-related sulfurtransferase"><magnitude>
<val> 1021.857321</val>
</magnitude>
</node>
<node name="SPX domain-containing protein involved in vacuolar polyphosphate accumulation"><magnitude>
<val> 49.035113</val>
</magnitude>
</node>
<node name="Siderophore-interacting protein"><magnitude>
<val> 8.990875</val>
</magnitude>
</node>
<node name="Sulfate permease and related transporters (MFS superfamily)"><magnitude>
<val> 342.223491</val>
</magnitude>
</node>
<node name="Sulfite reductase, alpha subunit (flavoprotein)"><magnitude>
<val> 151.791362</val>
</magnitude>
</node>
<node name="Sulfite reductase, beta subunit (hemoprotein)"><magnitude>
<val> 194.576991</val>
</magnitude>
</node>
<node name="Superoxide dismutase"><magnitude>
<val> 207.550694</val>
</magnitude>
</node>
<node name="Tellurite resistance protein"><magnitude>
<val> 17.499472</val>
</magnitude>
</node>
<node name="Tellurite resistance protein and related permeases"><magnitude>
<val> 27.471379</val>
</magnitude>
</node>
<node name="Trk-type K+ transport systems, membrane components"><magnitude>
<val> 319.187174</val>
</magnitude>
</node>
<node name="Uncharacterized Zn-ribbon-containing protein involved in phosphonate metabolism"><magnitude>
<val> 75.984312</val>
</magnitude>
</node>
<node name="Uncharacterized component of phosphonate metabolism"><magnitude>
<val> 78.52393</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein involved in intracellular sulfur reduction"><magnitude>
<val> 76.663707</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein involved in oxidation of intracellular sulfur"><magnitude>
<val> 25.804801</val>
</magnitude>
</node>
<node name="Uncharacterized copper-binding protein"><magnitude>
<val> 14.3892</val>
</magnitude>
</node>
<node name="Uncharacterized enzyme of phosphonate metabolism"><magnitude>
<val> 165.503744</val>
</magnitude>
</node>
<node name="Uncharacterized iron-regulated protein"><magnitude>
<val> 81.687235</val>
</magnitude>
</node>
<node name="Uncharacterized protein affecting Mg2+/Co2+ transport"><magnitude>
<val> 53.367929</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in cation transport"><magnitude>
<val> 16.608514</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in copper resistance"><magnitude>
<val> 142.109229</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in formation of periplasmic nitrate reductase"><magnitude>
<val> 12.460982</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in response to NO"><magnitude>
<val> 55.63389</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in tellurite resistance"><magnitude>
<val> 58.581623</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in the oxidation of intracellular sulfur"><magnitude>
<val> 77.142642</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in tolerance to divalent cations"><magnitude>
<val> 134.983592</val>
</magnitude>
</node>
<node name="Uncharacterized protein, possibly involved in nitrogen fixation"><magnitude>
<val> 5.074393</val>
</magnitude>
</node>
<node name="Uncharacterized protein/domain, possibly involved in tellurite resistance"><magnitude>
<val> 24.141628</val>
</magnitude>
</node>
</node>
<node name="Lipid transport and metabolism"><magnitude>
<val> 18089.560983</val>
</magnitude>
<node name="(acyl-carrier-protein) S-malonyltransferase"><magnitude>
<val> 318.507798</val>
</magnitude>
</node>
<node name="1-acyl-sn-glycerol-3-phosphate acyltransferase"><magnitude>
<val> 342.964913</val>
</magnitude>
</node>
<node name="1-deoxy-D-xylulose 5-phosphate reductoisomerase"><magnitude>
<val> 223.3867</val>
</magnitude>
</node>
<node name="2C-methyl-D-erythritol 2,4-cyclodiphosphate synthase"><magnitude>
<val> 136.487798</val>
</magnitude>
</node>
<node name="3-hydroxy-3-methylglutaryl CoA synthase"><magnitude>
<val> 159.140068</val>
</magnitude>
</node>
<node name="3-hydroxyacyl-CoA dehydrogenase"><magnitude>
<val> 260.10509</val>
</magnitude>
</node>
<node name="3-hydroxyisobutyrate dehydrogenase and related beta-hydroxyacid dehydrogenases"><magnitude>
<val> 230.640954</val>
</magnitude>
</node>
<node name="3-hydroxymyristoyl/3-hydroxydecanoyl-(acyl carrier protein) dehydratases"><magnitude>
<val> 280.33257</val>
</magnitude>
</node>
<node name="3-oxoacyl-(acyl-carrier-protein) synthase"><magnitude>
<val> 603.510647</val>
</magnitude>
</node>
<node name="3-oxoacyl-[acyl-carrier protein] reductase"><magnitude>
<val> 207.725201</val>
</magnitude>
</node>
<node name="3-oxoacyl-[acyl-carrier-protein] synthase III"><magnitude>
<val> 452.893851</val>
</magnitude>
</node>
<node name="3-phytase (myo-inositol-hexaphosphate 3-phosphohydrolase)"><magnitude>
<val> 12.677065</val>
</magnitude>
</node>
<node name="4-diphosphocytidyl-2-methyl-D-erithritol synthase"><magnitude>
<val> 174.038597</val>
</magnitude>
</node>
<node name="4-diphosphocytidyl-2C-methyl-D-erythritol 2-phosphate synthase"><magnitude>
<val> 181.290864</val>
</magnitude>
</node>
<node name="ABC-type long-chain fatty acid transport system, fused permease and ATPase components"><magnitude>
<val> 15.721756</val>
</magnitude>
</node>
<node name="Acetyl-CoA acetyltransferase"><magnitude>
<val> 406.66595</val>
</magnitude>
</node>
<node name="Acetyl-CoA carboxylase alpha subunit"><magnitude>
<val> 208.827145</val>
</magnitude>
</node>
<node name="Acetyl-CoA carboxylase beta subunit"><magnitude>
<val> 164.923914</val>
</magnitude>
</node>
<node name="Acetyl-CoA carboxylase, carboxyltransferase component (subunits alpha and beta)"><magnitude>
<val> 189.853326</val>
</magnitude>
</node>
<node name="Acetyl/propionyl-CoA carboxylase, alpha subunit"><magnitude>
<val> 390.709505</val>
</magnitude>
</node>
<node name="Activator of 2-hydroxyglutaryl-CoA dehydratase (HSP70-class ATPase domain)"><magnitude>
<val> 412.488191</val>
</magnitude>
</node>
<node name="Acyl CoA:acetate/3-ketoacid CoA transferase"><magnitude>
<val> 198.690785</val>
</magnitude>
</node>
<node name="Acyl CoA:acetate/3-ketoacid CoA transferase, alpha subunit"><magnitude>
<val> 126.160577</val>
</magnitude>
</node>
<node name="Acyl CoA:acetate/3-ketoacid CoA transferase, beta subunit"><magnitude>
<val> 140.292319</val>
</magnitude>
</node>
<node name="Acyl carrier protein"><magnitude>
<val> 541.612119</val>
</magnitude>
</node>
<node name="Acyl carrier protein phosphodiesterase"><magnitude>
<val> 34.924472</val>
</magnitude>
</node>
<node name="Acyl dehydratase"><magnitude>
<val> 120.399231</val>
</magnitude>
</node>
<node name="Acyl-ACP thioesterase"><magnitude>
<val> 38.260512</val>
</magnitude>
</node>
<node name="Acyl-CoA cholesterol acyltransferase"><magnitude>
<val> 36.741272</val>
</magnitude>
</node>
<node name="Acyl-CoA dehydrogenases"><magnitude>
<val> 578.164542</val>
</magnitude>
</node>
<node name="Acyl-CoA hydrolase"><magnitude>
<val> 148.756193</val>
</magnitude>
</node>
<node name="Acyl-CoA synthetases (AMP-forming)/AMP-acid ligases II"><magnitude>
<val> 954.579178</val>
</magnitude>
</node>
<node name="Acyl-CoA thioesterase"><magnitude>
<val> 65.772205</val>
</magnitude>
</node>
<node name="Acyl-CoA-binding protein"><magnitude>
<val> 51.581144</val>
</magnitude>
</node>
<node name="Acyl-coenzyme A synthetases/AMP-(fatty) acid ligases"><magnitude>
<val> 564.960988</val>
</magnitude>
</node>
<node name="Biotin carboxyl carrier protein"><magnitude>
<val> 162.351813</val>
</magnitude>
</node>
<node name="Biotin carboxylase"><magnitude>
<val> 154.409662</val>
</magnitude>
</node>
<node name="CDP-diacylglycerol pyrophosphatase"><magnitude>
<val> 5.73498</val>
</magnitude>
</node>
<node name="CDP-diglyceride synthetase"><magnitude>
<val> 214.456888</val>
</magnitude>
</node>
<node name="Carboxylesterase type B"><magnitude>
<val> 73.175383</val>
</magnitude>
</node>
<node name="Dehydrogenases with different specificities (related to short-chain alcohol dehydrogenases)"><magnitude>
<val> 1297.122711</val>
</magnitude>
</node>
<node name="Dolichol kinase"><magnitude>
<val> 158.862974</val>
</magnitude>
</node>
<node name="Enoyl reductase domain of yeast-type FAS1"><magnitude>
<val> 71.665369</val>
</magnitude>
</node>
<node name="Enoyl-CoA hydratase/carnithine racemase"><magnitude>
<val> 316.724273</val>
</magnitude>
</node>
<node name="Enoyl-[acyl-carrier-protein] reductase (NADH)"><magnitude>
<val> 132.028641</val>
</magnitude>
</node>
<node name="Enzyme involved in the deoxyxylulose pathway of isoprenoid biosynthesis"><magnitude>
<val> 162.461417</val>
</magnitude>
</node>
<node name="Esterase/lipase"><magnitude>
<val> 261.660547</val>
</magnitude>
</node>
<node name="Fatty acid desaturase"><magnitude>
<val> 24.426544</val>
</magnitude>
</node>
<node name="Fatty acid/phospholipid biosynthesis enzyme"><magnitude>
<val> 126.184082</val>
</magnitude>
</node>
<node name="Fatty-acid desaturase"><magnitude>
<val> 94.225566</val>
</magnitude>
</node>
<node name="Glycerol-3-phosphate O-acyltransferase"><magnitude>
<val> 125.576645</val>
</magnitude>
</node>
<node name="Hydroxymethylglutaryl-CoA reductase"><magnitude>
<val> 129.31766</val>
</magnitude>
</node>
<node name="Isopentenyldiphosphate isomerase"><magnitude>
<val> 77.505773</val>
</magnitude>
</node>
<node name="Long-chain acyl-CoA synthetases (AMP-forming)"><magnitude>
<val> 245.885126</val>
</magnitude>
</node>
<node name="Long-chain fatty acid transport protein"><magnitude>
<val> 319.325264</val>
</magnitude>
</node>
<node name="Lysophospholipase"><magnitude>
<val> 305.00327</val>
</magnitude>
</node>
<node name="Membrane-associated phospholipid phosphatase"><magnitude>
<val> 330.09504</val>
</magnitude>
</node>
<node name="Methylmalonyl-CoA mutase, C-terminal domain/subunit (cobalamin-binding)"><magnitude>
<val> 114.478204</val>
</magnitude>
</node>
<node name="Methylmalonyl-CoA mutase, N-terminal domain/subunit"><magnitude>
<val> 284.235165</val>
</magnitude>
</node>
<node name="Mevalonate kinase"><magnitude>
<val> 173.752102</val>
</magnitude>
</node>
<node name="Mevalonate pyrophosphate decarboxylase"><magnitude>
<val> 46.098871</val>
</magnitude>
</node>
<node name="Myo-inositol-1-phosphate synthase"><magnitude>
<val> 139.966068</val>
</magnitude>
</node>
<node name="N-myristoyl transferase"><magnitude>
<val> 28.683421</val>
</magnitude>
</node>
<node name="Oligoketide cyclase/lipid transport protein"><magnitude>
<val> 16.319029</val>
</magnitude>
</node>
<node name="Penicillin tolerance protein"><magnitude>
<val> 251.042848</val>
</magnitude>
</node>
<node name="Phosphatidylglycerophosphatase A and related proteins"><magnitude>
<val> 154.844662</val>
</magnitude>
</node>
<node name="Phosphatidylglycerophosphate synthase"><magnitude>
<val> 408.566602</val>
</magnitude>
</node>
<node name="Phosphatidylserine decarboxylase"><magnitude>
<val> 329.584609</val>
</magnitude>
</node>
<node name="Phosphatidylserine synthase"><magnitude>
<val> 192.121473</val>
</magnitude>
</node>
<node name="Phosphatidylserine/phosphatidylglycerophosphate/cardiolipin synthases and related enzymes"><magnitude>
<val> 507.07304</val>
</magnitude>
</node>
<node name="Phospholipase/lecithinase/hemolysin"><magnitude>
<val> 65.278057</val>
</magnitude>
</node>
<node name="Phospholipid N-methyltransferase"><magnitude>
<val> 40.572697</val>
</magnitude>
</node>
<node name="Phosphomevalonate kinase"><magnitude>
<val> 83.281711</val>
</magnitude>
</node>
<node name="Phosphopantetheinyl transferase (holo-ACP synthase)"><magnitude>
<val> 122.50877</val>
</magnitude>
</node>
<node name="Phytoene/squalene synthetase"><magnitude>
<val> 7.272321</val>
</magnitude>
</node>
<node name="Poly(3-hydroxyalkanoate) synthetase"><magnitude>
<val> 145.075635</val>
</magnitude>
</node>
<node name="Poly-beta-hydroxyalkanoate depolymerase"><magnitude>
<val> 82.897227</val>
</magnitude>
</node>
<node name="Predicted 3-hydroxylacyl-(acyl carrier protein) dehydratase"><magnitude>
<val> 29.425422</val>
</magnitude>
</node>
<node name="Predicted acyltransferases"><magnitude>
<val> 235.735958</val>
</magnitude>
</node>
<node name="Putative lipid carrier protein"><magnitude>
<val> 10.238222</val>
</magnitude>
</node>
<node name="Putative sterol carrier protein"><magnitude>
<val> 13.07967</val>
</magnitude>
</node>
<node name="S-adenosylmethionine:diacylglycerol 3-amino-3-carboxypropyl transferase"><magnitude>
<val> 71.856311</val>
</magnitude>
</node>
<node name="Short chain fatty acids transporter"><magnitude>
<val> 41.197742</val>
</magnitude>
</node>
<node name="Sphingosine kinase and enzymes related to eukaryotic diacylglycerol kinase"><magnitude>
<val> 144.985661</val>
</magnitude>
</node>
<node name="Squalene cyclase"><magnitude>
<val> 167.502473</val>
</magnitude>
</node>
<node name="Sterol desaturase"><magnitude>
<val> 38.983762</val>
</magnitude>
</node>
<node name="Undecaprenyl pyrophosphate synthase"><magnitude>
<val> 330.616103</val>
</magnitude>
</node>
<node name="sn-1,2-diacylglycerol ethanolamine- and cholinephosphotranferases"><magnitude>
<val> 50.300079</val>
</magnitude>
</node>
</node>
<node name="Nucleotide transport and metabolism"><magnitude>
<val> 17208.532829</val>
</magnitude>
<node name="5'-nucleotidase/2',3'-cyclic phosphodiesterase and related esterases"><magnitude>
<val> 264.343069</val>
</magnitude>
</node>
<node name="ADP-ribose pyrophosphatase"><magnitude>
<val> 422.278343</val>
</magnitude>
</node>
<node name="AICAR transformylase/IMP cyclohydrolase PurH (only IMP cyclohydrolase domain in Aful)"><magnitude>
<val> 255.26117</val>
</magnitude>
</node>
<node name="ATP adenylyltransferase (5',5'''-P-1,P-4-tetraphosphate phosphorylase II)"><magnitude>
<val> 104.540939</val>
</magnitude>
</node>
<node name="Adenine deaminase"><magnitude>
<val> 214.121574</val>
</magnitude>
</node>
<node name="Adenine/guanine phosphoribosyltransferases and related PRPP-binding proteins"><magnitude>
<val> 279.285616</val>
</magnitude>
</node>
<node name="Adenosine deaminase"><magnitude>
<val> 88.336852</val>
</magnitude>
</node>
<node name="Adenylate cyclase"><magnitude>
<val> 55.619723</val>
</magnitude>
</node>
<node name="Adenylate cyclase, class 2 (thermophilic)"><magnitude>
<val> 96.548321</val>
</magnitude>
</node>
<node name="Adenylate kinase and related kinases"><magnitude>
<val> 144.780657</val>
</magnitude>
</node>
<node name="Adenylosuccinate lyase"><magnitude>
<val> 335.836664</val>
</magnitude>
</node>
<node name="Adenylosuccinate synthase"><magnitude>
<val> 318.21649</val>
</magnitude>
</node>
<node name="Allantoicase"><magnitude>
<val> 26.543005</val>
</magnitude>
</node>
<node name="Archaeal IMP cyclohydrolase"><magnitude>
<val> 102.982235</val>
</magnitude>
</node>
<node name="Archaeal adenylate kinase"><magnitude>
<val> 50.212774</val>
</magnitude>
</node>
<node name="Aspartate carbamoyltransferase, catalytic chain"><magnitude>
<val> 221.768603</val>
</magnitude>
</node>
<node name="Aspartate carbamoyltransferase, regulatory subunit"><magnitude>
<val> 148.800572</val>
</magnitude>
</node>
<node name="CTP synthase (UTP-ammonia lyase)"><magnitude>
<val> 290.121492</val>
</magnitude>
</node>
<node name="Cytidine deaminase"><magnitude>
<val> 133.373991</val>
</magnitude>
</node>
<node name="Cytidylate kinase"><magnitude>
<val> 448.627737</val>
</magnitude>
</node>
<node name="Cytosine deaminase and related metal-dependent hydrolases"><magnitude>
<val> 517.061335</val>
</magnitude>
</node>
<node name="Cytosine/adenosine deaminases"><magnitude>
<val> 346.352404</val>
</magnitude>
</node>
<node name="Cytosine/uracil/thiamine/allantoin permeases"><magnitude>
<val> 78.832917</val>
</magnitude>
</node>
<node name="DNA/RNA endonuclease G, NUC1"><magnitude>
<val> 82.156333</val>
</magnitude>
</node>
<node name="Deoxycytidine deaminase"><magnitude>
<val> 135.046951</val>
</magnitude>
</node>
<node name="Deoxycytidylate deaminase"><magnitude>
<val> 174.778465</val>
</magnitude>
</node>
<node name="Deoxynucleoside kinases"><magnitude>
<val> 9.391606</val>
</magnitude>
</node>
<node name="Deoxyribose-phosphate aldolase"><magnitude>
<val> 132.690515</val>
</magnitude>
</node>
<node name="Diadenosine tetraphosphate (Ap4A) hydrolase and other HIT family hydrolases"><magnitude>
<val> 538.99727</val>
</magnitude>
</node>
<node name="Dihydroorotase"><magnitude>
<val> 118.671454</val>
</magnitude>
</node>
<node name="Dihydroorotase and related cyclic amidohydrolases"><magnitude>
<val> 343.444443</val>
</magnitude>
</node>
<node name="Dihydroorotate dehydrogenase"><magnitude>
<val> 322.043292</val>
</magnitude>
</node>
<node name="Exopolyphosphatase"><magnitude>
<val> 306.379448</val>
</magnitude>
</node>
<node name="Folate-dependent phosphoribosylglycinamide formyltransferase PurN"><magnitude>
<val> 267.020193</val>
</magnitude>
</node>
<node name="Formate-dependent phosphoribosylglycinamide formyltransferase (GAR transformylase)"><magnitude>
<val> 109.336525</val>
</magnitude>
</node>
<node name="Formyltetrahydrofolate hydrolase"><magnitude>
<val> 134.022714</val>
</magnitude>
</node>
<node name="Formyltetrahydrofolate synthetase"><magnitude>
<val> 149.755855</val>
</magnitude>
</node>
<node name="GMP synthase - Glutamine amidotransferase domain"><magnitude>
<val> 240.547405</val>
</magnitude>
</node>
<node name="GMP synthase, PP-ATPase domain/subunit"><magnitude>
<val> 294.865758</val>
</magnitude>
</node>
<node name="Glutamine phosphoribosylpyrophosphate amidotransferase"><magnitude>
<val> 384.120188</val>
</magnitude>
</node>
<node name="Guanyl-specific ribonuclease Sa"><magnitude>
<val> 19.082137</val>
</magnitude>
</node>
<node name="Guanylate kinase"><magnitude>
<val> 153.809569</val>
</magnitude>
</node>
<node name="Hydroxylase for synthesis of 2-methylthio-cis-ribozeatin in tRNA"><magnitude>
<val> 61.77566</val>
</magnitude>
</node>
<node name="Hypoxanthine-guanine phosphoribosyltransferase"><magnitude>
<val> 133.664887</val>
</magnitude>
</node>
<node name="IMP dehydrogenase/GMP reductase"><magnitude>
<val> 303.075102</val>
</magnitude>
</node>
<node name="Inosine-uridine nucleoside N-ribohydrolase"><magnitude>
<val> 65.841652</val>
</magnitude>
</node>
<node name="Nucleoside 2-deoxyribosyltransferase"><magnitude>
<val> 135.756078</val>
</magnitude>
</node>
<node name="Nucleoside diphosphate kinase"><magnitude>
<val> 180.870928</val>
</magnitude>
</node>
<node name="Nucleoside permease"><magnitude>
<val> 53.768815</val>
</magnitude>
</node>
<node name="Nucleoside phosphorylase"><magnitude>
<val> 215.642548</val>
</magnitude>
</node>
<node name="Orotate phosphoribosyltransferase"><magnitude>
<val> 235.666787</val>
</magnitude>
</node>
<node name="Orotate phosphoribosyltransferase homologs"><magnitude>
<val> 196.056033</val>
</magnitude>
</node>
<node name="Orotidine-5'-phosphate decarboxylase"><magnitude>
<val> 206.276885</val>
</magnitude>
</node>
<node name="Oxygen-sensitive ribonucleoside-triphosphate reductase"><magnitude>
<val> 529.929727</val>
</magnitude>
</node>
<node name="Phosphoribosylamine-glycine ligase"><magnitude>
<val> 305.446058</val>
</magnitude>
</node>
<node name="Phosphoribosylaminoimidazole (AIR) synthetase"><magnitude>
<val> 227.511807</val>
</magnitude>
</node>
<node name="Phosphoribosylaminoimidazole carboxylase (NCAIR synthetase)"><magnitude>
<val> 56.738093</val>
</magnitude>
</node>
<node name="Phosphoribosylaminoimidazolesuccinocarboxamide (SAICAR) synthase"><magnitude>
<val> 248.507461</val>
</magnitude>
</node>
<node name="Phosphoribosylcarboxyaminoimidazole (NCAIR) mutase"><magnitude>
<val> 257.156329</val>
</magnitude>
</node>
<node name="Phosphoribosylformylglycinamidine (FGAM) synthase, PurS component"><magnitude>
<val> 73.876581</val>
</magnitude>
</node>
<node name="Phosphoribosylformylglycinamidine (FGAM) synthase, glutamine amidotransferase domain"><magnitude>
<val> 222.411722</val>
</magnitude>
</node>
<node name="Phosphoribosylformylglycinamidine (FGAM) synthase, synthetase domain"><magnitude>
<val> 310.720208</val>
</magnitude>
</node>
<node name="Phosphoribosylpyrophosphate synthetase"><magnitude>
<val> 270.136554</val>
</magnitude>
</node>
<node name="Predicted alternative thymidylate synthase"><magnitude>
<val> 218.113929</val>
</magnitude>
</node>
<node name="Predicted nucleotide kinase"><magnitude>
<val> 82.644186</val>
</magnitude>
</node>
<node name="Predicted nucleotide kinase (related to CMP and AMP kinases)"><magnitude>
<val> 93.775442</val>
</magnitude>
</node>
<node name="Predicted secreted endonuclease distantly related to archaeal Holliday junction resolvase"><magnitude>
<val> 62.555649</val>
</magnitude>
</node>
<node name="Protein involved in ribonucleotide reduction"><magnitude>
<val> 37.087274</val>
</magnitude>
</node>
<node name="Purine nucleoside permease"><magnitude>
<val> 48.909436</val>
</magnitude>
</node>
<node name="Purine nucleoside phosphorylase"><magnitude>
<val> 204.168257</val>
</magnitude>
</node>
<node name="Purine-cytosine permease and related proteins"><magnitude>
<val> 106.206243</val>
</magnitude>
</node>
<node name="Purine-nucleoside phosphorylase"><magnitude>
<val> 93.507572</val>
</magnitude>
</node>
<node name="Pyrimidine operon attenuation protein/uracil phosphoribosyltransferase"><magnitude>
<val> 119.815097</val>
</magnitude>
</node>
<node name="Ribonucleotide reductase, alpha subunit"><magnitude>
<val> 263.211712</val>
</magnitude>
</node>
<node name="Ribonucleotide reductase, beta subunit"><magnitude>
<val> 56.949592</val>
</magnitude>
</node>
<node name="Thymidine kinase"><magnitude>
<val> 67.460349</val>
</magnitude>
</node>
<node name="Thymidine phosphorylase"><magnitude>
<val> 186.179401</val>
</magnitude>
</node>
<node name="Thymidylate kinase"><magnitude>
<val> 385.710505</val>
</magnitude>
</node>
<node name="Thymidylate synthase"><magnitude>
<val> 149.671128</val>
</magnitude>
</node>
<node name="Uracil phosphoribosyltransferase"><magnitude>
<val> 145.423793</val>
</magnitude>
</node>
<node name="Ureidoglycolate hydrolase"><magnitude>
<val> 42.327033</val>
</magnitude>
</node>
<node name="Uridine kinase"><magnitude>
<val> 128.331306</val>
</magnitude>
</node>
<node name="Uridine phosphorylase"><magnitude>
<val> 144.167812</val>
</magnitude>
</node>
<node name="Uridylate kinase"><magnitude>
<val> 275.697525</val>
</magnitude>
</node>
<node name="Xanthine dehydrogenase, iron-sulfur cluster and FAD-binding subunit A"><magnitude>
<val> 85.632747</val>
</magnitude>
</node>
<node name="Xanthine dehydrogenase, molybdopterin-binding subunit B"><magnitude>
<val> 160.668814</val>
</magnitude>
</node>
<node name="Xanthine/uracil permeases"><magnitude>
<val> 241.945419</val>
</magnitude>
</node>
<node name="Xanthosine triphosphate pyrophosphatase"><magnitude>
<val> 273.101551</val>
</magnitude>
</node>
<node name="dGTP triphosphohydrolase"><magnitude>
<val> 191.143861</val>
</magnitude>
</node>
<node name="dUTPase"><magnitude>
<val> 193.290677</val>
</magnitude>
</node>
</node>
<node name="Secondary metabolites biosynthesis, transport and catabolism"><magnitude>
<val> 7368.108573</val>
</magnitude>
<node name="2-hydroxychromene-2-carboxylate isomerase"><magnitude>
<val> 11.152134</val>
</magnitude>
</node>
<node name="2-keto-4-pentenoate hydratase"><magnitude>
<val> 26.592469</val>
</magnitude>
</node>
<node name="2-keto-4-pentenoate hydratase/2-oxohepta-3-ene-1,7-dioic acid hydratase (catechol pathway)"><magnitude>
<val> 301.314358</val>
</magnitude>
</node>
<node name="ABC-type siderophore export system, fused ATPase and permease components"><magnitude>
<val> 86.171131</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in resistance to organic solvents, ATPase component"><magnitude>
<val> 240.470936</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in resistance to organic solvents, auxiliary component"><magnitude>
<val> 130.737006</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in resistance to organic solvents, periplasmic component"><magnitude>
<val> 310.72782</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in resistance to organic solvents, permease component"><magnitude>
<val> 287.6046</val>
</magnitude>
</node>
<node name="Acetaldehyde dehydrogenase (acetylating)"><magnitude>
<val> 33.543067</val>
</magnitude>
</node>
<node name="Acetoacetate decarboxylase"><magnitude>
<val> 63.131146</val>
</magnitude>
</node>
<node name="Acetyl esterase (deacetylase)"><magnitude>
<val> 132.418531</val>
</magnitude>
</node>
<node name="Alkyl sulfatase and related hydrolases"><magnitude>
<val> 159.087429</val>
</magnitude>
</node>
<node name="Alpha-acetolactate decarboxylase"><magnitude>
<val> 44.217157</val>
</magnitude>
</node>
<node name="Amidases related to nicotinamidase"><magnitude>
<val> 542.055442</val>
</magnitude>
</node>
<node name="Aromatic ring hydroxylase"><magnitude>
<val> 150.981364</val>
</magnitude>
</node>
<node name="Aromatic ring-cleaving dioxygenase"><magnitude>
<val> 29.635717</val>
</magnitude>
</node>
<node name="Arylamine N-acetyltransferase"><magnitude>
<val> 34.290718</val>
</magnitude>
</node>
<node name="Carbon dioxide concentrating mechanism/carboxysome shell protein"><magnitude>
<val> 112.117872</val>
</magnitude>
</node>
<node name="Cu2+-containing amine oxidase"><magnitude>
<val> 104.944092</val>
</magnitude>
</node>
<node name="Cyanophycinase and related exopeptidases"><magnitude>
<val> 82.516313</val>
</magnitude>
</node>
<node name="Cytochrome P450"><magnitude>
<val> 55.075235</val>
</magnitude>
</node>
<node name="Dienelactone hydrolase and related enzymes"><magnitude>
<val> 293.098845</val>
</magnitude>
</node>
<node name="Gentisate 1,2-dioxygenase"><magnitude>
<val> 107.047168</val>
</magnitude>
</node>
<node name="Homogentisate 1,2-dioxygenase"><magnitude>
<val> 30.191319</val>
</magnitude>
</node>
<node name="Homospermidine synthase"><magnitude>
<val> 4.556351</val>
</magnitude>
</node>
<node name="Imidazolonepropionase and related amidohydrolases"><magnitude>
<val> 188.027728</val>
</magnitude>
</node>
<node name="Isochorismate hydrolase"><magnitude>
<val> 19.395933</val>
</magnitude>
</node>
<node name="Lignostilbene-alpha,beta-dioxygenase and related enzymes"><magnitude>
<val> 65.814036</val>
</magnitude>
</node>
<node name="Limonene-1,2-epoxide hydrolase"><magnitude>
<val> 41.485317</val>
</magnitude>
</node>
<node name="Lysine/ornithine N-monooxygenase"><magnitude>
<val> 70.40616</val>
</magnitude>
</node>
<node name="Maleate cis-trans isomerase"><magnitude>
<val> 66.459004</val>
</magnitude>
</node>
<node name="Muconolactone delta-isomerase"><magnitude>
<val> 14.094981</val>
</magnitude>
</node>
<node name="N-acyl-D-aspartate/D-glutamate deacylase"><magnitude>
<val> 64.738976</val>
</magnitude>
</node>
<node name="Non-ribosomal peptide synthetase modules and related proteins"><magnitude>
<val> 469.862607</val>
</magnitude>
</node>
<node name="O-Methyltransferase involved in polyketide biosynthesis"><magnitude>
<val> 76.982349</val>
</magnitude>
</node>
<node name="Oxidoreductase (NAD-binding), involved in siderophore biosynthesis"><magnitude>
<val> 57.193258</val>
</magnitude>
</node>
<node name="Peptide arylation enzymes"><magnitude>
<val> 67.014401</val>
</magnitude>
</node>
<node name="Phenolic acid decarboxylase"><magnitude>
<val> 4.978916</val>
</magnitude>
</node>
<node name="Phosphopantetheinyl transferase component of siderophore synthetase"><magnitude>
<val> 5.393927</val>
</magnitude>
</node>
<node name="Phytoene dehydrogenase and related proteins"><magnitude>
<val> 171.328146</val>
</magnitude>
</node>
<node name="Poly(3-hydroxybutyrate) depolymerase"><magnitude>
<val> 30.988491</val>
</magnitude>
</node>
<node name="Polyketide synthase modules and related proteins"><magnitude>
<val> 228.402035</val>
</magnitude>
</node>
<node name="Predicted ABC-type transport system involved in lysophospholipase L1 biosynthesis, ATPase component"><magnitude>
<val> 4.477985</val>
</magnitude>
</node>
<node name="Predicted ABC-type transport system involved in lysophospholipase L1 biosynthesis, permease component"><magnitude>
<val> 393.909986</val>
</magnitude>
</node>
<node name="Predicted dithiol-disulfide isomerase involved in polyketide biosynthesis"><magnitude>
<val> 83.353649</val>
</magnitude>
</node>
<node name="Predicted enzyme involved in methoxymalonyl-ACP biosynthesis"><magnitude>
<val> 44.002172</val>
</magnitude>
</node>
<node name="Predicted naringenin-chalcone synthase"><magnitude>
<val> 26.210382</val>
</magnitude>
</node>
<node name="Predicted thioesterase involved in non-ribosomal peptide biosynthesis"><magnitude>
<val> 68.019483</val>
</magnitude>
</node>
<node name="Probable taurine catabolism dioxygenase"><magnitude>
<val> 63.251959</val>
</magnitude>
</node>
<node name="Propanediol dehydratase, large subunit"><magnitude>
<val> 43.977425</val>
</magnitude>
</node>
<node name="Propanediol dehydratase, small subunit"><magnitude>
<val> 14.75906</val>
</magnitude>
</node>
<node name="Propanediol utilization protein"><magnitude>
<val> 9.613123</val>
</magnitude>
</node>
<node name="Protein involved in biosynthesis of mitomycin antibiotics/polyketide fumonisin"><magnitude>
<val> 8.517079</val>
</magnitude>
</node>
<node name="Protein involved in propanediol utilization, and related proteins (includes coumermycin biosynthetic protein), possible kinase"><magnitude>
<val> 19.169541</val>
</magnitude>
</node>
<node name="Protocatechuate 3,4-dioxygenase beta subunit"><magnitude>
<val> 143.755932</val>
</magnitude>
</node>
<node name="Putative dehydrogenase domain of multifunctional non-ribosomal peptide synthetases and related enzymes"><magnitude>
<val> 38.720442</val>
</magnitude>
</node>
<node name="Putative multicopper oxidases"><magnitude>
<val> 175.620831</val>
</magnitude>
</node>
<node name="Pyoverdine/dityrosine biosynthesis protein"><magnitude>
<val> 28.343854</val>
</magnitude>
</node>
<node name="RTX toxins and related Ca2+-binding proteins"><magnitude>
<val> 43.544357</val>
</magnitude>
</node>
<node name="SAM-dependent methyltransferases"><magnitude>
<val> 124.944714</val>
</magnitude>
</node>
<node name="Siderophore synthetase component"><magnitude>
<val> 51.667237</val>
</magnitude>
</node>
<node name="Small subunit of phenylpropionate dioxygenase"><magnitude>
<val> 6.697158</val>
</magnitude>
</node>
<node name="TRAP-type mannitol/chloroaromatic compound transport system, large permease component"><magnitude>
<val> 47.284611</val>
</magnitude>
</node>
<node name="TRAP-type mannitol/chloroaromatic compound transport system, periplasmic component"><magnitude>
<val> 8.561173</val>
</magnitude>
</node>
<node name="Thioesterase domains of type I polyketide synthases or non-ribosomal peptide synthetases"><magnitude>
<val> 102.216449</val>
</magnitude>
</node>
<node name="Transcriptional activator of acetoin/glycerol metabolism"><magnitude>
<val> 102.040706</val>
</magnitude>
</node>
<node name="Uncharacterized enzyme involved in pigment biosynthesis"><magnitude>
<val> 35.244942</val>
</magnitude>
</node>
<node name="Uncharacterized enzyme of phenylacetate metabolism"><magnitude>
<val> 5.374097</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in an early stage of isoprenoid biosynthesis"><magnitude>
<val> 60.495824</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in benzoate metabolism"><magnitude>
<val> 56.768688</val>
</magnitude>
</node>
<node name="Uncharacterized protein, possibly involved in aromatic compounds catabolism"><magnitude>
<val> 465.62702</val>
</magnitude>
</node>
<node name="Uricase (urate oxidase)"><magnitude>
<val> 75.694209</val>
</magnitude>
</node>
</node>
</node>
<node name="POORLY CHARACTERIZED"><magnitude>
<val> 194456.176271</val>
</magnitude>
<node name="Function unknown"><magnitude>
<val> 97381.993097</val>
</magnitude>
<node name="Acetolactate synthase (isozyme II), small (regulatory) subunit"><magnitude>
<val> 18.998936</val>
</magnitude>
</node>
<node name="Bacteriophage protein gp37"><magnitude>
<val> 30.135766</val>
</magnitude>
</node>
<node name="Conserved secreted protein"><magnitude>
<val> 147.716001</val>
</magnitude>
</node>
<node name="FOG: WD40-like repeat"><magnitude>
<val> 408.19862</val>
</magnitude>
</node>
<node name="Hemolysin-coregulated protein (uncharacterized)"><magnitude>
<val> 98.946566</val>
</magnitude>
</node>
<node name="Integral membrane protein"><magnitude>
<val> 32.738216</val>
</magnitude>
</node>
<node name="Ketosteroid isomerase homolog"><magnitude>
<val> 138.082994</val>
</magnitude>
</node>
<node name="Leucine-rich repeat (LRR) protein"><magnitude>
<val> 113.519041</val>
</magnitude>
</node>
<node name="Major membrane immunogen, membrane-anchored lipoprotein"><magnitude>
<val> 46.26731</val>
</magnitude>
</node>
<node name="Microcystin-dependent protein"><magnitude>
<val> 57.692807</val>
</magnitude>
</node>
<node name="Mu-like prophage protein gp16"><magnitude>
<val> 4.935791</val>
</magnitude>
</node>
<node name="Mu-like prophage protein gp29"><magnitude>
<val> 58.273582</val>
</magnitude>
</node>
<node name="Mu-like prophage protein gp36"><magnitude>
<val> 23.834298</val>
</magnitude>
</node>
<node name="Mu-like prophage protein gp45"><magnitude>
<val> 17.969008</val>
</magnitude>
</node>
<node name="Mu-like prophage protein gp46"><magnitude>
<val> 74.943602</val>
</magnitude>
</node>
<node name="Myosin-crossreactive antigen"><magnitude>
<val> 62.244989</val>
</magnitude>
</node>
<node name="Phage-related minor tail protein"><magnitude>
<val> 69.443587</val>
</magnitude>
</node>
<node name="Phage-related protein"><magnitude>
<val> 531.82876</val>
</magnitude>
</node>
<node name="Phage-related protein, tail component"><magnitude>
<val> 110.438348</val>
</magnitude>
</node>
<node name="Phage-related tail protein"><magnitude>
<val> 178.206001</val>
</magnitude>
</node>
<node name="Predicted archaeal membrane protein"><magnitude>
<val> 45.239471</val>
</magnitude>
</node>
<node name="Predicted integral membrane protein"><magnitude>
<val> 1797.771244</val>
</magnitude>
</node>
<node name="Predicted integral membrane proteins containing uncharacterized repeats"><magnitude>
<val> 61.127556</val>
</magnitude>
</node>
<node name="Predicted membrane protein"><magnitude>
<val> 14308.803732</val>
</magnitude>
</node>
<node name="Predicted membrane protein/domain"><magnitude>
<val> 178.129764</val>
</magnitude>
</node>
<node name="Predicted metal-binding protein"><magnitude>
<val> 270.224134</val>
</magnitude>
</node>
<node name="Predicted multitransmembrane protein"><magnitude>
<val> 102.901662</val>
</magnitude>
</node>
<node name="Predicted outer membrane lipoprotein"><magnitude>
<val> 15.958731</val>
</magnitude>
</node>
<node name="Predicted outer membrane protein"><magnitude>
<val> 35.247352</val>
</magnitude>
</node>
<node name="Predicted periplasmic protein"><magnitude>
<val> 271.584975</val>
</magnitude>
</node>
<node name="Predicted periplasmic/secreted protein"><magnitude>
<val> 50.164269</val>
</magnitude>
</node>
<node name="Predicted secreted (periplasmic) protein"><magnitude>
<val> 106.825282</val>
</magnitude>
</node>
<node name="Predicted secreted protein"><magnitude>
<val> 397.42872</val>
</magnitude>
</node>
<node name="Predicted small integral membrane protein"><magnitude>
<val> 177.072338</val>
</magnitude>
</node>
<node name="Predicted small metal-binding protein"><magnitude>
<val> .987952</val>
</magnitude>
</node>
<node name="Predicted small secreted protein"><magnitude>
<val> 109.290491</val>
</magnitude>
</node>
<node name="Putative transposon-encoded protein"><magnitude>
<val> 13.26833</val>
</magnitude>
</node>
<node name="Putative virion core protein (lumpy skin disease virus)"><magnitude>
<val> 161.498949</val>
</magnitude>
</node>
<node name="Small integral membrane protein"><magnitude>
<val> 146.591424</val>
</magnitude>
</node>
<node name="Uncharacterized N-terminal domain of the transcription elongation factor GreA"><magnitude>
<val> 108.96373</val>
</magnitude>
</node>
<node name="Uncharacterized archaeal coiled-coil protein"><magnitude>
<val> 126.53419</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein"><magnitude>
<val> 33488.1929090001</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein (small basic protein)"><magnitude>
<val> 68.179419</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein containing a coiled-coil domain"><magnitude>
<val> 43.107611</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein related to C-terminal domain of eukaryotic chaperone, SACSIN"><magnitude>
<val> 329.05609</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein related to MYG1 family"><magnitude>
<val> 54.00782</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein related to dihydrodipicolinate reductase"><magnitude>
<val> 28.105033</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein, contains S4-like domain"><magnitude>
<val> 35.391777</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein, contains Zn-ribbon-like motif"><magnitude>
<val> 4.879973</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein, contains double-stranded beta-helix domain"><magnitude>
<val> 660.238149</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein, homolog of nitrogen regulatory protein PII"><magnitude>
<val> 58.77984</val>
</magnitude>
</node>
<node name="Uncharacterized conserved small protein"><magnitude>
<val> 207.344762</val>
</magnitude>
</node>
<node name="Uncharacterized conserved small protein containing a coiled-coil domain"><magnitude>
<val> 15.41204</val>
</magnitude>
</node>
<node name="Uncharacterized distant relative of cell wall-associated hydrolases"><magnitude>
<val> 106.451165</val>
</magnitude>
</node>
<node name="Uncharacterized homolog of Blt101"><magnitude>
<val> 4.925065</val>
</magnitude>
</node>
<node name="Uncharacterized homolog of PSP1"><magnitude>
<val> 143.842526</val>
</magnitude>
</node>
<node name="Uncharacterized homolog of PrgY (pheromone shutdown protein)"><magnitude>
<val> 179.922417</val>
</magnitude>
</node>
<node name="Uncharacterized homolog of biotin synthetase"><magnitude>
<val> 16.08645</val>
</magnitude>
</node>
<node name="Uncharacterized homolog of gamma-carboxymuconolactone decarboxylase subunit"><magnitude>
<val> 722.751538</val>
</magnitude>
</node>
<node name="Uncharacterized homolog of phage Mu protein gp47"><magnitude>
<val> 48.715706</val>
</magnitude>
</node>
<node name="Uncharacterized homolog of plant Iojap protein"><magnitude>
<val> 137.884444</val>
</magnitude>
</node>
<node name="Uncharacterized homolog of the cytoplasmic domain of flagellar protein FhlB"><magnitude>
<val> 124.214151</val>
</magnitude>
</node>
<node name="Uncharacterized integral membrane protein"><magnitude>
<val> 156.843365</val>
</magnitude>
</node>
<node name="Uncharacterized iron-regulated membrane protein"><magnitude>
<val> 152.381852</val>
</magnitude>
</node>
<node name="Uncharacterized iron-regulated protein"><magnitude>
<val> 85.555215</val>
</magnitude>
</node>
<node name="Uncharacterized low-complexity proteins"><magnitude>
<val> 307.131089</val>
</magnitude>
</node>
<node name="Uncharacterized membrane protein"><magnitude>
<val> 662.362882</val>
</magnitude>
</node>
<node name="Uncharacterized membrane-anchored protein"><magnitude>
<val> 49.447899</val>
</magnitude>
</node>
<node name="Uncharacterized membrane-anchored protein conserved in bacteria"><magnitude>
<val> 238.177276</val>
</magnitude>
</node>
<node name="Uncharacterized membrane-associated protein"><magnitude>
<val> 397.318488</val>
</magnitude>
</node>
<node name="Uncharacterized membrane-associated protein/domain"><magnitude>
<val> 237.436509</val>
</magnitude>
</node>
<node name="Uncharacterized membrane-bound protein"><magnitude>
<val> 47.380432</val>
</magnitude>
</node>
<node name="Uncharacterized membrane-bound protein conserved in bacteria"><magnitude>
<val> 45.577257</val>
</magnitude>
</node>
<node name="Uncharacterized metal-binding protein"><magnitude>
<val> 3.088889</val>
</magnitude>
</node>
<node name="Uncharacterized paraquat-inducible protein A"><magnitude>
<val> 182.169082</val>
</magnitude>
</node>
<node name="Uncharacterized paraquat-inducible protein B"><magnitude>
<val> 85.027503</val>
</magnitude>
</node>
<node name="Uncharacterized phage-associated protein"><magnitude>
<val> 29.824732</val>
</magnitude>
</node>
<node name="Uncharacterized phage-encoded protein"><magnitude>
<val> 142.748275</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in archaea"><magnitude>
<val> 6816.394266</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria"><magnitude>
<val> 27742.756214</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria containing a divergent form of TPR repeats"><magnitude>
<val> 128.122884</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria containing a pentein-type domain"><magnitude>
<val> 53.45452</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria with a cystatin-like fold"><magnitude>
<val> 45.81075</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria with the myosin-like domain"><magnitude>
<val> 78.573033</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria, prophage-related"><magnitude>
<val> 24.381532</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria, putative lipoprotein"><magnitude>
<val> 66.823587</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria, putative virulence factor"><magnitude>
<val> 178.790416</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in cyanobacteria"><magnitude>
<val> 151.230097</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing LysM domain"><magnitude>
<val> 249.111257</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing a TIR (Toll-Interleukin 1-resistance) domain"><magnitude>
<val> 16.934973</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing a Zn-ribbon"><magnitude>
<val> 187.444353</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing a ferredoxin domain"><magnitude>
<val> 84.606072</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing double-stranded beta helix domain"><magnitude>
<val> 38.051582</val>
</magnitude>
</node>
<node name="Uncharacterized protein encoded in hypervariable junctions of pilus gene clusters"><magnitude>
<val> 106.461038</val>
</magnitude>
</node>
<node name="Uncharacterized protein related to the periplasmic component of the Tol biopolymer transport system"><magnitude>
<val> 95.956369</val>
</magnitude>
</node>
<node name="Uncharacterized protein with SCP/PR1 domains"><magnitude>
<val> 185.156601</val>
</magnitude>
</node>
<node name="Uncharacterized protein with a C-terminal OMP (outer membrane protein) domain"><magnitude>
<val> 151.200894</val>
</magnitude>
</node>
<node name="Uncharacterized protein with a bacterial SH3 domain homologue"><magnitude>
<val> 61.748752</val>
</magnitude>
</node>
<node name="Uncharacterized protein with conserved CXXC pairs"><magnitude>
<val> 79.901299</val>
</magnitude>
</node>
<node name="Uncharacterized protein, homolog of phage Mu protein gp30"><magnitude>
<val> 93.011184</val>
</magnitude>
</node>
<node name="Uncharacterized protein, linocin/CFP29 homolog"><magnitude>
<val> 112.386793</val>
</magnitude>
</node>
<node name="Uncharacterized protein/domain associated with GTPases"><magnitude>
<val> 18.31318</val>
</magnitude>
</node>
<node name="Uncharacterized proteins, LmbE homologs"><magnitude>
<val> 119.938774</val>
</magnitude>
</node>
<node name="Uncharacterized secreted protein"><magnitude>
<val> 9.866648</val>
</magnitude>
</node>
<node name="Uncharacterized small membrane protein"><magnitude>
<val> 17.921054</val>
</magnitude>
</node>
<node name="Uncharacterized small protein"><magnitude>
<val> 147.024266</val>
</magnitude>
</node>
<node name="Uncharacterized small protein containing a coiled-coil domain"><magnitude>
<val> 11.600574</val>
</magnitude>
</node>
<node name="Uncharacterized stress-induced protein"><magnitude>
<val> 247.450842</val>
</magnitude>
</node>
<node name="Virulence-associated protein and related proteins"><magnitude>
<val> 51.581144</val>
</magnitude>
</node>
</node>
<node name="General function prediction only"><magnitude>
<val> 97074.1831740001</val>
</magnitude>
<node name="5-bromo-4-chloroindolyl phosphate hydrolysis protein"><magnitude>
<val> 36.191098</val>
</magnitude>
</node>
<node name="AAA ATPase containing von Willebrand factor type A (vWA) domain"><magnitude>
<val> 536.448233</val>
</magnitude>
</node>
<node name="ABC-type (unclassified) transport system, ATPase component"><magnitude>
<val> 163.119632</val>
</magnitude>
</node>
<node name="ABC-type ATPase fused to a predicted acetyltransferase domain"><magnitude>
<val> 91.883294</val>
</magnitude>
</node>
<node name="ABC-type protease/lipase transport system, ATPase and permease components"><magnitude>
<val> 23.749894</val>
</magnitude>
</node>
<node name="ABC-type sugar transport system, auxiliary component"><magnitude>
<val> 36.877694</val>
</magnitude>
</node>
<node name="ABC-type transport system involved in multi-copper enzyme maturation, permease component"><magnitude>
<val> 97.585042</val>
</magnitude>
</node>
<node name="ABC-type uncharacterized transport system, ATPase component"><magnitude>
<val> 404.7399</val>
</magnitude>
</node>
<node name="ABC-type uncharacterized transport system, auxiliary component"><magnitude>
<val> 45.180965</val>
</magnitude>
</node>
<node name="ABC-type uncharacterized transport system, duplicated ATPase component"><magnitude>
<val> 88.833661</val>
</magnitude>
</node>
<node name="ABC-type uncharacterized transport system, periplasmic component"><magnitude>
<val> 477.097593</val>
</magnitude>
</node>
<node name="ABC-type uncharacterized transport system, permease and ATPase components"><magnitude>
<val> 162.748667</val>
</magnitude>
</node>
<node name="ABC-type uncharacterized transport system, permease component"><magnitude>
<val> 838.522496</val>
</magnitude>
</node>
<node name="ABC-type uncharacterized transport systems, ATPase components"><magnitude>
<val> 124.307607</val>
</magnitude>
</node>
<node name="ACT domain-containing protein"><magnitude>
<val> 373.73156</val>
</magnitude>
</node>
<node name="AT-rich DNA-binding protein"><magnitude>
<val> 137.319648</val>
</magnitude>
</node>
<node name="ATP-utilizing enzymes of ATP-grasp superfamily (probably carboligases)"><magnitude>
<val> 136.907333</val>
</magnitude>
</node>
<node name="ATP-utilizing enzymes of the PP-loop superfamily"><magnitude>
<val> 199.34875</val>
</magnitude>
</node>
<node name="ATPase (PilT family)"><magnitude>
<val> 187.880864</val>
</magnitude>
</node>
<node name="ATPase components of ABC transporters with duplicated ATPase domains"><magnitude>
<val> 850.023483</val>
</magnitude>
</node>
<node name="ATPase components of various ABC-type transport systems, contain duplicated ATPase"><magnitude>
<val> 348.843369</val>
</magnitude>
</node>
<node name="Accessory colonization factor AcfC, contains ABC-type periplasmic domain"><magnitude>
<val> 101.757321</val>
</magnitude>
</node>
<node name="Acetyltransferase (isoleucine patch superfamily)"><magnitude>
<val> 591.157129</val>
</magnitude>
</node>
<node name="Acetyltransferases"><magnitude>
<val> 1364.3969</val>
</magnitude>
</node>
<node name="Acid phosphatase (class B)"><magnitude>
<val> 28.835179</val>
</magnitude>
</node>
<node name="Acid-induced glycyl radical enzyme"><magnitude>
<val> 9.846862</val>
</magnitude>
</node>
<node name="Aldo/keto reductases, related to diketogulonate reductase"><magnitude>
<val> 72.015697</val>
</magnitude>
</node>
<node name="Anaerobic C4-dicarboxylate transporter"><magnitude>
<val> 85.815192</val>
</magnitude>
</node>
<node name="Antirestriction protein"><magnitude>
<val> 24.381532</val>
</magnitude>
</node>
<node name="Archaea-specific pyridoxal phosphate-dependent enzymes"><magnitude>
<val> 60.183063</val>
</magnitude>
</node>
<node name="Archaeal enzymes of ATP-grasp superfamily"><magnitude>
<val> 78.006719</val>
</magnitude>
</node>
<node name="Archaeal serine proteases"><magnitude>
<val> 69.481332</val>
</magnitude>
</node>
<node name="Arylsulfatase regulator (Fe-S oxidoreductase)"><magnitude>
<val> 629.559463</val>
</magnitude>
</node>
<node name="BRCT domain type II"><magnitude>
<val> 11.575068</val>
</magnitude>
</node>
<node name="Bacteriophage P2-related tail formation protein"><magnitude>
<val> 48.785918</val>
</magnitude>
</node>
<node name="Bacteriophage capsid portal protein"><magnitude>
<val> 59.899794</val>
</magnitude>
</node>
<node name="Bacteriophage capsid protein"><magnitude>
<val> 141.206971</val>
</magnitude>
</node>
<node name="Bacteriophage head-tail adaptor"><magnitude>
<val> 4.107829</val>
</magnitude>
</node>
<node name="Bacteriophage tail assembly protein"><magnitude>
<val> 207.169098</val>
</magnitude>
</node>
<node name="Bacteriorhodopsin"><magnitude>
<val> 65.771492</val>
</magnitude>
</node>
<node name="Beta-propeller domains of methanol dehydrogenase type"><magnitude>
<val> 209.916753</val>
</magnitude>
</node>
<node name="BioD-like N-terminal domain of phosphotransacetylase"><magnitude>
<val> 132.878486</val>
</magnitude>
</node>
<node name="Biotin synthase-related enzyme"><magnitude>
<val> 34.552385</val>
</magnitude>
</node>
<node name="C4-type Zn-finger protein"><magnitude>
<val> 127.435503</val>
</magnitude>
</node>
<node name="CCCH-type Zn-finger protein"><magnitude>
<val> 9.544273</val>
</magnitude>
</node>
<node name="Ca2+-dependent lipid-binding protein, contains C2 domain"><magnitude>
<val> 95.337475</val>
</magnitude>
</node>
<node name="Carbonic anhydrases/acetyltransferases, isoleucine patch superfamily"><magnitude>
<val> 235.817137</val>
</magnitude>
</node>
<node name="Cleavage and polyadenylation specificity factor (CPSF) Clipper subunit and related makorin family Zn-finger proteins"><magnitude>
<val> 3.120253</val>
</magnitude>
</node>
<node name="Collagen-binding surface adhesin SpaP (antigen I/II family)"><magnitude>
<val> 11.028067</val>
</magnitude>
</node>
<node name="Competence protein"><magnitude>
<val> 40.899127</val>
</magnitude>
</node>
<node name="Conserved protein containing a Zn-ribbon-like motif, possibly RNA-binding"><magnitude>
<val> 34.139666</val>
</magnitude>
</node>
<node name="Conserved protein/domain typically associated with flavoprotein oxygenases, DIM6/NTAB family"><magnitude>
<val> 595.799975</val>
</magnitude>
</node>
<node name="DMSO reductase anchor subunit"><magnitude>
<val> 27.159843</val>
</magnitude>
</node>
<node name="DNA uptake lipoprotein"><magnitude>
<val> 293.580993</val>
</magnitude>
</node>
<node name="DNA-binding protein"><magnitude>
<val> 61.224741</val>
</magnitude>
</node>
<node name="DNA-binding protein H-NS"><magnitude>
<val> 10.705817</val>
</magnitude>
</node>
<node name="DNA-binding protein, stimulates sugar fermentation"><magnitude>
<val> 197.541308</val>
</magnitude>
</node>
<node name="Dioxygenases related to 2-nitropropane dioxygenase"><magnitude>
<val> 249.736975</val>
</magnitude>
</node>
<node name="Distinct helicase family with a unique C-terminal domain including a metal-binding cysteine cluster"><magnitude>
<val> 379.754869</val>
</magnitude>
</node>
<node name="EMAP domain"><magnitude>
<val> 41.151346</val>
</magnitude>
</node>
<node name="Enzyme related to GTP cyclohydrolase I"><magnitude>
<val> 107.040548</val>
</magnitude>
</node>
<node name="Erythromycin esterase homolog"><magnitude>
<val> 57.337117</val>
</magnitude>
</node>
<node name="Esterase/lipase"><magnitude>
<val> 55.788328</val>
</magnitude>
</node>
<node name="Exopolyphosphatase-related proteins"><magnitude>
<val> 234.724546</val>
</magnitude>
</node>
<node name="FOG: Ankyrin repeat"><magnitude>
<val> 162.448958</val>
</magnitude>
</node>
<node name="FOG: CBS domain"><magnitude>
<val> 578.426426</val>
</magnitude>
</node>
<node name="FOG: Glucan-binding domain (YG repeat)"><magnitude>
<val> 256.069335</val>
</magnitude>
</node>
<node name="FOG: PKD repeat"><magnitude>
<val> 408.738049</val>
</magnitude>
</node>
<node name="FOG: TPR repeat"><magnitude>
<val> 1458.027482</val>
</magnitude>
</node>
<node name="FOG: TPR repeat, SEL1 subfamily"><magnitude>
<val> 229.935151</val>
</magnitude>
</node>
<node name="FOG: WD40 repeat"><magnitude>
<val> 166.351878</val>
</magnitude>
</node>
<node name="Fibronectin type 3 domain-containing protein"><magnitude>
<val> 266.827608</val>
</magnitude>
</node>
<node name="Fumarylacetoacetate (FAA) hydrolase family protein"><magnitude>
<val> 89.261864</val>
</magnitude>
</node>
<node name="GTPase"><magnitude>
<val> 288.615818</val>
</magnitude>
</node>
<node name="GTPase SAR1 and related small G proteins"><magnitude>
<val> 110.666045</val>
</magnitude>
</node>
<node name="GTPase-activating protein"><magnitude>
<val> 54.910409</val>
</magnitude>
</node>
<node name="GTPases"><magnitude>
<val> 187.874291</val>
</magnitude>
</node>
<node name="Gas vesicle protein"><magnitude>
<val> 237.523603</val>
</magnitude>
</node>
<node name="Glyoxylate carboligase"><magnitude>
<val> 98.759455</val>
</magnitude>
</node>
<node name="HD superfamily phosphohydrolases"><magnitude>
<val> 219.014864</val>
</magnitude>
</node>
<node name="Hemolysins and related proteins containing CBS domains"><magnitude>
<val> 470.76172</val>
</magnitude>
</node>
<node name="Hydrolases of the alpha/beta superfamily"><magnitude>
<val> 282.967397</val>
</magnitude>
</node>
<node name="Inhibitor of the KinA pathway to sporulation, predicted exonuclease"><magnitude>
<val> 16.9085</val>
</magnitude>
</node>
<node name="Integral membrane protein (PIN domain superfamily)"><magnitude>
<val> 187.385558</val>
</magnitude>
</node>
<node name="Integral membrane protein, interacts with FtsH"><magnitude>
<val> 41.544063</val>
</magnitude>
</node>
<node name="Invasion protein B, involved in pathogenesis"><magnitude>
<val> 127.784873</val>
</magnitude>
</node>
<node name="Iron only hydrogenase large subunit, C-terminal domain"><magnitude>
<val> 373.084004</val>
</magnitude>
</node>
<node name="Ketosteroid isomerase-related protein"><magnitude>
<val> 86.138203</val>
</magnitude>
</node>
<node name="Large extracellular alpha-helical protein"><magnitude>
<val> 539.517163</val>
</magnitude>
</node>
<node name="Lhr-like helicases"><magnitude>
<val> 234.844875</val>
</magnitude>
</node>
<node name="Lipoprotein NlpI, contains TPR repeats"><magnitude>
<val> 98.969595</val>
</magnitude>
</node>
<node name="Low specificity phosphatase (HAD superfamily)"><magnitude>
<val> 153.737162</val>
</magnitude>
</node>
<node name="Lysine efflux permease"><magnitude>
<val> 75.689791</val>
</magnitude>
</node>
<node name="Membrane protein involved in the export of O-antigen and teichoic acid"><magnitude>
<val> 1094.852999</val>
</magnitude>
</node>
<node name="Metal-dependent amidase/aminoacylase/carboxypeptidase"><magnitude>
<val> 349.997975</val>
</magnitude>
</node>
<node name="Metal-dependent hydrolase"><magnitude>
<val> 163.151599</val>
</magnitude>
</node>
<node name="Metal-dependent hydrolases of the beta-lactamase superfamily I"><magnitude>
<val> 363.637892</val>
</magnitude>
</node>
<node name="Metal-dependent hydrolases of the beta-lactamase superfamily II"><magnitude>
<val> 347.280973</val>
</magnitude>
</node>
<node name="Metal-dependent hydrolases of the beta-lactamase superfamily III"><magnitude>
<val> 370.930171</val>
</magnitude>
</node>
<node name="MoxR-like ATPases"><magnitude>
<val> 549.954038</val>
</magnitude>
</node>
<node name="Mu-like prophage DNA circulation protein"><magnitude>
<val> 10.867283</val>
</magnitude>
</node>
<node name="Mu-like prophage FluMu protein gp28"><magnitude>
<val> 110.393808</val>
</magnitude>
</node>
<node name="Mu-like prophage I protein"><magnitude>
<val> 58.75166</val>
</magnitude>
</node>
<node name="Mu-like prophage host-nuclease inhibitor protein Gam"><magnitude>
<val> 4.031579</val>
</magnitude>
</node>
<node name="Mu-like prophage major head subunit gpT"><magnitude>
<val> 24.519178</val>
</magnitude>
</node>
<node name="Mu-like prophage protein"><magnitude>
<val> 164.746883</val>
</magnitude>
</node>
<node name="Mu-like prophage protein Com"><magnitude>
<val> 28.72395</val>
</magnitude>
</node>
<node name="Mu-like prophage protein gp37"><magnitude>
<val> 11.415103</val>
</magnitude>
</node>
<node name="Mu-like prophage protein gpG"><magnitude>
<val> 76.037979</val>
</magnitude>
</node>
<node name="Mu-like prophage tail protein gpP"><magnitude>
<val> 49.025068</val>
</magnitude>
</node>
<node name="Mu-like prophage tail sheath protein gpL"><magnitude>
<val> 196.870762</val>
</magnitude>
</node>
<node name="Multimeric flavodoxin WrbA"><magnitude>
<val> 1219.697082</val>
</magnitude>
</node>
<node name="N-terminal domain of molybdenum-binding protein"><magnitude>
<val> 136.107884</val>
</magnitude>
</node>
<node name="NCAIR mutase (PurE)-related proteins"><magnitude>
<val> 260.050681</val>
</magnitude>
</node>
<node name="Na+-dependent transporters of the SNF family"><magnitude>
<val> 185.955515</val>
</magnitude>
</node>
<node name="Nuclear protein with HMG-like acidic region"><magnitude>
<val> 17.08062</val>
</magnitude>
</node>
<node name="Nucleoid-associated protein"><magnitude>
<val> 39.562882</val>
</magnitude>
</node>
<node name="P2-like prophage tail protein X"><magnitude>
<val> 15.49055</val>
</magnitude>
</node>
<node name="PAP2 (acid phosphatase) superfamily protein"><magnitude>
<val> 48.158993</val>
</magnitude>
</node>
<node name="PHD zinc finger-containing protein"><magnitude>
<val> 109.736572</val>
</magnitude>
</node>
<node name="Paraquat-inducible protein B"><magnitude>
<val> 119.726901</val>
</magnitude>
</node>
<node name="Patatin"><magnitude>
<val> 25.777992</val>
</magnitude>
</node>
<node name="Penicillin-binding protein-related factor A, putative recombinase"><magnitude>
<val> 15.079691</val>
</magnitude>
</node>
<node name="Permeases"><magnitude>
<val> 230.990628</val>
</magnitude>
</node>
<node name="Permeases of the major facilitator superfamily"><magnitude>
<val> 112.26594</val>
</magnitude>
</node>
<node name="Phage P2 baseplate assembly protein gpV"><magnitude>
<val> 16.913729</val>
</magnitude>
</node>
<node name="Phage baseplate assembly protein W"><magnitude>
<val> 123.438429</val>
</magnitude>
</node>
<node name="Phage envelope protein"><magnitude>
<val> 8.960576</val>
</magnitude>
</node>
<node name="Phage head maturation protease"><magnitude>
<val> 58.8789</val>
</magnitude>
</node>
<node name="Phage protein D"><magnitude>
<val> 121.582767</val>
</magnitude>
</node>
<node name="Phage protein U"><magnitude>
<val> 45.870092</val>
</magnitude>
</node>
<node name="Phage tail sheath protein FI"><magnitude>
<val> 163.000365</val>
</magnitude>
</node>
<node name="Phage tail tube protein FII"><magnitude>
<val> 22.315137</val>
</magnitude>
</node>
<node name="Phage terminase large subunit"><magnitude>
<val> 168.970096</val>
</magnitude>
</node>
<node name="Phage terminase-like protein, large subunit"><magnitude>
<val> 106.213621</val>
</magnitude>
</node>
<node name="Phage-related baseplate assembly protein"><magnitude>
<val> 12.625587</val>
</magnitude>
</node>
<node name="Phage-related holin (Lysis protein)"><magnitude>
<val> 23.866451</val>
</magnitude>
</node>
<node name="Phage-related lysozyme (muraminidase)"><magnitude>
<val> 93.165792</val>
</magnitude>
</node>
<node name="Phage-related replication protein"><magnitude>
<val> 21.943449</val>
</magnitude>
</node>
<node name="Phage-related tail fibre protein"><magnitude>
<val> 74.299223</val>
</magnitude>
</node>
<node name="Phage-related terminase"><magnitude>
<val> 48.351663</val>
</magnitude>
</node>
<node name="PhoPQ-activated pathogenicity-related protein"><magnitude>
<val> 56.963273</val>
</magnitude>
</node>
<node name="Phospholipid-binding protein"><magnitude>
<val> 105.685065</val>
</magnitude>
</node>
<node name="Pirin-related protein"><magnitude>
<val> 102.272949</val>
</magnitude>
</node>
<node name="Plasmid maintenance system antidote protein"><magnitude>
<val> 210.016208</val>
</magnitude>
</node>
<node name="Plasmid maintenance system killer protein"><magnitude>
<val> 40.761921</val>
</magnitude>
</node>
<node name="Plasmid stabilization system protein"><magnitude>
<val> 200.620588</val>
</magnitude>
</node>
<node name="Post-segregation antitoxin (ccd killing mechanism protein) encoded by the F plasmid"><magnitude>
<val> 57.832094</val>
</magnitude>
</node>
<node name="Predicted ABC-type sugar transport system, permease component"><magnitude>
<val> 15.17648</val>
</magnitude>
</node>
<node name="Predicted ATP-binding protein involved in virulence"><magnitude>
<val> 38.740166</val>
</magnitude>
</node>
<node name="Predicted ATP-dependent carboligase related to biotin carboxylase"><magnitude>
<val> 65.82071</val>
</magnitude>
</node>
<node name="Predicted ATP-grasp enzyme"><magnitude>
<val> 93.495343</val>
</magnitude>
</node>
<node name="Predicted ATP-utilizing enzyme (ATP-grasp superfamily)"><magnitude>
<val> 92.230872</val>
</magnitude>
</node>
<node name="Predicted ATPase"><magnitude>
<val> 1460.479092</val>
</magnitude>
</node>
<node name="Predicted ATPase (AAA+ superfamily)"><magnitude>
<val> 1438.338873</val>
</magnitude>
</node>
<node name="Predicted ATPase (PP-loop superfamily)"><magnitude>
<val> 51.860784</val>
</magnitude>
</node>
<node name="Predicted ATPase of the ABC class"><magnitude>
<val> 166.27557</val>
</magnitude>
</node>
<node name="Predicted ATPase or kinase"><magnitude>
<val> 112.126329</val>
</magnitude>
</node>
<node name="Predicted ATPase, RNase L inhibitor (RLI) homolog"><magnitude>
<val> 123.222778</val>
</magnitude>
</node>
<node name="Predicted ATPases"><magnitude>
<val> 45.277187</val>
</magnitude>
</node>
<node name="Predicted ATPases of PP-loop superfamily"><magnitude>
<val> 98.48188</val>
</magnitude>
</node>
<node name="Predicted CDP-diglyceride synthetase/phosphatidate cytidylyltransferase"><magnitude>
<val> 25.145089</val>
</magnitude>
</node>
<node name="Predicted CoA-binding protein"><magnitude>
<val> 143.52052</val>
</magnitude>
</node>
<node name="Predicted DNA binding protein"><magnitude>
<val> 128.149225</val>
</magnitude>
</node>
<node name="Predicted DNA-binding protein containing PIN domain"><magnitude>
<val> 81.086831</val>
</magnitude>
</node>
<node name="Predicted DNA-binding protein containing a Zn-ribbon domain"><magnitude>
<val> 404.316988</val>
</magnitude>
</node>
<node name="Predicted DNA-binding protein with PD1-like DNA-binding motif"><magnitude>
<val> 184.916588</val>
</magnitude>
</node>
<node name="Predicted DNA-binding protein with an HTH domain"><magnitude>
<val> 74.252288</val>
</magnitude>
</node>
<node name="Predicted DNA-binding protein with the Helix-hairpin-helix motif"><magnitude>
<val> 177.857203</val>
</magnitude>
</node>
<node name="Predicted DNA-binding proteins"><magnitude>
<val> 172.509493</val>
</magnitude>
</node>
<node name="Predicted Fe-S oxidoreductase"><magnitude>
<val> 491.055404</val>
</magnitude>
</node>
<node name="Predicted Fe-S oxidoreductases"><magnitude>
<val> 1576.548665</val>
</magnitude>
</node>
<node name="Predicted Fe-S protein"><magnitude>
<val> 104.447212</val>
</magnitude>
</node>
<node name="Predicted Fe-S-cluster oxidoreductase"><magnitude>
<val> 401.376012</val>
</magnitude>
</node>
<node name="Predicted Fe-S-cluster redox enzyme"><magnitude>
<val> 156.146253</val>
</magnitude>
</node>
<node name="Predicted GTPase"><magnitude>
<val> 1515.299835</val>
</magnitude>
</node>
<node name="Predicted GTPase or GTP-binding protein"><magnitude>
<val> 12.531199</val>
</magnitude>
</node>
<node name="Predicted GTPases"><magnitude>
<val> 612.238153</val>
</magnitude>
</node>
<node name="Predicted HD phosphohydrolase"><magnitude>
<val> 31.963742</val>
</magnitude>
</node>
<node name="Predicted HD superfamily hydrolase"><magnitude>
<val> 416.13297</val>
</magnitude>
</node>
<node name="Predicted HD-superfamily hydrolase"><magnitude>
<val> 229.32313</val>
</magnitude>
</node>
<node name="Predicted ICC-like phosphoesterases"><magnitude>
<val> 93.511954</val>
</magnitude>
</node>
<node name="Predicted NAD/FAD-binding protein"><magnitude>
<val> 110.61823</val>
</magnitude>
</node>
<node name="Predicted NAD/FAD-dependent oxidoreductase"><magnitude>
<val> 30.012164</val>
</magnitude>
</node>
<node name="Predicted NTP binding protein (contains STAS domain)"><magnitude>
<val> 6.335864</val>
</magnitude>
</node>
<node name="Predicted Na+-dependent transporter"><magnitude>
<val> 127.81818</val>
</magnitude>
</node>
<node name="Predicted Na+/dicarboxylate symporter"><magnitude>
<val> 73.967506</val>
</magnitude>
</node>
<node name="Predicted O-methyltransferase"><magnitude>
<val> 387.582432</val>
</magnitude>
</node>
<node name="Predicted P-loop ATPase"><magnitude>
<val> 112.9636</val>
</magnitude>
</node>
<node name="Predicted P-loop ATPase and inactivated derivatives"><magnitude>
<val> 212.829471</val>
</magnitude>
</node>
<node name="Predicted P-loop ATPase fused to an acetyltransferase"><magnitude>
<val> 63.006413</val>
</magnitude>
</node>
<node name="Predicted P-loop ATPase/GTPase"><magnitude>
<val> 18.093544</val>
</magnitude>
</node>
<node name="Predicted P-loop-containing kinase"><magnitude>
<val> 210.915107</val>
</magnitude>
</node>
<node name="Predicted PP-loop superfamily ATPase"><magnitude>
<val> 245.922675</val>
</magnitude>
</node>
<node name="Predicted RNA methylase"><magnitude>
<val> 58.187669</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein"><magnitude>
<val> 211.526423</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein (contains KH domain)"><magnitude>
<val> 241.054577</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein (contains KH domains)"><magnitude>
<val> 104.208347</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein containing a C-terminal EMAP domain"><magnitude>
<val> 101.069721</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein containing a PIN domain"><magnitude>
<val> 26.465264</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein, contains THUMP domain"><magnitude>
<val> 67.266385</val>
</magnitude>
</node>
<node name="Predicted RNA-binding protein, contains TRAM domain"><magnitude>
<val> 268.329414</val>
</magnitude>
</node>
<node name="Predicted RNA-binding proteins"><magnitude>
<val> 67.031884</val>
</magnitude>
</node>
<node name="Predicted Rossmann fold nucleotide-binding protein"><magnitude>
<val> 258.912347</val>
</magnitude>
</node>
<node name="Predicted S-adenosylmethionine-dependent methyltransferase"><magnitude>
<val> 121.118245</val>
</magnitude>
</node>
<node name="Predicted SAM-dependent methyltransferase"><magnitude>
<val> 88.377723</val>
</magnitude>
</node>
<node name="Predicted SAM-dependent methyltransferases"><magnitude>
<val> 237.620737</val>
</magnitude>
</node>
<node name="Predicted TIM-barrel enzyme"><magnitude>
<val> 46.938112</val>
</magnitude>
</node>
<node name="Predicted TIM-barrel enzyme, possibly a dioxygenase"><magnitude>
<val> 62.841996</val>
</magnitude>
</node>
<node name="Predicted Zn-dependent hydrolases of the beta-lactamase fold"><magnitude>
<val> 249.132137</val>
</magnitude>
</node>
<node name="Predicted Zn-dependent peptidases"><magnitude>
<val> 610.455975</val>
</magnitude>
</node>
<node name="Predicted Zn-dependent peptidases, insulinase-like"><magnitude>
<val> 253.373447</val>
</magnitude>
</node>
<node name="Predicted Zn-dependent protease"><magnitude>
<val> 64.699702</val>
</magnitude>
</node>
<node name="Predicted Zn-dependent proteases"><magnitude>
<val> 181.936202</val>
</magnitude>
</node>
<node name="Predicted Zn-dependent proteases and their inactivated homologs"><magnitude>
<val> 436.744834</val>
</magnitude>
</node>
<node name="Predicted acetyltransferase"><magnitude>
<val> 325.796692</val>
</magnitude>
</node>
<node name="Predicted acetyltransferase involved in intracellular survival and related acetyltransferases"><magnitude>
<val> 92.366522</val>
</magnitude>
</node>
<node name="Predicted acetyltransferases and hydrolases with the alpha/beta hydrolase fold"><magnitude>
<val> 131.354031</val>
</magnitude>
</node>
<node name="Predicted acid phosphatase"><magnitude>
<val> 232.037344</val>
</magnitude>
</node>
<node name="Predicted acyl esterases"><magnitude>
<val> 149.172007</val>
</magnitude>
</node>
<node name="Predicted acyltransferase"><magnitude>
<val> 159.460349</val>
</magnitude>
</node>
<node name="Predicted alpha/beta hydrolase"><magnitude>
<val> 19.834879</val>
</magnitude>
</node>
<node name="Predicted alternative tryptophan synthase beta-subunit (paralog of TrpB)"><magnitude>
<val> 261.817102</val>
</magnitude>
</node>
<node name="Predicted amidohydrolase"><magnitude>
<val> 588.805937</val>
</magnitude>
</node>
<node name="Predicted amidophosphoribosyltransferases"><magnitude>
<val> 355.293086</val>
</magnitude>
</node>
<node name="Predicted aminoglycoside phosphotransferase"><magnitude>
<val> 71.904071</val>
</magnitude>
</node>
<node name="Predicted aminomethyltransferase related to GcvT"><magnitude>
<val> 61.680868</val>
</magnitude>
</node>
<node name="Predicted aminopeptidase"><magnitude>
<val> 61.797784</val>
</magnitude>
</node>
<node name="Predicted aminopeptidase, Iap family"><magnitude>
<val> 67.722894</val>
</magnitude>
</node>
<node name="Predicted aminopeptidases"><magnitude>
<val> 146.67931</val>
</magnitude>
</node>
<node name="Predicted archaeal kinase"><magnitude>
<val> 105.910752</val>
</magnitude>
</node>
<node name="Predicted archaeal kinase (sugar kinase superfamily)"><magnitude>
<val> 100.835904</val>
</magnitude>
</node>
<node name="Predicted archaeal methyltransferase"><magnitude>
<val> 48.513599</val>
</magnitude>
</node>
<node name="Predicted archaeal sugar kinases"><magnitude>
<val> 116.798021</val>
</magnitude>
</node>
<node name="Predicted aspartyl protease"><magnitude>
<val> 31.363121</val>
</magnitude>
</node>
<node name="Predicted beta-xylosidase"><magnitude>
<val> 39.657432</val>
</magnitude>
</node>
<node name="Predicted cation transporter"><magnitude>
<val> 153.004418</val>
</magnitude>
</node>
<node name="Predicted chitinase"><magnitude>
<val> 58.496437</val>
</magnitude>
</node>
<node name="Predicted choloylglycine hydrolase"><magnitude>
<val> 83.697778</val>
</magnitude>
</node>
<node name="Predicted cobalamin binding protein"><magnitude>
<val> 177.872906</val>
</magnitude>
</node>
<node name="Predicted deacetylase"><magnitude>
<val> 16.691938</val>
</magnitude>
</node>
<node name="Predicted deacylase"><magnitude>
<val> 191.206857</val>
</magnitude>
</node>
<node name="Predicted dehydrogenase"><magnitude>
<val> 206.706529</val>
</magnitude>
</node>
<node name="Predicted dehydrogenase related to H2-forming N5,N10-methylenetetrahydromethanopterin dehydrogenase"><magnitude>
<val> 36.934053</val>
</magnitude>
</node>
<node name="Predicted dehydrogenases and related proteins"><magnitude>
<val> 772.643755</val>
</magnitude>
</node>
<node name="Predicted dienelactone hydrolase"><magnitude>
<val> 53.670562</val>
</magnitude>
</node>
<node name="Predicted dinucleotide-binding enzymes"><magnitude>
<val> 81.245061</val>
</magnitude>
</node>
<node name="Predicted dinucleotide-utilizing enzyme"><magnitude>
<val> 104.741809</val>
</magnitude>
</node>
<node name="Predicted dinucleotide-utilizing enzyme of the ThiF/HesA family"><magnitude>
<val> 14.908885</val>
</magnitude>
</node>
<node name="Predicted dioxygenase"><magnitude>
<val> 204.708498</val>
</magnitude>
</node>
<node name="Predicted dioxygenase of extradiol dioxygenase family"><magnitude>
<val> 9.439944</val>
</magnitude>
</node>
<node name="Predicted double-glycine peptidase"><magnitude>
<val> 77.587445</val>
</magnitude>
</node>
<node name="Predicted double-stranded RNA/RNA-DNA hybrid binding protein"><magnitude>
<val> 34.790096</val>
</magnitude>
</node>
<node name="Predicted drug exporters of the RND superfamily"><magnitude>
<val> 77.797107</val>
</magnitude>
</node>
<node name="Predicted enzyme of the cupin superfamily"><magnitude>
<val> 89.721005</val>
</magnitude>
</node>
<node name="Predicted enzyme related to lactoylglutathione lyase"><magnitude>
<val> 244.516714</val>
</magnitude>
</node>
<node name="Predicted enzyme with a TIM-barrel fold"><magnitude>
<val> 162.349341</val>
</magnitude>
</node>
<node name="Predicted epimerase, PhzC/PhzF homolog"><magnitude>
<val> 101.854635</val>
</magnitude>
</node>
<node name="Predicted ester cyclase"><magnitude>
<val> 34.488784</val>
</magnitude>
</node>
<node name="Predicted esterase"><magnitude>
<val> 110.919662</val>
</magnitude>
</node>
<node name="Predicted esterase of the alpha-beta hydrolase superfamily"><magnitude>
<val> 480.201707</val>
</magnitude>
</node>
<node name="Predicted esterase of the alpha/beta hydrolase fold"><magnitude>
<val> 30.174115</val>
</magnitude>
</node>
<node name="Predicted exporter"><magnitude>
<val> 99.5812</val>
</magnitude>
</node>
<node name="Predicted exporters of the RND superfamily"><magnitude>
<val> 349.350969</val>
</magnitude>
</node>
<node name="Predicted extracellular nuclease"><magnitude>
<val> 111.726478</val>
</magnitude>
</node>
<node name="Predicted flavin-nucleotide-binding protein"><magnitude>
<val> 211.219429</val>
</magnitude>
</node>
<node name="Predicted flavin-nucleotide-binding protein structurally related to pyridoxine 5'-phosphate oxidase"><magnitude>
<val> 149.786516</val>
</magnitude>
</node>
<node name="Predicted flavoprotein"><magnitude>
<val> 72.296125</val>
</magnitude>
</node>
<node name="Predicted flavoproteins"><magnitude>
<val> 197.77186</val>
</magnitude>
</node>
<node name="Predicted glutamine amidotransferase"><magnitude>
<val> 169.850536</val>
</magnitude>
</node>
<node name="Predicted glutamine amidotransferases"><magnitude>
<val> 105.421914</val>
</magnitude>
</node>
<node name="Predicted glycosyl hydrolase"><magnitude>
<val> 136.796687</val>
</magnitude>
</node>
<node name="Predicted glycosyl transferase"><magnitude>
<val> 54.206764</val>
</magnitude>
</node>
<node name="Predicted glycosyltransferases"><magnitude>
<val> 954.482106</val>
</magnitude>
</node>
<node name="Predicted helicase"><magnitude>
<val> 146.606555</val>
</magnitude>
</node>
<node name="Predicted helicases"><magnitude>
<val> 394.627534</val>
</magnitude>
</node>
<node name="Predicted heme/steroid binding protein"><magnitude>
<val> 26.258607</val>
</magnitude>
</node>
<node name="Predicted hydrocarbon binding protein (contains V4R domain)"><magnitude>
<val> 331.0824</val>
</magnitude>
</node>
<node name="Predicted hydrolase (HAD superfamily)"><magnitude>
<val> 567.303032</val>
</magnitude>
</node>
<node name="Predicted hydrolase (HD superfamily)"><magnitude>
<val> 87.447583</val>
</magnitude>
</node>
<node name="Predicted hydrolase (metallo-beta-lactamase superfamily)"><magnitude>
<val> 362.737276</val>
</magnitude>
</node>
<node name="Predicted hydrolase of alkaline phosphatase superfamily"><magnitude>
<val> 169.220651</val>
</magnitude>
</node>
<node name="Predicted hydrolase of the HAD superfamily"><magnitude>
<val> 47.678956</val>
</magnitude>
</node>
<node name="Predicted hydrolase of the HD superfamily (permuted catalytic motifs)"><magnitude>
<val> 108.470845</val>
</magnitude>
</node>
<node name="Predicted hydrolase of the alpha/beta superfamily"><magnitude>
<val> 160.926682</val>
</magnitude>
</node>
<node name="Predicted hydrolase of the alpha/beta-hydrolase fold"><magnitude>
<val> 99.714233</val>
</magnitude>
</node>
<node name="Predicted hydrolase of the metallo-beta-lactamase superfamily"><magnitude>
<val> 440.893789</val>
</magnitude>
</node>
<node name="Predicted hydrolases of HD superfamily"><magnitude>
<val> 124.574894</val>
</magnitude>
</node>
<node name="Predicted hydrolases of the HAD superfamily"><magnitude>
<val> 351.266136</val>
</magnitude>
</node>
<node name="Predicted hydrolases or acyltransferases (alpha/beta hydrolase superfamily)"><magnitude>
<val> 934.458688</val>
</magnitude>
</node>
<node name="Predicted integral membrane metal-binding protein"><magnitude>
<val> 24.693696</val>
</magnitude>
</node>
<node name="Predicted kinase"><magnitude>
<val> 203.616686</val>
</magnitude>
</node>
<node name="Predicted kinase related to dihydroxyacetone kinase"><magnitude>
<val> 99.266936</val>
</magnitude>
</node>
<node name="Predicted kinase related to galactokinase and mevalonate kinase"><magnitude>
<val> 121.495385</val>
</magnitude>
</node>
<node name="Predicted lactoylglutathione lyase"><magnitude>
<val> 55.448794</val>
</magnitude>
</node>
<node name="Predicted membrane metal-binding protein"><magnitude>
<val> 96.795692</val>
</magnitude>
</node>
<node name="Predicted membrane protein"><magnitude>
<val> 54.793579</val>
</magnitude>
</node>
<node name="Predicted membrane protein, hemolysin III homolog"><magnitude>
<val> 117.899393</val>
</magnitude>
</node>
<node name="Predicted membrane protein, putative toxin regulator"><magnitude>
<val> 27.27381</val>
</magnitude>
</node>
<node name="Predicted membrane-associated HD superfamily hydrolase"><magnitude>
<val> 555.878766</val>
</magnitude>
</node>
<node name="Predicted membrane-associated, metal-dependent hydrolase"><magnitude>
<val> 214.334593</val>
</magnitude>
</node>
<node name="Predicted membrane-bound metal-dependent hydrolases"><magnitude>
<val> 154.346579</val>
</magnitude>
</node>
<node name="Predicted metal-binding protein"><magnitude>
<val> 200.772887</val>
</magnitude>
</node>
<node name="Predicted metal-binding protein related to the C-terminal domain of SecA"><magnitude>
<val> 81.051736</val>
</magnitude>
</node>
<node name="Predicted metal-binding, possibly nucleic acid-binding protein"><magnitude>
<val> 157.102362</val>
</magnitude>
</node>
<node name="Predicted metal-dependent RNase, consists of a metallo-beta-lactamase domain and an RNA-binding KH domain"><magnitude>
<val> 195.521879</val>
</magnitude>
</node>
<node name="Predicted metal-dependent enzyme"><magnitude>
<val> 107.139797</val>
</magnitude>
</node>
<node name="Predicted metal-dependent enzyme of the double-stranded beta helix superfamily"><magnitude>
<val> 11.688107</val>
</magnitude>
</node>
<node name="Predicted metal-dependent hydrolase"><magnitude>
<val> 611.572696</val>
</magnitude>
</node>
<node name="Predicted metal-dependent hydrolase (urease superfamily)"><magnitude>
<val> 154.472583</val>
</magnitude>
</node>
<node name="Predicted metal-dependent hydrolase of the TIM-barrel fold"><magnitude>
<val> 307.886535</val>
</magnitude>
</node>
<node name="Predicted metal-dependent hydrolase with the TIM-barrel fold"><magnitude>
<val> 586.119848</val>
</magnitude>
</node>
<node name="Predicted metal-dependent hydrolases related to alanyl-tRNA synthetase HxxxH domain"><magnitude>
<val> 221.461575</val>
</magnitude>
</node>
<node name="Predicted metal-dependent hydrolases with the TIM-barrel fold"><magnitude>
<val> 72.96874</val>
</magnitude>
</node>
<node name="Predicted metal-dependent membrane protease"><magnitude>
<val> 490.054278</val>
</magnitude>
</node>
<node name="Predicted metal-dependent phosphoesterases (PHP family)"><magnitude>
<val> 386.216204</val>
</magnitude>
</node>
<node name="Predicted metal-dependent protease of the PAD1/JAB1 superfamily"><magnitude>
<val> 103.674973</val>
</magnitude>
</node>
<node name="Predicted metal-sulfur cluster biosynthetic enzyme"><magnitude>
<val> 69.172332</val>
</magnitude>
</node>
<node name="Predicted metallopeptidase"><magnitude>
<val> 53.895556</val>
</magnitude>
</node>
<node name="Predicted metalloprotease"><magnitude>
<val> 151.353299</val>
</magnitude>
</node>
<node name="Predicted methyltransferase"><magnitude>
<val> 389.675355</val>
</magnitude>
</node>
<node name="Predicted methyltransferase (contains TPR repeat)"><magnitude>
<val> 105.965682</val>
</magnitude>
</node>
<node name="Predicted methyltransferases"><magnitude>
<val> 365.370604</val>
</magnitude>
</node>
<node name="Predicted nuclease (RecB family)"><magnitude>
<val> 44.325911</val>
</magnitude>
</node>
<node name="Predicted nucleic acid binding protein containing the AN1-type Zn-finger"><magnitude>
<val> 14.756931</val>
</magnitude>
</node>
<node name="Predicted nucleic acid-binding protein, consists of a PIN domain and a Zn-ribbon module"><magnitude>
<val> 212.216296</val>
</magnitude>
</node>
<node name="Predicted nucleic acid-binding protein, contains PIN domain"><magnitude>
<val> 851.328445</val>
</magnitude>
</node>
<node name="Predicted nucleic-acid-binding protein (contains the HHH domain)"><magnitude>
<val> 151.570203</val>
</magnitude>
</node>
<node name="Predicted nucleic-acid-binding protein containing a Zn-ribbon"><magnitude>
<val> 89.438867</val>
</magnitude>
</node>
<node name="Predicted nucleic-acid-binding protein containing a Zn-ribbon domain"><magnitude>
<val> 14.327463</val>
</magnitude>
</node>
<node name="Predicted nucleic-acid-binding protein, contains PIN domain"><magnitude>
<val> 30.516102</val>
</magnitude>
</node>
<node name="Predicted nucleoside-diphosphate sugar epimerase"><magnitude>
<val> 60.223261</val>
</magnitude>
</node>
<node name="Predicted nucleotide-binding protein"><magnitude>
<val> 11.818332</val>
</magnitude>
</node>
<node name="Predicted nucleotide-utilizing enzyme related to molybdopterin-biosynthesis enzyme MoeA"><magnitude>
<val> 138.484644</val>
</magnitude>
</node>
<node name="Predicted nucleotidyltransferase"><magnitude>
<val> 525.503584</val>
</magnitude>
</node>
<node name="Predicted nucleotidyltransferases"><magnitude>
<val> 660.638621</val>
</magnitude>
</node>
<node name="Predicted oxidoreductase"><magnitude>
<val> 108.383372</val>
</magnitude>
</node>
<node name="Predicted oxidoreductase related to nitroreductase"><magnitude>
<val> 55.498392</val>
</magnitude>
</node>
<node name="Predicted oxidoreductases of the aldo/keto reductase family"><magnitude>
<val> 535.626548</val>
</magnitude>
</node>
<node name="Predicted peptidase"><magnitude>
<val> 106.576354</val>
</magnitude>
</node>
<node name="Predicted periplasmic lipoprotein"><magnitude>
<val> 193.230508</val>
</magnitude>
</node>
<node name="Predicted periplasmic or secreted lipoprotein"><magnitude>
<val> 138.506263</val>
</magnitude>
</node>
<node name="Predicted periplasmic or secreted protein"><magnitude>
<val> 65.343035</val>
</magnitude>
</node>
<node name="Predicted periplasmic protein"><magnitude>
<val> 28.260599</val>
</magnitude>
</node>
<node name="Predicted periplasmic solute-binding protein"><magnitude>
<val> 569.240204</val>
</magnitude>
</node>
<node name="Predicted permease"><magnitude>
<val> 1092.913921</val>
</magnitude>
</node>
<node name="Predicted permease, DMT superfamily"><magnitude>
<val> 63.490727</val>
</magnitude>
</node>
<node name="Predicted permeases"><magnitude>
<val> 1642.836519</val>
</magnitude>
</node>
<node name="Predicted peroxiredoxins"><magnitude>
<val> 54.940168</val>
</magnitude>
</node>
<node name="Predicted phage phi-C31 gp36 major capsid-like protein"><magnitude>
<val> 110.132507</val>
</magnitude>
</node>
<node name="Predicted phosphatase"><magnitude>
<val> 134.604825</val>
</magnitude>
</node>
<node name="Predicted phosphatase homologous to the C-terminal domain of histone macroH2A1"><magnitude>
<val> 197.295268</val>
</magnitude>
</node>
<node name="Predicted phosphatase/phosphohexomutase"><magnitude>
<val> 364.995491</val>
</magnitude>
</node>
<node name="Predicted phosphatases"><magnitude>
<val> 457.962438</val>
</magnitude>
</node>
<node name="Predicted phosphate-binding enzymes, TIM-barrel fold"><magnitude>
<val> 76.118553</val>
</magnitude>
</node>
<node name="Predicted phosphoadenosine phosphosulfate sulfotransferase"><magnitude>
<val> 65.693293</val>
</magnitude>
</node>
<node name="Predicted phosphoesterase"><magnitude>
<val> 332.294141</val>
</magnitude>
</node>
<node name="Predicted phosphoesterase (MutT family)"><magnitude>
<val> .992509</val>
</magnitude>
</node>
<node name="Predicted phosphoesterase or phosphohydrolase"><magnitude>
<val> 9.825899</val>
</magnitude>
</node>
<node name="Predicted phosphoesterases, related to the Icc protein"><magnitude>
<val> 85.359378</val>
</magnitude>
</node>
<node name="Predicted phosphohydrolase"><magnitude>
<val> 27.537806</val>
</magnitude>
</node>
<node name="Predicted phosphohydrolase (DHH superfamily)"><magnitude>
<val> 68.159273</val>
</magnitude>
</node>
<node name="Predicted phosphohydrolases"><magnitude>
<val> 653.285174</val>
</magnitude>
</node>
<node name="Predicted phosphoribosyltransferases"><magnitude>
<val> 121.996744</val>
</magnitude>
</node>
<node name="Predicted phosphotransacetylase"><magnitude>
<val> 96.472816</val>
</magnitude>
</node>
<node name="Predicted phosphotransferase related to Ser/Thr protein kinases"><magnitude>
<val> 66.091026</val>
</magnitude>
</node>
<node name="Predicted protease of the Abi (CAAX) family"><magnitude>
<val> 197.157362</val>
</magnitude>
</node>
<node name="Predicted protease with the C-terminal PDZ domain"><magnitude>
<val> 9.428169</val>
</magnitude>
</node>
<node name="Predicted protein tyrosine phosphatase"><magnitude>
<val> 10.271985</val>
</magnitude>
</node>
<node name="Predicted pyrophosphatase"><magnitude>
<val> 65.590474</val>
</magnitude>
</node>
<node name="Predicted regulator of amino acid metabolism, contains ACT domain"><magnitude>
<val> 71.856185</val>
</magnitude>
</node>
<node name="Predicted regulatory domain of a methyltransferase"><magnitude>
<val> 43.089456</val>
</magnitude>
</node>
<node name="Predicted ring-cleavage extradiol dioxygenase"><magnitude>
<val> 29.165655</val>
</magnitude>
</node>
<node name="Predicted secreted acid phosphatase"><magnitude>
<val> 9.026195</val>
</magnitude>
</node>
<node name="Predicted secreted hydrolase"><magnitude>
<val> 177.98511</val>
</magnitude>
</node>
<node name="Predicted small molecule binding protein (contains 3H domain)"><magnitude>
<val> 80.28454</val>
</magnitude>
</node>
<node name="Predicted solute binding protein"><magnitude>
<val> 576.895365</val>
</magnitude>
</node>
<node name="Predicted spermidine synthase with an N-terminal membrane domain"><magnitude>
<val> 118.82674</val>
</magnitude>
</node>
<node name="Predicted sugar phosphate isomerase"><magnitude>
<val> 72.962358</val>
</magnitude>
</node>
<node name="Predicted sulfurtransferase"><magnitude>
<val> 13.273845</val>
</magnitude>
</node>
<node name="Predicted symporter"><magnitude>
<val> 433.002384</val>
</magnitude>
</node>
<node name="Predicted thiamine-pyrophosphate-binding protein"><magnitude>
<val> 69.29961</val>
</magnitude>
</node>
<node name="Predicted thioesterase"><magnitude>
<val> 419.370476</val>
</magnitude>
</node>
<node name="Predicted transcriptional regulator"><magnitude>
<val> 87.564927</val>
</magnitude>
</node>
<node name="Predicted transglutaminase-like proteases"><magnitude>
<val> 12.116825</val>
</magnitude>
</node>
<node name="Predicted transporter"><magnitude>
<val> 95.543719</val>
</magnitude>
</node>
<node name="Predicted transporter component"><magnitude>
<val> 114.410332</val>
</magnitude>
</node>
<node name="Predicted unsaturated glucuronyl hydrolase involved in regulation of bacterial surface properties, and related proteins"><magnitude>
<val> 553.631011</val>
</magnitude>
</node>
<node name="Predicted unusual protein kinase"><magnitude>
<val> 255.1082</val>
</magnitude>
</node>
<node name="Prophage maintenance system killer protein"><magnitude>
<val> 54.144427</val>
</magnitude>
</node>
<node name="Protein affecting phage T7 exclusion by the F plasmid"><magnitude>
<val> 64.672372</val>
</magnitude>
</node>
<node name="Protein containing a metal-binding domain shared with formylmethanofuran dehydrogenase subunit E"><magnitude>
<val> 251.484191</val>
</magnitude>
</node>
<node name="Protein containing tetrapyrrole methyltransferase domain and MazG-like (predicted pyrophosphatase) domain"><magnitude>
<val> 239.038628</val>
</magnitude>
</node>
<node name="Protein containing von Willebrand factor type A (vWA) domain"><magnitude>
<val> 66.017809</val>
</magnitude>
</node>
<node name="Protein distantly related to bacterial ferritins"><magnitude>
<val> 61.639648</val>
</magnitude>
</node>
<node name="Protein involved in catabolism of external DNA"><magnitude>
<val> 63.115875</val>
</magnitude>
</node>
<node name="Protein involved in propagation of M2 dsRNA satellite of L-A virus"><magnitude>
<val> 16.642108</val>
</magnitude>
</node>
<node name="Protein involved in sex pheromone biosynthesis"><magnitude>
<val> 26.694039</val>
</magnitude>
</node>
<node name="Protein of avirulence locus involved in temperature-dependent protein secretion"><magnitude>
<val> 26.956989</val>
</magnitude>
</node>
<node name="Protein related to penicillin acylase"><magnitude>
<val> 98.737352</val>
</magnitude>
</node>
<node name="Protein tyrosine phosphatase-like protein (contains Pro instead of catalytic Arg)"><magnitude>
<val> 30.367188</val>
</magnitude>
</node>
<node name="Proteins containing SET domain"><magnitude>
<val> 20.163545</val>
</magnitude>
</node>
<node name="Putative GTPases (G3E family)"><magnitude>
<val> 130.943429</val>
</magnitude>
</node>
<node name="Putative NADH-flavin reductase"><magnitude>
<val> 38.253107</val>
</magnitude>
</node>
<node name="Putative NADP-dependent oxidoreductases"><magnitude>
<val> 64.269706</val>
</magnitude>
</node>
<node name="Putative NADPH-quinone reductase (modulator of drug activity B)"><magnitude>
<val> 298.061608</val>
</magnitude>
</node>
<node name="Putative Zn-dependent protease"><magnitude>
<val> 31.936328</val>
</magnitude>
</node>
<node name="Putative Zn-dependent protease, contains TPR repeats"><magnitude>
<val> 717.82051</val>
</magnitude>
</node>
<node name="Putative ammonia monooxygenase"><magnitude>
<val> 74.244557</val>
</magnitude>
</node>
<node name="Putative cation transport regulator"><magnitude>
<val> 27.956951</val>
</magnitude>
</node>
<node name="Putative effector of murein hydrolase LrgA"><magnitude>
<val> 66.40556</val>
</magnitude>
</node>
<node name="Putative hemolysin"><magnitude>
<val> 209.522126</val>
</magnitude>
</node>
<node name="Putative homoserine kinase type II (protein kinase fold)"><magnitude>
<val> 147.200315</val>
</magnitude>
</node>
<node name="Putative intracellular protease/amidase"><magnitude>
<val> 321.271476</val>
</magnitude>
</node>
<node name="Putative lipoprotein"><magnitude>
<val> 65.579087</val>
</magnitude>
</node>
<node name="Putative secretion activating protein"><magnitude>
<val> 39.78706</val>
</magnitude>
</node>
<node name="Pyocin large subunit"><magnitude>
<val> 32.140046</val>
</magnitude>
</node>
<node name="RNA-binding proteins (RRM domain)"><magnitude>
<val> 204.071521</val>
</magnitude>
</node>
<node name="Rad5p-binding protein"><magnitude>
<val> 10.022744</val>
</magnitude>
</node>
<node name="SEC7 domain proteins"><magnitude>
<val> 13.11531</val>
</magnitude>
</node>
<node name="SLT domain proteins"><magnitude>
<val> 55.295696</val>
</magnitude>
</node>
<node name="Secreted protein containing C-terminal beta-propeller domain distantly related to WD-40 repeats"><magnitude>
<val> 124.299109</val>
</magnitude>
</node>
<node name="Selenophosphate synthetase-related proteins"><magnitude>
<val> 109.323235</val>
</magnitude>
</node>
<node name="Serine protease inhibitor ecotin"><magnitude>
<val> 11.816334</val>
</magnitude>
</node>
<node name="Serine/threonine protein kinase"><magnitude>
<val> 334.955521</val>
</magnitude>
</node>
<node name="Short-chain alcohol dehydrogenase of unknown specificity"><magnitude>
<val> 132.036177</val>
</magnitude>
</node>
<node name="Short-chain dehydrogenases of various substrate specificities"><magnitude>
<val> 167.669689</val>
</magnitude>
</node>
<node name="Soluble P-type ATPase"><magnitude>
<val> 94.645738</val>
</magnitude>
</node>
<node name="Spore germination protein"><magnitude>
<val> 49.416316</val>
</magnitude>
</node>
<node name="Sporulation control protein"><magnitude>
<val> 41.526938</val>
</magnitude>
</node>
<node name="Stringent starvation protein B"><magnitude>
<val> 4.898681</val>
</magnitude>
</node>
<node name="SufE protein probably involved in Fe-S center assembly"><magnitude>
<val> 77.053021</val>
</magnitude>
</node>
<node name="Sulfite oxidase and related enzymes"><magnitude>
<val> 189.828182</val>
</magnitude>
</node>
<node name="Superfamily I DNA and RNA helicases"><magnitude>
<val> 268.978013</val>
</magnitude>
</node>
<node name="Superfamily II helicase"><magnitude>
<val> 144.713208</val>
</magnitude>
</node>
<node name="Superfamily II helicase, archaea-specific"><magnitude>
<val> 205.285216</val>
</magnitude>
</node>
<node name="Suppressor of G2 allele of skp1 and related proteins"><magnitude>
<val> 112.913866</val>
</magnitude>
</node>
<node name="Surface antigen"><magnitude>
<val> 4.255698</val>
</magnitude>
</node>
<node name="TRAP-type uncharacterized transport system, fused permease components"><magnitude>
<val> 63.987096</val>
</magnitude>
</node>
<node name="TRAP-type uncharacterized transport system, periplasmic component"><magnitude>
<val> 118.657359</val>
</magnitude>
</node>
<node name="Trans-aconitate methyltransferase"><magnitude>
<val> 135.590219</val>
</magnitude>
</node>
<node name="Transthyretin-like protein"><magnitude>
<val> 16.779635</val>
</magnitude>
</node>
<node name="U1-like Zn-finger-containing protein"><magnitude>
<val> 2.280702</val>
</magnitude>
</node>
<node name="Uncharacterized ABC-type transport system, ATPase component"><magnitude>
<val> 21.162845</val>
</magnitude>
</node>
<node name="Uncharacterized ABC-type transport system, periplasmic component/surface lipoprotein"><magnitude>
<val> 153.582573</val>
</magnitude>
</node>
<node name="Uncharacterized ABC-type transport system, permease component"><magnitude>
<val> 215.585832</val>
</magnitude>
</node>
<node name="Uncharacterized ATPase, putative transposase"><magnitude>
<val> 30.29242</val>
</magnitude>
</node>
<node name="Uncharacterized C-terminal domain of topoisomerase IA"><magnitude>
<val> 86.216496</val>
</magnitude>
</node>
<node name="Uncharacterized FAD-dependent dehydrogenases"><magnitude>
<val> 288.771049</val>
</magnitude>
</node>
<node name="Uncharacterized Fe-S center protein"><magnitude>
<val> 390.123484</val>
</magnitude>
</node>
<node name="Uncharacterized Fe-S protein"><magnitude>
<val> 58.441798</val>
</magnitude>
</node>
<node name="Uncharacterized Fe-S protein PflX, homolog of pyruvate formate lyase activating proteins"><magnitude>
<val> 257.919726</val>
</magnitude>
</node>
<node name="Uncharacterized FlgJ-related protein"><magnitude>
<val> 60.111649</val>
</magnitude>
</node>
<node name="Uncharacterized MobA-related protein"><magnitude>
<val> 203.573786</val>
</magnitude>
</node>
<node name="Uncharacterized NAD(FAD)-dependent dehydrogenases"><magnitude>
<val> 373.278885</val>
</magnitude>
</node>
<node name="Uncharacterized Rossmann fold enzyme"><magnitude>
<val> 63.132613</val>
</magnitude>
</node>
<node name="Uncharacterized Zn ribbon-containing protein"><magnitude>
<val> 14.438967</val>
</magnitude>
</node>
<node name="Uncharacterized Zn-finger containing protein"><magnitude>
<val> 18.551917</val>
</magnitude>
</node>
<node name="Uncharacterized anaerobic dehydrogenase"><magnitude>
<val> 816.41382</val>
</magnitude>
</node>
<node name="Uncharacterized archaeal Zn-finger protein"><magnitude>
<val> 243.968239</val>
</magnitude>
</node>
<node name="Uncharacterized archaeal kinase related to aspartokinases, uridylate kinases"><magnitude>
<val> 66.666683</val>
</magnitude>
</node>
<node name="Uncharacterized component of anaerobic dehydrogenases"><magnitude>
<val> 14.496976</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein"><magnitude>
<val> 208.455887</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein (some members contain a von Willebrand factor type A (vWA) domain)"><magnitude>
<val> 275.587067</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein related to pyruvate formate-lyase activating enzyme"><magnitude>
<val> 124.619071</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein, contains CCCH-type Zn-finger protein"><magnitude>
<val> 51.240393</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein, contains HAT (Half-A-TPR) repeat"><magnitude>
<val> 44.444344</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein, contains RING Zn-finger"><magnitude>
<val> 162.98291</val>
</magnitude>
</node>
<node name="Uncharacterized conserved protein, contains RING and CCCH-type Zn-fingers"><magnitude>
<val> 39.773547</val>
</magnitude>
</node>
<node name="Uncharacterized distant relative of homeotic protein bithoraxoid"><magnitude>
<val> 706.067239</val>
</magnitude>
</node>
<node name="Uncharacterized enzyme involved in biosynthesis of extracellular polysaccharides"><magnitude>
<val> 21.736646</val>
</magnitude>
</node>
<node name="Uncharacterized host factor I protein"><magnitude>
<val> 97.624111</val>
</magnitude>
</node>
<node name="Uncharacterized membrane protein (homolog of Drosophila rhomboid)"><magnitude>
<val> 345.948618</val>
</magnitude>
</node>
<node name="Uncharacterized membrane protein affecting hemolysin expression"><magnitude>
<val> 44.155175</val>
</magnitude>
</node>
<node name="Uncharacterized membrane protein required for alginate biosynthesis"><magnitude>
<val> 19.851799</val>
</magnitude>
</node>
<node name="Uncharacterized membrane protein, possible Na+ channel or pump"><magnitude>
<val> 100.611858</val>
</magnitude>
</node>
<node name="Uncharacterized membrane protein, putative virulence factor"><magnitude>
<val> 222.23049</val>
</magnitude>
</node>
<node name="Uncharacterized membrane protein, required for N-linked glycosylation"><magnitude>
<val> 410.647039</val>
</magnitude>
</node>
<node name="Uncharacterized membrane protein, required for colicin V production"><magnitude>
<val> 128.553471</val>
</magnitude>
</node>
<node name="Uncharacterized membrane protein, required for spore maturation in B.subtilis."><magnitude>
<val> 72.835748</val>
</magnitude>
</node>
<node name="Uncharacterized metal-binding protein"><magnitude>
<val> 242.176535</val>
</magnitude>
</node>
<node name="Uncharacterized metal-binding protein conserved in archaea"><magnitude>
<val> 111.654573</val>
</magnitude>
</node>
<node name="Uncharacterized protein (ATP-grasp superfamily)"><magnitude>
<val> 121.83344</val>
</magnitude>
</node>
<node name="Uncharacterized protein (competence- and mitomycin-induced)"><magnitude>
<val> 117.174098</val>
</magnitude>
</node>
<node name="Uncharacterized protein SCO1/SenC/PrrC, involved in biogenesis of respiratory and photosynthetic systems"><magnitude>
<val> 225.281099</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria"><magnitude>
<val> 181.353335</val>
</magnitude>
</node>
<node name="Uncharacterized protein conserved in bacteria with an aminopeptidase-like domain"><magnitude>
<val> 37.058826</val>
</magnitude>
</node>
<node name="Uncharacterized protein contain chitin-binding domain type 3"><magnitude>
<val> 9.519386</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing DHHC-type Zn finger"><magnitude>
<val> 15.61148</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing SIS (Sugar ISomerase) phosphosugar binding domain"><magnitude>
<val> 38.52965</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing TOPRIM domain, potential nuclease"><magnitude>
<val> 68.356089</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing a NRPS condensation (elongation) domain"><magnitude>
<val> 39.303427</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing a divergent version of the methyl-accepting chemotaxis-like domain"><magnitude>
<val> 62.126019</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing a von Willebrand factor type A (vWA) domain"><magnitude>
<val> 196.562723</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing archaeal-type C2H2 Zn-finger"><magnitude>
<val> 11.927114</val>
</magnitude>
</node>
<node name="Uncharacterized protein containing caspase domain"><magnitude>
<val> 100.539623</val>
</magnitude>
</node>
<node name="Uncharacterized protein encoded in toxicity protection region of plasmid R478, contains von Willebrand factor (vWF) domain"><magnitude>
<val> 56.198601</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in cell differentiation/sexual development"><magnitude>
<val> 3.801199</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in plasmid maintenance"><magnitude>
<val> 47.714091</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in propionate catabolism"><magnitude>
<val> 169.814154</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in purine metabolism"><magnitude>
<val> 18.462331</val>
</magnitude>
</node>
<node name="Uncharacterized protein involved in stress response"><magnitude>
<val> 5.929334</val>
</magnitude>
</node>
<node name="Uncharacterized protein related to arylsulfate sulfotransferase involved in siderophore biosynthesis"><magnitude>
<val> 43.008415</val>
</magnitude>
</node>
<node name="Uncharacterized protein related to capsule biosynthesis enzymes"><magnitude>
<val> 166.339305</val>
</magnitude>
</node>
<node name="Uncharacterized protein related to deoxyribodipyrimidine photolyase"><magnitude>
<val> 36.382216</val>
</magnitude>
</node>
<node name="Uncharacterized protein related to glutamine synthetase"><magnitude>
<val> 153.429701</val>
</magnitude>
</node>
<node name="Uncharacterized protein related to methyl coenzyme M reductase subunit C"><magnitude>
<val> 159.753811</val>
</magnitude>
</node>
<node name="Uncharacterized protein related to plant photosystem II stability/assembly factor"><magnitude>
<val> 89.315698</val>
</magnitude>
</node>
<node name="Uncharacterized protein related to proFAR isomerase (HisA)"><magnitude>
<val> 71.907074</val>
</magnitude>
</node>
<node name="Uncharacterized protein with a von Willebrand factor type A (vWA) domain"><magnitude>
<val> 19.929316</val>
</magnitude>
</node>
<node name="Uncharacterized protein with an alpha/beta hydrolase fold"><magnitude>
<val> 57.396343</val>
</magnitude>
</node>
<node name="Uncharacterized protein with protein kinase and helix-hairpin-helix DNA-binding domains"><magnitude>
<val> 76.394088</val>
</magnitude>
</node>
<node name="Uncharacterized protein, 4-oxalocrotonate tautomerase homolog"><magnitude>
<val> 256.622501</val>
</magnitude>
</node>
<node name="Uncharacterized protein, contains Trp-Asp (WD) repeat"><magnitude>
<val> 41.434641</val>
</magnitude>
</node>
<node name="Uncharacterized protein, homolog of Cu resistance protein CopC"><magnitude>
<val> 24.300854</val>
</magnitude>
</node>
<node name="Uncharacterized protein, possibly involved in glyoxylate utilization"><magnitude>
<val> 6.27627</val>
</magnitude>
</node>
<node name="Uncharacterized protein, possibly involved in utilization of glycolate and propanediol"><magnitude>
<val> 39.2197</val>
</magnitude>
</node>
<node name="Uncharacterized protein, putative amidase"><magnitude>
<val> 215.458237</val>
</magnitude>
</node>
<node name="Uncharacterized protein, similar to the N-terminal domain of Lon protease"><magnitude>
<val> 9.068371</val>
</magnitude>
</node>
<node name="Uncharacterized proteins of PilT N-term./Vapc superfamily"><magnitude>
<val> 91.781228</val>
</magnitude>
</node>
<node name="Uncharacterized proteins of the AP superfamily"><magnitude>
<val> 115.928318</val>
</magnitude>
</node>
<node name="Uncharacterized proteins, homologs of lactam utilization protein B"><magnitude>
<val> 85.647008</val>
</magnitude>
</node>
<node name="Uncharacterized relative of glutathione S-transferase, MAPEG superfamily"><magnitude>
<val> 47.495353</val>
</magnitude>
</node>
<node name="Uncharacterized stress protein (general stress protein 26)"><magnitude>
<val> 201.112652</val>
</magnitude>
</node>
<node name="Virulence protein"><magnitude>
<val> 278.235423</val>
</magnitude>
</node>
<node name="Zn finger protein HypA/HybF (possibly regulating hydrogenase expression)"><magnitude>
<val> 193.234832</val>
</magnitude>
</node>
<node name="Zn-dependent alcohol dehydrogenases"><magnitude>
<val> 210.96911</val>
</magnitude>
</node>
<node name="Zn-dependent hydrolases, including glyoxylases"><magnitude>
<val> 975.67054</val>
</magnitude>
</node>
<node name="Zn-dependent proteases"><magnitude>
<val> 242.503755</val>
</magnitude>
</node>
<node name="Zn-ribbon containing protein"><magnitude>
<val> 98.870186</val>
</magnitude>
</node>
<node name="Zn-ribbon protein, possibly nucleic acid-binding"><magnitude>
<val> 138.629281</val>
</magnitude>
</node>
<node name="Zn-ribbon-containing, possibly RNA-binding protein and truncated derivatives"><magnitude>
<val> 27.701467</val>
</magnitude>
</node>
<node name="Zn-ribbon-containing, possibly nucleic-acid-binding protein"><magnitude>
<val> 11.85847</val>
</magnitude>
</node>
<node name="Zonula occludens toxin"><magnitude>
<val> 85.298025</val>
</magnitude>
</node>
</node>
</node>
</node>
    </krona></div>
    </body>
</html>
